--- a/net/minecraft/server/BiomeBaseSub.java
+++ b/net/minecraft/server/BiomeBaseSub.java
@@ -48,7 +48,7 @@
         return this.aE.a(random);
     }
 
-    public Class l() {
+    public Class<? extends BiomeBase> l() {
         return this.aE.l();
     }
 
@@ -56,7 +56,7 @@
         return this.aE.a(biomebase);
     }
 
-    public EnumTemperature m() {
+    public BiomeBase.EnumTemperature m() {
         return this.aE.m();
     }
 }
