--- a/net/minecraft/server/BiomeHell.java
+++ b/net/minecraft/server/BiomeHell.java
@@ -8,8 +8,8 @@
         this.au.clear();
         this.av.clear();
         this.aw.clear();
-        this.at.add(new BiomeMeta(EntityGhast.class, 50, 4, 4));
-        this.at.add(new BiomeMeta(EntityPigZombie.class, 100, 4, 4));
-        this.at.add(new BiomeMeta(EntityMagmaCube.class, 1, 4, 4));
+        this.at.add(new BiomeBase.BiomeMeta(EntityGhast.class, 50, 4, 4));
+        this.at.add(new BiomeBase.BiomeMeta(EntityPigZombie.class, 100, 4, 4));
+        this.at.add(new BiomeBase.BiomeMeta(EntityMagmaCube.class, 1, 4, 4));
     }
 }
