--- a/net/minecraft/server/BlockDropper.java
+++ b/net/minecraft/server/BlockDropper.java
@@ -2,19 +2,19 @@
 
 public class BlockDropper extends BlockDispenser {
 
-    private final IDispenseBehavior O = new DispenseBehaviorItem();
+    private final IDispenseBehavior P = new DispenseBehaviorItem();
 
     public BlockDropper() {}
 
     protected IDispenseBehavior a(ItemStack itemstack) {
-        return this.O;
+        return this.P;
     }
 
     public TileEntity a(World world, int i) {
         return new TileEntityDropper();
     }
 
-    public void dispense(World world, BlockPosition blockposition) {
+    protected void dispense(World world, BlockPosition blockposition) {
         SourceBlock sourceblock = new SourceBlock(world, blockposition);
         TileEntityDispenser tileentitydispenser = (TileEntityDispenser) sourceblock.getTileEntity();
 
@@ -33,7 +33,7 @@
                     ItemStack itemstack1;
 
                     if (iinventory == null) {
-                        itemstack1 = this.O.a(sourceblock, itemstack);
+                        itemstack1 = this.P.a(sourceblock, itemstack);
                         if (itemstack1 != null && itemstack1.count == 0) {
                             itemstack1 = null;
                         }
