--- a/net/minecraft/server/BlockLeaves1.java
+++ b/net/minecraft/server/BlockLeaves1.java
@@ -4,25 +4,33 @@
 
 public class BlockLeaves1 extends BlockLeaves {
 
-    public static final BlockStateEnum VARIANT = BlockStateEnum.a("variant", EnumLogVariant.class, (Predicate) (new BlockLeaves1InnerClass1()));
+    public static final BlockStateEnum<BlockWood.EnumLogVariant> VARIANT = BlockStateEnum.a("variant", BlockWood.EnumLogVariant.class, new Predicate() {
+        public boolean a(BlockWood.EnumLogVariant blockwood_enumlogvariant) {
+            return blockwood_enumlogvariant.a() < 4;
+        }
+
+        public boolean apply(Object object) {
+            return this.a((BlockWood.EnumLogVariant) object);
+        }
+    });
 
     public BlockLeaves1() {
-        this.j(this.blockStateList.getBlockData().set(BlockLeaves1.VARIANT, EnumLogVariant.OAK).set(BlockLeaves1.CHECK_DECAY, Boolean.valueOf(true)).set(BlockLeaves1.DECAYABLE, Boolean.valueOf(true)));
+        this.j(this.blockStateList.getBlockData().set(BlockLeaves1.VARIANT, BlockWood.EnumLogVariant.OAK).set(BlockLeaves1.CHECK_DECAY, Boolean.valueOf(true)).set(BlockLeaves1.DECAYABLE, Boolean.valueOf(true)));
     }
 
     protected void a(World world, BlockPosition blockposition, IBlockData iblockdata, int i) {
-        if (iblockdata.get(BlockLeaves1.VARIANT) == EnumLogVariant.OAK && world.random.nextInt(i) == 0) {
+        if (iblockdata.get(BlockLeaves1.VARIANT) == BlockWood.EnumLogVariant.OAK && world.random.nextInt(i) == 0) {
             a(world, blockposition, new ItemStack(Items.APPLE, 1, 0));
         }
 
     }
 
     protected int d(IBlockData iblockdata) {
-        return iblockdata.get(BlockLeaves1.VARIANT) == EnumLogVariant.JUNGLE ? 40 : super.d(iblockdata);
+        return iblockdata.get(BlockLeaves1.VARIANT) == BlockWood.EnumLogVariant.JUNGLE ? 40 : super.d(iblockdata);
     }
 
     protected ItemStack i(IBlockData iblockdata) {
-        return new ItemStack(Item.getItemOf(this), 1, ((EnumLogVariant) iblockdata.get(BlockLeaves1.VARIANT)).a());
+        return new ItemStack(Item.getItemOf(this), 1, ((BlockWood.EnumLogVariant) iblockdata.get(BlockLeaves1.VARIANT)).a());
     }
 
     public IBlockData fromLegacyData(int i) {
@@ -31,7 +39,7 @@
 
     public int toLegacyData(IBlockData iblockdata) {
         byte b0 = 0;
-        int i = b0 | ((EnumLogVariant) iblockdata.get(BlockLeaves1.VARIANT)).a();
+        int i = b0 | ((BlockWood.EnumLogVariant) iblockdata.get(BlockLeaves1.VARIANT)).a();
 
         if (!((Boolean) iblockdata.get(BlockLeaves1.DECAYABLE)).booleanValue()) {
             i |= 4;
@@ -44,8 +52,8 @@
         return i;
     }
 
-    public EnumLogVariant b(int i) {
-        return EnumLogVariant.a((i & 3) % 4);
+    public BlockWood.EnumLogVariant b(int i) {
+        return BlockWood.EnumLogVariant.a((i & 3) % 4);
     }
 
     protected BlockStateList getStateList() {
@@ -53,13 +61,13 @@
     }
 
     public int getDropData(IBlockData iblockdata) {
-        return ((EnumLogVariant) iblockdata.get(BlockLeaves1.VARIANT)).a();
+        return ((BlockWood.EnumLogVariant) iblockdata.get(BlockLeaves1.VARIANT)).a();
     }
 
     public void a(World world, EntityHuman entityhuman, BlockPosition blockposition, IBlockData iblockdata, TileEntity tileentity) {
-        if (!world.isStatic && entityhuman.bY() != null && entityhuman.bY().getItem() == Items.SHEARS) {
+        if (!world.isClientSide && entityhuman.bZ() != null && entityhuman.bZ().getItem() == Items.SHEARS) {
             entityhuman.b(StatisticList.MINE_BLOCK_COUNT[Block.getId(this)]);
-            a(world, blockposition, new ItemStack(Item.getItemOf(this), 1, ((EnumLogVariant) iblockdata.get(BlockLeaves1.VARIANT)).a()));
+            a(world, blockposition, new ItemStack(Item.getItemOf(this), 1, ((BlockWood.EnumLogVariant) iblockdata.get(BlockLeaves1.VARIANT)).a()));
         } else {
             super.a(world, entityhuman, blockposition, iblockdata, tileentity);
         }
