--- a/net/minecraft/server/BlockCloth.java
+++ b/net/minecraft/server/BlockCloth.java
@@ -2,7 +2,7 @@
 
 public class BlockCloth extends Block {
 
-    public static final BlockStateEnum COLOR = BlockStateEnum.of("color", EnumColor.class);
+    public static final BlockStateEnum<EnumColor> COLOR = BlockStateEnum.of("color", EnumColor.class);
 
     public BlockCloth(Material material) {
         super(material);
