--- a/net/minecraft/server/BlockRotatable.java
+++ b/net/minecraft/server/BlockRotatable.java
@@ -2,9 +2,13 @@
 
 public abstract class BlockRotatable extends Block {
 
-    public static final BlockStateEnum AXIS = BlockStateEnum.of("axis", EnumAxis.class);
+    public static final BlockStateEnum<EnumDirection.EnumAxis> AXIS = BlockStateEnum.of("axis", EnumDirection.EnumAxis.class);
 
     protected BlockRotatable(Material material) {
-        super(material);
+        super(material, material.r());
+    }
+
+    protected BlockRotatable(Material material, MaterialMapColor materialmapcolor) {
+        super(material, materialmapcolor);
     }
 }
