--- a/net/minecraft/server/BlockStateInteger.java
+++ b/net/minecraft/server/BlockStateInteger.java
@@ -5,9 +5,9 @@
 import java.util.Collection;
 import java.util.HashSet;
 
-public class BlockStateInteger extends BlockState {
+public class BlockStateInteger extends BlockState<Integer> {
 
-    private final ImmutableSet a;
+    private final ImmutableSet<Integer> a;
 
     protected BlockStateInteger(String s, int i, int j) {
         super(s, Integer.class);
@@ -26,7 +26,7 @@
         }
     }
 
-    public Collection c() {
+    public Collection<Integer> c() {
         return this.a;
     }
 
