--- a/net/minecraft/server/BlockWoodenStep.java
+++ b/net/minecraft/server/BlockWoodenStep.java
@@ -4,41 +4,45 @@
 
 public abstract class BlockWoodenStep extends BlockStepAbstract {
 
-    public static final BlockStateEnum VARIANT = BlockStateEnum.of("variant", EnumLogVariant.class);
+    public static final BlockStateEnum<BlockWood.EnumLogVariant> VARIANT = BlockStateEnum.of("variant", BlockWood.EnumLogVariant.class);
 
     public BlockWoodenStep() {
         super(Material.WOOD);
         IBlockData iblockdata = this.blockStateList.getBlockData();
 
-        if (!this.j()) {
-            iblockdata = iblockdata.set(BlockWoodenStep.HALF, EnumSlabHalf.BOTTOM);
+        if (!this.l()) {
+            iblockdata = iblockdata.set(BlockWoodenStep.HALF, BlockStepAbstract.EnumSlabHalf.BOTTOM);
         }
 
-        this.j(iblockdata.set(BlockWoodenStep.VARIANT, EnumLogVariant.OAK));
+        this.j(iblockdata.set(BlockWoodenStep.VARIANT, BlockWood.EnumLogVariant.OAK));
         this.a(CreativeModeTab.b);
     }
 
+    public MaterialMapColor g(IBlockData iblockdata) {
+        return ((BlockWood.EnumLogVariant) iblockdata.get(BlockWoodenStep.VARIANT)).c();
+    }
+
     public Item getDropType(IBlockData iblockdata, Random random, int i) {
         return Item.getItemOf(Blocks.WOODEN_SLAB);
     }
 
     public String b(int i) {
-        return super.a() + "." + EnumLogVariant.a(i).c();
+        return super.a() + "." + BlockWood.EnumLogVariant.a(i).d();
     }
 
-    public IBlockState l() {
+    public IBlockState<?> n() {
         return BlockWoodenStep.VARIANT;
     }
 
     public Object a(ItemStack itemstack) {
-        return EnumLogVariant.a(itemstack.getData() & 7);
+        return BlockWood.EnumLogVariant.a(itemstack.getData() & 7);
     }
 
     public IBlockData fromLegacyData(int i) {
-        IBlockData iblockdata = this.getBlockData().set(BlockWoodenStep.VARIANT, EnumLogVariant.a(i & 7));
+        IBlockData iblockdata = this.getBlockData().set(BlockWoodenStep.VARIANT, BlockWood.EnumLogVariant.a(i & 7));
 
-        if (!this.j()) {
-            iblockdata = iblockdata.set(BlockWoodenStep.HALF, (i & 8) == 0 ? EnumSlabHalf.BOTTOM : EnumSlabHalf.TOP);
+        if (!this.l()) {
+            iblockdata = iblockdata.set(BlockWoodenStep.HALF, (i & 8) == 0 ? BlockStepAbstract.EnumSlabHalf.BOTTOM : BlockStepAbstract.EnumSlabHalf.TOP);
         }
 
         return iblockdata;
@@ -46,9 +50,9 @@
 
     public int toLegacyData(IBlockData iblockdata) {
         byte b0 = 0;
-        int i = b0 | ((EnumLogVariant) iblockdata.get(BlockWoodenStep.VARIANT)).a();
+        int i = b0 | ((BlockWood.EnumLogVariant) iblockdata.get(BlockWoodenStep.VARIANT)).a();
 
-        if (!this.j() && iblockdata.get(BlockWoodenStep.HALF) == EnumSlabHalf.TOP) {
+        if (!this.l() && iblockdata.get(BlockWoodenStep.HALF) == BlockStepAbstract.EnumSlabHalf.TOP) {
             i |= 8;
         }
 
@@ -56,10 +60,10 @@
     }
 
     protected BlockStateList getStateList() {
-        return this.j() ? new BlockStateList(this, new IBlockState[] { BlockWoodenStep.VARIANT}) : new BlockStateList(this, new IBlockState[] { BlockWoodenStep.HALF, BlockWoodenStep.VARIANT});
+        return this.l() ? new BlockStateList(this, new IBlockState[] { BlockWoodenStep.VARIANT}) : new BlockStateList(this, new IBlockState[] { BlockWoodenStep.HALF, BlockWoodenStep.VARIANT});
     }
 
     public int getDropData(IBlockData iblockdata) {
-        return ((EnumLogVariant) iblockdata.get(BlockWoodenStep.VARIANT)).a();
+        return ((BlockWood.EnumLogVariant) iblockdata.get(BlockWoodenStep.VARIANT)).a();
     }
 }
