--- a/net/minecraft/server/BlockFlowing.java
+++ b/net/minecraft/server/BlockFlowing.java
@@ -35,7 +35,7 @@
 
             EnumDirection enumdirection;
 
-            for (Iterator iterator = EnumDirectionLimit.HORIZONTAL.iterator(); iterator.hasNext(); l = this.a(world, blockposition.shift(enumdirection), l)) {
+            for (Iterator iterator = EnumDirection.EnumDirectionLimit.HORIZONTAL.iterator(); iterator.hasNext(); l = this.a(world, blockposition.shift(enumdirection), l)) {
                 enumdirection = (EnumDirection) iterator.next();
             }
 
@@ -45,8 +45,8 @@
                 i1 = -1;
             }
 
-            if (this.e((IBlockAccess) world, blockposition.up()) >= 0) {
-                k = this.e((IBlockAccess) world, blockposition.up());
+            if (this.e(world, blockposition.up()) >= 0) {
+                k = this.e(world, blockposition.up());
                 if (k >= 8) {
                     i1 = k;
                 } else {
@@ -100,7 +100,7 @@
                 this.flow(world, blockposition.down(), iblockdata2, i + 8);
             }
         } else if (i >= 0 && (i == 0 || this.g(world, blockposition.down(), iblockdata2))) {
-            Set set = this.e(world, blockposition);
+            Set set = this.f(world, blockposition);
 
             k = i + b0;
             if (i >= 8) {
@@ -139,7 +139,7 @@
 
     private int a(World world, BlockPosition blockposition, int i, EnumDirection enumdirection) {
         int j = 1000;
-        Iterator iterator = EnumDirectionLimit.HORIZONTAL.iterator();
+        Iterator iterator = EnumDirection.EnumDirectionLimit.HORIZONTAL.iterator();
 
         while (iterator.hasNext()) {
             EnumDirection enumdirection1 = (EnumDirection) iterator.next();
@@ -167,10 +167,10 @@
         return j;
     }
 
-    private Set e(World world, BlockPosition blockposition) {
+    private Set<EnumDirection> f(World world, BlockPosition blockposition) {
         int i = 1000;
         EnumSet enumset = EnumSet.noneOf(EnumDirection.class);
-        Iterator iterator = EnumDirectionLimit.HORIZONTAL.iterator();
+        Iterator iterator = EnumDirection.EnumDirectionLimit.HORIZONTAL.iterator();
 
         while (iterator.hasNext()) {
             EnumDirection enumdirection = (EnumDirection) iterator.next();
@@ -207,7 +207,7 @@
     }
 
     protected int a(World world, BlockPosition blockposition, int i) {
-        int j = this.e((IBlockAccess) world, blockposition);
+        int j = this.e(world, blockposition);
 
         if (j < 0) {
             return i;
