--- a/net/minecraft/server/ChunkProviderTheEnd.java
+++ b/net/minecraft/server/ChunkProviderTheEnd.java
@@ -252,7 +252,7 @@
         return "RandomLevelSource";
     }
 
-    public List getMobsFor(EnumCreatureType enumcreaturetype, BlockPosition blockposition) {
+    public List<BiomeBase.BiomeMeta> getMobsFor(EnumCreatureType enumcreaturetype, BlockPosition blockposition) {
         return this.l.getBiome(blockposition).getMobs(enumcreaturetype);
     }
 
