--- a/net/minecraft/server/CommandToggleDownfall.java
+++ b/net/minecraft/server/CommandToggleDownfall.java
@@ -16,7 +16,7 @@
         return "commands.downfall.usage";
     }
 
-    public void execute(ICommandListener icommandlistener, String[] astring) {
+    public void execute(ICommandListener icommandlistener, String[] astring) throws CommandException {
         this.d();
         a(icommandlistener, this, "commands.downfall.success", new Object[0]);
     }
