--- a/net/minecraft/server/CommandList.java
+++ b/net/minecraft/server/CommandList.java
@@ -16,11 +16,11 @@
         return "commands.players.usage";
     }
 
-    public void execute(ICommandListener icommandlistener, String[] astring) {
-        int i = MinecraftServer.getServer().G();
+    public void execute(ICommandListener icommandlistener, String[] astring) throws CommandException {
+        int i = MinecraftServer.getServer().H();
 
-        icommandlistener.sendMessage(new ChatMessage("commands.players.list", new Object[] { Integer.valueOf(i), Integer.valueOf(MinecraftServer.getServer().H())}));
-        icommandlistener.sendMessage(new ChatComponentText(MinecraftServer.getServer().getPlayerList().f()));
-        icommandlistener.a(EnumCommandResult.QUERY_RESULT, i);
+        icommandlistener.sendMessage(new ChatMessage("commands.players.list", new Object[] { Integer.valueOf(i), Integer.valueOf(MinecraftServer.getServer().I())}));
+        icommandlistener.sendMessage(new ChatComponentText(MinecraftServer.getServer().getPlayerList().b(astring.length > 0 && "uuids".equalsIgnoreCase(astring[0]))));
+        icommandlistener.a(CommandObjectiveExecutor.EnumCommandResult.QUERY_RESULT, i);
     }
 }
