--- a/net/minecraft/server/CommandTp.java
+++ b/net/minecraft/server/CommandTp.java
@@ -19,7 +19,7 @@
         return "commands.tp.usage";
     }
 
-    public void execute(ICommandListener icommandlistener, String[] astring) {
+    public void execute(ICommandListener icommandlistener, String[] astring) throws CommandException {
         if (astring.length < 1) {
             throw new ExceptionUsage("commands.tp.usage", new Object[0]);
         } else {
@@ -38,44 +38,44 @@
                     throw new ExceptionUsage("commands.tp.usage", new Object[0]);
                 } else if (((Entity) object).world != null) {
                     int i = b0 + 1;
-                    CommandNumber commandnumber = a(((Entity) object).locX, astring[b0], true);
-                    CommandNumber commandnumber1 = a(((Entity) object).locY, astring[i++], 0, 0, false);
-                    CommandNumber commandnumber2 = a(((Entity) object).locZ, astring[i++], true);
-                    CommandNumber commandnumber3 = a((double) ((Entity) object).yaw, astring.length > i ? astring[i++] : "~", false);
-                    CommandNumber commandnumber4 = a((double) ((Entity) object).pitch, astring.length > i ? astring[i] : "~", false);
+                    CommandAbstract.CommandNumber commandabstract_commandnumber = a(((Entity) object).locX, astring[b0], true);
+                    CommandAbstract.CommandNumber commandabstract_commandnumber1 = a(((Entity) object).locY, astring[i++], 0, 0, false);
+                    CommandAbstract.CommandNumber commandabstract_commandnumber2 = a(((Entity) object).locZ, astring[i++], true);
+                    CommandAbstract.CommandNumber commandabstract_commandnumber3 = a((double) ((Entity) object).yaw, astring.length > i ? astring[i++] : "~", false);
+                    CommandAbstract.CommandNumber commandabstract_commandnumber4 = a((double) ((Entity) object).pitch, astring.length > i ? astring[i] : "~", false);
                     float f;
 
                     if (object instanceof EntityPlayer) {
-                        EnumSet enumset = EnumSet.noneOf(EnumPlayerTeleportFlags.class);
+                        EnumSet enumset = EnumSet.noneOf(PacketPlayOutPosition.EnumPlayerTeleportFlags.class);
 
-                        if (commandnumber.c()) {
-                            enumset.add(EnumPlayerTeleportFlags.X);
+                        if (commandabstract_commandnumber.c()) {
+                            enumset.add(PacketPlayOutPosition.EnumPlayerTeleportFlags.X);
                         }
 
-                        if (commandnumber1.c()) {
-                            enumset.add(EnumPlayerTeleportFlags.Y);
+                        if (commandabstract_commandnumber1.c()) {
+                            enumset.add(PacketPlayOutPosition.EnumPlayerTeleportFlags.Y);
                         }
 
-                        if (commandnumber2.c()) {
-                            enumset.add(EnumPlayerTeleportFlags.Z);
+                        if (commandabstract_commandnumber2.c()) {
+                            enumset.add(PacketPlayOutPosition.EnumPlayerTeleportFlags.Z);
                         }
 
-                        if (commandnumber4.c()) {
-                            enumset.add(EnumPlayerTeleportFlags.X_ROT);
+                        if (commandabstract_commandnumber4.c()) {
+                            enumset.add(PacketPlayOutPosition.EnumPlayerTeleportFlags.X_ROT);
                         }
 
-                        if (commandnumber3.c()) {
-                            enumset.add(EnumPlayerTeleportFlags.Y_ROT);
+                        if (commandabstract_commandnumber3.c()) {
+                            enumset.add(PacketPlayOutPosition.EnumPlayerTeleportFlags.Y_ROT);
                         }
 
-                        f = (float) commandnumber3.b();
-                        if (!commandnumber3.c()) {
+                        f = (float) commandabstract_commandnumber3.b();
+                        if (!commandabstract_commandnumber3.c()) {
                             f = MathHelper.g(f);
                         }
 
-                        float f1 = (float) commandnumber4.b();
+                        float f1 = (float) commandabstract_commandnumber4.b();
 
-                        if (!commandnumber4.c()) {
+                        if (!commandabstract_commandnumber4.c()) {
                             f1 = MathHelper.g(f1);
                         }
 
@@ -85,22 +85,22 @@
                         }
 
                         ((Entity) object).mount((Entity) null);
-                        ((EntityPlayer) object).playerConnection.a(commandnumber.b(), commandnumber1.b(), commandnumber2.b(), f, f1, enumset);
+                        ((EntityPlayer) object).playerConnection.a(commandabstract_commandnumber.b(), commandabstract_commandnumber1.b(), commandabstract_commandnumber2.b(), f, f1, enumset);
                         ((Entity) object).f(f);
                     } else {
-                        float f2 = (float) MathHelper.g(commandnumber3.a());
+                        float f2 = (float) MathHelper.g(commandabstract_commandnumber3.a());
 
-                        f = (float) MathHelper.g(commandnumber4.a());
+                        f = (float) MathHelper.g(commandabstract_commandnumber4.a());
                         if (f > 90.0F || f < -90.0F) {
                             f = MathHelper.g(180.0F - f);
                             f2 = MathHelper.g(f2 + 180.0F);
                         }
 
-                        ((Entity) object).setPositionRotation(commandnumber.a(), commandnumber1.a(), commandnumber2.a(), f2, f);
+                        ((Entity) object).setPositionRotation(commandabstract_commandnumber.a(), commandabstract_commandnumber1.a(), commandabstract_commandnumber2.a(), f2, f);
                         ((Entity) object).f(f2);
                     }
 
-                    a(icommandlistener, this, "commands.tp.success.coordinates", new Object[] { ((Entity) object).getName(), Double.valueOf(commandnumber.a()), Double.valueOf(commandnumber1.a()), Double.valueOf(commandnumber2.a())});
+                    a(icommandlistener, this, "commands.tp.success.coordinates", new Object[] { ((Entity) object).getName(), Double.valueOf(commandabstract_commandnumber.a()), Double.valueOf(commandabstract_commandnumber1.a()), Double.valueOf(commandabstract_commandnumber2.a())});
                 }
             } else {
                 Entity entity = b(icommandlistener, astring[astring.length - 1]);
@@ -121,7 +121,7 @@
         }
     }
 
-    public List tabComplete(ICommandListener icommandlistener, String[] astring, BlockPosition blockposition) {
+    public List<String> tabComplete(ICommandListener icommandlistener, String[] astring, BlockPosition blockposition) {
         return astring.length != 1 && astring.length != 2 ? null : a(astring, MinecraftServer.getServer().getPlayers());
     }
 
