--- a/net/minecraft/server/CommandTell.java
+++ b/net/minecraft/server/CommandTell.java
@@ -7,7 +7,7 @@
 
     public CommandTell() {}
 
-    public List b() {
+    public List<String> b() {
         return Arrays.asList(new String[] { "w", "msg"});
     }
 
@@ -23,7 +23,7 @@
         return "commands.message.usage";
     }
 
-    public void execute(ICommandListener icommandlistener, String[] astring) {
+    public void execute(ICommandListener icommandlistener, String[] astring) throws CommandException {
         if (astring.length < 2) {
             throw new ExceptionUsage("commands.message.usage", new Object[0]);
         } else {
@@ -44,7 +44,7 @@
         }
     }
 
-    public List tabComplete(ICommandListener icommandlistener, String[] astring, BlockPosition blockposition) {
+    public List<String> tabComplete(ICommandListener icommandlistener, String[] astring, BlockPosition blockposition) {
         return a(astring, MinecraftServer.getServer().getPlayers());
     }
 
