--- a/net/minecraft/server/EnchantmentThorns.java
+++ b/net/minecraft/server/EnchantmentThorns.java
@@ -26,12 +26,15 @@
     }
 
     public void b(EntityLiving entityliving, Entity entity, int i) {
-        Random random = entityliving.bb();
+        Random random = entityliving.bc();
         ItemStack itemstack = EnchantmentManager.a(Enchantment.THORNS, entityliving);
 
         if (a(i, random)) {
-            entity.damageEntity(DamageSource.a(entityliving), (float) b(i, random));
-            entity.makeSound("damage.thorns", 0.5F, 1.0F);
+            if (entity != null) {
+                entity.damageEntity(DamageSource.a(entityliving), (float) b(i, random));
+                entity.makeSound("damage.thorns", 0.5F, 1.0F);
+            }
+
             if (itemstack != null) {
                 itemstack.damage(3, entityliving);
             }
