--- a/net/minecraft/server/EnchantmentWeaponDamage.java
+++ b/net/minecraft/server/EnchantmentWeaponDamage.java
@@ -46,7 +46,7 @@
             EntityLiving entityliving1 = (EntityLiving) entity;
 
             if (this.a == 2 && entityliving1.getMonsterType() == EnumMonsterType.ARTHROPOD) {
-                int j = 20 + entityliving.bb().nextInt(10 * i);
+                int j = 20 + entityliving.bc().nextInt(10 * i);
 
                 entityliving1.addEffect(new MobEffect(MobEffectList.SLOWER_MOVEMENT.id, j, 3));
             }
