--- a/net/minecraft/server/EntityMinecartTNT.java
+++ b/net/minecraft/server/EntityMinecartTNT.java
@@ -12,16 +12,16 @@
         super(world, d0, d1, d2);
     }
 
-    public EnumMinecartType s() {
-        return EnumMinecartType.TNT;
+    public EntityMinecartAbstract.EnumMinecartType s() {
+        return EntityMinecartAbstract.EnumMinecartType.TNT;
     }
 
     public IBlockData u() {
         return Blocks.TNT.getBlockData();
     }
 
-    public void s_() {
-        super.s_();
+    public void t_() {
+        super.t_();
         if (this.a > 0) {
             --this.a;
             this.world.addParticle(EnumParticle.SMOKE_NORMAL, this.locX, this.locY + 0.5D, this.locZ, 0.0D, 0.0D, 0.0D, new int[0]);
@@ -57,7 +57,7 @@
         super.a(damagesource);
         double d0 = this.motX * this.motX + this.motZ * this.motZ;
 
-        if (!damagesource.isExplosion()) {
+        if (!damagesource.isExplosion() && this.world.getGameRules().getBoolean("doEntityDrops")) {
             this.a(new ItemStack(Blocks.TNT, 1), 0.0F);
         }
 
@@ -68,7 +68,7 @@
     }
 
     protected void b(double d0) {
-        if (!this.world.isStatic) {
+        if (!this.world.isClientSide) {
             double d1 = Math.sqrt(d0);
 
             if (d1 > 5.0D) {
@@ -100,7 +100,7 @@
 
     public void j() {
         this.a = 80;
-        if (!this.world.isStatic) {
+        if (!this.world.isClientSide) {
             this.world.broadcastEntityEffect(this, (byte) 10);
             if (!this.R()) {
                 this.world.makeSound(this, "game.tnt.primed", 1.0F, 1.0F);
@@ -114,11 +114,11 @@
     }
 
     public float a(Explosion explosion, World world, BlockPosition blockposition, IBlockData iblockdata) {
-        return this.y() && (BlockMinecartTrackAbstract.d(iblockdata) || BlockMinecartTrackAbstract.d(world, blockposition.up())) ? 0.0F : super.a(explosion, world, blockposition, iblockdata);
+        return this.y() && (BlockMinecartTrackAbstract.d(iblockdata) || BlockMinecartTrackAbstract.e(world, blockposition.up())) ? 0.0F : super.a(explosion, world, blockposition, iblockdata);
     }
 
     public boolean a(Explosion explosion, World world, BlockPosition blockposition, IBlockData iblockdata, float f) {
-        return this.y() && (BlockMinecartTrackAbstract.d(iblockdata) || BlockMinecartTrackAbstract.d(world, blockposition.up())) ? false : super.a(explosion, world, blockposition, iblockdata, f);
+        return this.y() && (BlockMinecartTrackAbstract.d(iblockdata) || BlockMinecartTrackAbstract.e(world, blockposition.up())) ? false : super.a(explosion, world, blockposition, iblockdata, f);
     }
 
     protected void a(NBTTagCompound nbttagcompound) {
