--- a/net/minecraft/server/EntityWaterAnimal.java
+++ b/net/minecraft/server/EntityWaterAnimal.java
@@ -6,11 +6,11 @@
         super(world);
     }
 
-    public boolean aX() {
+    public boolean aY() {
         return true;
     }
 
-    public boolean bQ() {
+    public boolean bR() {
         return true;
     }
 
@@ -47,7 +47,7 @@
 
     }
 
-    public boolean aK() {
+    public boolean aL() {
         return false;
     }
 }
