--- a/net/minecraft/server/EntityMinecartMobSpawner.java
+++ b/net/minecraft/server/EntityMinecartMobSpawner.java
@@ -2,7 +2,19 @@
 
 public class EntityMinecartMobSpawner extends EntityMinecartAbstract {
 
-    private final MobSpawnerAbstract a = new MobSpawnerMinecart(this);
+    private final MobSpawnerAbstract a = new MobSpawnerAbstract() {
+        public void a(int i) {
+            EntityMinecartMobSpawner.this.world.broadcastEntityEffect(EntityMinecartMobSpawner.this, (byte) i);
+        }
+
+        public World a() {
+            return EntityMinecartMobSpawner.this.world;
+        }
+
+        public BlockPosition b() {
+            return new BlockPosition(EntityMinecartMobSpawner.this);
+        }
+    };
 
     public EntityMinecartMobSpawner(World world) {
         super(world);
@@ -12,8 +24,8 @@
         super(world, d0, d1, d2);
     }
 
-    public EnumMinecartType s() {
-        return EnumMinecartType.SPAWNER;
+    public EntityMinecartAbstract.EnumMinecartType s() {
+        return EntityMinecartAbstract.EnumMinecartType.SPAWNER;
     }
 
     public IBlockData u() {
@@ -30,8 +42,8 @@
         this.a.b(nbttagcompound);
     }
 
-    public void s_() {
-        super.s_();
+    public void t_() {
+        super.t_();
         this.a.c();
     }
 }
