--- a/net/minecraft/server/EntityEnderPearl.java
+++ b/net/minecraft/server/EntityEnderPearl.java
@@ -2,14 +2,25 @@
 
 public class EntityEnderPearl extends EntityProjectile {
 
+    private EntityLiving c;
+
+    public EntityEnderPearl(World world) {
+        super(world);
+    }
+
     public EntityEnderPearl(World world, EntityLiving entityliving) {
         super(world, entityliving);
+        this.c = entityliving;
     }
 
     protected void a(MovingObjectPosition movingobjectposition) {
         EntityLiving entityliving = this.getShooter();
 
         if (movingobjectposition.entity != null) {
+            if (movingobjectposition.entity == this.c) {
+                return;
+            }
+
             movingobjectposition.entity.damageEntity(DamageSource.projectile(this, entityliving), 0.0F);
         }
 
@@ -17,7 +28,7 @@
             this.world.addParticle(EnumParticle.PORTAL, this.locX, this.locY + this.random.nextDouble() * 2.0D, this.locZ, this.random.nextGaussian(), 0.0D, this.random.nextGaussian(), new int[0]);
         }
 
-        if (!this.world.isStatic) {
+        if (!this.world.isClientSide) {
             if (entityliving instanceof EntityPlayer) {
                 EntityPlayer entityplayer = (EntityPlayer) entityliving;
 
@@ -30,7 +41,7 @@
                         this.world.addEntity(entityendermite);
                     }
 
-                    if (entityliving.av()) {
+                    if (entityliving.au()) {
                         entityliving.mount((Entity) null);
                     }
 
@@ -38,6 +49,9 @@
                     entityliving.fallDistance = 0.0F;
                     entityliving.damageEntity(DamageSource.FALL, 5.0F);
                 }
+            } else if (entityliving != null) {
+                entityliving.enderTeleportTo(this.locX, this.locY, this.locZ);
+                entityliving.fallDistance = 0.0F;
             }
 
             this.die();
@@ -45,13 +59,13 @@
 
     }
 
-    public void s_() {
+    public void t_() {
         EntityLiving entityliving = this.getShooter();
 
         if (entityliving != null && entityliving instanceof EntityHuman && !entityliving.isAlive()) {
             this.die();
         } else {
-            super.s_();
+            super.t_();
         }
 
     }
