--- a/net/minecraft/server/EntityEnderCrystal.java
+++ b/net/minecraft/server/EntityEnderCrystal.java
@@ -8,12 +8,12 @@
     public EntityEnderCrystal(World world) {
         super(world);
         this.k = true;
-        this.a(2.0F, 2.0F);
+        this.setSize(2.0F, 2.0F);
         this.b = 5;
         this.a = this.random.nextInt(100000);
     }
 
-    protected boolean r_() {
+    protected boolean s_() {
         return false;
     }
 
@@ -21,7 +21,7 @@
         this.datawatcher.a(8, Integer.valueOf(this.b));
     }
 
-    public void s_() {
+    public void t_() {
         this.lastX = this.locX;
         this.lastY = this.locY;
         this.lastZ = this.locZ;
@@ -49,11 +49,11 @@
         if (this.isInvulnerable(damagesource)) {
             return false;
         } else {
-            if (!this.dead && !this.world.isStatic) {
+            if (!this.dead && !this.world.isClientSide) {
                 this.b = 0;
                 if (this.b <= 0) {
                     this.die();
-                    if (!this.world.isStatic) {
+                    if (!this.world.isClientSide) {
                         this.world.explode((Entity) null, this.locX, this.locY, this.locZ, 6.0F, true);
                     }
                 }
