--- a/net/minecraft/server/EntityTameableAnimal.java
+++ b/net/minecraft/server/EntityTameableAnimal.java
@@ -4,11 +4,11 @@
 
 public abstract class EntityTameableAnimal extends EntityAnimal implements EntityOwnable {
 
-    protected PathfinderGoalSit bk = new PathfinderGoalSit(this);
+    protected PathfinderGoalSit bm = new PathfinderGoalSit(this);
 
     public EntityTameableAnimal(World world) {
         super(world);
-        this.ck();
+        this.cm();
     }
 
     protected void h() {
@@ -45,7 +45,7 @@
             this.setTamed(true);
         }
 
-        this.bk.setSitting(nbttagcompound.getBoolean("Sitting"));
+        this.bm.setSitting(nbttagcompound.getBoolean("Sitting"));
         this.setSitting(nbttagcompound.getBoolean("Sitting"));
     }
 
@@ -79,10 +79,10 @@
             this.datawatcher.watch(16, Byte.valueOf((byte) (b0 & -5)));
         }
 
-        this.ck();
+        this.cm();
     }
 
-    protected void ck() {}
+    protected void cm() {}
 
     public boolean isSitting() {
         return (this.datawatcher.getByte(16) & 1) != 0;
@@ -122,7 +122,7 @@
     }
 
     public PathfinderGoalSit getGoalSit() {
-        return this.bk;
+        return this.bm;
     }
 
     public boolean a(EntityLiving entityliving, EntityLiving entityliving1) {
@@ -158,8 +158,8 @@
     }
 
     public void die(DamageSource damagesource) {
-        if (!this.world.isStatic && this.world.getGameRules().getBoolean("showDeathMessages") && this.hasCustomName() && this.getOwner() instanceof EntityPlayer) {
-            ((EntityPlayer) this.getOwner()).sendMessage(this.br().b());
+        if (!this.world.isClientSide && this.world.getGameRules().getBoolean("showDeathMessages") && this.hasCustomName() && this.getOwner() instanceof EntityPlayer) {
+            ((EntityPlayer) this.getOwner()).sendMessage(this.bs().b());
         }
 
         super.die(damagesource);
