--- a/net/minecraft/server/EntitySheep.java
+++ b/net/minecraft/server/EntitySheep.java
@@ -6,47 +6,51 @@
 
 public class EntitySheep extends EntityAnimal {
 
-    private final InventoryCrafting bk = new InventoryCrafting(new ContainerSheepBreed(this), 2, 1);
-    private static final Map bm = Maps.newEnumMap(EnumColor.class);
-    private int bn;
-    private PathfinderGoalEatTile bo = new PathfinderGoalEatTile(this);
+    private final InventoryCrafting bm = new InventoryCrafting(new Container() {
+        public boolean a(EntityHuman entityhuman) {
+            return false;
+        }
+    }, 2, 1);
+    private static final Map<EnumColor, float[]> bo = Maps.newEnumMap(EnumColor.class);
+    private int bp;
+    private PathfinderGoalEatTile bq = new PathfinderGoalEatTile(this);
 
     public static float[] a(EnumColor enumcolor) {
-        return (float[]) EntitySheep.bm.get(enumcolor);
+        return (float[]) EntitySheep.bo.get(enumcolor);
     }
 
     public EntitySheep(World world) {
         super(world);
-        this.a(0.9F, 1.3F);
+        this.setSize(0.9F, 1.3F);
         ((Navigation) this.getNavigation()).a(true);
         this.goalSelector.a(0, new PathfinderGoalFloat(this));
         this.goalSelector.a(1, new PathfinderGoalPanic(this, 1.25D));
         this.goalSelector.a(2, new PathfinderGoalBreed(this, 1.0D));
         this.goalSelector.a(3, new PathfinderGoalTempt(this, 1.1D, Items.WHEAT, false));
         this.goalSelector.a(4, new PathfinderGoalFollowParent(this, 1.1D));
-        this.goalSelector.a(5, this.bo);
+        this.goalSelector.a(5, this.bq);
         this.goalSelector.a(6, new PathfinderGoalRandomStroll(this, 1.0D));
         this.goalSelector.a(7, new PathfinderGoalLookAtPlayer(this, EntityHuman.class, 6.0F));
         this.goalSelector.a(8, new PathfinderGoalRandomLookaround(this));
-        this.bk.setItem(0, new ItemStack(Items.DYE, 1, 0));
-        this.bk.setItem(1, new ItemStack(Items.DYE, 1, 0));
+        this.bm.setItem(0, new ItemStack(Items.DYE, 1, 0));
+        this.bm.setItem(1, new ItemStack(Items.DYE, 1, 0));
     }
 
     protected void E() {
-        this.bn = this.bo.f();
+        this.bp = this.bq.f();
         super.E();
     }
 
     public void m() {
-        if (this.world.isStatic) {
-            this.bn = Math.max(0, this.bn - 1);
+        if (this.world.isClientSide) {
+            this.bp = Math.max(0, this.bp - 1);
         }
 
         super.m();
     }
 
-    protected void aW() {
-        super.aW();
+    protected void initAttributes() {
+        super.initAttributes();
         this.getAttributeInstance(GenericAttributes.maxHealth).setValue(8.0D);
         this.getAttributeInstance(GenericAttributes.d).setValue(0.23000000417232513D);
     }
@@ -81,7 +85,7 @@
         ItemStack itemstack = entityhuman.inventory.getItemInHand();
 
         if (itemstack != null && itemstack.getItem() == Items.SHEARS && !this.isSheared() && !this.isBaby()) {
-            if (!this.world.isStatic) {
+            if (!this.world.isClientSide) {
                 this.setSheared(true);
                 int i = 1 + this.random.nextInt(3);
 
@@ -117,11 +121,11 @@
         return "mob.sheep.say";
     }
 
-    protected String bn() {
+    protected String bo() {
         return "mob.sheep.say";
     }
 
-    protected String bo() {
+    protected String bp() {
         return "mob.sheep.say";
     }
 
@@ -186,9 +190,9 @@
         int i = ((EntitySheep) entityanimal).getColor().getInvColorIndex();
         int j = ((EntitySheep) entityanimal1).getColor().getInvColorIndex();
 
-        this.bk.getItem(0).setData(i);
-        this.bk.getItem(1).setData(j);
-        ItemStack itemstack = CraftingManager.getInstance().craft(this.bk, ((EntitySheep) entityanimal).world);
+        this.bm.getItem(0).setData(i);
+        this.bm.getItem(1).setData(j);
+        ItemStack itemstack = CraftingManager.getInstance().craft(this.bm, ((EntitySheep) entityanimal).world);
         int k;
 
         if (itemstack != null && itemstack.getItem() == Items.DYE) {
@@ -209,21 +213,21 @@
     }
 
     static {
-        EntitySheep.bm.put(EnumColor.WHITE, new float[] { 1.0F, 1.0F, 1.0F});
-        EntitySheep.bm.put(EnumColor.ORANGE, new float[] { 0.85F, 0.5F, 0.2F});
-        EntitySheep.bm.put(EnumColor.MAGENTA, new float[] { 0.7F, 0.3F, 0.85F});
-        EntitySheep.bm.put(EnumColor.LIGHT_BLUE, new float[] { 0.4F, 0.6F, 0.85F});
-        EntitySheep.bm.put(EnumColor.YELLOW, new float[] { 0.9F, 0.9F, 0.2F});
-        EntitySheep.bm.put(EnumColor.LIME, new float[] { 0.5F, 0.8F, 0.1F});
-        EntitySheep.bm.put(EnumColor.PINK, new float[] { 0.95F, 0.5F, 0.65F});
-        EntitySheep.bm.put(EnumColor.GRAY, new float[] { 0.3F, 0.3F, 0.3F});
-        EntitySheep.bm.put(EnumColor.SILVER, new float[] { 0.6F, 0.6F, 0.6F});
-        EntitySheep.bm.put(EnumColor.CYAN, new float[] { 0.3F, 0.5F, 0.6F});
-        EntitySheep.bm.put(EnumColor.PURPLE, new float[] { 0.5F, 0.25F, 0.7F});
-        EntitySheep.bm.put(EnumColor.BLUE, new float[] { 0.2F, 0.3F, 0.7F});
-        EntitySheep.bm.put(EnumColor.BROWN, new float[] { 0.4F, 0.3F, 0.2F});
-        EntitySheep.bm.put(EnumColor.GREEN, new float[] { 0.4F, 0.5F, 0.2F});
-        EntitySheep.bm.put(EnumColor.RED, new float[] { 0.6F, 0.2F, 0.2F});
-        EntitySheep.bm.put(EnumColor.BLACK, new float[] { 0.1F, 0.1F, 0.1F});
+        EntitySheep.bo.put(EnumColor.WHITE, new float[] { 1.0F, 1.0F, 1.0F});
+        EntitySheep.bo.put(EnumColor.ORANGE, new float[] { 0.85F, 0.5F, 0.2F});
+        EntitySheep.bo.put(EnumColor.MAGENTA, new float[] { 0.7F, 0.3F, 0.85F});
+        EntitySheep.bo.put(EnumColor.LIGHT_BLUE, new float[] { 0.4F, 0.6F, 0.85F});
+        EntitySheep.bo.put(EnumColor.YELLOW, new float[] { 0.9F, 0.9F, 0.2F});
+        EntitySheep.bo.put(EnumColor.LIME, new float[] { 0.5F, 0.8F, 0.1F});
+        EntitySheep.bo.put(EnumColor.PINK, new float[] { 0.95F, 0.5F, 0.65F});
+        EntitySheep.bo.put(EnumColor.GRAY, new float[] { 0.3F, 0.3F, 0.3F});
+        EntitySheep.bo.put(EnumColor.SILVER, new float[] { 0.6F, 0.6F, 0.6F});
+        EntitySheep.bo.put(EnumColor.CYAN, new float[] { 0.3F, 0.5F, 0.6F});
+        EntitySheep.bo.put(EnumColor.PURPLE, new float[] { 0.5F, 0.25F, 0.7F});
+        EntitySheep.bo.put(EnumColor.BLUE, new float[] { 0.2F, 0.3F, 0.7F});
+        EntitySheep.bo.put(EnumColor.BROWN, new float[] { 0.4F, 0.3F, 0.2F});
+        EntitySheep.bo.put(EnumColor.GREEN, new float[] { 0.4F, 0.5F, 0.2F});
+        EntitySheep.bo.put(EnumColor.RED, new float[] { 0.6F, 0.2F, 0.2F});
+        EntitySheep.bo.put(EnumColor.BLACK, new float[] { 0.1F, 0.1F, 0.1F});
     }
 }
