--- a/net/minecraft/server/EntitySquid.java
+++ b/net/minecraft/server/EntitySquid.java
@@ -5,28 +5,28 @@
     public float a;
     public float b;
     public float c;
-    public float bi;
-    public float bj;
     public float bk;
     public float bl;
     public float bm;
-    private float bn;
-    private float bo;
+    public float bn;
+    public float bo;
     private float bp;
     private float bq;
     private float br;
     private float bs;
+    private float bt;
+    private float bu;
 
     public EntitySquid(World world) {
         super(world);
-        this.a(0.95F, 0.95F);
+        this.setSize(0.95F, 0.95F);
         this.random.setSeed((long) (1 + this.getId()));
-        this.bo = 1.0F / (this.random.nextFloat() + 1.0F) * 0.2F;
-        this.goalSelector.a(0, new PathfinderGoalSquid(this));
+        this.bq = 1.0F / (this.random.nextFloat() + 1.0F) * 0.2F;
+        this.goalSelector.a(0, new EntitySquid.PathfinderGoalSquid(this));
     }
 
-    protected void aW() {
-        super.aW();
+    protected void initAttributes() {
+        super.initAttributes();
         this.getAttributeInstance(GenericAttributes.maxHealth).setValue(10.0D);
     }
 
@@ -38,15 +38,15 @@
         return null;
     }
 
-    protected String bn() {
+    protected String bo() {
         return null;
     }
 
-    protected String bo() {
+    protected String bp() {
         return null;
     }
 
-    protected float bA() {
+    protected float bB() {
         return 0.4F;
     }
 
@@ -54,7 +54,7 @@
         return null;
     }
 
-    protected boolean r_() {
+    protected boolean s_() {
         return false;
     }
 
@@ -74,17 +74,17 @@
     public void m() {
         super.m();
         this.b = this.a;
-        this.bi = this.c;
-        this.bk = this.bj;
+        this.bk = this.c;
         this.bm = this.bl;
-        this.bj += this.bo;
-        if ((double) this.bj > 6.283185307179586D) {
-            if (this.world.isStatic) {
-                this.bj = 6.2831855F;
+        this.bo = this.bn;
+        this.bl += this.bq;
+        if ((double) this.bl > 6.283185307179586D) {
+            if (this.world.isClientSide) {
+                this.bl = 6.2831855F;
             } else {
-                this.bj = (float) ((double) this.bj - 6.283185307179586D);
+                this.bl = (float) ((double) this.bl - 6.283185307179586D);
                 if (this.random.nextInt(10) == 0) {
-                    this.bo = 1.0F / (this.random.nextFloat() + 1.0F) * 0.2F;
+                    this.bq = 1.0F / (this.random.nextFloat() + 1.0F) * 0.2F;
                 }
 
                 this.world.broadcastEntityEffect(this, (byte) 19);
@@ -94,35 +94,35 @@
         if (this.inWater) {
             float f;
 
-            if (this.bj < 3.1415927F) {
-                f = this.bj / 3.1415927F;
-                this.bl = MathHelper.sin(f * f * 3.1415927F) * 3.1415927F * 0.25F;
+            if (this.bl < 3.1415927F) {
+                f = this.bl / 3.1415927F;
+                this.bn = MathHelper.sin(f * f * 3.1415927F) * 3.1415927F * 0.25F;
                 if ((double) f > 0.75D) {
-                    this.bn = 1.0F;
                     this.bp = 1.0F;
+                    this.br = 1.0F;
                 } else {
-                    this.bp *= 0.8F;
+                    this.br *= 0.8F;
                 }
             } else {
-                this.bl = 0.0F;
-                this.bn *= 0.9F;
-                this.bp *= 0.99F;
+                this.bn = 0.0F;
+                this.bp *= 0.9F;
+                this.br *= 0.99F;
             }
 
-            if (!this.world.isStatic) {
-                this.motX = (double) (this.bq * this.bn);
-                this.motY = (double) (this.br * this.bn);
-                this.motZ = (double) (this.bs * this.bn);
+            if (!this.world.isClientSide) {
+                this.motX = (double) (this.bs * this.bp);
+                this.motY = (double) (this.bt * this.bp);
+                this.motZ = (double) (this.bu * this.bp);
             }
 
             f = MathHelper.sqrt(this.motX * this.motX + this.motZ * this.motZ);
-            this.aG += (-((float) Math.atan2(this.motX, this.motZ)) * 180.0F / 3.1415927F - this.aG) * 0.1F;
-            this.yaw = this.aG;
-            this.c = (float) ((double) this.c + 3.141592653589793D * (double) this.bp * 1.5D);
-            this.a += (-((float) Math.atan2((double) f, this.motY)) * 180.0F / 3.1415927F - this.a) * 0.1F;
+            this.aI += (-((float) MathHelper.b(this.motX, this.motZ)) * 180.0F / 3.1415927F - this.aI) * 0.1F;
+            this.yaw = this.aI;
+            this.c = (float) ((double) this.c + 3.141592653589793D * (double) this.br * 1.5D);
+            this.a += (-((float) MathHelper.b((double) f, this.motY)) * 180.0F / 3.1415927F - this.a) * 0.1F;
         } else {
-            this.bl = MathHelper.e(MathHelper.sin(this.bj)) * 3.1415927F * 0.25F;
-            if (!this.world.isStatic) {
+            this.bn = MathHelper.e(MathHelper.sin(this.bl)) * 3.1415927F * 0.25F;
+            if (!this.world.isClientSide) {
                 this.motX = 0.0D;
                 this.motY -= 0.08D;
                 this.motY *= 0.9800000190734863D;
@@ -138,21 +138,46 @@
         this.move(this.motX, this.motY, this.motZ);
     }
 
-    public boolean bQ() {
-        return this.locY > 45.0D && this.locY < 63.0D && super.bQ();
+    public boolean bR() {
+        return this.locY > 45.0D && this.locY < (double) this.world.F() && super.bR();
     }
 
     public void b(float f, float f1, float f2) {
-        this.bq = f;
-        this.br = f1;
-        this.bs = f2;
+        this.bs = f;
+        this.bt = f1;
+        this.bu = f2;
     }
 
     public boolean n() {
-        return this.bq != 0.0F || this.br != 0.0F || this.bs != 0.0F;
+        return this.bs != 0.0F || this.bt != 0.0F || this.bu != 0.0F;
     }
 
-    static boolean a(EntitySquid entitysquid) {
-        return entitysquid.inWater;
+    static class PathfinderGoalSquid extends PathfinderGoal {
+
+        private EntitySquid a;
+
+        public PathfinderGoalSquid(EntitySquid entitysquid) {
+            this.a = entitysquid;
+        }
+
+        public boolean a() {
+            return true;
+        }
+
+        public void e() {
+            int i = this.a.bh();
+
+            if (i > 100) {
+                this.a.b(0.0F, 0.0F, 0.0F);
+            } else if (this.a.bc().nextInt(50) == 0 || !this.a.inWater || !this.a.n()) {
+                float f = this.a.bc().nextFloat() * 3.1415927F * 2.0F;
+                float f1 = MathHelper.cos(f) * 0.2F;
+                float f2 = -0.1F + this.a.bc().nextFloat() * 0.2F;
+                float f3 = MathHelper.sin(f) * 0.2F;
+
+                this.a.b(f1, f2, f3);
+            }
+
+        }
     }
 }
