--- a/net/minecraft/server/FoodMetaData.java
+++ b/net/minecraft/server/FoodMetaData.java
@@ -2,10 +2,10 @@
 
 public class FoodMetaData {
 
-    public int foodLevel = 20;
-    public float saturationLevel = 5.0F;
-    public float exhaustionLevel;
-    public int foodTickTimer;
+    private int foodLevel = 20;
+    private float saturationLevel = 5.0F;
+    private float exhaustionLevel;
+    private int foodTickTimer;
     private int e = 20;
 
     public FoodMetaData() {}
@@ -32,7 +32,7 @@
             }
         }
 
-        if (entityhuman.world.getGameRules().getBoolean("naturalRegeneration") && this.foodLevel >= 18 && entityhuman.cl()) {
+        if (entityhuman.world.getGameRules().getBoolean("naturalRegeneration") && this.foodLevel >= 18 && entityhuman.cm()) {
             ++this.foodTickTimer;
             if (this.foodTickTimer >= 80) {
                 entityhuman.heal(1.0F);
