--- a/net/minecraft/server/IChunkLoader.java
+++ b/net/minecraft/server/IChunkLoader.java
@@ -1,12 +1,14 @@
 package net.minecraft.server;
 
+import java.io.IOException;
+
 public interface IChunkLoader {
 
-    Chunk a(World world, int i, int j);
+    Chunk a(World world, int i, int j) throws IOException;
 
-    void a(World world, Chunk chunk);
+    void a(World world, Chunk chunk) throws IOException, ExceptionWorldConflict;
 
-    void b(World world, Chunk chunk);
+    void b(World world, Chunk chunk) throws IOException;
 
     void a();
 
