--- a/net/minecraft/server/ICommand.java
+++ b/net/minecraft/server/ICommand.java
@@ -2,19 +2,19 @@
 
 import java.util.List;
 
-public interface ICommand extends Comparable {
+public interface ICommand extends Comparable<ICommand> {
 
     String getCommand();
 
     String getUsage(ICommandListener icommandlistener);
 
-    List b();
+    List<String> b();
 
-    void execute(ICommandListener icommandlistener, String[] astring);
+    void execute(ICommandListener icommandlistener, String[] astring) throws CommandException;
 
     boolean canUse(ICommandListener icommandlistener);
 
-    List tabComplete(ICommandListener icommandlistener, String[] astring, BlockPosition blockposition);
+    List<String> tabComplete(ICommandListener icommandlistener, String[] astring, BlockPosition blockposition);
 
     boolean isListStart(String[] astring, int i);
 }
