--- a/net/minecraft/server/IMonster.java
+++ b/net/minecraft/server/IMonster.java
@@ -4,6 +4,22 @@
 
 public interface IMonster extends IAnimal {
 
-    Predicate d = new EntitySelectorMonster();
-    Predicate e = new EntitySelectorVisibleMonster();
+    Predicate<Entity> d = new Predicate() {
+        public boolean a(Entity entity) {
+            return entity instanceof IMonster;
+        }
+
+        public boolean apply(Object object) {
+            return this.a((Entity) object);
+        }
+    };
+    Predicate<Entity> e = new Predicate() {
+        public boolean a(Entity entity) {
+            return entity instanceof IMonster && !entity.isInvisible();
+        }
+
+        public boolean apply(Object object) {
+            return this.a((Entity) object);
+        }
+    };
 }
