--- a/net/minecraft/server/ItemRedstone.java
+++ b/net/minecraft/server/ItemRedstone.java
@@ -7,7 +7,7 @@
     }
 
     public boolean interactWith(ItemStack itemstack, EntityHuman entityhuman, World world, BlockPosition blockposition, EnumDirection enumdirection, float f, float f1, float f2) {
-        boolean flag = world.getType(blockposition).getBlock().f(world, blockposition);
+        boolean flag = world.getType(blockposition).getBlock().a(world, blockposition);
         BlockPosition blockposition1 = flag ? blockposition : blockposition.shift(enumdirection);
 
         if (!entityhuman.a(blockposition1, enumdirection, itemstack)) {
