--- a/net/minecraft/server/ItemEnderEye.java
+++ b/net/minecraft/server/ItemEnderEye.java
@@ -10,7 +10,7 @@
         IBlockData iblockdata = world.getType(blockposition);
 
         if (entityhuman.a(blockposition.shift(enumdirection), enumdirection, itemstack) && iblockdata.getBlock() == Blocks.END_PORTAL_FRAME && !((Boolean) iblockdata.get(BlockEnderPortalFrame.EYE)).booleanValue()) {
-            if (world.isStatic) {
+            if (world.isClientSide) {
                 return true;
             } else {
                 world.setTypeAndData(blockposition, iblockdata.set(BlockEnderPortalFrame.EYE, Boolean.valueOf(true)), 2);
@@ -107,10 +107,10 @@
     public ItemStack a(ItemStack itemstack, World world, EntityHuman entityhuman) {
         MovingObjectPosition movingobjectposition = this.a(world, entityhuman, false);
 
-        if (movingobjectposition != null && movingobjectposition.type == EnumMovingObjectType.BLOCK && world.getType(movingobjectposition.a()).getBlock() == Blocks.END_PORTAL_FRAME) {
+        if (movingobjectposition != null && movingobjectposition.type == MovingObjectPosition.EnumMovingObjectType.BLOCK && world.getType(movingobjectposition.a()).getBlock() == Blocks.END_PORTAL_FRAME) {
             return itemstack;
         } else {
-            if (!world.isStatic) {
+            if (!world.isClientSide) {
                 BlockPosition blockposition = world.a("Stronghold", new BlockPosition(entityhuman));
 
                 if (blockposition != null) {
