--- a/net/minecraft/server/ItemAxe.java
+++ b/net/minecraft/server/ItemAxe.java
@@ -5,10 +5,10 @@
 
 public class ItemAxe extends ItemTool {
 
-    private static final Set c = Sets.newHashSet(new Block[] { Blocks.PLANKS, Blocks.BOOKSHELF, Blocks.LOG, Blocks.LOG2, Blocks.CHEST, Blocks.PUMPKIN, Blocks.LIT_PUMPKIN, Blocks.MELON_BLOCK, Blocks.LADDER});
+    private static final Set<Block> c = Sets.newHashSet(new Block[] { Blocks.PLANKS, Blocks.BOOKSHELF, Blocks.LOG, Blocks.LOG2, Blocks.CHEST, Blocks.PUMPKIN, Blocks.LIT_PUMPKIN, Blocks.MELON_BLOCK, Blocks.LADDER});
 
-    protected ItemAxe(EnumToolMaterial enumtoolmaterial) {
-        super(3.0F, enumtoolmaterial, ItemAxe.c);
+    protected ItemAxe(Item.EnumToolMaterial item_enumtoolmaterial) {
+        super(3.0F, item_enumtoolmaterial, ItemAxe.c);
     }
 
     public float getDestroySpeed(ItemStack itemstack, Block block) {
