--- a/net/minecraft/server/ItemMultiTexture.java
+++ b/net/minecraft/server/ItemMultiTexture.java
@@ -5,9 +5,9 @@
 public class ItemMultiTexture extends ItemBlock {
 
     protected final Block b;
-    protected final Function c;
+    protected final Function<ItemStack, String> c;
 
-    public ItemMultiTexture(Block block, Block block1, Function function) {
+    public ItemMultiTexture(Block block, Block block1, Function<ItemStack, String> function) {
         super(block);
         this.b = block1;
         this.c = function;
@@ -15,8 +15,22 @@
         this.a(true);
     }
 
-    public ItemMultiTexture(Block block, Block block1, String[] astring) {
-        this(block, block1, (Function) (new ItemMultiTextureInnerClass1(astring)));
+    public ItemMultiTexture(Block block, Block block1, final String[] astring) {
+        this(block, block1, new Function() {
+            public String a(ItemStack itemstack) {
+                int i = itemstack.getData();
+
+                if (i < 0 || i >= astring.length) {
+                    i = 0;
+                }
+
+                return astring[i];
+            }
+
+            public Object apply(Object object) {
+                return this.a((ItemStack) object);
+            }
+        });
     }
 
     public int filterData(int i) {
