--- a/net/minecraft/server/ItemTool.java
+++ b/net/minecraft/server/ItemTool.java
@@ -5,18 +5,18 @@
 
 public class ItemTool extends Item {
 
-    private Set c;
+    private Set<Block> c;
     protected float a = 4.0F;
     private float d;
-    protected EnumToolMaterial b;
+    protected Item.EnumToolMaterial b;
 
-    protected ItemTool(float f, EnumToolMaterial enumtoolmaterial, Set set) {
-        this.b = enumtoolmaterial;
+    protected ItemTool(float f, Item.EnumToolMaterial item_enumtoolmaterial, Set<Block> set) {
+        this.b = item_enumtoolmaterial;
         this.c = set;
         this.maxStackSize = 1;
-        this.setMaxDurability(enumtoolmaterial.a());
-        this.a = enumtoolmaterial.b();
-        this.d = f + enumtoolmaterial.c();
+        this.setMaxDurability(item_enumtoolmaterial.a());
+        this.a = item_enumtoolmaterial.b();
+        this.d = f + item_enumtoolmaterial.c();
         this.a(CreativeModeTab.i);
     }
 
@@ -37,7 +37,7 @@
         return true;
     }
 
-    public EnumToolMaterial g() {
+    public Item.EnumToolMaterial g() {
         return this.b;
     }
 
@@ -53,7 +53,7 @@
         return this.b.f() == itemstack1.getItem() ? true : super.a(itemstack, itemstack1);
     }
 
-    public Multimap i() {
+    public Multimap<String, AttributeModifier> i() {
         Multimap multimap = super.i();
 
         multimap.put(GenericAttributes.e.getName(), new AttributeModifier(ItemTool.f, "Tool modifier", (double) this.d, 0));
