--- a/net/minecraft/server/JsonListEntry.java
+++ b/net/minecraft/server/JsonListEntry.java
@@ -2,19 +2,19 @@
 
 import com.google.gson.JsonObject;
 
-public class JsonListEntry {
+public class JsonListEntry<T> {
 
-    private final Object a;
+    private final T a;
 
-    public JsonListEntry(Object object) {
-        this.a = object;
+    public JsonListEntry(T t0) {
+        this.a = t0;
     }
 
-    protected JsonListEntry(Object object, JsonObject jsonobject) {
-        this.a = object;
+    protected JsonListEntry(T t0, JsonObject jsonobject) {
+        this.a = t0;
     }
 
-    public Object getKey() {
+    T getKey() {
         return this.a;
     }
 
