--- a/net/minecraft/server/LazyInitVar.java
+++ b/net/minecraft/server/LazyInitVar.java
@@ -1,13 +1,13 @@
 package net.minecraft.server;
 
-public abstract class LazyInitVar {
+public abstract class LazyInitVar<T> {
 
-    private Object a;
+    private T a;
     private boolean b = false;
 
     public LazyInitVar() {}
 
-    public Object c() {
+    public T c() {
         if (!this.b) {
             this.b = true;
             this.a = this.init();
@@ -16,5 +16,5 @@
         return this.a;
     }
 
-    protected abstract Object init();
+    protected abstract T init();
 }
