--- a/net/minecraft/server/MaterialMapColor.java
+++ b/net/minecraft/server/MaterialMapColor.java
@@ -6,17 +6,17 @@
     public static final MaterialMapColor b = new MaterialMapColor(0, 0);
     public static final MaterialMapColor c = new MaterialMapColor(1, 8368696);
     public static final MaterialMapColor d = new MaterialMapColor(2, 16247203);
-    public static final MaterialMapColor e = new MaterialMapColor(3, 10987431);
+    public static final MaterialMapColor e = new MaterialMapColor(3, 13092807);
     public static final MaterialMapColor f = new MaterialMapColor(4, 16711680);
     public static final MaterialMapColor g = new MaterialMapColor(5, 10526975);
     public static final MaterialMapColor h = new MaterialMapColor(6, 10987431);
     public static final MaterialMapColor i = new MaterialMapColor(7, 31744);
     public static final MaterialMapColor j = new MaterialMapColor(8, 16777215);
     public static final MaterialMapColor k = new MaterialMapColor(9, 10791096);
-    public static final MaterialMapColor l = new MaterialMapColor(10, 12020271);
+    public static final MaterialMapColor l = new MaterialMapColor(10, 9923917);
     public static final MaterialMapColor m = new MaterialMapColor(11, 7368816);
     public static final MaterialMapColor n = new MaterialMapColor(12, 4210943);
-    public static final MaterialMapColor o = new MaterialMapColor(13, 6837042);
+    public static final MaterialMapColor o = new MaterialMapColor(13, 9402184);
     public static final MaterialMapColor p = new MaterialMapColor(14, 16776437);
     public static final MaterialMapColor q = new MaterialMapColor(15, 14188339);
     public static final MaterialMapColor r = new MaterialMapColor(16, 11685080);
@@ -37,7 +37,7 @@
     public static final MaterialMapColor G = new MaterialMapColor(31, 6085589);
     public static final MaterialMapColor H = new MaterialMapColor(32, 4882687);
     public static final MaterialMapColor I = new MaterialMapColor(33, '\ud93a');
-    public static final MaterialMapColor J = new MaterialMapColor(34, 1381407);
+    public static final MaterialMapColor J = new MaterialMapColor(34, 8476209);
     public static final MaterialMapColor K = new MaterialMapColor(35, 7340544);
     public final int L;
     public final int M;
