--- a/net/minecraft/server/Navigation.java
+++ b/net/minecraft/server/Navigation.java
@@ -18,7 +18,7 @@
     }
 
     protected boolean b() {
-        return this.b.onGround || this.h() && this.o() || this.b.av() && this.b instanceof EntityZombie && this.b.vehicle instanceof EntityChicken;
+        return this.b.onGround || this.h() && this.o() || this.b.au() && this.b instanceof EntityZombie && this.b.vehicle instanceof EntityChicken;
     }
 
     protected Vec3D c() {
