--- a/net/minecraft/server/NBTBase.java
+++ b/net/minecraft/server/NBTBase.java
@@ -2,14 +2,15 @@
 
 import java.io.DataInput;
 import java.io.DataOutput;
+import java.io.IOException;
 
 public abstract class NBTBase {
 
     public static final String[] a = new String[] { "END", "BYTE", "SHORT", "INT", "LONG", "FLOAT", "DOUBLE", "BYTE[]", "STRING", "LIST", "COMPOUND", "INT[]"};
 
-    abstract void write(DataOutput dataoutput);
+    abstract void write(DataOutput dataoutput) throws IOException;
 
-    abstract void load(DataInput datainput, int i, NBTReadLimiter nbtreadlimiter);
+    abstract void load(DataInput datainput, int i, NBTReadLimiter nbtreadlimiter) throws IOException;
 
     public abstract String toString();
 
@@ -83,4 +84,21 @@
     protected String a_() {
         return this.toString();
     }
+
+    public abstract static class NBTNumber extends NBTBase {
+
+        protected NBTNumber() {}
+
+        public abstract long c();
+
+        public abstract int d();
+
+        public abstract short e();
+
+        public abstract byte f();
+
+        public abstract double g();
+
+        public abstract float h();
+    }
 }
