--- a/net/minecraft/server/NBTTagString.java
+++ b/net/minecraft/server/NBTTagString.java
@@ -2,6 +2,7 @@
 
 import java.io.DataInput;
 import java.io.DataOutput;
+import java.io.IOException;
 
 public class NBTTagString extends NBTBase {
 
@@ -18,11 +19,11 @@
         }
     }
 
-    void write(DataOutput dataoutput) {
+    void write(DataOutput dataoutput) throws IOException {
         dataoutput.writeUTF(this.data);
     }
 
-    void load(DataInput datainput, int i, NBTReadLimiter nbtreadlimiter) {
+    void load(DataInput datainput, int i, NBTReadLimiter nbtreadlimiter) throws IOException {
         this.data = datainput.readUTF();
         nbtreadlimiter.a((long) (16 * this.data.length()));
     }
