--- a/net/minecraft/server/NBTTagDouble.java
+++ b/net/minecraft/server/NBTTagDouble.java
@@ -2,8 +2,9 @@
 
 import java.io.DataInput;
 import java.io.DataOutput;
+import java.io.IOException;
 
-public class NBTTagDouble extends NBTNumber {
+public class NBTTagDouble extends NBTBase.NBTNumber {
 
     private double data;
 
@@ -13,11 +14,11 @@
         this.data = d0;
     }
 
-    void write(DataOutput dataoutput) {
+    void write(DataOutput dataoutput) throws IOException {
         dataoutput.writeDouble(this.data);
     }
 
-    void load(DataInput datainput, int i, NBTReadLimiter nbtreadlimiter) {
+    void load(DataInput datainput, int i, NBTReadLimiter nbtreadlimiter) throws IOException {
         nbtreadlimiter.a(64L);
         this.data = datainput.readDouble();
     }
