--- a/net/minecraft/server/NBTTagLong.java
+++ b/net/minecraft/server/NBTTagLong.java
@@ -2,8 +2,9 @@
 
 import java.io.DataInput;
 import java.io.DataOutput;
+import java.io.IOException;
 
-public class NBTTagLong extends NBTNumber {
+public class NBTTagLong extends NBTBase.NBTNumber {
 
     private long data;
 
@@ -13,11 +14,11 @@
         this.data = i;
     }
 
-    void write(DataOutput dataoutput) {
+    void write(DataOutput dataoutput) throws IOException {
         dataoutput.writeLong(this.data);
     }
 
-    void load(DataInput datainput, int i, NBTReadLimiter nbtreadlimiter) {
+    void load(DataInput datainput, int i, NBTReadLimiter nbtreadlimiter) throws IOException {
         nbtreadlimiter.a(64L);
         this.data = datainput.readLong();
     }
