--- a/net/minecraft/server/NBTTagFloat.java
+++ b/net/minecraft/server/NBTTagFloat.java
@@ -2,8 +2,9 @@
 
 import java.io.DataInput;
 import java.io.DataOutput;
+import java.io.IOException;
 
-public class NBTTagFloat extends NBTNumber {
+public class NBTTagFloat extends NBTBase.NBTNumber {
 
     private float data;
 
@@ -13,11 +14,11 @@
         this.data = f;
     }
 
-    void write(DataOutput dataoutput) {
+    void write(DataOutput dataoutput) throws IOException {
         dataoutput.writeFloat(this.data);
     }
 
-    void load(DataInput datainput, int i, NBTReadLimiter nbtreadlimiter) {
+    void load(DataInput datainput, int i, NBTReadLimiter nbtreadlimiter) throws IOException {
         nbtreadlimiter.a(32L);
         this.data = datainput.readFloat();
     }
