--- a/net/minecraft/server/PacketPlayOutEntityDestroy.java
+++ b/net/minecraft/server/PacketPlayOutEntityDestroy.java
@@ -1,6 +1,8 @@
 package net.minecraft.server;
 
-public class PacketPlayOutEntityDestroy implements Packet {
+import java.io.IOException;
+
+public class PacketPlayOutEntityDestroy implements Packet<PacketListenerPlayOut> {
 
     private int[] a;
 
@@ -10,7 +12,7 @@
         this.a = aint;
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = new int[packetdataserializer.e()];
 
         for (int i = 0; i < this.a.length; ++i) {
@@ -19,7 +21,7 @@
 
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.b(this.a.length);
 
         for (int i = 0; i < this.a.length; ++i) {
