--- a/net/minecraft/server/PacketPlayInSpectate.java
+++ b/net/minecraft/server/PacketPlayInSpectate.java
@@ -1,8 +1,9 @@
 package net.minecraft.server;
 
+import java.io.IOException;
 import java.util.UUID;
 
-public class PacketPlayInSpectate implements Packet {
+public class PacketPlayInSpectate implements Packet<PacketListenerPlayIn> {
 
     private UUID a;
 
@@ -12,11 +13,11 @@
         this.a = uuid;
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.g();
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.a(this.a);
     }
 
