--- a/net/minecraft/server/PacketPlayInCloseWindow.java
+++ b/net/minecraft/server/PacketPlayInCloseWindow.java
@@ -1,6 +1,8 @@
 package net.minecraft.server;
 
-public class PacketPlayInCloseWindow implements Packet {
+import java.io.IOException;
+
+public class PacketPlayInCloseWindow implements Packet<PacketListenerPlayIn> {
 
     private int id;
 
@@ -10,11 +12,11 @@
         packetlistenerplayin.a(this);
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.id = packetdataserializer.readByte();
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.writeByte(this.id);
     }
 
