--- a/net/minecraft/server/PacketPlayOutKickDisconnect.java
+++ b/net/minecraft/server/PacketPlayOutKickDisconnect.java
@@ -1,6 +1,8 @@
 package net.minecraft.server;
 
-public class PacketPlayOutKickDisconnect implements Packet {
+import java.io.IOException;
+
+public class PacketPlayOutKickDisconnect implements Packet<PacketListenerPlayOut> {
 
     private IChatBaseComponent a;
 
@@ -10,11 +12,11 @@
         this.a = ichatbasecomponent;
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.d();
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.a(this.a);
     }
 
