--- a/net/minecraft/server/PacketPrepender.java
+++ b/net/minecraft/server/PacketPrepender.java
@@ -4,11 +4,11 @@
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.handler.codec.MessageToByteEncoder;
 
-public class PacketPrepender extends MessageToByteEncoder {
+public class PacketPrepender extends MessageToByteEncoder<ByteBuf> {
 
     public PacketPrepender() {}
 
-    protected void a(ChannelHandlerContext channelhandlercontext, ByteBuf bytebuf, ByteBuf bytebuf1) {
+    protected void a(ChannelHandlerContext channelhandlercontext, ByteBuf bytebuf, ByteBuf bytebuf1) throws Exception {
         int i = bytebuf.readableBytes();
         int j = PacketDataSerializer.a(i);
 
@@ -23,7 +23,7 @@
         }
     }
 
-    protected void encode(ChannelHandlerContext channelhandlercontext, Object object, ByteBuf bytebuf) {
+    protected void encode(ChannelHandlerContext channelhandlercontext, Object object, ByteBuf bytebuf) throws Exception {
         this.a(channelhandlercontext, (ByteBuf) object, bytebuf);
     }
 }
