--- a/net/minecraft/server/PacketPlayOutBlockChange.java
+++ b/net/minecraft/server/PacketPlayOutBlockChange.java
@@ -1,9 +1,11 @@
 package net.minecraft.server;
 
-public class PacketPlayOutBlockChange implements Packet {
+import java.io.IOException;
+
+public class PacketPlayOutBlockChange implements Packet<PacketListenerPlayOut> {
 
     private BlockPosition a;
-    public IBlockData block;
+    private IBlockData block;
 
     public PacketPlayOutBlockChange() {}
 
@@ -12,12 +14,12 @@
         this.block = world.getType(blockposition);
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.c();
         this.block = (IBlockData) Block.d.a(packetdataserializer.e());
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.a(this.a);
         packetdataserializer.b(Block.d.b(this.block));
     }
