--- a/net/minecraft/server/PacketStatusInPing.java
+++ b/net/minecraft/server/PacketStatusInPing.java
@@ -1,16 +1,18 @@
 package net.minecraft.server;
 
-public class PacketStatusInPing implements Packet {
+import java.io.IOException;
+
+public class PacketStatusInPing implements Packet<PacketStatusInListener> {
 
     private long a;
 
     public PacketStatusInPing() {}
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.readLong();
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.writeLong(this.a);
     }
 
