--- a/net/minecraft/server/PacketPlayOutScoreboardObjective.java
+++ b/net/minecraft/server/PacketPlayOutScoreboardObjective.java
@@ -1,10 +1,12 @@
 package net.minecraft.server;
 
-public class PacketPlayOutScoreboardObjective implements Packet {
+import java.io.IOException;
+
+public class PacketPlayOutScoreboardObjective implements Packet<PacketListenerPlayOut> {
 
     private String a;
     private String b;
-    private EnumScoreboardHealthDisplay c;
+    private IScoreboardCriteria.EnumScoreboardHealthDisplay c;
     private int d;
 
     public PacketPlayOutScoreboardObjective() {}
@@ -16,17 +18,17 @@
         this.d = i;
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.c(16);
         this.d = packetdataserializer.readByte();
         if (this.d == 0 || this.d == 2) {
             this.b = packetdataserializer.c(32);
-            this.c = EnumScoreboardHealthDisplay.a(packetdataserializer.c(16));
+            this.c = IScoreboardCriteria.EnumScoreboardHealthDisplay.a(packetdataserializer.c(16));
         }
 
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.a(this.a);
         packetdataserializer.writeByte(this.d);
         if (this.d == 0 || this.d == 2) {
