--- a/net/minecraft/server/PacketPlayInClientCommand.java
+++ b/net/minecraft/server/PacketPlayInClientCommand.java
@@ -1,20 +1,22 @@
 package net.minecraft.server;
 
-public class PacketPlayInClientCommand implements Packet {
+import java.io.IOException;
 
-    private EnumClientCommand a;
+public class PacketPlayInClientCommand implements Packet<PacketListenerPlayIn> {
+
+    private PacketPlayInClientCommand.EnumClientCommand a;
 
     public PacketPlayInClientCommand() {}
 
-    public PacketPlayInClientCommand(EnumClientCommand enumclientcommand) {
-        this.a = enumclientcommand;
+    public PacketPlayInClientCommand(PacketPlayInClientCommand.EnumClientCommand packetplayinclientcommand_enumclientcommand) {
+        this.a = packetplayinclientcommand_enumclientcommand;
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
-        this.a = (EnumClientCommand) packetdataserializer.a(EnumClientCommand.class);
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
+        this.a = (PacketPlayInClientCommand.EnumClientCommand) packetdataserializer.a(PacketPlayInClientCommand.EnumClientCommand.class);
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.a((Enum) this.a);
     }
 
@@ -22,11 +24,18 @@
         packetlistenerplayin.a(this);
     }
 
-    public EnumClientCommand a() {
+    public PacketPlayInClientCommand.EnumClientCommand a() {
         return this.a;
     }
 
     public void a(PacketListener packetlistener) {
         this.a((PacketListenerPlayIn) packetlistener);
     }
+
+    public static enum EnumClientCommand {
+
+        PERFORM_RESPAWN, REQUEST_STATS, OPEN_INVENTORY_ACHIEVEMENT;
+
+        private EnumClientCommand() {}
+    }
 }
