--- a/net/minecraft/server/PacketPlayInResourcePackStatus.java
+++ b/net/minecraft/server/PacketPlayInResourcePackStatus.java
@@ -1,18 +1,20 @@
 package net.minecraft.server;
 
-public class PacketPlayInResourcePackStatus implements Packet {
+import java.io.IOException;
+
+public class PacketPlayInResourcePackStatus implements Packet<PacketListenerPlayIn> {
 
     private String a;
-    private EnumResourcePackStatus b;
+    private PacketPlayInResourcePackStatus.EnumResourcePackStatus b;
 
     public PacketPlayInResourcePackStatus() {}
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.c(40);
-        this.b = (EnumResourcePackStatus) packetdataserializer.a(EnumResourcePackStatus.class);
+        this.b = (PacketPlayInResourcePackStatus.EnumResourcePackStatus) packetdataserializer.a(PacketPlayInResourcePackStatus.EnumResourcePackStatus.class);
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.a(this.a);
         packetdataserializer.a((Enum) this.b);
     }
@@ -24,4 +26,11 @@
     public void a(PacketListener packetlistener) {
         this.a((PacketListenerPlayIn) packetlistener);
     }
+
+    public static enum EnumResourcePackStatus {
+
+        SUCCESSFULLY_LOADED, DECLINED, FAILED_DOWNLOAD, ACCEPTED;
+
+        private EnumResourcePackStatus() {}
+    }
 }
