--- a/net/minecraft/server/PacketPlayOutTransaction.java
+++ b/net/minecraft/server/PacketPlayOutTransaction.java
@@ -1,6 +1,8 @@
 package net.minecraft.server;
 
-public class PacketPlayOutTransaction implements Packet {
+import java.io.IOException;
+
+public class PacketPlayOutTransaction implements Packet<PacketListenerPlayOut> {
 
     private int a;
     private short b;
@@ -18,13 +20,13 @@
         packetlistenerplayout.a(this);
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.readUnsignedByte();
         this.b = packetdataserializer.readShort();
         this.c = packetdataserializer.readBoolean();
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.writeByte(this.a);
         packetdataserializer.writeShort(this.b);
         packetdataserializer.writeBoolean(this.c);
