--- a/net/minecraft/server/PacketSplitter.java
+++ b/net/minecraft/server/PacketSplitter.java
@@ -11,7 +11,7 @@
 
     public PacketSplitter() {}
 
-    protected void decode(ChannelHandlerContext channelhandlercontext, ByteBuf bytebuf, List list) {
+    protected void decode(ChannelHandlerContext channelhandlercontext, ByteBuf bytebuf, List<Object> list) throws Exception {
         bytebuf.markReaderIndex();
         byte[] abyte = new byte[3];
 
