--- a/net/minecraft/server/PacketPlayOutScoreboardDisplayObjective.java
+++ b/net/minecraft/server/PacketPlayOutScoreboardDisplayObjective.java
@@ -1,6 +1,8 @@
 package net.minecraft.server;
 
-public class PacketPlayOutScoreboardDisplayObjective implements Packet {
+import java.io.IOException;
+
+public class PacketPlayOutScoreboardDisplayObjective implements Packet<PacketListenerPlayOut> {
 
     private int a;
     private String b;
@@ -17,12 +19,12 @@
 
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.readByte();
         this.b = packetdataserializer.c(16);
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.writeByte(this.a);
         packetdataserializer.a(this.b);
     }
