--- a/net/minecraft/server/PacketDecompressor.java
+++ b/net/minecraft/server/PacketDecompressor.java
@@ -18,7 +18,7 @@
         this.a = new Inflater();
     }
 
-    protected void decode(ChannelHandlerContext channelhandlercontext, ByteBuf bytebuf, List list) {
+    protected void decode(ChannelHandlerContext channelhandlercontext, ByteBuf bytebuf, List<Object> list) throws Exception {
         if (bytebuf.readableBytes() != 0) {
             PacketDataSerializer packetdataserializer = new PacketDataSerializer(bytebuf);
             int i = packetdataserializer.e();
