--- a/net/minecraft/server/PacketPlayInUpdateSign.java
+++ b/net/minecraft/server/PacketPlayInUpdateSign.java
@@ -1,13 +1,15 @@
 package net.minecraft.server;
 
-public class PacketPlayInUpdateSign implements Packet {
+import java.io.IOException;
+
+public class PacketPlayInUpdateSign implements Packet<PacketListenerPlayIn> {
 
     private BlockPosition a;
     private IChatBaseComponent[] b;
 
     public PacketPlayInUpdateSign() {}
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.c();
         this.b = new IChatBaseComponent[4];
 
@@ -17,7 +19,7 @@
 
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.a(this.a);
 
         for (int i = 0; i < 4; ++i) {
