--- a/net/minecraft/server/PacketPlayOutUpdateTime.java
+++ b/net/minecraft/server/PacketPlayOutUpdateTime.java
@@ -1,6 +1,8 @@
 package net.minecraft.server;
 
-public class PacketPlayOutUpdateTime implements Packet {
+import java.io.IOException;
+
+public class PacketPlayOutUpdateTime implements Packet<PacketListenerPlayOut> {
 
     private long a;
     private long b;
@@ -19,12 +21,12 @@
 
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.readLong();
         this.b = packetdataserializer.readLong();
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.writeLong(this.a);
         packetdataserializer.writeLong(this.b);
     }
