--- a/net/minecraft/server/PacketPlayOutCamera.java
+++ b/net/minecraft/server/PacketPlayOutCamera.java
@@ -1,6 +1,8 @@
 package net.minecraft.server;
 
-public class PacketPlayOutCamera implements Packet {
+import java.io.IOException;
+
+public class PacketPlayOutCamera implements Packet<PacketListenerPlayOut> {
 
     public int a;
 
@@ -10,11 +12,11 @@
         this.a = entity.getId();
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.e();
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.b(this.a);
     }
 
