--- a/net/minecraft/server/PacketPlayInBlockDig.java
+++ b/net/minecraft/server/PacketPlayInBlockDig.java
@@ -1,20 +1,22 @@
 package net.minecraft.server;
 
-public class PacketPlayInBlockDig implements Packet {
+import java.io.IOException;
+
+public class PacketPlayInBlockDig implements Packet<PacketListenerPlayIn> {
 
     private BlockPosition a;
     private EnumDirection b;
-    private EnumPlayerDigType c;
+    private PacketPlayInBlockDig.EnumPlayerDigType c;
 
     public PacketPlayInBlockDig() {}
 
-    public void a(PacketDataSerializer packetdataserializer) {
-        this.c = (EnumPlayerDigType) packetdataserializer.a(EnumPlayerDigType.class);
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
+        this.c = (PacketPlayInBlockDig.EnumPlayerDigType) packetdataserializer.a(PacketPlayInBlockDig.EnumPlayerDigType.class);
         this.a = packetdataserializer.c();
         this.b = EnumDirection.fromType1(packetdataserializer.readUnsignedByte());
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.a((Enum) this.c);
         packetdataserializer.a(this.a);
         packetdataserializer.writeByte(this.b.a());
@@ -32,11 +34,18 @@
         return this.b;
     }
 
-    public EnumPlayerDigType c() {
+    public PacketPlayInBlockDig.EnumPlayerDigType c() {
         return this.c;
     }
 
     public void a(PacketListener packetlistener) {
         this.a((PacketListenerPlayIn) packetlistener);
     }
+
+    public static enum EnumPlayerDigType {
+
+        START_DESTROY_BLOCK, ABORT_DESTROY_BLOCK, STOP_DESTROY_BLOCK, DROP_ALL_ITEMS, DROP_ITEM, RELEASE_USE_ITEM;
+
+        private EnumPlayerDigType() {}
+    }
 }
