--- a/net/minecraft/server/PacketEncrypter.java
+++ b/net/minecraft/server/PacketEncrypter.java
@@ -5,7 +5,7 @@
 import io.netty.handler.codec.MessageToByteEncoder;
 import javax.crypto.Cipher;
 
-public class PacketEncrypter extends MessageToByteEncoder {
+public class PacketEncrypter extends MessageToByteEncoder<ByteBuf> {
 
     private final PacketEncryptionHandler a;
 
@@ -13,11 +13,11 @@
         this.a = new PacketEncryptionHandler(cipher);
     }
 
-    protected void a(ChannelHandlerContext channelhandlercontext, ByteBuf bytebuf, ByteBuf bytebuf1) {
+    protected void a(ChannelHandlerContext channelhandlercontext, ByteBuf bytebuf, ByteBuf bytebuf1) throws Exception {
         this.a.a(bytebuf, bytebuf1);
     }
 
-    protected void encode(ChannelHandlerContext channelhandlercontext, Object object, ByteBuf bytebuf) {
+    protected void encode(ChannelHandlerContext channelhandlercontext, Object object, ByteBuf bytebuf) throws Exception {
         this.a(channelhandlercontext, (ByteBuf) object, bytebuf);
     }
 }
