--- a/net/minecraft/server/PacketPlayOutEntityEquipment.java
+++ b/net/minecraft/server/PacketPlayOutEntityEquipment.java
@@ -1,6 +1,8 @@
 package net.minecraft.server;
 
-public class PacketPlayOutEntityEquipment implements Packet {
+import java.io.IOException;
+
+public class PacketPlayOutEntityEquipment implements Packet<PacketListenerPlayOut> {
 
     private int a;
     private int b;
@@ -14,13 +16,13 @@
         this.c = itemstack == null ? null : itemstack.cloneItemStack();
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.e();
         this.b = packetdataserializer.readShort();
         this.c = packetdataserializer.i();
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.b(this.a);
         packetdataserializer.writeShort(this.b);
         packetdataserializer.a(this.c);
