--- a/net/minecraft/server/PacketPlayInTransaction.java
+++ b/net/minecraft/server/PacketPlayInTransaction.java
@@ -1,6 +1,8 @@
 package net.minecraft.server;
 
-public class PacketPlayInTransaction implements Packet {
+import java.io.IOException;
+
+public class PacketPlayInTransaction implements Packet<PacketListenerPlayIn> {
 
     private int a;
     private short b;
@@ -12,13 +14,13 @@
         packetlistenerplayin.a(this);
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.readByte();
         this.b = packetdataserializer.readShort();
         this.c = packetdataserializer.readByte() != 0;
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.writeByte(this.a);
         packetdataserializer.writeShort(this.b);
         packetdataserializer.writeByte(this.c ? 1 : 0);
