--- a/net/minecraft/server/PacketPlayOutEntityStatus.java
+++ b/net/minecraft/server/PacketPlayOutEntityStatus.java
@@ -1,6 +1,8 @@
 package net.minecraft.server;
 
-public class PacketPlayOutEntityStatus implements Packet {
+import java.io.IOException;
+
+public class PacketPlayOutEntityStatus implements Packet<PacketListenerPlayOut> {
 
     private int a;
     private byte b;
@@ -12,12 +14,12 @@
         this.b = b0;
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.readInt();
         this.b = packetdataserializer.readByte();
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.writeInt(this.a);
         packetdataserializer.writeByte(this.b);
     }
