--- a/net/minecraft/server/PacketLoginOutEncryptionBegin.java
+++ b/net/minecraft/server/PacketLoginOutEncryptionBegin.java
@@ -1,8 +1,9 @@
 package net.minecraft.server;
 
+import java.io.IOException;
 import java.security.PublicKey;
 
-public class PacketLoginOutEncryptionBegin implements Packet {
+public class PacketLoginOutEncryptionBegin implements Packet<PacketLoginOutListener> {
 
     private String a;
     private PublicKey b;
@@ -16,13 +17,13 @@
         this.c = abyte;
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.c(20);
         this.b = MinecraftEncryption.a(packetdataserializer.a());
         this.c = packetdataserializer.a();
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.a(this.a);
         packetdataserializer.a(this.b.getEncoded());
         packetdataserializer.a(this.c);
