--- a/net/minecraft/server/PacketStatusInStart.java
+++ b/net/minecraft/server/PacketStatusInStart.java
@@ -1,12 +1,14 @@
 package net.minecraft.server;
 
-public class PacketStatusInStart implements Packet {
+import java.io.IOException;
+
+public class PacketStatusInStart implements Packet<PacketStatusInListener> {
 
     public PacketStatusInStart() {}
 
-    public void a(PacketDataSerializer packetdataserializer) {}
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {}
 
-    public void b(PacketDataSerializer packetdataserializer) {}
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {}
 
     public void a(PacketStatusInListener packetstatusinlistener) {
         packetstatusinlistener.a(this);
