--- a/net/minecraft/server/PacketPlayOutRespawn.java
+++ b/net/minecraft/server/PacketPlayOutRespawn.java
@@ -1,18 +1,20 @@
 package net.minecraft.server;
 
-public class PacketPlayOutRespawn implements Packet {
+import java.io.IOException;
+
+public class PacketPlayOutRespawn implements Packet<PacketListenerPlayOut> {
 
     private int a;
     private EnumDifficulty b;
-    private EnumGamemode c;
+    private WorldSettings.EnumGamemode c;
     private WorldType d;
 
     public PacketPlayOutRespawn() {}
 
-    public PacketPlayOutRespawn(int i, EnumDifficulty enumdifficulty, WorldType worldtype, EnumGamemode enumgamemode) {
+    public PacketPlayOutRespawn(int i, EnumDifficulty enumdifficulty, WorldType worldtype, WorldSettings.EnumGamemode worldsettings_enumgamemode) {
         this.a = i;
         this.b = enumdifficulty;
-        this.c = enumgamemode;
+        this.c = worldsettings_enumgamemode;
         this.d = worldtype;
     }
 
@@ -20,10 +22,10 @@
         packetlistenerplayout.a(this);
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.readInt();
         this.b = EnumDifficulty.getById(packetdataserializer.readUnsignedByte());
-        this.c = EnumGamemode.getById(packetdataserializer.readUnsignedByte());
+        this.c = WorldSettings.EnumGamemode.getById(packetdataserializer.readUnsignedByte());
         this.d = WorldType.getType(packetdataserializer.c(16));
         if (this.d == null) {
             this.d = WorldType.NORMAL;
@@ -31,7 +33,7 @@
 
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.writeInt(this.a);
         packetdataserializer.writeByte(this.b.a());
         packetdataserializer.writeByte(this.c.getId());
