--- a/net/minecraft/server/PacketStatusOutPong.java
+++ b/net/minecraft/server/PacketStatusOutPong.java
@@ -1,6 +1,8 @@
 package net.minecraft.server;
 
-public class PacketStatusOutPong implements Packet {
+import java.io.IOException;
+
+public class PacketStatusOutPong implements Packet<PacketStatusOutListener> {
 
     private long a;
 
@@ -10,11 +12,11 @@
         this.a = i;
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.readLong();
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.writeLong(this.a);
     }
 
