--- a/net/minecraft/server/PacketPlayOutScoreboardScore.java
+++ b/net/minecraft/server/PacketPlayOutScoreboardScore.java
@@ -1,11 +1,13 @@
 package net.minecraft.server;
 
-public class PacketPlayOutScoreboardScore implements Packet {
+import java.io.IOException;
+
+public class PacketPlayOutScoreboardScore implements Packet<PacketListenerPlayOut> {
 
     private String a = "";
     private String b = "";
     private int c;
-    private EnumScoreboardAction d;
+    private PacketPlayOutScoreboardScore.EnumScoreboardAction d;
 
     public PacketPlayOutScoreboardScore() {}
 
@@ -13,38 +15,38 @@
         this.a = scoreboardscore.getPlayerName();
         this.b = scoreboardscore.getObjective().getName();
         this.c = scoreboardscore.getScore();
-        this.d = EnumScoreboardAction.CHANGE;
+        this.d = PacketPlayOutScoreboardScore.EnumScoreboardAction.CHANGE;
     }
 
     public PacketPlayOutScoreboardScore(String s) {
         this.a = s;
         this.b = "";
         this.c = 0;
-        this.d = EnumScoreboardAction.REMOVE;
+        this.d = PacketPlayOutScoreboardScore.EnumScoreboardAction.REMOVE;
     }
 
     public PacketPlayOutScoreboardScore(String s, ScoreboardObjective scoreboardobjective) {
         this.a = s;
         this.b = scoreboardobjective.getName();
         this.c = 0;
-        this.d = EnumScoreboardAction.REMOVE;
+        this.d = PacketPlayOutScoreboardScore.EnumScoreboardAction.REMOVE;
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.c(40);
-        this.d = (EnumScoreboardAction) packetdataserializer.a(EnumScoreboardAction.class);
+        this.d = (PacketPlayOutScoreboardScore.EnumScoreboardAction) packetdataserializer.a(PacketPlayOutScoreboardScore.EnumScoreboardAction.class);
         this.b = packetdataserializer.c(16);
-        if (this.d != EnumScoreboardAction.REMOVE) {
+        if (this.d != PacketPlayOutScoreboardScore.EnumScoreboardAction.REMOVE) {
             this.c = packetdataserializer.e();
         }
 
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.a(this.a);
         packetdataserializer.a((Enum) this.d);
         packetdataserializer.a(this.b);
-        if (this.d != EnumScoreboardAction.REMOVE) {
+        if (this.d != PacketPlayOutScoreboardScore.EnumScoreboardAction.REMOVE) {
             packetdataserializer.b(this.c);
         }
 
@@ -57,4 +59,11 @@
     public void a(PacketListener packetlistener) {
         this.a((PacketListenerPlayOut) packetlistener);
     }
+
+    public static enum EnumScoreboardAction {
+
+        CHANGE, REMOVE;
+
+        private EnumScoreboardAction() {}
+    }
 }
