--- a/net/minecraft/server/PacketPlayOutWorldEvent.java
+++ b/net/minecraft/server/PacketPlayOutWorldEvent.java
@@ -1,6 +1,8 @@
 package net.minecraft.server;
 
-public class PacketPlayOutWorldEvent implements Packet {
+import java.io.IOException;
+
+public class PacketPlayOutWorldEvent implements Packet<PacketListenerPlayOut> {
 
     private int a;
     private BlockPosition b;
@@ -16,14 +18,14 @@
         this.d = flag;
     }
 
-    public void a(PacketDataSerializer packetdataserializer) {
+    public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.readInt();
         this.b = packetdataserializer.c();
         this.c = packetdataserializer.readInt();
         this.d = packetdataserializer.readBoolean();
     }
 
-    public void b(PacketDataSerializer packetdataserializer) {
+    public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.writeInt(this.a);
         packetdataserializer.a(this.b);
         packetdataserializer.writeInt(this.c);
