--- a/net/minecraft/server/PathfinderGoalPassengerCarrotStick.java
+++ b/net/minecraft/server/PathfinderGoalPassengerCarrotStick.java
@@ -25,7 +25,7 @@
     }
 
     public boolean a() {
-        return this.a.isAlive() && this.a.passenger != null && this.a.passenger instanceof EntityHuman && (this.d || this.a.bV());
+        return this.a.isAlive() && this.a.passenger != null && this.a.passenger instanceof EntityHuman && (this.d || this.a.bW());
     }
 
     public void e() {
@@ -72,7 +72,7 @@
         float f3 = 0.16277136F / (f2 * f2 * f2);
         float f4 = MathHelper.sin(entitycreature.yaw * 3.1415927F / 180.0F);
         float f5 = MathHelper.cos(entitycreature.yaw * 3.1415927F / 180.0F);
-        float f6 = entitycreature.bH() * f3;
+        float f6 = entitycreature.bI() * f3;
         float f7 = Math.max(f1, 1.0F);
 
         f7 = f6 / f7;
@@ -116,8 +116,8 @@
             }
         }
 
-        if (!entityhuman.abilities.canInstantlyBuild && this.c >= this.b * 0.5F && this.a.bb().nextFloat() < 0.006F && !this.d) {
-            ItemStack itemstack = entityhuman.bz();
+        if (!entityhuman.abilities.canInstantlyBuild && this.c >= this.b * 0.5F && this.a.bc().nextFloat() < 0.006F && !this.d) {
+            ItemStack itemstack = entityhuman.bA();
 
             if (itemstack != null && itemstack.getItem() == Items.CARROT_ON_A_STICK) {
                 itemstack.damage(1, entityhuman);
@@ -144,7 +144,7 @@
     public void g() {
         this.d = true;
         this.e = 0;
-        this.f = this.a.bb().nextInt(841) + 140;
+        this.f = this.a.bc().nextInt(841) + 140;
     }
 
     public boolean h() {
