--- a/net/minecraft/server/PathfinderGoalMeleeAttack.java
+++ b/net/minecraft/server/PathfinderGoalMeleeAttack.java
@@ -8,13 +8,13 @@
     double d;
     boolean e;
     PathEntity f;
-    Class g;
+    Class<? extends Entity> g;
     private int h;
     private double i;
     private double j;
     private double k;
 
-    public PathfinderGoalMeleeAttack(EntityCreature entitycreature, Class oclass, double d0, boolean flag) {
+    public PathfinderGoalMeleeAttack(EntityCreature entitycreature, Class<? extends Entity> oclass, double d0, boolean flag) {
         this(entitycreature, d0, flag);
         this.g = oclass;
     }
@@ -45,7 +45,7 @@
     public boolean b() {
         EntityLiving entityliving = this.b.getGoalTarget();
 
-        return entityliving == null ? false : (!entityliving.isAlive() ? false : (!this.e ? !this.b.getNavigation().m() : this.b.d(new BlockPosition(entityliving))));
+        return entityliving == null ? false : (!entityliving.isAlive() ? false : (!this.e ? !this.b.getNavigation().m() : this.b.e(new BlockPosition(entityliving))));
     }
 
     public void c() {
@@ -65,11 +65,11 @@
         double d1 = this.a(entityliving);
 
         --this.h;
-        if ((this.e || this.b.getEntitySenses().a(entityliving)) && this.h <= 0 && (this.i == 0.0D && this.j == 0.0D && this.k == 0.0D || entityliving.e(this.i, this.j, this.k) >= 1.0D || this.b.bb().nextFloat() < 0.05F)) {
+        if ((this.e || this.b.getEntitySenses().a(entityliving)) && this.h <= 0 && (this.i == 0.0D && this.j == 0.0D && this.k == 0.0D || entityliving.e(this.i, this.j, this.k) >= 1.0D || this.b.bc().nextFloat() < 0.05F)) {
             this.i = entityliving.locX;
             this.j = entityliving.getBoundingBox().b;
             this.k = entityliving.locZ;
-            this.h = 4 + this.b.bb().nextInt(7);
+            this.h = 4 + this.b.bc().nextInt(7);
             if (d0 > 1024.0D) {
                 this.h += 10;
             } else if (d0 > 256.0D) {
@@ -84,8 +84,8 @@
         this.c = Math.max(this.c - 1, 0);
         if (d0 <= d1 && this.c <= 0) {
             this.c = 20;
-            if (this.b.bz() != null) {
-                this.b.bv();
+            if (this.b.bA() != null) {
+                this.b.bw();
             }
 
             this.b.r(entityliving);
