--- a/net/minecraft/server/PathfinderGoalInteractVillagers.java
+++ b/net/minecraft/server/PathfinderGoalInteractVillagers.java
@@ -12,7 +12,7 @@
 
     public void c() {
         super.c();
-        if (this.f.cq() && this.b instanceof EntityVillager && ((EntityVillager) this.b).cr()) {
+        if (this.f.cs() && this.b instanceof EntityVillager && ((EntityVillager) this.b).ct()) {
             this.e = 10;
         } else {
             this.e = 0;
@@ -25,7 +25,7 @@
         if (this.e > 0) {
             --this.e;
             if (this.e == 0) {
-                InventorySubcontainer inventorysubcontainer = this.f.co();
+                InventorySubcontainer inventorysubcontainer = this.f.cq();
 
                 for (int i = 0; i < inventorysubcontainer.getSize(); ++i) {
                     ItemStack itemstack = inventorysubcontainer.getItem(i);
@@ -56,7 +56,7 @@
                         double d0 = this.f.locY - 0.30000001192092896D + (double) this.f.getHeadHeight();
                         EntityItem entityitem = new EntityItem(this.f.world, this.f.locX, d0, this.f.locZ, itemstack1);
                         float f = 0.3F;
-                        float f1 = this.f.aI;
+                        float f1 = this.f.aK;
                         float f2 = this.f.pitch;
 
                         entityitem.motX = (double) (-MathHelper.sin(f1 / 180.0F * 3.1415927F) * MathHelper.cos(f2 / 180.0F * 3.1415927F) * f);
