--- a/net/minecraft/server/PathfinderGoalInteract.java
+++ b/net/minecraft/server/PathfinderGoalInteract.java
@@ -2,7 +2,7 @@
 
 public class PathfinderGoalInteract extends PathfinderGoalLookAtPlayer {
 
-    public PathfinderGoalInteract(EntityInsentient entityinsentient, Class oclass, float f, float f1) {
+    public PathfinderGoalInteract(EntityInsentient entityinsentient, Class<? extends Entity> oclass, float f, float f1) {
         super(entityinsentient, oclass, f, f1);
         this.a(3);
     }
