--- a/net/minecraft/server/PathfinderGoalMoveIndoors.java
+++ b/net/minecraft/server/PathfinderGoalMoveIndoors.java
@@ -16,7 +16,7 @@
         BlockPosition blockposition = new BlockPosition(this.a);
 
         if ((!this.a.world.w() || this.a.world.S() && !this.a.world.getBiome(blockposition).e()) && !this.a.world.worldProvider.o()) {
-            if (this.a.bb().nextInt(50) != 0) {
+            if (this.a.bc().nextInt(50) != 0) {
                 return false;
             } else if (this.c != -1 && this.a.e((double) this.c, this.a.locY, (double) this.d) < 4.0D) {
                 return false;
