--- a/net/minecraft/server/PathfinderGoalTakeFlower.java
+++ b/net/minecraft/server/PathfinderGoalTakeFlower.java
@@ -31,7 +31,7 @@
                 while (iterator.hasNext()) {
                     EntityIronGolem entityirongolem = (EntityIronGolem) iterator.next();
 
-                    if (entityirongolem.ck() > 0) {
+                    if (entityirongolem.cm() > 0) {
                         this.b = entityirongolem;
                         break;
                     }
@@ -43,11 +43,11 @@
     }
 
     public boolean b() {
-        return this.b.ck() > 0;
+        return this.b.cm() > 0;
     }
 
     public void c() {
-        this.c = this.a.bb().nextInt(320);
+        this.c = this.a.bc().nextInt(320);
         this.d = false;
         this.b.getNavigation().n();
     }
@@ -59,7 +59,7 @@
 
     public void e() {
         this.a.getControllerLook().a(this.b, 30.0F, 30.0F);
-        if (this.b.ck() == this.c) {
+        if (this.b.cm() == this.c) {
             this.a.getNavigation().a((Entity) this.b, 0.5D);
             this.d = true;
         }
