--- a/net/minecraft/server/PathfinderGoalTempt.java
+++ b/net/minecraft/server/PathfinderGoalTempt.java
@@ -36,7 +36,7 @@
             if (this.h == null) {
                 return false;
             } else {
-                ItemStack itemstack = this.h.bY();
+                ItemStack itemstack = this.h.bZ();
 
                 return itemstack == null ? false : itemstack.getItem() == this.k;
             }
@@ -84,7 +84,7 @@
     }
 
     public void e() {
-        this.a.getControllerLook().a(this.h, 30.0F, (float) this.a.bP());
+        this.a.getControllerLook().a(this.h, 30.0F, (float) this.a.bQ());
         if (this.a.h(this.h) < 6.25D) {
             this.a.getNavigation().n();
         } else {
