--- a/net/minecraft/server/PersistentScoreboard.java
+++ b/net/minecraft/server/PersistentScoreboard.java
@@ -64,19 +64,19 @@
                 scoreboardteam.setCanSeeFriendlyInvisibles(nbttagcompound.getBoolean("SeeFriendlyInvisibles"));
             }
 
-            EnumNameTagVisibility enumnametagvisibility;
+            ScoreboardTeamBase.EnumNameTagVisibility scoreboardteambase_enumnametagvisibility;
 
             if (nbttagcompound.hasKeyOfType("NameTagVisibility", 8)) {
-                enumnametagvisibility = EnumNameTagVisibility.a(nbttagcompound.getString("NameTagVisibility"));
-                if (enumnametagvisibility != null) {
-                    scoreboardteam.a(enumnametagvisibility);
+                scoreboardteambase_enumnametagvisibility = ScoreboardTeamBase.EnumNameTagVisibility.a(nbttagcompound.getString("NameTagVisibility"));
+                if (scoreboardteambase_enumnametagvisibility != null) {
+                    scoreboardteam.setNameTagVisibility(scoreboardteambase_enumnametagvisibility);
                 }
             }
 
             if (nbttagcompound.hasKeyOfType("DeathMessageVisibility", 8)) {
-                enumnametagvisibility = EnumNameTagVisibility.a(nbttagcompound.getString("DeathMessageVisibility"));
-                if (enumnametagvisibility != null) {
-                    scoreboardteam.b(enumnametagvisibility);
+                scoreboardteambase_enumnametagvisibility = ScoreboardTeamBase.EnumNameTagVisibility.a(nbttagcompound.getString("DeathMessageVisibility"));
+                if (scoreboardteambase_enumnametagvisibility != null) {
+                    scoreboardteam.b(scoreboardteambase_enumnametagvisibility);
                 }
             }
 
@@ -113,7 +113,7 @@
                 ScoreboardObjective scoreboardobjective = this.c.registerObjective(nbttagcompound.getString("Name"), iscoreboardcriteria);
 
                 scoreboardobjective.setDisplayName(nbttagcompound.getString("DisplayName"));
-                scoreboardobjective.a(EnumScoreboardHealthDisplay.a(nbttagcompound.getString("RenderType")));
+                scoreboardobjective.a(IScoreboardCriteria.EnumScoreboardHealthDisplay.a(nbttagcompound.getString("RenderType")));
             }
         }
 
@@ -163,7 +163,7 @@
             nbttagcompound.setString("Suffix", scoreboardteam.getSuffix());
             nbttagcompound.setBoolean("AllowFriendlyFire", scoreboardteam.allowFriendlyFire());
             nbttagcompound.setBoolean("SeeFriendlyInvisibles", scoreboardteam.canSeeFriendlyInvisibles());
-            nbttagcompound.setString("NameTagVisibility", scoreboardteam.i().e);
+            nbttagcompound.setString("NameTagVisibility", scoreboardteam.getNameTagVisibility().e);
             nbttagcompound.setString("DeathMessageVisibility", scoreboardteam.j().e);
             NBTTagList nbttaglist1 = new NBTTagList();
             Iterator iterator1 = scoreboardteam.getPlayerNameSet().iterator();
