--- a/net/minecraft/server/PlayerListBox.java
+++ b/net/minecraft/server/PlayerListBox.java
@@ -17,8 +17,8 @@
         if (this.b++ % 20 == 0) {
             Vector vector = new Vector();
 
-            for (int i = 0; i < this.a.getPlayerList().players.size(); ++i) {
-                vector.add(((EntityPlayer) this.a.getPlayerList().players.get(i)).getName());
+            for (int i = 0; i < this.a.getPlayerList().v().size(); ++i) {
+                vector.add(((EntityPlayer) this.a.getPlayerList().v().get(i)).getName());
             }
 
             this.setListData(vector);
