--- a/net/minecraft/server/RemoteStatusReply.java
+++ b/net/minecraft/server/RemoteStatusReply.java
@@ -2,6 +2,7 @@
 
 import java.io.ByteArrayOutputStream;
 import java.io.DataOutputStream;
+import java.io.IOException;
 
 public class RemoteStatusReply {
 
@@ -13,20 +14,20 @@
         this.stream = new DataOutputStream(this.buffer);
     }
 
-    public void a(byte[] abyte) {
+    public void a(byte[] abyte) throws IOException {
         this.stream.write(abyte, 0, abyte.length);
     }
 
-    public void a(String s) {
+    public void a(String s) throws IOException {
         this.stream.writeBytes(s);
         this.stream.write(0);
     }
 
-    public void a(int i) {
+    public void a(int i) throws IOException {
         this.stream.write(i);
     }
 
-    public void a(short short0) {
+    public void a(short short0) throws IOException {
         this.stream.writeShort(Short.reverseBytes(short0));
     }
 
