--- a/net/minecraft/server/ScoreboardTeam.java
+++ b/net/minecraft/server/ScoreboardTeam.java
@@ -8,19 +8,19 @@
 
     private final Scoreboard a;
     private final String b;
-    private final Set c = Sets.newHashSet();
+    private final Set<String> c = Sets.newHashSet();
     private String d;
     private String e = "";
     private String f = "";
     private boolean g = true;
     private boolean h = true;
-    private EnumNameTagVisibility i;
-    private EnumNameTagVisibility j;
+    private ScoreboardTeamBase.EnumNameTagVisibility i;
+    private ScoreboardTeamBase.EnumNameTagVisibility j;
     private EnumChatFormat k;
 
     public ScoreboardTeam(Scoreboard scoreboard, String s) {
-        this.i = EnumNameTagVisibility.ALWAYS;
-        this.j = EnumNameTagVisibility.ALWAYS;
+        this.i = ScoreboardTeamBase.EnumNameTagVisibility.ALWAYS;
+        this.j = ScoreboardTeamBase.EnumNameTagVisibility.ALWAYS;
         this.k = EnumChatFormat.RESET;
         this.a = scoreboard;
         this.b = s;
@@ -44,7 +44,7 @@
         }
     }
 
-    public Collection getPlayerNameSet() {
+    public Collection<String> getPlayerNameSet() {
         return this.c;
     }
 
@@ -96,21 +96,21 @@
         this.a.handleTeamChanged(this);
     }
 
-    public EnumNameTagVisibility i() {
+    public ScoreboardTeamBase.EnumNameTagVisibility getNameTagVisibility() {
         return this.i;
     }
 
-    public EnumNameTagVisibility j() {
+    public ScoreboardTeamBase.EnumNameTagVisibility j() {
         return this.j;
     }
 
-    public void a(EnumNameTagVisibility enumnametagvisibility) {
-        this.i = enumnametagvisibility;
+    public void setNameTagVisibility(ScoreboardTeamBase.EnumNameTagVisibility scoreboardteambase_enumnametagvisibility) {
+        this.i = scoreboardteambase_enumnametagvisibility;
         this.a.handleTeamChanged(this);
     }
 
-    public void b(EnumNameTagVisibility enumnametagvisibility) {
-        this.j = enumnametagvisibility;
+    public void b(ScoreboardTeamBase.EnumNameTagVisibility scoreboardteambase_enumnametagvisibility) {
+        this.j = scoreboardteambase_enumnametagvisibility;
         this.a.handleTeamChanged(this);
     }
 
