--- a/net/minecraft/server/ServerStatisticManager.java
+++ b/net/minecraft/server/ServerStatisticManager.java
@@ -24,7 +24,7 @@
     private static final Logger b = LogManager.getLogger();
     private final MinecraftServer c;
     private final File d;
-    private final Set e = Sets.newHashSet();
+    private final Set<Statistic> e = Sets.newHashSet();
     private int f = -300;
     private boolean g = false;
 
@@ -63,21 +63,21 @@
         this.e.add(statistic);
         if (statistic.d() && j == 0 && i > 0) {
             this.g = true;
-            if (this.c.az()) {
+            if (this.c.aA()) {
                 this.c.getPlayerList().sendMessage(new ChatMessage("chat.type.achievement", new Object[] { entityhuman.getScoreboardDisplayName(), statistic.j()}));
             }
         }
 
         if (statistic.d() && j > 0 && i == 0) {
             this.g = true;
-            if (this.c.az()) {
+            if (this.c.aA()) {
                 this.c.getPlayerList().sendMessage(new ChatMessage("chat.type.achievement.taken", new Object[] { entityhuman.getScoreboardDisplayName(), statistic.j()}));
             }
         }
 
     }
 
-    public Set c() {
+    public Set<Statistic> c() {
         HashSet hashset = Sets.newHashSet(this.e);
 
         this.e.clear();
@@ -85,7 +85,7 @@
         return hashset;
     }
 
-    public Map a(String s) {
+    public Map<Statistic, StatisticWrapper> a(String s) {
         JsonElement jsonelement = (new JsonParser()).parse(s);
 
         if (!jsonelement.isJsonObject()) {
@@ -134,7 +134,7 @@
         }
     }
 
-    public static String a(Map map) {
+    public static String a(Map<Statistic, StatisticWrapper> map) {
         JsonObject jsonobject = new JsonObject();
         Iterator iterator = map.entrySet().iterator();
 
@@ -173,7 +173,7 @@
     }
 
     public void a(EntityPlayer entityplayer) {
-        int i = this.c.ar();
+        int i = this.c.as();
         HashMap hashmap = Maps.newHashMap();
 
         if (this.g || i - this.f > 300) {
