--- a/net/minecraft/server/ShapeDetectorBlock.java
+++ b/net/minecraft/server/ShapeDetectorBlock.java
@@ -6,37 +6,47 @@
 
     private final World a;
     private final BlockPosition b;
-    private IBlockData c;
-    private TileEntity d;
-    private boolean e;
+    private final boolean c;
+    private IBlockData d;
+    private TileEntity e;
+    private boolean f;
 
-    public ShapeDetectorBlock(World world, BlockPosition blockposition) {
+    public ShapeDetectorBlock(World world, BlockPosition blockposition, boolean flag) {
         this.a = world;
         this.b = blockposition;
+        this.c = flag;
     }
 
     public IBlockData a() {
-        if (this.c == null && this.a.isLoaded(this.b)) {
-            this.c = this.a.getType(this.b);
+        if (this.d == null && (this.c || this.a.isLoaded(this.b))) {
+            this.d = this.a.getType(this.b);
         }
 
-        return this.c;
+        return this.d;
     }
 
     public TileEntity b() {
-        if (this.d == null && !this.e) {
-            this.d = this.a.getTileEntity(this.b);
-            this.e = true;
+        if (this.e == null && !this.f) {
+            this.e = this.a.getTileEntity(this.b);
+            this.f = true;
         }
 
-        return this.d;
+        return this.e;
     }
 
     public BlockPosition d() {
         return this.b;
     }
 
-    public static Predicate a(Predicate predicate) {
-        return new ShapeDetectorBlockInnerClass1(predicate);
+    public static Predicate<ShapeDetectorBlock> a(final Predicate<IBlockData> predicate) {
+        return new Predicate() {
+            public boolean a(ShapeDetectorBlock shapedetectorblock) {
+                return shapedetectorblock != null && predicate.apply(shapedetectorblock.a());
+            }
+
+            public boolean apply(Object object) {
+                return this.a((ShapeDetectorBlock) object);
+            }
+        };
     }
 }
