--- a/net/minecraft/server/StatisticManager.java
+++ b/net/minecraft/server/StatisticManager.java
@@ -5,7 +5,7 @@
 
 public class StatisticManager {
 
-    protected final Map a = Maps.newConcurrentMap();
+    protected final Map<Statistic, StatisticWrapper> a = Maps.newConcurrentMap();
 
     public StatisticManager() {}
 
@@ -40,13 +40,13 @@
         return statisticwrapper == null ? 0 : statisticwrapper.a();
     }
 
-    public IJsonStatistic b(Statistic statistic) {
+    public <T extends IJsonStatistic> T b(Statistic statistic) {
         StatisticWrapper statisticwrapper = (StatisticWrapper) this.a.get(statistic);
 
         return statisticwrapper != null ? statisticwrapper.b() : null;
     }
 
-    public IJsonStatistic a(Statistic statistic, IJsonStatistic ijsonstatistic) {
+    public <T extends IJsonStatistic> T a(Statistic statistic, T t0) {
         StatisticWrapper statisticwrapper = (StatisticWrapper) this.a.get(statistic);
 
         if (statisticwrapper == null) {
@@ -54,7 +54,7 @@
             this.a.put(statistic, statisticwrapper);
         }
 
-        statisticwrapper.a(ijsonstatistic);
-        return ijsonstatistic;
+        statisticwrapper.a(t0);
+        return t0;
     }
 }
