--- a/net/minecraft/server/WorldGenTreeAbstract.java
+++ b/net/minecraft/server/WorldGenTreeAbstract.java
@@ -9,7 +9,9 @@
     }
 
     protected boolean a(Block block) {
-        return block.getMaterial() == Material.AIR || block.getMaterial() == Material.LEAVES || block == Blocks.GRASS || block == Blocks.DIRT || block == Blocks.LOG || block == Blocks.LOG2 || block == Blocks.SAPLING || block == Blocks.VINE;
+        Material material = block.getMaterial();
+
+        return material == Material.AIR || material == Material.LEAVES || block == Blocks.GRASS || block == Blocks.DIRT || block == Blocks.LOG || block == Blocks.LOG2 || block == Blocks.SAPLING || block == Blocks.VINE;
     }
 
     public void a(World world, Random random, BlockPosition blockposition) {}
