--- a/net/minecraft/server/WorldGenFlowers.java
+++ b/net/minecraft/server/WorldGenFlowers.java
@@ -7,13 +7,13 @@
     private BlockFlowers a;
     private IBlockData b;
 
-    public WorldGenFlowers(BlockFlowers blockflowers, EnumFlowerVarient enumflowervarient) {
-        this.a(blockflowers, enumflowervarient);
+    public WorldGenFlowers(BlockFlowers blockflowers, BlockFlowers.EnumFlowerVarient blockflowers_enumflowervarient) {
+        this.a(blockflowers, blockflowers_enumflowervarient);
     }
 
-    public void a(BlockFlowers blockflowers, EnumFlowerVarient enumflowervarient) {
+    public void a(BlockFlowers blockflowers, BlockFlowers.EnumFlowerVarient blockflowers_enumflowervarient) {
         this.a = blockflowers;
-        this.b = blockflowers.getBlockData().set(blockflowers.l(), enumflowervarient);
+        this.b = blockflowers.getBlockData().set(blockflowers.n(), blockflowers_enumflowervarient);
     }
 
     public boolean generate(World world, Random random, BlockPosition blockposition) {
