--- a/net/minecraft/server/BiomeTheEnd.java
+++ b/net/minecraft/server/BiomeTheEnd.java
@@ -2,15 +2,15 @@
 
 public class BiomeTheEnd extends BiomeBase {
 
-    public BiomeTheEnd(int i) {
-        super(i);
-        this.at.clear();
-        this.au.clear();
-        this.av.clear();
-        this.aw.clear();
-        this.at.add(new BiomeBase.BiomeMeta(EntityEnderman.class, 10, 4, 4));
-        this.ak = Blocks.DIRT.getBlockData();
-        this.al = Blocks.DIRT.getBlockData();
-        this.as = new BiomeTheEndDecorator();
+    public BiomeTheEnd(BiomeBase.a biomebase_a) {
+        super(biomebase_a);
+        this.u.clear();
+        this.v.clear();
+        this.w.clear();
+        this.x.clear();
+        this.u.add(new BiomeBase.BiomeMeta(EntityEnderman.class, 10, 4, 4));
+        this.r = Blocks.DIRT.getBlockData();
+        this.s = Blocks.DIRT.getBlockData();
+        this.t = new BiomeTheEndDecorator();
     }
 }
