--- a/net/minecraft/server/BlockWorkbench.java
+++ b/net/minecraft/server/BlockWorkbench.java
@@ -7,12 +7,12 @@
         this.a(CreativeModeTab.c);
     }
 
-    public boolean interact(World world, BlockPosition blockposition, IBlockData iblockdata, EntityHuman entityhuman, EnumDirection enumdirection, float f, float f1, float f2) {
+    public boolean interact(World world, BlockPosition blockposition, IBlockData iblockdata, EntityHuman entityhuman, EnumHand enumhand, ItemStack itemstack, EnumDirection enumdirection, float f, float f1, float f2) {
         if (world.isClientSide) {
             return true;
         } else {
             entityhuman.openTileEntity(new BlockWorkbench.TileEntityContainerWorkbench(world, blockposition));
-            entityhuman.b(StatisticList.Z);
+            entityhuman.b(StatisticList.ab);
             return true;
         }
     }
