--- a/net/minecraft/server/BlockSlime.java
+++ b/net/minecraft/server/BlockSlime.java
@@ -22,11 +22,14 @@
             super.a(world, entity);
         } else if (entity.motY < 0.0D) {
             entity.motY = -entity.motY;
+            if (!(entity instanceof EntityLiving)) {
+                entity.motY *= 0.8D;
+            }
         }
 
     }
 
-    public void a(World world, BlockPosition blockposition, Entity entity) {
+    public void stepOn(World world, BlockPosition blockposition, Entity entity) {
         if (Math.abs(entity.motY) < 0.1D && !entity.isSneaking()) {
             double d0 = 0.4D + Math.abs(entity.motY) * 0.2D;
 
@@ -34,6 +37,6 @@
             entity.motZ *= d0;
         }
 
-        super.a(world, blockposition, entity);
+        super.stepOn(world, blockposition, entity);
     }
 }
