--- a/net/minecraft/server/BlockPowered.java
+++ b/net/minecraft/server/BlockPowered.java
@@ -6,11 +6,11 @@
         super(material, materialmapcolor);
     }
 
-    public boolean isPowerSource() {
+    public boolean isPowerSource(IBlockData iblockdata) {
         return true;
     }
 
-    public int a(IBlockAccess iblockaccess, BlockPosition blockposition, IBlockData iblockdata, EnumDirection enumdirection) {
+    public int b(IBlockData iblockdata, IBlockAccess iblockaccess, BlockPosition blockposition, EnumDirection enumdirection) {
         return 15;
     }
 }
