--- a/net/minecraft/server/BlockStationary.java
+++ b/net/minecraft/server/BlockStationary.java
@@ -37,10 +37,14 @@
 
                     for (int j = 0; j < i; ++j) {
                         blockposition1 = blockposition1.a(random.nextInt(3) - 1, 1, random.nextInt(3) - 1);
+                        if (blockposition1.getY() >= 0 && blockposition1.getY() < 256 && !world.isLoaded(blockposition1)) {
+                            return;
+                        }
+
                         Block block = world.getType(blockposition1).getBlock();
 
                         if (block.material == Material.AIR) {
-                            if (this.f(world, blockposition1)) {
+                            if (this.c(world, blockposition1)) {
                                 world.setTypeUpdate(blockposition1, Blocks.FIRE.getBlockData());
                                 return;
                             }
@@ -52,7 +56,11 @@
                     for (int k = 0; k < 3; ++k) {
                         BlockPosition blockposition2 = blockposition.a(random.nextInt(3) - 1, 0, random.nextInt(3) - 1);
 
-                        if (world.isEmpty(blockposition2.up()) && this.m(world, blockposition2)) {
+                        if (blockposition2.getY() >= 0 && blockposition2.getY() < 256 && !world.isLoaded(blockposition2)) {
+                            return;
+                        }
+
+                        if (world.isEmpty(blockposition2.up()) && this.d(world, blockposition2)) {
                             world.setTypeUpdate(blockposition2.up(), Blocks.FIRE.getBlockData());
                         }
                     }
@@ -62,14 +70,14 @@
         }
     }
 
-    protected boolean f(World world, BlockPosition blockposition) {
+    protected boolean c(World world, BlockPosition blockposition) {
         EnumDirection[] aenumdirection = EnumDirection.values();
         int i = aenumdirection.length;
 
         for (int j = 0; j < i; ++j) {
             EnumDirection enumdirection = aenumdirection[j];
 
-            if (this.m(world, blockposition.shift(enumdirection))) {
+            if (this.d(world, blockposition.shift(enumdirection))) {
                 return true;
             }
         }
@@ -77,7 +85,7 @@
         return false;
     }
 
-    private boolean m(World world, BlockPosition blockposition) {
-        return world.getType(blockposition).getBlock().getMaterial().isBurnable();
+    private boolean d(World world, BlockPosition blockposition) {
+        return blockposition.getY() >= 0 && blockposition.getY() < 256 && !world.isLoaded(blockposition) ? false : world.getType(blockposition).getMaterial().isBurnable();
     }
 }
