--- a/net/minecraft/server/BlockWaterLily.java
+++ b/net/minecraft/server/BlockWaterLily.java
@@ -4,34 +4,41 @@
 
 public class BlockWaterLily extends BlockPlant {
 
-    protected BlockWaterLily() {
-        float f = 0.5F;
-        float f1 = 0.015625F;
+    protected static final AxisAlignedBB a = new AxisAlignedBB(0.0625D, 0.0D, 0.0625D, 0.9375D, 0.09375D, 0.9375D);
 
-        this.a(0.5F - f, 0.0F, 0.5F - f, 0.5F + f, f1, 0.5F + f);
+    protected BlockWaterLily() {
         this.a(CreativeModeTab.c);
     }
 
-    public void a(World world, BlockPosition blockposition, IBlockData iblockdata, AxisAlignedBB axisalignedbb, List<AxisAlignedBB> list, Entity entity) {
-        if (entity == null || !(entity instanceof EntityBoat)) {
-            super.a(world, blockposition, iblockdata, axisalignedbb, list, entity);
+    public void a(IBlockData iblockdata, World world, BlockPosition blockposition, AxisAlignedBB axisalignedbb, List<AxisAlignedBB> list, Entity entity) {
+        if (!(entity instanceof EntityBoat)) {
+            a(blockposition, axisalignedbb, list, BlockWaterLily.a);
+        }
+
+    }
+
+    public void a(World world, BlockPosition blockposition, IBlockData iblockdata, Entity entity) {
+        super.a(world, blockposition, iblockdata, entity);
+        if (entity instanceof EntityBoat) {
+            world.setAir(new BlockPosition(blockposition), true);
         }
 
     }
 
-    public AxisAlignedBB a(World world, BlockPosition blockposition, IBlockData iblockdata) {
-        return new AxisAlignedBB((double) blockposition.getX() + this.minX, (double) blockposition.getY() + this.minY, (double) blockposition.getZ() + this.minZ, (double) blockposition.getX() + this.maxX, (double) blockposition.getY() + this.maxY, (double) blockposition.getZ() + this.maxZ);
+    public AxisAlignedBB a(IBlockData iblockdata, IBlockAccess iblockaccess, BlockPosition blockposition) {
+        return BlockWaterLily.a;
     }
 
-    protected boolean c(Block block) {
-        return block == Blocks.WATER;
+    protected boolean i(IBlockData iblockdata) {
+        return iblockdata.getBlock() == Blocks.WATER || iblockdata.getMaterial() == Material.ICE;
     }
 
     public boolean f(World world, BlockPosition blockposition, IBlockData iblockdata) {
         if (blockposition.getY() >= 0 && blockposition.getY() < 256) {
             IBlockData iblockdata1 = world.getType(blockposition.down());
+            Material material = iblockdata1.getMaterial();
 
-            return iblockdata1.getBlock().getMaterial() == Material.WATER && ((Integer) iblockdata1.get(BlockFluids.LEVEL)).intValue() == 0;
+            return material == Material.WATER && ((Integer) iblockdata1.get(BlockFluids.LEVEL)).intValue() == 0 || material == Material.ICE;
         } else {
             return false;
         }
