--- a/net/minecraft/server/BlockEnchantmentTable.java
+++ b/net/minecraft/server/BlockEnchantmentTable.java
@@ -1,31 +1,36 @@
 package net.minecraft.server;
 
-public class BlockEnchantmentTable extends BlockContainer {
+public class BlockEnchantmentTable extends BlockTileEntity {
+
+    protected static final AxisAlignedBB a = new AxisAlignedBB(0.0D, 0.0D, 0.0D, 1.0D, 0.75D, 1.0D);
 
     protected BlockEnchantmentTable() {
         super(Material.STONE, MaterialMapColor.D);
-        this.a(0.0F, 0.0F, 0.0F, 1.0F, 0.75F, 1.0F);
-        this.e(0);
+        this.d(0);
         this.a(CreativeModeTab.c);
     }
 
-    public boolean d() {
+    public AxisAlignedBB a(IBlockData iblockdata, IBlockAccess iblockaccess, BlockPosition blockposition) {
+        return BlockEnchantmentTable.a;
+    }
+
+    public boolean c(IBlockData iblockdata) {
         return false;
     }
 
-    public boolean c() {
+    public boolean b(IBlockData iblockdata) {
         return false;
     }
 
-    public int b() {
-        return 3;
+    public EnumRenderType a(IBlockData iblockdata) {
+        return EnumRenderType.MODEL;
     }
 
     public TileEntity a(World world, int i) {
         return new TileEntityEnchantTable();
     }
 
-    public boolean interact(World world, BlockPosition blockposition, IBlockData iblockdata, EntityHuman entityhuman, EnumDirection enumdirection, float f, float f1, float f2) {
+    public boolean interact(World world, BlockPosition blockposition, IBlockData iblockdata, EntityHuman entityhuman, EnumHand enumhand, ItemStack itemstack, EnumDirection enumdirection, float f, float f1, float f2) {
         if (world.isClientSide) {
             return true;
         } else {
