--- a/net/minecraft/server/BlockSlowSand.java
+++ b/net/minecraft/server/BlockSlowSand.java
@@ -2,15 +2,15 @@
 
 public class BlockSlowSand extends Block {
 
+    protected static final AxisAlignedBB a = new AxisAlignedBB(0.0D, 0.0D, 0.0D, 1.0D, 0.875D, 1.0D);
+
     public BlockSlowSand() {
         super(Material.SAND, MaterialMapColor.B);
         this.a(CreativeModeTab.b);
     }
 
-    public AxisAlignedBB a(World world, BlockPosition blockposition, IBlockData iblockdata) {
-        float f = 0.125F;
-
-        return new AxisAlignedBB((double) blockposition.getX(), (double) blockposition.getY(), (double) blockposition.getZ(), (double) (blockposition.getX() + 1), (double) ((float) (blockposition.getY() + 1) - f), (double) (blockposition.getZ() + 1));
+    public AxisAlignedBB a(IBlockData iblockdata, World world, BlockPosition blockposition) {
+        return BlockSlowSand.a;
     }
 
     public void a(World world, BlockPosition blockposition, IBlockData iblockdata, Entity entity) {
