--- a/net/minecraft/server/BlockIce.java
+++ b/net/minecraft/server/BlockIce.java
@@ -11,25 +11,25 @@
         this.a(CreativeModeTab.b);
     }
 
-    public void a(World world, EntityHuman entityhuman, BlockPosition blockposition, IBlockData iblockdata, TileEntity tileentity) {
-        entityhuman.b(StatisticList.MINE_BLOCK_COUNT[Block.getId(this)]);
+    public void a(World world, EntityHuman entityhuman, BlockPosition blockposition, IBlockData iblockdata, TileEntity tileentity, ItemStack itemstack) {
+        entityhuman.b(StatisticList.a((Block) this));
         entityhuman.applyExhaustion(0.025F);
-        if (this.I() && EnchantmentManager.hasSilkTouchEnchantment(entityhuman)) {
-            ItemStack itemstack = this.i(iblockdata);
+        if (this.o() && EnchantmentManager.getEnchantmentLevel(Enchantments.SILK_TOUCH, itemstack) > 0) {
+            ItemStack itemstack1 = this.u(iblockdata);
 
-            if (itemstack != null) {
-                a(world, blockposition, itemstack);
+            if (itemstack1 != null) {
+                a(world, blockposition, itemstack1);
             }
         } else {
-            if (world.worldProvider.n()) {
+            if (world.worldProvider.l()) {
                 world.setAir(blockposition);
                 return;
             }
 
-            int i = EnchantmentManager.getBonusBlockLootEnchantmentLevel(entityhuman);
+            int i = EnchantmentManager.getEnchantmentLevel(Enchantments.LOOT_BONUS_BLOCKS, itemstack);
 
             this.b(world, blockposition, iblockdata, i);
-            Material material = world.getType(blockposition.down()).getBlock().getMaterial();
+            Material material = world.getType(blockposition.down()).getMaterial();
 
             if (material.isSolid() || material.isLiquid()) {
                 world.setTypeUpdate(blockposition, Blocks.FLOWING_WATER.getBlockData());
@@ -43,17 +43,23 @@
     }
 
     public void b(World world, BlockPosition blockposition, IBlockData iblockdata, Random random) {
-        if (world.b(EnumSkyBlock.BLOCK, blockposition) > 11 - this.p()) {
-            if (world.worldProvider.n()) {
-                world.setAir(blockposition);
-            } else {
-                this.b(world, blockposition, world.getType(blockposition), 0);
-                world.setTypeUpdate(blockposition, Blocks.WATER.getBlockData());
-            }
+        if (world.b(EnumSkyBlock.BLOCK, blockposition) > 11 - this.getBlockData().c()) {
+            this.b(world, blockposition);
         }
+
     }
 
-    public int k() {
-        return 0;
+    protected void b(World world, BlockPosition blockposition) {
+        if (world.worldProvider.l()) {
+            world.setAir(blockposition);
+        } else {
+            this.b(world, blockposition, world.getType(blockposition), 0);
+            world.setTypeUpdate(blockposition, Blocks.WATER.getBlockData());
+            world.e(blockposition, Blocks.WATER);
+        }
+    }
+
+    public EnumPistonReaction h(IBlockData iblockdata) {
+        return EnumPistonReaction.NORMAL;
     }
 }
