--- a/net/minecraft/server/BlockSand.java
+++ b/net/minecraft/server/BlockSand.java
@@ -5,14 +5,14 @@
     public static final BlockStateEnum<BlockSand.EnumSandVariant> VARIANT = BlockStateEnum.of("variant", BlockSand.EnumSandVariant.class);
 
     public BlockSand() {
-        this.j(this.blockStateList.getBlockData().set(BlockSand.VARIANT, BlockSand.EnumSandVariant.SAND));
+        this.w(this.blockStateList.getBlockData().set(BlockSand.VARIANT, BlockSand.EnumSandVariant.SAND));
     }
 
     public int getDropData(IBlockData iblockdata) {
         return ((BlockSand.EnumSandVariant) iblockdata.get(BlockSand.VARIANT)).a();
     }
 
-    public MaterialMapColor g(IBlockData iblockdata) {
+    public MaterialMapColor r(IBlockData iblockdata) {
         return ((BlockSand.EnumSandVariant) iblockdata.get(BlockSand.VARIANT)).c();
     }
 
