--- a/net/minecraft/server/BlockFloorSign.java
+++ b/net/minecraft/server/BlockFloorSign.java
@@ -5,11 +5,11 @@
     public static final BlockStateInteger ROTATION = BlockStateInteger.of("rotation", 0, 15);
 
     public BlockFloorSign() {
-        this.j(this.blockStateList.getBlockData().set(BlockFloorSign.ROTATION, Integer.valueOf(0)));
+        this.w(this.blockStateList.getBlockData().set(BlockFloorSign.ROTATION, Integer.valueOf(0)));
     }
 
     public void doPhysics(World world, BlockPosition blockposition, IBlockData iblockdata, Block block) {
-        if (!world.getType(blockposition.down()).getBlock().getMaterial().isBuildable()) {
+        if (!world.getType(blockposition.down()).getMaterial().isBuildable()) {
             this.b(world, blockposition, iblockdata, 0);
             world.setAir(blockposition);
         }
@@ -25,6 +25,14 @@
         return ((Integer) iblockdata.get(BlockFloorSign.ROTATION)).intValue();
     }
 
+    public IBlockData a(IBlockData iblockdata, EnumBlockRotation enumblockrotation) {
+        return iblockdata.set(BlockFloorSign.ROTATION, Integer.valueOf(enumblockrotation.a(((Integer) iblockdata.get(BlockFloorSign.ROTATION)).intValue(), 16)));
+    }
+
+    public IBlockData a(IBlockData iblockdata, EnumBlockMirror enumblockmirror) {
+        return iblockdata.set(BlockFloorSign.ROTATION, Integer.valueOf(enumblockmirror.a(((Integer) iblockdata.get(BlockFloorSign.ROTATION)).intValue(), 16)));
+    }
+
     protected BlockStateList getStateList() {
         return new BlockStateList(this, new IBlockState[] { BlockFloorSign.ROTATION});
     }
