--- a/net/minecraft/server/BlockRedSandstone.java
+++ b/net/minecraft/server/BlockRedSandstone.java
@@ -6,7 +6,7 @@
 
     public BlockRedSandstone() {
         super(Material.STONE, BlockSand.EnumSandVariant.RED_SAND.c());
-        this.j(this.blockStateList.getBlockData().set(BlockRedSandstone.TYPE, BlockRedSandstone.EnumRedSandstoneVariant.DEFAULT));
+        this.w(this.blockStateList.getBlockData().set(BlockRedSandstone.TYPE, BlockRedSandstone.EnumRedSandstoneVariant.DEFAULT));
         this.a(CreativeModeTab.b);
     }
 
