--- a/net/minecraft/server/BlockFlowing.java
+++ b/net/minecraft/server/BlockFlowing.java
@@ -21,7 +21,7 @@
         int i = ((Integer) iblockdata.get(BlockFlowing.LEVEL)).intValue();
         byte b0 = 1;
 
-        if (this.material == Material.LAVA && !world.worldProvider.n()) {
+        if (this.material == Material.LAVA && !world.worldProvider.l()) {
             b0 = 2;
         }
 
@@ -45,8 +45,8 @@
                 i1 = -1;
             }
 
-            if (this.e(world, blockposition.up()) >= 0) {
-                k = this.e(world, blockposition.up());
+            if (this.c((IBlockAccess) world, blockposition.up()) >= 0) {
+                k = this.c((IBlockAccess) world, blockposition.up());
                 if (k >= 8) {
                     i1 = k;
                 } else {
@@ -57,9 +57,9 @@
             if (this.a >= 2 && this.material == Material.WATER) {
                 IBlockData iblockdata1 = world.getType(blockposition.down());
 
-                if (iblockdata1.getBlock().getMaterial().isBuildable()) {
+                if (iblockdata1.getMaterial().isBuildable()) {
                     i1 = 0;
-                } else if (iblockdata1.getBlock().getMaterial() == this.material && ((Integer) iblockdata1.get(BlockFlowing.LEVEL)).intValue() == 0) {
+                } else if (iblockdata1.getMaterial() == this.material && ((Integer) iblockdata1.get(BlockFlowing.LEVEL)).intValue() == 0) {
                     i1 = 0;
                 }
             }
@@ -88,7 +88,7 @@
         IBlockData iblockdata2 = world.getType(blockposition.down());
 
         if (this.h(world, blockposition.down(), iblockdata2)) {
-            if (this.material == Material.LAVA && world.getType(blockposition.down()).getBlock().getMaterial() == Material.WATER) {
+            if (this.material == Material.LAVA && world.getType(blockposition.down()).getMaterial() == Material.WATER) {
                 world.setTypeUpdate(blockposition.down(), Blocks.STONE.getBlockData());
                 this.fizz(world, blockposition.down());
                 return;
@@ -100,7 +100,7 @@
                 this.flow(world, blockposition.down(), iblockdata2, i + 8);
             }
         } else if (i >= 0 && (i == 0 || this.g(world, blockposition.down(), iblockdata2))) {
-            Set set = this.f(world, blockposition);
+            Set set = this.c(world, blockposition);
 
             k = i + b0;
             if (i >= 8) {
@@ -148,12 +148,12 @@
                 BlockPosition blockposition1 = blockposition.shift(enumdirection1);
                 IBlockData iblockdata = world.getType(blockposition1);
 
-                if (!this.g(world, blockposition1, iblockdata) && (iblockdata.getBlock().getMaterial() != this.material || ((Integer) iblockdata.get(BlockFlowing.LEVEL)).intValue() > 0)) {
+                if (!this.g(world, blockposition1, iblockdata) && (iblockdata.getMaterial() != this.material || ((Integer) iblockdata.get(BlockFlowing.LEVEL)).intValue() > 0)) {
                     if (!this.g(world, blockposition1.down(), iblockdata)) {
                         return i;
                     }
 
-                    if (i < 4) {
+                    if (i < this.b(world)) {
                         int k = this.a(world, blockposition1, i + 1, enumdirection1.opposite());
 
                         if (k < j) {
@@ -167,7 +167,11 @@
         return j;
     }
 
-    private Set<EnumDirection> f(World world, BlockPosition blockposition) {
+    private int b(World world) {
+        return this.material == Material.LAVA && !world.worldProvider.l() ? 2 : 4;
+    }
+
+    private Set<EnumDirection> c(World world, BlockPosition blockposition) {
         int i = 1000;
         EnumSet enumset = EnumSet.noneOf(EnumDirection.class);
         Iterator iterator = EnumDirection.EnumDirectionLimit.HORIZONTAL.iterator();
@@ -177,7 +181,7 @@
             BlockPosition blockposition1 = blockposition.shift(enumdirection);
             IBlockData iblockdata = world.getType(blockposition1);
 
-            if (!this.g(world, blockposition1, iblockdata) && (iblockdata.getBlock().getMaterial() != this.material || ((Integer) iblockdata.get(BlockFlowing.LEVEL)).intValue() > 0)) {
+            if (!this.g(world, blockposition1, iblockdata) && (iblockdata.getMaterial() != this.material || ((Integer) iblockdata.get(BlockFlowing.LEVEL)).intValue() > 0)) {
                 int j;
 
                 if (this.g(world, blockposition1.down(), world.getType(blockposition1.down()))) {
@@ -207,7 +211,7 @@
     }
 
     protected int a(World world, BlockPosition blockposition, int i) {
-        int j = this.e(world, blockposition);
+        int j = this.c((IBlockAccess) world, blockposition);
 
         if (j < 0) {
             return i;
@@ -225,7 +229,7 @@
     }
 
     private boolean h(World world, BlockPosition blockposition, IBlockData iblockdata) {
-        Material material = iblockdata.getBlock().getMaterial();
+        Material material = iblockdata.getMaterial();
 
         return material != this.material && material != Material.LAVA && !this.g(world, blockposition, iblockdata);
     }
