--- a/net/minecraft/server/BlockQuartz.java
+++ b/net/minecraft/server/BlockQuartz.java
@@ -6,7 +6,7 @@
 
     public BlockQuartz() {
         super(Material.STONE);
-        this.j(this.blockStateList.getBlockData().set(BlockQuartz.VARIANT, BlockQuartz.EnumQuartzVariant.DEFAULT));
+        this.w(this.blockStateList.getBlockData().set(BlockQuartz.VARIANT, BlockQuartz.EnumQuartzVariant.DEFAULT));
         this.a(CreativeModeTab.b);
     }
 
@@ -34,13 +34,13 @@
         return blockquartz_enumquartzvariant != BlockQuartz.EnumQuartzVariant.LINES_X && blockquartz_enumquartzvariant != BlockQuartz.EnumQuartzVariant.LINES_Z ? blockquartz_enumquartzvariant.a() : BlockQuartz.EnumQuartzVariant.LINES_Y.a();
     }
 
-    protected ItemStack i(IBlockData iblockdata) {
+    protected ItemStack u(IBlockData iblockdata) {
         BlockQuartz.EnumQuartzVariant blockquartz_enumquartzvariant = (BlockQuartz.EnumQuartzVariant) iblockdata.get(BlockQuartz.VARIANT);
 
-        return blockquartz_enumquartzvariant != BlockQuartz.EnumQuartzVariant.LINES_X && blockquartz_enumquartzvariant != BlockQuartz.EnumQuartzVariant.LINES_Z ? super.i(iblockdata) : new ItemStack(Item.getItemOf(this), 1, BlockQuartz.EnumQuartzVariant.LINES_Y.a());
+        return blockquartz_enumquartzvariant != BlockQuartz.EnumQuartzVariant.LINES_X && blockquartz_enumquartzvariant != BlockQuartz.EnumQuartzVariant.LINES_Z ? super.u(iblockdata) : new ItemStack(Item.getItemOf(this), 1, BlockQuartz.EnumQuartzVariant.LINES_Y.a());
     }
 
-    public MaterialMapColor g(IBlockData iblockdata) {
+    public MaterialMapColor r(IBlockData iblockdata) {
         return MaterialMapColor.p;
     }
 
@@ -52,33 +52,83 @@
         return ((BlockQuartz.EnumQuartzVariant) iblockdata.get(BlockQuartz.VARIANT)).a();
     }
 
+    public IBlockData a(IBlockData iblockdata, EnumBlockRotation enumblockrotation) {
+        switch (BlockQuartz.SyntheticClass_1.c[enumblockrotation.ordinal()]) {
+        case 1:
+        case 2:
+            switch (BlockQuartz.SyntheticClass_1.b[((BlockQuartz.EnumQuartzVariant) iblockdata.get(BlockQuartz.VARIANT)).ordinal()]) {
+            case 1:
+                return iblockdata.set(BlockQuartz.VARIANT, BlockQuartz.EnumQuartzVariant.LINES_Z);
+
+            case 2:
+                return iblockdata.set(BlockQuartz.VARIANT, BlockQuartz.EnumQuartzVariant.LINES_X);
+
+            default:
+                return iblockdata;
+            }
+
+        default:
+            return iblockdata;
+        }
+    }
+
     protected BlockStateList getStateList() {
         return new BlockStateList(this, new IBlockState[] { BlockQuartz.VARIANT});
     }
 
     static class SyntheticClass_1 {
 
-        static final int[] a = new int[EnumDirection.EnumAxis.values().length];
+        static final int[] a;
+        static final int[] b;
+        static final int[] c = new int[EnumBlockRotation.values().length];
 
         static {
             try {
-                BlockQuartz.SyntheticClass_1.a[EnumDirection.EnumAxis.Z.ordinal()] = 1;
+                BlockQuartz.SyntheticClass_1.c[EnumBlockRotation.COUNTERCLOCKWISE_90.ordinal()] = 1;
             } catch (NoSuchFieldError nosuchfielderror) {
                 ;
             }
 
             try {
-                BlockQuartz.SyntheticClass_1.a[EnumDirection.EnumAxis.X.ordinal()] = 2;
+                BlockQuartz.SyntheticClass_1.c[EnumBlockRotation.CLOCKWISE_90.ordinal()] = 2;
             } catch (NoSuchFieldError nosuchfielderror1) {
                 ;
             }
 
+            b = new int[BlockQuartz.EnumQuartzVariant.values().length];
+
             try {
-                BlockQuartz.SyntheticClass_1.a[EnumDirection.EnumAxis.Y.ordinal()] = 3;
+                BlockQuartz.SyntheticClass_1.b[BlockQuartz.EnumQuartzVariant.LINES_X.ordinal()] = 1;
             } catch (NoSuchFieldError nosuchfielderror2) {
                 ;
             }
 
+            try {
+                BlockQuartz.SyntheticClass_1.b[BlockQuartz.EnumQuartzVariant.LINES_Z.ordinal()] = 2;
+            } catch (NoSuchFieldError nosuchfielderror3) {
+                ;
+            }
+
+            a = new int[EnumDirection.EnumAxis.values().length];
+
+            try {
+                BlockQuartz.SyntheticClass_1.a[EnumDirection.EnumAxis.Z.ordinal()] = 1;
+            } catch (NoSuchFieldError nosuchfielderror4) {
+                ;
+            }
+
+            try {
+                BlockQuartz.SyntheticClass_1.a[EnumDirection.EnumAxis.X.ordinal()] = 2;
+            } catch (NoSuchFieldError nosuchfielderror5) {
+                ;
+            }
+
+            try {
+                BlockQuartz.SyntheticClass_1.a[EnumDirection.EnumAxis.Y.ordinal()] = 3;
+            } catch (NoSuchFieldError nosuchfielderror6) {
+                ;
+            }
+
         }
     }
 
