--- a/net/minecraft/server/ChunkGenerator.java
+++ b/net/minecraft/server/ChunkGenerator.java
@@ -0,0 +1,18 @@
+package net.minecraft.server;
+
+import java.util.List;
+
+public interface ChunkGenerator {
+
+    Chunk getOrCreateChunk(int i, int j);
+
+    void recreateStructures(int i, int j);
+
+    boolean a(Chunk chunk, int i, int j);
+
+    List<BiomeBase.BiomeMeta> getMobsFor(EnumCreatureType enumcreaturetype, BlockPosition blockposition);
+
+    BlockPosition findNearestMapFeature(World world, String s, BlockPosition blockposition);
+
+    void recreateStructures(Chunk chunk, int i, int j);
+}
