--- a/net/minecraft/server/CommandSetWorldSpawn.java
+++ b/net/minecraft/server/CommandSetWorldSpawn.java
@@ -1,5 +1,6 @@
 package net.minecraft.server;
 
+import java.util.Collections;
 import java.util.List;
 
 public class CommandSetWorldSpawn extends CommandAbstract {
@@ -18,11 +19,11 @@
         return "commands.setworldspawn.usage";
     }
 
-    public void execute(ICommandListener icommandlistener, String[] astring) throws CommandException {
+    public void execute(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring) throws CommandException {
         BlockPosition blockposition;
 
         if (astring.length == 0) {
-            blockposition = b(icommandlistener).getChunkCoordinates();
+            blockposition = a(icommandlistener).getChunkCoordinates();
         } else {
             if (astring.length != 3 || icommandlistener.getWorld() == null) {
                 throw new ExceptionUsage("commands.setworldspawn.usage", new Object[0]);
@@ -31,12 +32,12 @@
             blockposition = a(icommandlistener, astring, 0, true);
         }
 
-        icommandlistener.getWorld().B(blockposition);
-        MinecraftServer.getServer().getPlayerList().sendAll(new PacketPlayOutSpawnPosition(blockposition));
-        a(icommandlistener, this, "commands.setworldspawn.success", new Object[] { Integer.valueOf(blockposition.getX()), Integer.valueOf(blockposition.getY()), Integer.valueOf(blockposition.getZ())});
+        icommandlistener.getWorld().A(blockposition);
+        minecraftserver.getPlayerList().sendAll(new PacketPlayOutSpawnPosition(blockposition));
+        a(icommandlistener, (ICommand) this, "commands.setworldspawn.success", new Object[] { Integer.valueOf(blockposition.getX()), Integer.valueOf(blockposition.getY()), Integer.valueOf(blockposition.getZ())});
     }
 
-    public List<String> tabComplete(ICommandListener icommandlistener, String[] astring, BlockPosition blockposition) {
-        return astring.length > 0 && astring.length <= 3 ? a(astring, 0, blockposition) : null;
+    public List<String> tabComplete(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring, BlockPosition blockposition) {
+        return astring.length > 0 && astring.length <= 3 ? a(astring, 0, blockposition) : Collections.emptyList();
     }
 }
