--- a/net/minecraft/server/CommandXp.java
+++ b/net/minecraft/server/CommandXp.java
@@ -1,5 +1,6 @@
 package net.minecraft.server;
 
+import java.util.Collections;
 import java.util.List;
 
 public class CommandXp extends CommandAbstract {
@@ -18,7 +19,7 @@
         return "commands.xp.usage";
     }
 
-    public void execute(ICommandListener icommandlistener, String[] astring) throws CommandException {
+    public void execute(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring) throws CommandException {
         if (astring.length <= 0) {
             throw new ExceptionUsage("commands.xp.usage", new Object[0]);
         } else {
@@ -36,16 +37,16 @@
                 i *= -1;
             }
 
-            EntityPlayer entityplayer = astring.length > 1 ? a(icommandlistener, astring[1]) : b(icommandlistener);
+            EntityPlayer entityplayer = astring.length > 1 ? a(minecraftserver, icommandlistener, astring[1]) : a(icommandlistener);
 
             if (flag) {
                 icommandlistener.a(CommandObjectiveExecutor.EnumCommandResult.QUERY_RESULT, entityplayer.expLevel);
                 if (flag1) {
                     entityplayer.levelDown(-i);
-                    a(icommandlistener, this, "commands.xp.success.negative.levels", new Object[] { Integer.valueOf(i), entityplayer.getName()});
+                    a(icommandlistener, (ICommand) this, "commands.xp.success.negative.levels", new Object[] { Integer.valueOf(i), entityplayer.getName()});
                 } else {
                     entityplayer.levelDown(i);
-                    a(icommandlistener, this, "commands.xp.success.levels", new Object[] { Integer.valueOf(i), entityplayer.getName()});
+                    a(icommandlistener, (ICommand) this, "commands.xp.success.levels", new Object[] { Integer.valueOf(i), entityplayer.getName()});
                 }
             } else {
                 icommandlistener.a(CommandObjectiveExecutor.EnumCommandResult.QUERY_RESULT, entityplayer.expTotal);
@@ -54,18 +55,14 @@
                 }
 
                 entityplayer.giveExp(i);
-                a(icommandlistener, this, "commands.xp.success", new Object[] { Integer.valueOf(i), entityplayer.getName()});
+                a(icommandlistener, (ICommand) this, "commands.xp.success", new Object[] { Integer.valueOf(i), entityplayer.getName()});
             }
 
         }
     }
 
-    public List<String> tabComplete(ICommandListener icommandlistener, String[] astring, BlockPosition blockposition) {
-        return astring.length == 2 ? a(astring, this.d()) : null;
-    }
-
-    protected String[] d() {
-        return MinecraftServer.getServer().getPlayers();
+    public List<String> tabComplete(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring, BlockPosition blockposition) {
+        return astring.length == 2 ? a(astring, minecraftserver.getPlayers()) : Collections.emptyList();
     }
 
     public boolean isListStart(String[] astring, int i) {
