--- a/net/minecraft/server/CommandList.java
+++ b/net/minecraft/server/CommandList.java
@@ -16,11 +16,11 @@
         return "commands.players.usage";
     }
 
-    public void execute(ICommandListener icommandlistener, String[] astring) throws CommandException {
-        int i = MinecraftServer.getServer().I();
+    public void execute(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring) throws CommandException {
+        int i = minecraftserver.H();
 
-        icommandlistener.sendMessage(new ChatMessage("commands.players.list", new Object[] { Integer.valueOf(i), Integer.valueOf(MinecraftServer.getServer().J())}));
-        icommandlistener.sendMessage(new ChatComponentText(MinecraftServer.getServer().getPlayerList().b(astring.length > 0 && "uuids".equalsIgnoreCase(astring[0]))));
+        icommandlistener.sendMessage(new ChatMessage("commands.players.list", new Object[] { Integer.valueOf(i), Integer.valueOf(minecraftserver.I())}));
+        icommandlistener.sendMessage(new ChatComponentText(minecraftserver.getPlayerList().b(astring.length > 0 && "uuids".equalsIgnoreCase(astring[0]))));
         icommandlistener.a(CommandObjectiveExecutor.EnumCommandResult.QUERY_RESULT, i);
     }
 }
