--- a/net/minecraft/server/CommandTestFor.java
+++ b/net/minecraft/server/CommandTestFor.java
@@ -1,5 +1,6 @@
 package net.minecraft.server;
 
+import java.util.Collections;
 import java.util.List;
 
 public class CommandTestFor extends CommandAbstract {
@@ -18,11 +19,11 @@
         return "commands.testfor.usage";
     }
 
-    public void execute(ICommandListener icommandlistener, String[] astring) throws CommandException {
+    public void execute(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring) throws CommandException {
         if (astring.length < 1) {
             throw new ExceptionUsage("commands.testfor.usage", new Object[0]);
         } else {
-            Entity entity = b(icommandlistener, astring[0]);
+            Entity entity = b(minecraftserver, icommandlistener, astring[0]);
             NBTTagCompound nbttagcompound = null;
 
             if (astring.length >= 2) {
@@ -34,15 +35,14 @@
             }
 
             if (nbttagcompound != null) {
-                NBTTagCompound nbttagcompound1 = new NBTTagCompound();
+                NBTTagCompound nbttagcompound1 = a(entity);
 
-                entity.e(nbttagcompound1);
                 if (!GameProfileSerializer.a(nbttagcompound, nbttagcompound1, true)) {
                     throw new CommandException("commands.testfor.failure", new Object[] { entity.getName()});
                 }
             }
 
-            a(icommandlistener, this, "commands.testfor.success", new Object[] { entity.getName()});
+            a(icommandlistener, (ICommand) this, "commands.testfor.success", new Object[] { entity.getName()});
         }
     }
 
@@ -50,7 +50,7 @@
         return i == 0;
     }
 
-    public List<String> tabComplete(ICommandListener icommandlistener, String[] astring, BlockPosition blockposition) {
-        return astring.length == 1 ? a(astring, MinecraftServer.getServer().getPlayers()) : null;
+    public List<String> tabComplete(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring, BlockPosition blockposition) {
+        return astring.length == 1 ? a(astring, minecraftserver.getPlayers()) : Collections.emptyList();
     }
 }
