--- a/net/minecraft/server/CommandWeather.java
+++ b/net/minecraft/server/CommandWeather.java
@@ -1,5 +1,6 @@
 package net.minecraft.server;
 
+import java.util.Collections;
 import java.util.List;
 import java.util.Random;
 
@@ -19,7 +20,7 @@
         return "commands.weather.usage";
     }
 
-    public void execute(ICommandListener icommandlistener, String[] astring) throws CommandException {
+    public void execute(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring) throws CommandException {
         if (astring.length >= 1 && astring.length <= 2) {
             int i = (300 + (new Random()).nextInt(600)) * 20;
 
@@ -27,7 +28,7 @@
                 i = a(astring[1], 1, 1000000) * 20;
             }
 
-            WorldServer worldserver = MinecraftServer.getServer().worldServer[0];
+            WorldServer worldserver = minecraftserver.worldServer[0];
             WorldData worlddata = worldserver.getWorldData();
 
             if ("clear".equalsIgnoreCase(astring[0])) {
@@ -36,14 +37,14 @@
                 worlddata.setThunderDuration(0);
                 worlddata.setStorm(false);
                 worlddata.setThundering(false);
-                a(icommandlistener, this, "commands.weather.clear", new Object[0]);
+                a(icommandlistener, (ICommand) this, "commands.weather.clear", new Object[0]);
             } else if ("rain".equalsIgnoreCase(astring[0])) {
                 worlddata.i(0);
                 worlddata.setWeatherDuration(i);
                 worlddata.setThunderDuration(i);
                 worlddata.setStorm(true);
                 worlddata.setThundering(false);
-                a(icommandlistener, this, "commands.weather.rain", new Object[0]);
+                a(icommandlistener, (ICommand) this, "commands.weather.rain", new Object[0]);
             } else {
                 if (!"thunder".equalsIgnoreCase(astring[0])) {
                     throw new ExceptionUsage("commands.weather.usage", new Object[0]);
@@ -54,7 +55,7 @@
                 worlddata.setThunderDuration(i);
                 worlddata.setStorm(true);
                 worlddata.setThundering(true);
-                a(icommandlistener, this, "commands.weather.thunder", new Object[0]);
+                a(icommandlistener, (ICommand) this, "commands.weather.thunder", new Object[0]);
             }
 
         } else {
@@ -62,7 +63,7 @@
         }
     }
 
-    public List<String> tabComplete(ICommandListener icommandlistener, String[] astring, BlockPosition blockposition) {
-        return astring.length == 1 ? a(astring, new String[] { "clear", "rain", "thunder"}) : null;
+    public List<String> tabComplete(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring, BlockPosition blockposition) {
+        return astring.length == 1 ? a(astring, new String[] { "clear", "rain", "thunder"}) : Collections.emptyList();
     }
 }
