--- a/net/minecraft/server/CommandStop.java
+++ b/net/minecraft/server/CommandStop.java
@@ -12,11 +12,11 @@
         return "commands.stop.usage";
     }
 
-    public void execute(ICommandListener icommandlistener, String[] astring) throws CommandException {
-        if (MinecraftServer.getServer().worldServer != null) {
-            a(icommandlistener, this, "commands.stop.start", new Object[0]);
+    public void execute(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring) throws CommandException {
+        if (minecraftserver.worldServer != null) {
+            a(icommandlistener, (ICommand) this, "commands.stop.start", new Object[0]);
         }
 
-        MinecraftServer.getServer().safeShutdown();
+        minecraftserver.safeShutdown();
     }
 }
