--- a/net/minecraft/server/Convertable.java
+++ b/net/minecraft/server/Convertable.java
@@ -1,14 +1,14 @@
 package net.minecraft.server;
 
+import java.io.File;
+
 public interface Convertable {
 
     IDataManager a(String s, boolean flag);
 
-    void d();
-
-    boolean e(String s);
-
     boolean isConvertable(String s);
 
     boolean convert(String s, IProgressUpdate iprogressupdate);
+
+    File b(String s, String s1);
 }
