--- a/net/minecraft/server/CreativeModeTab.java
+++ b/net/minecraft/server/CreativeModeTab.java
@@ -20,7 +20,7 @@
     public static final CreativeModeTab i = (new CreativeModeTab(7, "tools") {
     }).a(new EnchantmentSlotType[] { EnchantmentSlotType.DIGGER, EnchantmentSlotType.FISHING_ROD, EnchantmentSlotType.BREAKABLE});
     public static final CreativeModeTab j = (new CreativeModeTab(8, "combat") {
-    }).a(new EnchantmentSlotType[] { EnchantmentSlotType.ARMOR, EnchantmentSlotType.ARMOR_FEET, EnchantmentSlotType.ARMOR_HEAD, EnchantmentSlotType.ARMOR_LEGS, EnchantmentSlotType.ARMOR_TORSO, EnchantmentSlotType.BOW, EnchantmentSlotType.WEAPON});
+    }).a(new EnchantmentSlotType[] { EnchantmentSlotType.ARMOR, EnchantmentSlotType.ARMOR_FEET, EnchantmentSlotType.ARMOR_HEAD, EnchantmentSlotType.ARMOR_LEGS, EnchantmentSlotType.ARMOR_CHEST, EnchantmentSlotType.BOW, EnchantmentSlotType.WEAPON});
     public static final CreativeModeTab k = new CreativeModeTab(9, "brewing") {
     };
     public static final CreativeModeTab l = new CreativeModeTab(10, "materials") {
