--- a/net/minecraft/server/DamageSource.java
+++ b/net/minecraft/server/DamageSource.java
@@ -11,27 +11,33 @@
     public static DamageSource STARVE = (new DamageSource("starve")).setIgnoreArmor().m();
     public static DamageSource CACTUS = new DamageSource("cactus");
     public static DamageSource FALL = (new DamageSource("fall")).setIgnoreArmor();
+    public static DamageSource j = (new DamageSource("flyIntoWall")).setIgnoreArmor();
     public static DamageSource OUT_OF_WORLD = (new DamageSource("outOfWorld")).setIgnoreArmor().l();
     public static DamageSource GENERIC = (new DamageSource("generic")).setIgnoreArmor();
     public static DamageSource MAGIC = (new DamageSource("magic")).setIgnoreArmor().setMagic();
     public static DamageSource WITHER = (new DamageSource("wither")).setIgnoreArmor();
     public static DamageSource ANVIL = new DamageSource("anvil");
     public static DamageSource FALLING_BLOCK = new DamageSource("fallingBlock");
-    private boolean q;
-    private boolean r;
+    public static DamageSource DRAGON_BREATH = (new DamageSource("dragonBreath")).setIgnoreArmor();
     private boolean s;
-    private float t = 0.3F;
+    private boolean t;
     private boolean u;
-    private boolean v;
+    private float v = 0.3F;
     private boolean w;
     private boolean x;
     private boolean y;
+    private boolean z;
+    private boolean A;
     public String translationIndex;
 
     public static DamageSource mobAttack(EntityLiving entityliving) {
         return new EntityDamageSource("mob", entityliving);
     }
 
+    public static DamageSource a(Entity entity, EntityLiving entityliving) {
+        return new EntityDamageSourceIndirect("mob", entity, entityliving);
+    }
+
     public static DamageSource playerAttack(EntityHuman entityhuman) {
         return new EntityDamageSource("player", entityhuman);
     }
@@ -53,45 +59,49 @@
     }
 
     public static DamageSource a(Entity entity) {
-        return (new EntityDamageSource("thorns", entity)).v().setMagic();
+        return (new EntityDamageSource("thorns", entity)).w().setMagic();
     }
 
     public static DamageSource explosion(Explosion explosion) {
         return explosion != null && explosion.getSource() != null ? (new EntityDamageSource("explosion.player", explosion.getSource())).q().d() : (new DamageSource("explosion")).q().d();
     }
 
+    public static DamageSource b(EntityLiving entityliving) {
+        return entityliving != null ? (new EntityDamageSource("explosion.player", entityliving)).q().d() : (new DamageSource("explosion")).q().d();
+    }
+
     public boolean a() {
-        return this.v;
+        return this.x;
     }
 
     public DamageSource b() {
-        this.v = true;
+        this.x = true;
         return this;
     }
 
     public boolean isExplosion() {
-        return this.y;
+        return this.A;
     }
 
     public DamageSource d() {
-        this.y = true;
+        this.A = true;
         return this;
     }
 
     public boolean ignoresArmor() {
-        return this.q;
+        return this.s;
     }
 
     public float getExhaustionCost() {
-        return this.t;
+        return this.v;
     }
 
     public boolean ignoresInvulnerability() {
-        return this.r;
+        return this.t;
     }
 
     public boolean isStarvation() {
-        return this.s;
+        return this.u;
     }
 
     protected DamageSource(String s) {
@@ -107,29 +117,29 @@
     }
 
     protected DamageSource setIgnoreArmor() {
-        this.q = true;
-        this.t = 0.0F;
+        this.s = true;
+        this.v = 0.0F;
         return this;
     }
 
     protected DamageSource l() {
-        this.r = true;
+        this.t = true;
         return this;
     }
 
     protected DamageSource m() {
-        this.s = true;
-        this.t = 0.0F;
+        this.u = true;
+        this.v = 0.0F;
         return this;
     }
 
     protected DamageSource setExplosion() {
-        this.u = true;
+        this.w = true;
         return this;
     }
 
     public IChatBaseComponent getLocalizedDeathMessage(EntityLiving entityliving) {
-        EntityLiving entityliving1 = entityliving.bt();
+        EntityLiving entityliving1 = entityliving.bV();
         String s = "death.attack." + this.translationIndex;
         String s1 = s + ".player";
 
@@ -137,7 +147,7 @@
     }
 
     public boolean o() {
-        return this.u;
+        return this.w;
     }
 
     public String p() {
@@ -145,20 +155,20 @@
     }
 
     public DamageSource q() {
-        this.w = true;
+        this.y = true;
         return this;
     }
 
     public boolean r() {
-        return this.w;
+        return this.y;
     }
 
     public boolean isMagic() {
-        return this.x;
+        return this.z;
     }
 
     public DamageSource setMagic() {
-        this.x = true;
+        this.z = true;
         return this;
     }
 
@@ -167,4 +177,8 @@
 
         return entity instanceof EntityHuman && ((EntityHuman) entity).abilities.canInstantlyBuild;
     }
+
+    public Vec3D v() {
+        return null;
+    }
 }
