--- a/net/minecraft/server/DispenseBehaviorProjectile.java
+++ b/net/minecraft/server/DispenseBehaviorProjectile.java
@@ -7,8 +7,8 @@
     public ItemStack b(ISourceBlock isourceblock, ItemStack itemstack) {
         World world = isourceblock.getWorld();
         IPosition iposition = BlockDispenser.a(isourceblock);
-        EnumDirection enumdirection = BlockDispenser.b(isourceblock.f());
-        IProjectile iprojectile = this.a(world, iposition);
+        EnumDirection enumdirection = BlockDispenser.e(isourceblock.f());
+        IProjectile iprojectile = this.a(world, iposition, itemstack);
 
         iprojectile.shoot((double) enumdirection.getAdjacentX(), (double) ((float) enumdirection.getAdjacentY() + 0.1F), (double) enumdirection.getAdjacentZ(), this.getPower(), this.a());
         world.addEntity((Entity) iprojectile);
@@ -20,7 +20,7 @@
         isourceblock.getWorld().triggerEffect(1002, isourceblock.getBlockPosition(), 0);
     }
 
-    protected abstract IProjectile a(World world, IPosition iposition);
+    protected abstract IProjectile a(World world, IPosition iposition, ItemStack itemstack);
 
     protected float a() {
         return 6.0F;
