--- a/net/minecraft/server/EnchantmentDigging.java
+++ b/net/minecraft/server/EnchantmentDigging.java
@@ -2,8 +2,8 @@
 
 public class EnchantmentDigging extends Enchantment {
 
-    protected EnchantmentDigging(int i, MinecraftKey minecraftkey, int j) {
-        super(i, minecraftkey, j, EnchantmentSlotType.DIGGER);
+    protected EnchantmentDigging(Enchantment.Rarity enchantment_rarity, EnumItemSlot... aenumitemslot) {
+        super(enchantment_rarity, EnchantmentSlotType.DIGGER, aenumitemslot);
         this.c("digging");
     }
 
