--- a/net/minecraft/server/EnchantmentSlotType.java
+++ b/net/minecraft/server/EnchantmentSlotType.java
@@ -2,7 +2,7 @@
 
 public enum EnchantmentSlotType {
 
-    ALL, ARMOR, ARMOR_FEET, ARMOR_LEGS, ARMOR_TORSO, ARMOR_HEAD, WEAPON, DIGGER, FISHING_ROD, BREAKABLE, BOW;
+    ALL, ARMOR, ARMOR_FEET, ARMOR_LEGS, ARMOR_CHEST, ARMOR_HEAD, WEAPON, DIGGER, FISHING_ROD, BREAKABLE, BOW;
 
     private EnchantmentSlotType() {}
 
@@ -17,7 +17,7 @@
             } else {
                 ItemArmor itemarmor = (ItemArmor) item;
 
-                return itemarmor.b == 0 ? this == EnchantmentSlotType.ARMOR_HEAD : (itemarmor.b == 2 ? this == EnchantmentSlotType.ARMOR_LEGS : (itemarmor.b == 1 ? this == EnchantmentSlotType.ARMOR_TORSO : (itemarmor.b == 3 ? this == EnchantmentSlotType.ARMOR_FEET : false)));
+                return itemarmor.c == EnumItemSlot.HEAD ? this == EnchantmentSlotType.ARMOR_HEAD : (itemarmor.c == EnumItemSlot.LEGS ? this == EnchantmentSlotType.ARMOR_LEGS : (itemarmor.c == EnumItemSlot.CHEST ? this == EnchantmentSlotType.ARMOR_CHEST : (itemarmor.c == EnumItemSlot.FEET ? this == EnchantmentSlotType.ARMOR_FEET : false)));
             }
         } else {
             return item instanceof ItemSword ? this == EnchantmentSlotType.WEAPON : (item instanceof ItemTool ? this == EnchantmentSlotType.DIGGER : (item instanceof ItemBow ? this == EnchantmentSlotType.BOW : (item instanceof ItemFishingRod ? this == EnchantmentSlotType.FISHING_ROD : false)));
