--- a/net/minecraft/server/EnchantmentKnockback.java
+++ b/net/minecraft/server/EnchantmentKnockback.java
@@ -2,8 +2,8 @@
 
 public class EnchantmentKnockback extends Enchantment {
 
-    protected EnchantmentKnockback(int i, MinecraftKey minecraftkey, int j) {
-        super(i, minecraftkey, j, EnchantmentSlotType.WEAPON);
+    protected EnchantmentKnockback(Enchantment.Rarity enchantment_rarity, EnumItemSlot... aenumitemslot) {
+        super(enchantment_rarity, EnchantmentSlotType.WEAPON, aenumitemslot);
         this.c("knockback");
     }
 
