--- a/net/minecraft/server/EntityPositionTypes.java
+++ b/net/minecraft/server/EntityPositionTypes.java
@@ -1,13 +1,13 @@
 package net.minecraft.server;
 
 import com.google.common.collect.Maps;
-import java.util.HashMap;
+import java.util.Map;
 
 public class EntityPositionTypes {
 
-    private static final HashMap<Class, EntityInsentient.EnumEntityPositionType> a = Maps.newHashMap();
+    private static final Map<Class<?>, EntityInsentient.EnumEntityPositionType> a = Maps.newHashMap();
 
-    public static EntityInsentient.EnumEntityPositionType a(Class oclass) {
+    public static EntityInsentient.EnumEntityPositionType a(Class<?> oclass) {
         return (EntityInsentient.EnumEntityPositionType) EntityPositionTypes.a.get(oclass);
     }
 
