--- a/net/minecraft/server/EntityEnderSignal.java
+++ b/net/minecraft/server/EntityEnderSignal.java
@@ -13,7 +13,7 @@
         this.setSize(0.25F, 0.25F);
     }
 
-    protected void h() {}
+    protected void i() {}
 
     public EntityEnderSignal(World world, double d0, double d1, double d2) {
         super(world);
@@ -44,19 +44,19 @@
         this.e = this.random.nextInt(5) > 0;
     }
 
-    public void t_() {
-        this.P = this.locX;
-        this.Q = this.locY;
-        this.R = this.locZ;
-        super.t_();
+    public void m() {
+        this.M = this.locX;
+        this.N = this.locY;
+        this.O = this.locZ;
+        super.m();
         this.locX += this.motX;
         this.locY += this.motY;
         this.locZ += this.motZ;
         float f = MathHelper.sqrt(this.motX * this.motX + this.motZ * this.motZ);
 
-        this.yaw = (float) (MathHelper.b(this.motX, this.motZ) * 180.0D / 3.1415927410125732D);
+        this.yaw = (float) (MathHelper.b(this.motX, this.motZ) * 57.2957763671875D);
 
-        for (this.pitch = (float) (MathHelper.b(this.motY, (double) f) * 180.0D / 3.1415927410125732D); this.pitch - this.lastPitch < -180.0F; this.lastPitch -= 360.0F) {
+        for (this.pitch = (float) (MathHelper.b(this.motY, (double) f) * 57.2957763671875D); this.pitch - this.lastPitch < -180.0F; this.lastPitch -= 360.0F) {
             ;
         }
 
@@ -97,7 +97,7 @@
 
         float f3 = 0.25F;
 
-        if (this.V()) {
+        if (this.isInWater()) {
             for (int i = 0; i < 4; ++i) {
                 this.world.addParticle(EnumParticle.WATER_BUBBLE, this.locX - this.motX * (double) f3, this.locY - this.motY * (double) f3, this.locZ - this.motZ * (double) f3, this.motX, this.motY, this.motZ, new int[0]);
             }
@@ -124,11 +124,11 @@
 
     public void a(NBTTagCompound nbttagcompound) {}
 
-    public float c(float f) {
+    public float e(float f) {
         return 1.0F;
     }
 
-    public boolean aD() {
+    public boolean aT() {
         return false;
     }
 }
