--- a/net/minecraft/server/FoodMetaData.java
+++ b/net/minecraft/server/FoodMetaData.java
@@ -32,11 +32,22 @@
             }
         }
 
-        if (entityhuman.world.getGameRules().getBoolean("naturalRegeneration") && this.foodLevel >= 18 && entityhuman.cm()) {
+        boolean flag = entityhuman.world.getGameRules().getBoolean("naturalRegeneration");
+
+        if (flag && this.saturationLevel > 0.0F && entityhuman.cT() && this.foodLevel >= 20) {
+            ++this.foodTickTimer;
+            if (this.foodTickTimer >= 10) {
+                float f = Math.min(this.saturationLevel, 4.0F);
+
+                entityhuman.heal(f / 4.0F);
+                this.a(f);
+                this.foodTickTimer = 0;
+            }
+        } else if (flag && this.foodLevel >= 18 && entityhuman.cT()) {
             ++this.foodTickTimer;
             if (this.foodTickTimer >= 80) {
                 entityhuman.heal(1.0F);
-                this.a(3.0F);
+                this.a(4.0F);
                 this.foodTickTimer = 0;
             }
         } else if (this.foodLevel <= 0) {
