--- a/net/minecraft/server/GenLayerMushroomIsland.java
+++ b/net/minecraft/server/GenLayerMushroomIsland.java
@@ -25,7 +25,7 @@
 
                 this.a((long) (j2 + i), (long) (i2 + j));
                 if (k3 == 0 && k2 == 0 && l2 == 0 && i3 == 0 && j3 == 0 && this.a(100) == 0) {
-                    aint1[j2 + i2 * k] = BiomeBase.MUSHROOM_ISLAND.id;
+                    aint1[j2 + i2 * k] = BiomeBase.a(Biomes.p);
                 } else {
                     aint1[j2 + i2 * k] = k3;
                 }
