--- a/net/minecraft/server/GenLayerRiver.java
+++ b/net/minecraft/server/GenLayerRiver.java
@@ -26,7 +26,7 @@
                 if (k3 == k2 && k3 == i3 && k3 == l2 && k3 == j3) {
                     aint1[j2 + i2 * k] = -1;
                 } else {
-                    aint1[j2 + i2 * k] = BiomeBase.RIVER.id;
+                    aint1[j2 + i2 * k] = BiomeBase.a(Biomes.i);
                 }
             }
         }
