--- a/net/minecraft/server/IMerchant.java
+++ b/net/minecraft/server/IMerchant.java
@@ -2,15 +2,15 @@
 
 public interface IMerchant {
 
-    void a_(EntityHuman entityhuman);
+    void setTradingPlayer(EntityHuman entityhuman);
 
-    EntityHuman v_();
+    EntityHuman t_();
 
     MerchantRecipeList getOffers(EntityHuman entityhuman);
 
     void a(MerchantRecipe merchantrecipe);
 
-    void a_(ItemStack itemstack);
+    void a(ItemStack itemstack);
 
     IChatBaseComponent getScoreboardDisplayName();
 }
