--- a/net/minecraft/server/ItemSeedFood.java
+++ b/net/minecraft/server/ItemSeedFood.java
@@ -11,17 +11,13 @@
         this.c = block1;
     }
 
-    public boolean interactWith(ItemStack itemstack, EntityHuman entityhuman, World world, BlockPosition blockposition, EnumDirection enumdirection, float f, float f1, float f2) {
-        if (enumdirection != EnumDirection.UP) {
-            return false;
-        } else if (!entityhuman.a(blockposition.shift(enumdirection), enumdirection, itemstack)) {
-            return false;
-        } else if (world.getType(blockposition).getBlock() == this.c && world.isEmpty(blockposition.up())) {
-            world.setTypeUpdate(blockposition.up(), this.b.getBlockData());
+    public EnumInteractionResult a(ItemStack itemstack, EntityHuman entityhuman, World world, BlockPosition blockposition, EnumHand enumhand, EnumDirection enumdirection, float f, float f1, float f2) {
+        if (enumdirection == EnumDirection.UP && entityhuman.a(blockposition.shift(enumdirection), enumdirection, itemstack) && world.getType(blockposition).getBlock() == this.c && world.isEmpty(blockposition.up())) {
+            world.setTypeAndData(blockposition.up(), this.b.getBlockData(), 11);
             --itemstack.count;
-            return true;
+            return EnumInteractionResult.SUCCESS;
         } else {
-            return false;
+            return EnumInteractionResult.FAIL;
         }
     }
 }
