--- a/net/minecraft/server/ItemMapEmpty.java
+++ b/net/minecraft/server/ItemMapEmpty.java
@@ -6,7 +6,7 @@
         this.a(CreativeModeTab.f);
     }
 
-    public ItemStack a(ItemStack itemstack, World world, EntityHuman entityhuman) {
+    public InteractionResultWrapper<ItemStack> a(ItemStack itemstack, World world, EntityHuman entityhuman, EnumHand enumhand) {
         ItemStack itemstack1 = new ItemStack(Items.FILLED_MAP, 1, world.b("map"));
         String s = "map_" + itemstack1.getData();
         WorldMap worldmap = new WorldMap(s);
@@ -14,18 +14,19 @@
         world.a(s, (PersistentBase) worldmap);
         worldmap.scale = 0;
         worldmap.a(entityhuman.locX, entityhuman.locZ, worldmap.scale);
-        worldmap.map = (byte) world.worldProvider.getDimension();
+        worldmap.map = (byte) world.worldProvider.getDimensionManager().getDimensionID();
+        worldmap.track = true;
         worldmap.c();
         --itemstack.count;
         if (itemstack.count <= 0) {
-            return itemstack1;
+            return new InteractionResultWrapper(EnumInteractionResult.SUCCESS, itemstack1);
         } else {
             if (!entityhuman.inventory.pickup(itemstack1.cloneItemStack())) {
                 entityhuman.drop(itemstack1, false);
             }
 
-            entityhuman.b(StatisticList.USE_ITEM_COUNT[Item.getId(this)]);
-            return itemstack;
+            entityhuman.b(StatisticList.b((Item) this));
+            return new InteractionResultWrapper(EnumInteractionResult.SUCCESS, itemstack);
         }
     }
 }
