--- a/net/minecraft/server/ItemFireworks.java
+++ b/net/minecraft/server/ItemFireworks.java
@@ -4,7 +4,7 @@
 
     public ItemFireworks() {}
 
-    public boolean interactWith(ItemStack itemstack, EntityHuman entityhuman, World world, BlockPosition blockposition, EnumDirection enumdirection, float f, float f1, float f2) {
+    public EnumInteractionResult a(ItemStack itemstack, EntityHuman entityhuman, World world, BlockPosition blockposition, EnumHand enumhand, EnumDirection enumdirection, float f, float f1, float f2) {
         if (!world.isClientSide) {
             EntityFireworks entityfireworks = new EntityFireworks(world, (double) ((float) blockposition.getX() + f), (double) ((float) blockposition.getY() + f1), (double) ((float) blockposition.getZ() + f2), itemstack);
 
@@ -12,10 +12,8 @@
             if (!entityhuman.abilities.canInstantlyBuild) {
                 --itemstack.count;
             }
-
-            return true;
-        } else {
-            return false;
         }
+
+        return EnumInteractionResult.SUCCESS;
     }
 }
