--- a/net/minecraft/server/ItemSword.java
+++ b/net/minecraft/server/ItemSword.java
@@ -4,7 +4,7 @@
 
 public class ItemSword extends Item {
 
-    private float a;
+    private final float a;
     private final Item.EnumToolMaterial b;
 
     public ItemSword(Item.EnumToolMaterial item_enumtoolmaterial) {
@@ -12,18 +12,20 @@
         this.maxStackSize = 1;
         this.setMaxDurability(item_enumtoolmaterial.a());
         this.a(CreativeModeTab.j);
-        this.a = 4.0F + item_enumtoolmaterial.c();
+        this.a = 3.0F + item_enumtoolmaterial.c();
     }
 
     public float g() {
         return this.b.c();
     }
 
-    public float getDestroySpeed(ItemStack itemstack, Block block) {
+    public float getDestroySpeed(ItemStack itemstack, IBlockData iblockdata) {
+        Block block = iblockdata.getBlock();
+
         if (block == Blocks.WEB) {
             return 15.0F;
         } else {
-            Material material = block.getMaterial();
+            Material material = iblockdata.getMaterial();
 
             return material != Material.PLANT && material != Material.REPLACEABLE_PLANT && material != Material.CORAL && material != Material.LEAVES && material != Material.PUMPKIN ? 1.0F : 1.5F;
         }
@@ -34,32 +36,19 @@
         return true;
     }
 
-    public boolean a(ItemStack itemstack, World world, Block block, BlockPosition blockposition, EntityLiving entityliving) {
-        if ((double) block.g(world, blockposition) != 0.0D) {
+    public boolean a(ItemStack itemstack, World world, IBlockData iblockdata, BlockPosition blockposition, EntityLiving entityliving) {
+        if ((double) iblockdata.b(world, blockposition) != 0.0D) {
             itemstack.damage(2, entityliving);
         }
 
         return true;
     }
 
-    public EnumAnimation e(ItemStack itemstack) {
-        return EnumAnimation.BLOCK;
-    }
-
-    public int d(ItemStack itemstack) {
-        return 72000;
-    }
-
-    public ItemStack a(ItemStack itemstack, World world, EntityHuman entityhuman) {
-        entityhuman.a(itemstack, this.d(itemstack));
-        return itemstack;
+    public boolean canDestroySpecialBlock(IBlockData iblockdata) {
+        return iblockdata.getBlock() == Blocks.WEB;
     }
 
-    public boolean canDestroySpecialBlock(Block block) {
-        return block == Blocks.WEB;
-    }
-
-    public int b() {
+    public int c() {
         return this.b.e();
     }
 
@@ -71,10 +60,14 @@
         return this.b.f() == itemstack1.getItem() ? true : super.a(itemstack, itemstack1);
     }
 
-    public Multimap<String, AttributeModifier> i() {
-        Multimap multimap = super.i();
+    public Multimap<String, AttributeModifier> a(EnumItemSlot enumitemslot) {
+        Multimap multimap = super.a(enumitemslot);
+
+        if (enumitemslot == EnumItemSlot.MAINHAND) {
+            multimap.put(GenericAttributes.ATTACK_DAMAGE.getName(), new AttributeModifier(ItemSword.g, "Weapon modifier", (double) this.a, 0));
+            multimap.put(GenericAttributes.f.getName(), new AttributeModifier(ItemSword.h, "Weapon modifier", -2.4000000953674316D, 0));
+        }
 
-        multimap.put(GenericAttributes.ATTACK_DAMAGE.getName(), new AttributeModifier(ItemSword.f, "Weapon modifier", (double) this.a, 0));
         return multimap;
     }
 }
