--- a/net/minecraft/server/ItemSoup.java
+++ b/net/minecraft/server/ItemSoup.java
@@ -4,11 +4,11 @@
 
     public ItemSoup(int i) {
         super(i, false);
-        this.c(1);
+        this.d(1);
     }
 
-    public ItemStack b(ItemStack itemstack, World world, EntityHuman entityhuman) {
-        super.b(itemstack, world, entityhuman);
+    public ItemStack a(ItemStack itemstack, World world, EntityLiving entityliving) {
+        super.a(itemstack, world, entityliving);
         return new ItemStack(Items.BOWL);
     }
 }
