--- a/net/minecraft/server/ItemGoldenApple.java
+++ b/net/minecraft/server/ItemGoldenApple.java
@@ -11,19 +11,18 @@
         return itemstack.getData() == 0 ? EnumItemRarity.RARE : EnumItemRarity.EPIC;
     }
 
-    protected void c(ItemStack itemstack, World world, EntityHuman entityhuman) {
+    protected void a(ItemStack itemstack, World world, EntityHuman entityhuman) {
         if (!world.isClientSide) {
-            entityhuman.addEffect(new MobEffect(MobEffectList.ABSORBTION.id, 2400, 0));
-        }
-
-        if (itemstack.getData() > 0) {
-            if (!world.isClientSide) {
-                entityhuman.addEffect(new MobEffect(MobEffectList.REGENERATION.id, 600, 4));
-                entityhuman.addEffect(new MobEffect(MobEffectList.RESISTANCE.id, 6000, 0));
-                entityhuman.addEffect(new MobEffect(MobEffectList.FIRE_RESISTANCE.id, 6000, 0));
+            if (itemstack.getData() > 0) {
+                entityhuman.b((Statistic) AchievementList.M);
+                entityhuman.addEffect(new MobEffect(MobEffects.REGENERATION, 400, 1));
+                entityhuman.addEffect(new MobEffect(MobEffects.RESISTANCE, 6000, 0));
+                entityhuman.addEffect(new MobEffect(MobEffects.FIRE_RESISTANCE, 6000, 0));
+                entityhuman.addEffect(new MobEffect(MobEffects.ABSORBTION, 2400, 3));
+            } else {
+                entityhuman.addEffect(new MobEffect(MobEffects.REGENERATION, 100, 1));
+                entityhuman.addEffect(new MobEffect(MobEffects.ABSORBTION, 2400, 0));
             }
-        } else {
-            super.c(itemstack, world, entityhuman);
         }
 
     }
