--- a/net/minecraft/server/ItemLeash.java
+++ b/net/minecraft/server/ItemLeash.java
@@ -9,18 +9,17 @@
         this.a(CreativeModeTab.i);
     }
 
-    public boolean interactWith(ItemStack itemstack, EntityHuman entityhuman, World world, BlockPosition blockposition, EnumDirection enumdirection, float f, float f1, float f2) {
+    public EnumInteractionResult a(ItemStack itemstack, EntityHuman entityhuman, World world, BlockPosition blockposition, EnumHand enumhand, EnumDirection enumdirection, float f, float f1, float f2) {
         Block block = world.getType(blockposition).getBlock();
 
-        if (block instanceof BlockFence) {
-            if (world.isClientSide) {
-                return true;
-            } else {
+        if (!(block instanceof BlockFence)) {
+            return EnumInteractionResult.PASS;
+        } else {
+            if (!world.isClientSide) {
                 a(entityhuman, world, blockposition);
-                return true;
             }
-        } else {
-            return false;
+
+            return EnumInteractionResult.SUCCESS;
         }
     }
 
@@ -37,7 +36,7 @@
         while (iterator.hasNext()) {
             EntityInsentient entityinsentient = (EntityInsentient) iterator.next();
 
-            if (entityinsentient.cc() && entityinsentient.getLeashHolder() == entityhuman) {
+            if (entityinsentient.isLeashed() && entityinsentient.getLeashHolder() == entityhuman) {
                 if (entityleash == null) {
                     entityleash = EntityLeash.a(world, blockposition);
                 }
