--- a/net/minecraft/server/ItemRecord.java
+++ b/net/minecraft/server/ItemRecord.java
@@ -5,31 +5,32 @@
 
 public class ItemRecord extends Item {
 
-    private static final Map<String, ItemRecord> b = Maps.newHashMap();
-    public final String a;
-
-    protected ItemRecord(String s) {
-        this.a = s;
+    private static final Map<SoundEffect, ItemRecord> a = Maps.newHashMap();
+    private final SoundEffect b;
+    private final String c;
+
+    protected ItemRecord(String s, SoundEffect soundeffect) {
+        this.c = "item.record." + s + ".desc";
+        this.b = soundeffect;
         this.maxStackSize = 1;
         this.a(CreativeModeTab.f);
-        ItemRecord.b.put("records." + s, this);
+        ItemRecord.a.put(this.b, this);
     }
 
-    public boolean interactWith(ItemStack itemstack, EntityHuman entityhuman, World world, BlockPosition blockposition, EnumDirection enumdirection, float f, float f1, float f2) {
+    public EnumInteractionResult a(ItemStack itemstack, EntityHuman entityhuman, World world, BlockPosition blockposition, EnumHand enumhand, EnumDirection enumdirection, float f, float f1, float f2) {
         IBlockData iblockdata = world.getType(blockposition);
 
         if (iblockdata.getBlock() == Blocks.JUKEBOX && !((Boolean) iblockdata.get(BlockJukeBox.HAS_RECORD)).booleanValue()) {
-            if (world.isClientSide) {
-                return true;
-            } else {
+            if (!world.isClientSide) {
                 ((BlockJukeBox) Blocks.JUKEBOX).a(world, blockposition, iblockdata, itemstack);
-                world.a((EntityHuman) null, 1005, blockposition, Item.getId(this));
+                world.a((EntityHuman) null, 1010, blockposition, Item.getId(this));
                 --itemstack.count;
-                entityhuman.b(StatisticList.X);
-                return true;
+                entityhuman.b(StatisticList.Z);
             }
+
+            return EnumInteractionResult.SUCCESS;
         } else {
-            return false;
+            return EnumInteractionResult.PASS;
         }
     }
 
