--- a/net/minecraft/server/IWorldAccess.java
+++ b/net/minecraft/server/IWorldAccess.java
@@ -2,15 +2,15 @@
 
 public interface IWorldAccess {
 
-    void a(BlockPosition blockposition);
+    void a(World world, BlockPosition blockposition, IBlockData iblockdata, IBlockData iblockdata1, int i);
 
-    void b(BlockPosition blockposition);
+    void a(BlockPosition blockposition);
 
     void a(int i, int j, int k, int l, int i1, int j1);
 
-    void a(String s, double d0, double d1, double d2, float f, float f1);
+    void a(EntityHuman entityhuman, SoundEffect soundeffect, SoundCategory soundcategory, double d0, double d1, double d2, float f, float f1);
 
-    void a(EntityHuman entityhuman, String s, double d0, double d1, double d2, float f, float f1);
+    void a(SoundEffect soundeffect, BlockPosition blockposition);
 
     void a(int i, boolean flag, double d0, double d1, double d2, double d3, double d4, double d5, int... aint);
 
@@ -18,8 +18,6 @@
 
     void b(Entity entity);
 
-    void a(String s, BlockPosition blockposition);
-
     void a(int i, BlockPosition blockposition, int j);
 
     void a(EntityHuman entityhuman, int i, BlockPosition blockposition, int j);
