--- a/net/minecraft/server/MobEffectAbsorption.java
+++ b/net/minecraft/server/MobEffectAbsorption.java
@@ -2,8 +2,8 @@
 
 public class MobEffectAbsorption extends MobEffectList {
 
-    protected MobEffectAbsorption(int i, MinecraftKey minecraftkey, boolean flag, int j) {
-        super(i, minecraftkey, flag, j);
+    protected MobEffectAbsorption(boolean flag, int i) {
+        super(flag, i);
     }
 
     public void a(EntityLiving entityliving, AttributeMapBase attributemapbase, int i) {
