--- a/net/minecraft/server/OpList.java
+++ b/net/minecraft/server/OpList.java
@@ -28,6 +28,12 @@
         return astring;
     }
 
+    public int a(GameProfile gameprofile) {
+        OpListEntry oplistentry = (OpListEntry) this.get(gameprofile);
+
+        return oplistentry != null ? oplistentry.a() : 0;
+    }
+
     public boolean b(GameProfile gameprofile) {
         OpListEntry oplistentry = (OpListEntry) this.get(gameprofile);
 
