--- a/net/minecraft/server/PacketPlayOutBed.java
+++ b/net/minecraft/server/PacketPlayOutBed.java
@@ -15,8 +15,8 @@
     }
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
-        this.a = packetdataserializer.e();
-        this.b = packetdataserializer.c();
+        this.a = packetdataserializer.g();
+        this.b = packetdataserializer.e();
     }
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
@@ -27,8 +27,4 @@
     public void a(PacketListenerPlayOut packetlistenerplayout) {
         packetlistenerplayout.a(this);
     }
-
-    public void a(PacketListener packetlistener) {
-        this.a((PacketListenerPlayOut) packetlistener);
-    }
 }
