--- a/net/minecraft/server/PacketPlayOutSpawnEntityPainting.java
+++ b/net/minecraft/server/PacketPlayOutSpawnEntityPainting.java
@@ -1,42 +1,43 @@
 package net.minecraft.server;
 
 import java.io.IOException;
+import java.util.UUID;
 
 public class PacketPlayOutSpawnEntityPainting implements Packet<PacketListenerPlayOut> {
 
     private int a;
-    private BlockPosition b;
-    private EnumDirection c;
-    private String d;
+    private UUID b;
+    private BlockPosition c;
+    private EnumDirection d;
+    private String e;
 
     public PacketPlayOutSpawnEntityPainting() {}
 
     public PacketPlayOutSpawnEntityPainting(EntityPainting entitypainting) {
         this.a = entitypainting.getId();
-        this.b = entitypainting.getBlockPosition();
-        this.c = entitypainting.direction;
-        this.d = entitypainting.art.B;
+        this.b = entitypainting.getUniqueID();
+        this.c = entitypainting.getBlockPosition();
+        this.d = entitypainting.direction;
+        this.e = entitypainting.art.B;
     }
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
-        this.a = packetdataserializer.e();
-        this.d = packetdataserializer.c(EntityPainting.EnumArt.A);
-        this.b = packetdataserializer.c();
-        this.c = EnumDirection.fromType2(packetdataserializer.readUnsignedByte());
+        this.a = packetdataserializer.g();
+        this.b = packetdataserializer.i();
+        this.e = packetdataserializer.c(EntityPainting.EnumArt.A);
+        this.c = packetdataserializer.e();
+        this.d = EnumDirection.fromType2(packetdataserializer.readUnsignedByte());
     }
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.b(this.a);
-        packetdataserializer.a(this.d);
         packetdataserializer.a(this.b);
-        packetdataserializer.writeByte(this.c.b());
+        packetdataserializer.a(this.e);
+        packetdataserializer.a(this.c);
+        packetdataserializer.writeByte(this.d.get2DRotationValue());
     }
 
     public void a(PacketListenerPlayOut packetlistenerplayout) {
         packetlistenerplayout.a(this);
     }
-
-    public void a(PacketListener packetlistener) {
-        this.a((PacketListenerPlayOut) packetlistener);
-    }
 }
