--- a/net/minecraft/server/PacketPlayOutServerDifficulty.java
+++ b/net/minecraft/server/PacketPlayOutServerDifficulty.java
@@ -25,8 +25,4 @@
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.writeByte(this.a.a());
     }
-
-    public void a(PacketListener packetlistener) {
-        this.a((PacketListenerPlayOut) packetlistener);
-    }
 }
