--- a/net/minecraft/server/PacketPlayOutEntityVelocity.java
+++ b/net/minecraft/server/PacketPlayOutEntityVelocity.java
@@ -49,7 +49,7 @@
     }
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
-        this.a = packetdataserializer.e();
+        this.a = packetdataserializer.g();
         this.b = packetdataserializer.readShort();
         this.c = packetdataserializer.readShort();
         this.d = packetdataserializer.readShort();
@@ -65,8 +65,4 @@
     public void a(PacketListenerPlayOut packetlistenerplayout) {
         packetlistenerplayout.a(this);
     }
-
-    public void a(PacketListener packetlistener) {
-        this.a((PacketListenerPlayOut) packetlistener);
-    }
 }
