--- a/net/minecraft/server/PacketPlayOutEntityEquipment.java
+++ b/net/minecraft/server/PacketPlayOutEntityEquipment.java
@@ -5,34 +5,30 @@
 public class PacketPlayOutEntityEquipment implements Packet<PacketListenerPlayOut> {
 
     private int a;
-    private int b;
+    private EnumItemSlot b;
     private ItemStack c;
 
     public PacketPlayOutEntityEquipment() {}
 
-    public PacketPlayOutEntityEquipment(int i, int j, ItemStack itemstack) {
+    public PacketPlayOutEntityEquipment(int i, EnumItemSlot enumitemslot, ItemStack itemstack) {
         this.a = i;
-        this.b = j;
+        this.b = enumitemslot;
         this.c = itemstack == null ? null : itemstack.cloneItemStack();
     }
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
-        this.a = packetdataserializer.e();
-        this.b = packetdataserializer.readShort();
-        this.c = packetdataserializer.i();
+        this.a = packetdataserializer.g();
+        this.b = (EnumItemSlot) packetdataserializer.a(EnumItemSlot.class);
+        this.c = packetdataserializer.k();
     }
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.b(this.a);
-        packetdataserializer.writeShort(this.b);
+        packetdataserializer.a((Enum) this.b);
         packetdataserializer.a(this.c);
     }
 
     public void a(PacketListenerPlayOut packetlistenerplayout) {
         packetlistenerplayout.a(this);
     }
-
-    public void a(PacketListener packetlistener) {
-        this.a((PacketListenerPlayOut) packetlistener);
-    }
 }
