--- a/net/minecraft/server/PacketLoginInEncryptionBegin.java
+++ b/net/minecraft/server/PacketLoginInEncryptionBegin.java
@@ -32,8 +32,4 @@
     public byte[] b(PrivateKey privatekey) {
         return privatekey == null ? this.b : MinecraftEncryption.b(privatekey, this.b);
     }
-
-    public void a(PacketListener packetlistener) {
-        this.a((PacketLoginInListener) packetlistener);
-    }
 }
