--- a/net/minecraft/server/PacketPlayInSetCreativeSlot.java
+++ b/net/minecraft/server/PacketPlayInSetCreativeSlot.java
@@ -15,7 +15,7 @@
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.slot = packetdataserializer.readShort();
-        this.b = packetdataserializer.i();
+        this.b = packetdataserializer.k();
     }
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
@@ -30,8 +30,4 @@
     public ItemStack getItemStack() {
         return this.b;
     }
-
-    public void a(PacketListener packetlistener) {
-        this.a((PacketListenerPlayIn) packetlistener);
-    }
 }
