--- a/net/minecraft/server/PacketPlayOutOpenSignEditor.java
+++ b/net/minecraft/server/PacketPlayOutOpenSignEditor.java
@@ -17,14 +17,10 @@
     }
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
-        this.a = packetdataserializer.c();
+        this.a = packetdataserializer.e();
     }
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.a(this.a);
     }
-
-    public void a(PacketListener packetlistener) {
-        this.a((PacketListenerPlayOut) packetlistener);
-    }
 }
