--- a/net/minecraft/server/PacketPlayInSettings.java
+++ b/net/minecraft/server/PacketPlayInSettings.java
@@ -9,23 +9,26 @@
     private EntityHuman.EnumChatVisibility c;
     private boolean d;
     private int e;
+    private EnumMainHand f;
 
     public PacketPlayInSettings() {}
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.c(7);
         this.b = packetdataserializer.readByte();
-        this.c = EntityHuman.EnumChatVisibility.a(packetdataserializer.readByte());
+        this.c = (EntityHuman.EnumChatVisibility) packetdataserializer.a(EntityHuman.EnumChatVisibility.class);
         this.d = packetdataserializer.readBoolean();
         this.e = packetdataserializer.readUnsignedByte();
+        this.f = (EnumMainHand) packetdataserializer.a(EnumMainHand.class);
     }
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.a(this.a);
         packetdataserializer.writeByte(this.b);
-        packetdataserializer.writeByte(this.c.a());
+        packetdataserializer.a((Enum) this.c);
         packetdataserializer.writeBoolean(this.d);
         packetdataserializer.writeByte(this.e);
+        packetdataserializer.a((Enum) this.f);
     }
 
     public void a(PacketListenerPlayIn packetlistenerplayin) {
@@ -48,7 +51,7 @@
         return this.e;
     }
 
-    public void a(PacketListener packetlistener) {
-        this.a((PacketListenerPlayIn) packetlistener);
+    public EnumMainHand f() {
+        return this.f;
     }
 }
