--- a/net/minecraft/server/PacketPlayOutPlayerListHeaderFooter.java
+++ b/net/minecraft/server/PacketPlayOutPlayerListHeaderFooter.java
@@ -14,8 +14,8 @@
     }
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
-        this.a = packetdataserializer.d();
-        this.b = packetdataserializer.d();
+        this.a = packetdataserializer.f();
+        this.b = packetdataserializer.f();
     }
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
@@ -26,8 +26,4 @@
     public void a(PacketListenerPlayOut packetlistenerplayout) {
         packetlistenerplayout.a(this);
     }
-
-    public void a(PacketListener packetlistener) {
-        this.a((PacketListenerPlayOut) packetlistener);
-    }
 }
