--- a/net/minecraft/server/PacketPlayOutTabComplete.java
+++ b/net/minecraft/server/PacketPlayOutTabComplete.java
@@ -13,7 +13,7 @@
     }
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
-        this.a = new String[packetdataserializer.e()];
+        this.a = new String[packetdataserializer.g()];
 
         for (int i = 0; i < this.a.length; ++i) {
             this.a[i] = packetdataserializer.c(32767);
@@ -37,8 +37,4 @@
     public void a(PacketListenerPlayOut packetlistenerplayout) {
         packetlistenerplayout.a(this);
     }
-
-    public void a(PacketListener packetlistener) {
-        this.a((PacketListenerPlayOut) packetlistener);
-    }
 }
