--- a/net/minecraft/server/PacketSplitter.java
+++ b/net/minecraft/server/PacketSplitter.java
@@ -26,7 +26,7 @@
                 PacketDataSerializer packetdataserializer = new PacketDataSerializer(Unpooled.wrappedBuffer(abyte));
 
                 try {
-                    int j = packetdataserializer.e();
+                    int j = packetdataserializer.g();
 
                     if (bytebuf.readableBytes() >= j) {
                         list.add(bytebuf.readBytes(j));
