--- a/net/minecraft/server/PacketPlayOutEntityTeleport.java
+++ b/net/minecraft/server/PacketPlayOutEntityTeleport.java
@@ -5,9 +5,9 @@
 public class PacketPlayOutEntityTeleport implements Packet<PacketListenerPlayOut> {
 
     private int a;
-    private int b;
-    private int c;
-    private int d;
+    private double b;
+    private double c;
+    private double d;
     private byte e;
     private byte f;
     private boolean g;
@@ -16,29 +16,19 @@
 
     public PacketPlayOutEntityTeleport(Entity entity) {
         this.a = entity.getId();
-        this.b = MathHelper.floor(entity.locX * 32.0D);
-        this.c = MathHelper.floor(entity.locY * 32.0D);
-        this.d = MathHelper.floor(entity.locZ * 32.0D);
+        this.b = entity.locX;
+        this.c = entity.locY;
+        this.d = entity.locZ;
         this.e = (byte) ((int) (entity.yaw * 256.0F / 360.0F));
         this.f = (byte) ((int) (entity.pitch * 256.0F / 360.0F));
         this.g = entity.onGround;
     }
 
-    public PacketPlayOutEntityTeleport(int i, int j, int k, int l, byte b0, byte b1, boolean flag) {
-        this.a = i;
-        this.b = j;
-        this.c = k;
-        this.d = l;
-        this.e = b0;
-        this.f = b1;
-        this.g = flag;
-    }
-
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
-        this.a = packetdataserializer.e();
-        this.b = packetdataserializer.readInt();
-        this.c = packetdataserializer.readInt();
-        this.d = packetdataserializer.readInt();
+        this.a = packetdataserializer.g();
+        this.b = packetdataserializer.readDouble();
+        this.c = packetdataserializer.readDouble();
+        this.d = packetdataserializer.readDouble();
         this.e = packetdataserializer.readByte();
         this.f = packetdataserializer.readByte();
         this.g = packetdataserializer.readBoolean();
@@ -46,9 +36,9 @@
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.b(this.a);
-        packetdataserializer.writeInt(this.b);
-        packetdataserializer.writeInt(this.c);
-        packetdataserializer.writeInt(this.d);
+        packetdataserializer.writeDouble(this.b);
+        packetdataserializer.writeDouble(this.c);
+        packetdataserializer.writeDouble(this.d);
         packetdataserializer.writeByte(this.e);
         packetdataserializer.writeByte(this.f);
         packetdataserializer.writeBoolean(this.g);
@@ -57,8 +47,4 @@
     public void a(PacketListenerPlayOut packetlistenerplayout) {
         packetlistenerplayout.a(this);
     }
-
-    public void a(PacketListener packetlistener) {
-        this.a((PacketListenerPlayOut) packetlistener);
-    }
 }
