--- a/net/minecraft/server/PacketPlayInWindowClick.java
+++ b/net/minecraft/server/PacketPlayInWindowClick.java
@@ -9,7 +9,7 @@
     private int button;
     private short d;
     private ItemStack item;
-    private int shift;
+    private InventoryClickType shift;
 
     public PacketPlayInWindowClick() {}
 
@@ -22,8 +22,8 @@
         this.slot = packetdataserializer.readShort();
         this.button = packetdataserializer.readByte();
         this.d = packetdataserializer.readShort();
-        this.shift = packetdataserializer.readByte();
-        this.item = packetdataserializer.i();
+        this.shift = (InventoryClickType) packetdataserializer.a(InventoryClickType.class);
+        this.item = packetdataserializer.k();
     }
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
@@ -31,7 +31,7 @@
         packetdataserializer.writeShort(this.slot);
         packetdataserializer.writeByte(this.button);
         packetdataserializer.writeShort(this.d);
-        packetdataserializer.writeByte(this.shift);
+        packetdataserializer.a((Enum) this.shift);
         packetdataserializer.a(this.item);
     }
 
@@ -55,11 +55,7 @@
         return this.item;
     }
 
-    public int f() {
+    public InventoryClickType f() {
         return this.shift;
     }
-
-    public void a(PacketListener packetlistener) {
-        this.a((PacketListenerPlayIn) packetlistener);
-    }
 }
