--- a/net/minecraft/server/PacketPlayOutEntityHeadRotation.java
+++ b/net/minecraft/server/PacketPlayOutEntityHeadRotation.java
@@ -15,7 +15,7 @@
     }
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
-        this.a = packetdataserializer.e();
+        this.a = packetdataserializer.g();
         this.b = packetdataserializer.readByte();
     }
 
@@ -27,8 +27,4 @@
     public void a(PacketListenerPlayOut packetlistenerplayout) {
         packetlistenerplayout.a(this);
     }
-
-    public void a(PacketListener packetlistener) {
-        this.a((PacketListenerPlayOut) packetlistener);
-    }
 }
