--- a/net/minecraft/server/PacketPlayInBlockDig.java
+++ b/net/minecraft/server/PacketPlayInBlockDig.java
@@ -12,7 +12,7 @@
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.c = (PacketPlayInBlockDig.EnumPlayerDigType) packetdataserializer.a(PacketPlayInBlockDig.EnumPlayerDigType.class);
-        this.a = packetdataserializer.c();
+        this.a = packetdataserializer.e();
         this.b = EnumDirection.fromType1(packetdataserializer.readUnsignedByte());
     }
 
@@ -38,13 +38,9 @@
         return this.c;
     }
 
-    public void a(PacketListener packetlistener) {
-        this.a((PacketListenerPlayIn) packetlistener);
-    }
-
     public static enum EnumPlayerDigType {
 
-        START_DESTROY_BLOCK, ABORT_DESTROY_BLOCK, STOP_DESTROY_BLOCK, DROP_ALL_ITEMS, DROP_ITEM, RELEASE_USE_ITEM;
+        START_DESTROY_BLOCK, ABORT_DESTROY_BLOCK, STOP_DESTROY_BLOCK, DROP_ALL_ITEMS, DROP_ITEM, RELEASE_USE_ITEM, SWAP_HELD_ITEMS;
 
         private EnumPlayerDigType() {}
     }
