--- a/net/minecraft/server/PacketPlayOutTileEntityData.java
+++ b/net/minecraft/server/PacketPlayOutTileEntityData.java
@@ -17,9 +17,9 @@
     }
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
-        this.a = packetdataserializer.c();
+        this.a = packetdataserializer.e();
         this.b = packetdataserializer.readUnsignedByte();
-        this.c = packetdataserializer.h();
+        this.c = packetdataserializer.j();
     }
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
@@ -31,8 +31,4 @@
     public void a(PacketListenerPlayOut packetlistenerplayout) {
         packetlistenerplayout.a(this);
     }
-
-    public void a(PacketListener packetlistener) {
-        this.a((PacketListenerPlayOut) packetlistener);
-    }
 }
