--- a/net/minecraft/server/PacketPlayOutExplosion.java
+++ b/net/minecraft/server/PacketPlayOutExplosion.java
@@ -25,9 +25,9 @@
         this.d = f;
         this.e = Lists.newArrayList(list);
         if (vec3d != null) {
-            this.f = (float) vec3d.a;
-            this.g = (float) vec3d.b;
-            this.h = (float) vec3d.c;
+            this.f = (float) vec3d.x;
+            this.g = (float) vec3d.y;
+            this.h = (float) vec3d.z;
         }
 
     }
@@ -87,8 +87,4 @@
     public void a(PacketListenerPlayOut packetlistenerplayout) {
         packetlistenerplayout.a(this);
     }
-
-    public void a(PacketListener packetlistener) {
-        this.a((PacketListenerPlayOut) packetlistener);
-    }
 }
