--- a/net/minecraft/server/PathfinderGoalTakeFlower.java
+++ b/net/minecraft/server/PathfinderGoalTakeFlower.java
@@ -18,7 +18,7 @@
     public boolean a() {
         if (this.a.getAge() >= 0) {
             return false;
-        } else if (!this.a.world.w()) {
+        } else if (!this.a.world.B()) {
             return false;
         } else {
             List list = this.a.world.a(EntityIronGolem.class, this.a.getBoundingBox().grow(6.0D, 2.0D, 6.0D));
@@ -31,7 +31,7 @@
                 while (iterator.hasNext()) {
                     EntityIronGolem entityirongolem = (EntityIronGolem) iterator.next();
 
-                    if (entityirongolem.cm() > 0) {
+                    if (entityirongolem.da() > 0) {
                         this.b = entityirongolem;
                         break;
                     }
@@ -43,30 +43,30 @@
     }
 
     public boolean b() {
-        return this.b.cm() > 0;
+        return this.b.da() > 0;
     }
 
     public void c() {
-        this.c = this.a.bc().nextInt(320);
+        this.c = this.a.getRandom().nextInt(320);
         this.d = false;
-        this.b.getNavigation().n();
+        this.b.getNavigation().o();
     }
 
     public void d() {
         this.b = null;
-        this.a.getNavigation().n();
+        this.a.getNavigation().o();
     }
 
     public void e() {
         this.a.getControllerLook().a(this.b, 30.0F, 30.0F);
-        if (this.b.cm() == this.c) {
+        if (this.b.da() == this.c) {
             this.a.getNavigation().a((Entity) this.b, 0.5D);
             this.d = true;
         }
 
         if (this.d && this.a.h(this.b) < 4.0D) {
             this.b.a(false);
-            this.a.getNavigation().n();
+            this.a.getNavigation().o();
         }
 
     }
