--- a/net/minecraft/server/PathfinderGoalSit.java
+++ b/net/minecraft/server/PathfinderGoalSit.java
@@ -13,7 +13,7 @@
     public boolean a() {
         if (!this.entity.isTamed()) {
             return false;
-        } else if (this.entity.V()) {
+        } else if (this.entity.isInWater()) {
             return false;
         } else if (!this.entity.onGround) {
             return false;
@@ -25,7 +25,7 @@
     }
 
     public void c() {
-        this.entity.getNavigation().n();
+        this.entity.getNavigation().o();
         this.entity.setSitting(true);
     }
 
