--- a/net/minecraft/server/PathfinderGoalRandomTargetNonTamed.java
+++ b/net/minecraft/server/PathfinderGoalRandomTargetNonTamed.java
@@ -2,16 +2,16 @@
 
 import com.google.common.base.Predicate;
 
-public class PathfinderGoalRandomTargetNonTamed<T extends EntityLiving> extends PathfinderGoalNearestAttackableTarget {
+public class PathfinderGoalRandomTargetNonTamed<T extends EntityLiving> extends PathfinderGoalNearestAttackableTarget<T> {
 
-    private EntityTameableAnimal g;
+    private EntityTameableAnimal i;
 
     public PathfinderGoalRandomTargetNonTamed(EntityTameableAnimal entitytameableanimal, Class<T> oclass, boolean flag, Predicate<? super T> predicate) {
         super(entitytameableanimal, oclass, 10, flag, false, predicate);
-        this.g = entitytameableanimal;
+        this.i = entitytameableanimal;
     }
 
     public boolean a() {
-        return !this.g.isTamed() && super.a();
+        return !this.i.isTamed() && super.a();
     }
 }
