--- a/net/minecraft/server/PathfinderGoalHurtByTarget.java
+++ b/net/minecraft/server/PathfinderGoalHurtByTarget.java
@@ -5,26 +5,29 @@
 
 public class PathfinderGoalHurtByTarget extends PathfinderGoalTarget {
 
-    private boolean a;
+    private final boolean a;
     private int b;
-    private final Class[] c;
+    private final Class<?>[] c;
 
-    public PathfinderGoalHurtByTarget(EntityCreature entitycreature, boolean flag, Class... aclass) {
-        super(entitycreature, false);
+    public PathfinderGoalHurtByTarget(EntityCreature entitycreature, boolean flag, Class<?>... aclass) {
+        super(entitycreature, true);
         this.a = flag;
         this.c = aclass;
         this.a(1);
     }
 
     public boolean a() {
-        int i = this.e.be();
+        int i = this.e.bH();
+        EntityLiving entityliving = this.e.getLastDamager();
 
-        return i != this.b && this.a(this.e.getLastDamager(), false);
+        return i != this.b && entityliving != null && this.a(entityliving, false);
     }
 
     public void c() {
         this.e.setGoalTarget(this.e.getLastDamager());
-        this.b = this.e.be();
+        this.g = this.e.getGoalTarget();
+        this.b = this.e.bH();
+        this.h = 300;
         if (this.a) {
             double d0 = this.f();
             List list = this.e.world.a(this.e.getClass(), (new AxisAlignedBB(this.e.locX, this.e.locY, this.e.locZ, this.e.locX + 1.0D, this.e.locY + 1.0D, this.e.locZ + 1.0D)).grow(d0, 10.0D, d0));
@@ -33,7 +36,7 @@
             while (iterator.hasNext()) {
                 EntityCreature entitycreature = (EntityCreature) iterator.next();
 
-                if (this.e != entitycreature && entitycreature.getGoalTarget() == null && !entitycreature.c(this.e.getLastDamager())) {
+                if (this.e != entitycreature && entitycreature.getGoalTarget() == null && (!(this.e instanceof EntityTameableAnimal) || ((EntityTameableAnimal) this.e).getOwner() == ((EntityTameableAnimal) entitycreature).getOwner()) && !entitycreature.r(this.e.getLastDamager())) {
                     boolean flag = false;
                     Class[] aclass = this.c;
                     int i = aclass.length;
