--- a/net/minecraft/server/PersistentCollection.java
+++ b/net/minecraft/server/PersistentCollection.java
@@ -164,9 +164,8 @@
 
                     while (iterator.hasNext()) {
                         String s1 = (String) iterator.next();
-                        short short0 = ((Short) this.d.get(s1)).shortValue();
 
-                        nbttagcompound.setShort(s1, short0);
+                        nbttagcompound.setShort(s1, ((Short) this.d.get(s1)).shortValue());
                     }
 
                     DataOutputStream dataoutputstream = new DataOutputStream(new FileOutputStream(file));
