--- a/net/minecraft/server/RemoteStatusReply.java
+++ b/net/minecraft/server/RemoteStatusReply.java
@@ -6,36 +6,36 @@
 
 public class RemoteStatusReply {
 
-    private ByteArrayOutputStream buffer;
-    private DataOutputStream stream;
+    private ByteArrayOutputStream a;
+    private DataOutputStream b;
 
     public RemoteStatusReply(int i) {
-        this.buffer = new ByteArrayOutputStream(i);
-        this.stream = new DataOutputStream(this.buffer);
+        this.a = new ByteArrayOutputStream(i);
+        this.b = new DataOutputStream(this.a);
     }
 
     public void a(byte[] abyte) throws IOException {
-        this.stream.write(abyte, 0, abyte.length);
+        this.b.write(abyte, 0, abyte.length);
     }
 
     public void a(String s) throws IOException {
-        this.stream.writeBytes(s);
-        this.stream.write(0);
+        this.b.writeBytes(s);
+        this.b.write(0);
     }
 
     public void a(int i) throws IOException {
-        this.stream.write(i);
+        this.b.write(i);
     }
 
     public void a(short short0) throws IOException {
-        this.stream.writeShort(Short.reverseBytes(short0));
+        this.b.writeShort(Short.reverseBytes(short0));
     }
 
     public byte[] a() {
-        return this.buffer.toByteArray();
+        return this.a.toByteArray();
     }
 
     public void b() {
-        this.buffer.reset();
+        this.a.reset();
     }
 }
