--- a/net/minecraft/server/SystemUtils.java
+++ b/net/minecraft/server/SystemUtils.java
@@ -1,5 +1,6 @@
 package net.minecraft.server;
 
+import java.util.List;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.FutureTask;
 import org.apache.logging.log4j.Logger;
@@ -18,4 +19,8 @@
 
         return null;
     }
+
+    public static <T> T a(List<T> list) {
+        return list.get(list.size() - 1);
+    }
 }
