--- a/net/minecraft/server/TileEntityLightDetector.java
+++ b/net/minecraft/server/TileEntityLightDetector.java
@@ -1,14 +1,14 @@
 package net.minecraft.server;
 
-public class TileEntityLightDetector extends TileEntity implements IUpdatePlayerListBox {
+public class TileEntityLightDetector extends TileEntity implements ITickable {
 
     public TileEntityLightDetector() {}
 
     public void c() {
         if (this.world != null && !this.world.isClientSide && this.world.getTime() % 20L == 0L) {
-            this.e = this.w();
+            this.e = this.getBlock();
             if (this.e instanceof BlockDaylightDetector) {
-                ((BlockDaylightDetector) this.e).f(this.world, this.position);
+                ((BlockDaylightDetector) this.e).c(this.world, this.position);
             }
         }
 
