--- a/net/minecraft/server/TileEntityComparator.java
+++ b/net/minecraft/server/TileEntityComparator.java
@@ -6,8 +6,8 @@
 
     public TileEntityComparator() {}
 
-    public void b(NBTTagCompound nbttagcompound) {
-        super.b(nbttagcompound);
+    public void save(NBTTagCompound nbttagcompound) {
+        super.save(nbttagcompound);
         nbttagcompound.setInt("OutputSignal", this.a);
     }
 
