--- a/net/minecraft/server/VillageSiege.java
+++ b/net/minecraft/server/VillageSiege.java
@@ -20,7 +20,7 @@
     }
 
     public void a() {
-        if (this.a.w()) {
+        if (this.a.B()) {
             this.c = 0;
         } else if (this.c != 2) {
             if (this.c == 0) {
@@ -70,7 +70,7 @@
             EntityHuman entityhuman = (EntityHuman) iterator.next();
 
             if (!entityhuman.isSpectator()) {
-                this.f = this.a.ae().getClosestVillage(new BlockPosition(entityhuman), 1);
+                this.f = this.a.ai().getClosestVillage(new BlockPosition(entityhuman), 1);
                 if (this.f != null && this.f.c() >= 10 && this.f.d() >= 20 && this.f.e() >= 20) {
                     BlockPosition blockposition = this.f.a();
                     float f = (float) this.f.b();
@@ -79,13 +79,13 @@
 
                     while (true) {
                         if (i < 10) {
-                            float f1 = this.a.random.nextFloat() * 3.1415927F * 2.0F;
+                            float f1 = this.a.random.nextFloat() * 6.2831855F;
 
                             this.g = blockposition.getX() + (int) ((double) (MathHelper.cos(f1) * f) * 0.9D);
                             this.h = blockposition.getY();
                             this.i = blockposition.getZ() + (int) ((double) (MathHelper.sin(f1) * f) * 0.9D);
                             flag = false;
-                            Iterator iterator1 = this.a.ae().getVillages().iterator();
+                            Iterator iterator1 = this.a.ai().getVillages().iterator();
 
                             while (iterator1.hasNext()) {
                                 Village village = (Village) iterator1.next();
@@ -132,14 +132,14 @@
 
             try {
                 entityzombie = new EntityZombie(this.a);
-                entityzombie.prepare(this.a.E(new BlockPosition(entityzombie)), (GroupDataEntity) null);
-                entityzombie.setVillager(false);
+                entityzombie.prepare(this.a.D(new BlockPosition(entityzombie)), (GroupDataEntity) null);
+                entityzombie.clearVillagerType();
             } catch (Exception exception) {
                 exception.printStackTrace();
                 return false;
             }
 
-            entityzombie.setPositionRotation(vec3d.a, vec3d.b, vec3d.c, this.a.random.nextFloat() * 360.0F, 0.0F);
+            entityzombie.setPositionRotation(vec3d.x, vec3d.y, vec3d.z, this.a.random.nextFloat() * 360.0F, 0.0F);
             this.a.addEntity(entityzombie);
             BlockPosition blockposition = this.f.a();
 
