--- a/net/minecraft/server/WorldGenPackedIce2.java
+++ b/net/minecraft/server/WorldGenPackedIce2.java
@@ -37,15 +37,17 @@
                         float f2 = (float) MathHelper.a(j1) - 0.25F;
 
                         if ((i1 == 0 && j1 == 0 || f1 * f1 + f2 * f2 <= f * f) && (i1 != -l && i1 != l && j1 != -l && j1 != l || random.nextFloat() <= 0.75F)) {
-                            Block block = world.getType(blockposition.a(i1, k, j1)).getBlock();
+                            IBlockData iblockdata = world.getType(blockposition.a(i1, k, j1));
+                            Block block = iblockdata.getBlock();
 
-                            if (block.getMaterial() == Material.AIR || block == Blocks.DIRT || block == Blocks.SNOW || block == Blocks.ICE) {
+                            if (iblockdata.getMaterial() == Material.AIR || block == Blocks.DIRT || block == Blocks.SNOW || block == Blocks.ICE) {
                                 this.a(world, blockposition.a(i1, k, j1), Blocks.PACKED_ICE.getBlockData());
                             }
 
                             if (k != 0 && l > 1) {
-                                block = world.getType(blockposition.a(i1, -k, j1)).getBlock();
-                                if (block.getMaterial() == Material.AIR || block == Blocks.DIRT || block == Blocks.SNOW || block == Blocks.ICE) {
+                                iblockdata = world.getType(blockposition.a(i1, -k, j1));
+                                block = iblockdata.getBlock();
+                                if (iblockdata.getMaterial() == Material.AIR || block == Blocks.DIRT || block == Blocks.SNOW || block == Blocks.ICE) {
                                     this.a(world, blockposition.a(i1, -k, j1), Blocks.PACKED_ICE.getBlockData());
                                 }
                             }
@@ -74,9 +76,10 @@
 
                     while (true) {
                         if (blockposition1.getY() > 50) {
-                            Block block1 = world.getType(blockposition1).getBlock();
+                            IBlockData iblockdata1 = world.getType(blockposition1);
+                            Block block1 = iblockdata1.getBlock();
 
-                            if (block1.getMaterial() == Material.AIR || block1 == Blocks.DIRT || block1 == Blocks.SNOW || block1 == Blocks.ICE || block1 == Blocks.PACKED_ICE) {
+                            if (iblockdata1.getMaterial() == Material.AIR || block1 == Blocks.DIRT || block1 == Blocks.SNOW || block1 == Blocks.ICE || block1 == Blocks.PACKED_ICE) {
                                 this.a(world, blockposition1, Blocks.PACKED_ICE.getBlockData());
                                 blockposition1 = blockposition1.down();
                                 --l1;
