--- a/net/minecraft/server/WorldGenMineshaftStart.java
+++ b/net/minecraft/server/WorldGenMineshaftStart.java
@@ -1,6 +1,5 @@
 package net.minecraft.server;
 
-import java.util.List;
 import java.util.Random;
 
 public class WorldGenMineshaftStart extends StructureStart {
@@ -12,8 +11,8 @@
         WorldGenMineshaftPieces.WorldGenMineshaftRoom worldgenmineshaftpieces_worldgenmineshaftroom = new WorldGenMineshaftPieces.WorldGenMineshaftRoom(0, random, (i << 4) + 2, (j << 4) + 2);
 
         this.a.add(worldgenmineshaftpieces_worldgenmineshaftroom);
-        worldgenmineshaftpieces_worldgenmineshaftroom.a((StructurePiece) worldgenmineshaftpieces_worldgenmineshaftroom, (List) this.a, random);
-        this.c();
+        worldgenmineshaftpieces_worldgenmineshaftroom.a((StructurePiece) worldgenmineshaftpieces_worldgenmineshaftroom, this.a, random);
+        this.d();
         this.a(world, random, 10);
     }
 }
