--- a/net/minecraft/server/WorldGenGroundBush.java
+++ b/net/minecraft/server/WorldGenGroundBush.java
@@ -14,15 +14,15 @@
     }
 
     public boolean generate(World world, Random random, BlockPosition blockposition) {
-        Block block;
+        IBlockData iblockdata;
 
-        while (((block = world.getType(blockposition).getBlock()).getMaterial() == Material.AIR || block.getMaterial() == Material.LEAVES) && blockposition.getY() > 0) {
+        while (((iblockdata = world.getType(blockposition)).getMaterial() == Material.AIR || iblockdata.getMaterial() == Material.LEAVES) && blockposition.getY() > 0) {
             blockposition = blockposition.down();
         }
 
-        Block block1 = world.getType(blockposition).getBlock();
+        Block block = world.getType(blockposition).getBlock();
 
-        if (block1 == Blocks.DIRT || block1 == Blocks.GRASS) {
+        if (block == Blocks.DIRT || block == Blocks.GRASS) {
             blockposition = blockposition.up();
             this.a(world, blockposition, this.b);
 
@@ -38,8 +38,9 @@
 
                         if (Math.abs(i1) != k || Math.abs(k1) != k || random.nextInt(2) != 0) {
                             BlockPosition blockposition1 = new BlockPosition(l, i, j1);
+                            Material material = world.getType(blockposition1).getMaterial();
 
-                            if (!world.getType(blockposition1).getBlock().o()) {
+                            if (material == Material.AIR || material == Material.LEAVES) {
                                 this.a(world, blockposition1, this.a);
                             }
                         }
