--- a/net/minecraft/server/WorldGenBonusChest.java
+++ b/net/minecraft/server/WorldGenBonusChest.java
@@ -1,22 +1,15 @@
 package net.minecraft.server;
 
-import java.util.List;
 import java.util.Random;
 
 public class WorldGenBonusChest extends WorldGenerator {
 
-    private final List<StructurePieceTreasure> a;
-    private final int b;
-
-    public WorldGenBonusChest(List<StructurePieceTreasure> list, int i) {
-        this.a = list;
-        this.b = i;
-    }
+    public WorldGenBonusChest() {}
 
     public boolean generate(World world, Random random, BlockPosition blockposition) {
-        Block block;
+        IBlockData iblockdata;
 
-        while (((block = world.getType(blockposition).getBlock()).getMaterial() == Material.AIR || block.getMaterial() == Material.LEAVES) && blockposition.getY() > 1) {
+        while (((iblockdata = world.getType(blockposition)).getMaterial() == Material.AIR || iblockdata.getMaterial() == Material.LEAVES) && blockposition.getY() > 1) {
             blockposition = blockposition.down();
         }
 
@@ -28,12 +21,12 @@
             for (int i = 0; i < 4; ++i) {
                 BlockPosition blockposition1 = blockposition.a(random.nextInt(4) - random.nextInt(4), random.nextInt(3) - random.nextInt(3), random.nextInt(4) - random.nextInt(4));
 
-                if (world.isEmpty(blockposition1) && World.a((IBlockAccess) world, blockposition1.down())) {
+                if (world.isEmpty(blockposition1) && world.getType(blockposition1.down()).q()) {
                     world.setTypeAndData(blockposition1, Blocks.CHEST.getBlockData(), 2);
                     TileEntity tileentity = world.getTileEntity(blockposition1);
 
                     if (tileentity instanceof TileEntityChest) {
-                        StructurePieceTreasure.a(random, this.a, (IInventory) ((TileEntityChest) tileentity), this.b);
+                        ((TileEntityChest) tileentity).a(LootTables.b, random.nextLong());
                     }
 
                     BlockPosition blockposition2 = blockposition1.east();
@@ -41,19 +34,19 @@
                     BlockPosition blockposition4 = blockposition1.north();
                     BlockPosition blockposition5 = blockposition1.south();
 
-                    if (world.isEmpty(blockposition3) && World.a((IBlockAccess) world, blockposition3.down())) {
+                    if (world.isEmpty(blockposition3) && world.getType(blockposition3.down()).q()) {
                         world.setTypeAndData(blockposition3, Blocks.TORCH.getBlockData(), 2);
                     }
 
-                    if (world.isEmpty(blockposition2) && World.a((IBlockAccess) world, blockposition2.down())) {
+                    if (world.isEmpty(blockposition2) && world.getType(blockposition2.down()).q()) {
                         world.setTypeAndData(blockposition2, Blocks.TORCH.getBlockData(), 2);
                     }
 
-                    if (world.isEmpty(blockposition4) && World.a((IBlockAccess) world, blockposition4.down())) {
+                    if (world.isEmpty(blockposition4) && world.getType(blockposition4.down()).q()) {
                         world.setTypeAndData(blockposition4, Blocks.TORCH.getBlockData(), 2);
                     }
 
-                    if (world.isEmpty(blockposition5) && World.a((IBlockAccess) world, blockposition5.down())) {
+                    if (world.isEmpty(blockposition5) && world.getType(blockposition5.down()).q()) {
                         world.setTypeAndData(blockposition5, Blocks.TORCH.getBlockData(), 2);
                     }
 
