--- a/net/minecraft/server/WorldGenCanyon.java
+++ b/net/minecraft/server/WorldGenCanyon.java
@@ -4,7 +4,9 @@
 
 public class WorldGenCanyon extends WorldGenBase {
 
-    private float[] d = new float[1024];
+    protected static final IBlockData a = Blocks.FLOWING_LAVA.getBlockData();
+    protected static final IBlockData b = Blocks.AIR.getBlockData();
+    private float[] c = new float[1024];
 
     public WorldGenCanyon() {}
 
@@ -16,7 +18,7 @@
         float f4 = 0.0F;
 
         if (i1 <= 0) {
-            int j1 = this.a * 16 - 16;
+            int j1 = this.e * 16 - 16;
 
             i1 = j1 - random.nextInt(j1 / 4);
         }
@@ -32,14 +34,14 @@
 
         for (int k1 = 0; k1 < 256; ++k1) {
             if (k1 == 0 || random.nextInt(3) == 0) {
-                f5 = 1.0F + random.nextFloat() * random.nextFloat() * 1.0F;
+                f5 = 1.0F + random.nextFloat() * random.nextFloat();
             }
 
-            this.d[k1] = f5 * f5;
+            this.c[k1] = f5 * f5;
         }
 
         for (; l < i1; ++l) {
-            double d6 = 1.5D + (double) (MathHelper.sin((float) l * 3.1415927F / (float) i1) * f * 1.0F);
+            double d6 = 1.5D + (double) (MathHelper.sin((float) l * 3.1415927F / (float) i1) * f);
             double d7 = d6 * d3;
 
             d6 *= (double) random.nextFloat() * 0.25D + 0.75D;
@@ -135,7 +137,7 @@
                                     for (int j4 = k2; j4 > j2; --j4) {
                                         double d14 = ((double) (j4 - 1) + 0.5D - d1) / d7;
 
-                                        if ((d12 * d12 + d13 * d13) * (double) this.d[j4 - 1] + d14 * d14 / 6.0D < 1.0D) {
+                                        if ((d12 * d12 + d13 * d13) * (double) this.c[j4 - 1] + d14 * d14 / 6.0D < 1.0D) {
                                             IBlockData iblockdata1 = chunksnapshot.a(j3, j4, i4);
 
                                             if (iblockdata1.getBlock() == Blocks.GRASS) {
@@ -144,12 +146,12 @@
 
                                             if (iblockdata1.getBlock() == Blocks.STONE || iblockdata1.getBlock() == Blocks.DIRT || iblockdata1.getBlock() == Blocks.GRASS) {
                                                 if (j4 - 1 < 10) {
-                                                    chunksnapshot.a(j3, j4, i4, Blocks.FLOWING_LAVA.getBlockData());
+                                                    chunksnapshot.a(j3, j4, i4, WorldGenCanyon.a);
                                                 } else {
-                                                    chunksnapshot.a(j3, j4, i4, Blocks.AIR.getBlockData());
+                                                    chunksnapshot.a(j3, j4, i4, WorldGenCanyon.b);
                                                     if (flag2 && chunksnapshot.a(j3, j4 - 1, i4).getBlock() == Blocks.DIRT) {
                                                         blockposition_mutableblockposition.c(j3 + j * 16, 0, i4 + k * 16);
-                                                        chunksnapshot.a(j3, j4 - 1, i4, this.c.getBiome(blockposition_mutableblockposition).ak);
+                                                        chunksnapshot.a(j3, j4 - 1, i4, this.g.getBiome(blockposition_mutableblockposition).r);
                                                     }
                                                 }
                                             }
@@ -170,18 +172,18 @@
     }
 
     protected void a(World world, int i, int j, int k, int l, ChunkSnapshot chunksnapshot) {
-        if (this.b.nextInt(50) == 0) {
-            double d0 = (double) (i * 16 + this.b.nextInt(16));
-            double d1 = (double) (this.b.nextInt(this.b.nextInt(40) + 8) + 20);
-            double d2 = (double) (j * 16 + this.b.nextInt(16));
+        if (this.f.nextInt(50) == 0) {
+            double d0 = (double) (i * 16 + this.f.nextInt(16));
+            double d1 = (double) (this.f.nextInt(this.f.nextInt(40) + 8) + 20);
+            double d2 = (double) (j * 16 + this.f.nextInt(16));
             byte b0 = 1;
 
             for (int i1 = 0; i1 < b0; ++i1) {
-                float f = this.b.nextFloat() * 3.1415927F * 2.0F;
-                float f1 = (this.b.nextFloat() - 0.5F) * 2.0F / 8.0F;
-                float f2 = (this.b.nextFloat() * 2.0F + this.b.nextFloat()) * 2.0F;
+                float f = this.f.nextFloat() * 6.2831855F;
+                float f1 = (this.f.nextFloat() - 0.5F) * 2.0F / 8.0F;
+                float f2 = (this.f.nextFloat() * 2.0F + this.f.nextFloat()) * 2.0F;
 
-                this.a(this.b.nextLong(), k, l, chunksnapshot, d0, d1, d2, f2, f, f1, 0, 0, 3.0D);
+                this.a(this.f.nextLong(), k, l, chunksnapshot, d0, d1, d2, f2, f, f1, 0, 0, 3.0D);
             }
 
         }
