--- a/net/minecraft/server/WorldGenMineshaftPieces.java
+++ b/net/minecraft/server/WorldGenMineshaftPieces.java
@@ -7,8 +7,6 @@
 
 public class WorldGenMineshaftPieces {
 
-    private static final List<StructurePieceTreasure> a = Lists.newArrayList(new StructurePieceTreasure[] { new StructurePieceTreasure(Items.IRON_INGOT, 0, 1, 5, 10), new StructurePieceTreasure(Items.GOLD_INGOT, 0, 1, 3, 5), new StructurePieceTreasure(Items.REDSTONE, 0, 4, 9, 5), new StructurePieceTreasure(Items.DYE, EnumColor.BLUE.getInvColorIndex(), 4, 9, 5), new StructurePieceTreasure(Items.DIAMOND, 0, 1, 2, 3), new StructurePieceTreasure(Items.COAL, 0, 3, 8, 10), new StructurePieceTreasure(Items.BREAD, 0, 1, 3, 15), new StructurePieceTreasure(Items.IRON_PICKAXE, 0, 1, 1, 1), new StructurePieceTreasure(Item.getItemOf(Blocks.RAIL), 0, 4, 8, 1), new StructurePieceTreasure(Items.MELON_SEEDS, 0, 2, 4, 10), new StructurePieceTreasure(Items.PUMPKIN_SEEDS, 0, 2, 4, 10), new StructurePieceTreasure(Items.SADDLE, 0, 1, 1, 3), new StructurePieceTreasure(Items.IRON_HORSE_ARMOR, 0, 1, 1, 1)});
-
     public static void a() {
         WorldGenFactory.a(WorldGenMineshaftPieces.WorldGenMineshaftCorridor.class, "MSCorridor");
         WorldGenFactory.a(WorldGenMineshaftPieces.WorldGenMineshaftCross.class, "MSCrossing");
@@ -95,7 +93,7 @@
 
         public WorldGenMineshaftStairs(int i, Random random, StructureBoundingBox structureboundingbox, EnumDirection enumdirection) {
             super(i);
-            this.m = enumdirection;
+            this.a(enumdirection);
             this.l = structureboundingbox;
         }
 
@@ -132,9 +130,10 @@
 
         public void a(StructurePiece structurepiece, List<StructurePiece> list, Random random) {
             int i = this.d();
+            EnumDirection enumdirection = this.e();
 
-            if (this.m != null) {
-                switch (WorldGenMineshaftPieces.SyntheticClass_1.a[this.m.ordinal()]) {
+            if (enumdirection != null) {
+                switch (WorldGenMineshaftPieces.SyntheticClass_1.a[enumdirection.ordinal()]) {
                 case 1:
                     WorldGenMineshaftPieces.b(structurepiece, list, random, this.l.a, this.l.b, this.l.c - 1, EnumDirection.NORTH, i);
                     break;
@@ -179,7 +178,7 @@
 
         protected void a(NBTTagCompound nbttagcompound) {
             nbttagcompound.setBoolean("tf", this.b);
-            nbttagcompound.setInt("D", this.a.b());
+            nbttagcompound.setInt("D", this.a.get2DRotationValue());
         }
 
         protected void b(NBTTagCompound nbttagcompound) {
@@ -211,7 +210,7 @@
             case 2:
                 structureboundingbox.a = i - 1;
                 structureboundingbox.d = i + 3;
-                structureboundingbox.f = k + 4;
+                structureboundingbox.f = k + 3 + 1;
                 break;
 
             case 3:
@@ -221,7 +220,7 @@
                 break;
 
             case 4:
-                structureboundingbox.d = i + 4;
+                structureboundingbox.d = i + 3 + 1;
                 structureboundingbox.c = k - 1;
                 structureboundingbox.f = k + 3;
             }
@@ -299,7 +298,7 @@
 
                 for (int i = this.l.a; i <= this.l.d; ++i) {
                     for (int j = this.l.c; j <= this.l.f; ++j) {
-                        if (this.a(world, i, this.l.b - 1, j, structureboundingbox).getBlock().getMaterial() == Material.AIR) {
+                        if (this.a(world, i, this.l.b - 1, j, structureboundingbox).getMaterial() == Material.AIR) {
                             this.a(world, Blocks.PLANKS.getBlockData(), i, this.l.b - 1, j, structureboundingbox);
                         }
                     }
@@ -335,14 +334,14 @@
 
         public WorldGenMineshaftCorridor(int i, Random random, StructureBoundingBox structureboundingbox, EnumDirection enumdirection) {
             super(i);
-            this.m = enumdirection;
+            this.a(enumdirection);
             this.l = structureboundingbox;
             this.a = random.nextInt(3) == 0;
             this.b = !this.a && random.nextInt(23) == 0;
-            if (this.m != EnumDirection.NORTH && this.m != EnumDirection.SOUTH) {
-                this.d = structureboundingbox.c() / 5;
-            } else {
+            if (this.e().k() == EnumDirection.EnumAxis.Z) {
                 this.d = structureboundingbox.e() / 5;
+            } else {
+                this.d = structureboundingbox.c() / 5;
             }
 
         }
@@ -387,12 +386,13 @@
         public void a(StructurePiece structurepiece, List<StructurePiece> list, Random random) {
             int i = this.d();
             int j = random.nextInt(4);
+            EnumDirection enumdirection = this.e();
 
-            if (this.m != null) {
-                switch (WorldGenMineshaftPieces.SyntheticClass_1.a[this.m.ordinal()]) {
+            if (enumdirection != null) {
+                switch (WorldGenMineshaftPieces.SyntheticClass_1.a[enumdirection.ordinal()]) {
                 case 1:
                     if (j <= 1) {
-                        WorldGenMineshaftPieces.b(structurepiece, list, random, this.l.a, this.l.b - 1 + random.nextInt(3), this.l.c - 1, this.m, i);
+                        WorldGenMineshaftPieces.b(structurepiece, list, random, this.l.a, this.l.b - 1 + random.nextInt(3), this.l.c - 1, enumdirection, i);
                     } else if (j == 2) {
                         WorldGenMineshaftPieces.b(structurepiece, list, random, this.l.a - 1, this.l.b - 1 + random.nextInt(3), this.l.c, EnumDirection.WEST, i);
                     } else {
@@ -402,7 +402,7 @@
 
                 case 2:
                     if (j <= 1) {
-                        WorldGenMineshaftPieces.b(structurepiece, list, random, this.l.a, this.l.b - 1 + random.nextInt(3), this.l.f + 1, this.m, i);
+                        WorldGenMineshaftPieces.b(structurepiece, list, random, this.l.a, this.l.b - 1 + random.nextInt(3), this.l.f + 1, enumdirection, i);
                     } else if (j == 2) {
                         WorldGenMineshaftPieces.b(structurepiece, list, random, this.l.a - 1, this.l.b - 1 + random.nextInt(3), this.l.f - 3, EnumDirection.WEST, i);
                     } else {
@@ -412,7 +412,7 @@
 
                 case 3:
                     if (j <= 1) {
-                        WorldGenMineshaftPieces.b(structurepiece, list, random, this.l.a - 1, this.l.b - 1 + random.nextInt(3), this.l.c, this.m, i);
+                        WorldGenMineshaftPieces.b(structurepiece, list, random, this.l.a - 1, this.l.b - 1 + random.nextInt(3), this.l.c, enumdirection, i);
                     } else if (j == 2) {
                         WorldGenMineshaftPieces.b(structurepiece, list, random, this.l.a, this.l.b - 1 + random.nextInt(3), this.l.c - 1, EnumDirection.NORTH, i);
                     } else {
@@ -422,7 +422,7 @@
 
                 case 4:
                     if (j <= 1) {
-                        WorldGenMineshaftPieces.b(structurepiece, list, random, this.l.d + 1, this.l.b - 1 + random.nextInt(3), this.l.c, this.m, i);
+                        WorldGenMineshaftPieces.b(structurepiece, list, random, this.l.d + 1, this.l.b - 1 + random.nextInt(3), this.l.c, enumdirection, i);
                     } else if (j == 2) {
                         WorldGenMineshaftPieces.b(structurepiece, list, random, this.l.d - 3, this.l.b - 1 + random.nextInt(3), this.l.c - 1, EnumDirection.NORTH, i);
                     } else {
@@ -435,7 +435,7 @@
                 int k;
                 int l;
 
-                if (this.m != EnumDirection.NORTH && this.m != EnumDirection.SOUTH) {
+                if (enumdirection != EnumDirection.NORTH && enumdirection != EnumDirection.SOUTH) {
                     for (k = this.l.a + 3; k + 3 <= this.l.d; k += 5) {
                         l = random.nextInt(5);
                         if (l == 0) {
@@ -458,16 +458,16 @@
 
         }
 
-        protected boolean a(World world, StructureBoundingBox structureboundingbox, Random random, int i, int j, int k, List<StructurePieceTreasure> list, int l) {
+        protected boolean a(World world, StructureBoundingBox structureboundingbox, Random random, int i, int j, int k, MinecraftKey minecraftkey) {
             BlockPosition blockposition = new BlockPosition(this.a(i, k), this.d(j), this.b(i, k));
 
-            if (structureboundingbox.b((BaseBlockPosition) blockposition) && world.getType(blockposition).getBlock().getMaterial() == Material.AIR) {
-                int i1 = random.nextBoolean() ? 1 : 0;
+            if (structureboundingbox.b((BaseBlockPosition) blockposition) && world.getType(blockposition).getMaterial() == Material.AIR) {
+                IBlockData iblockdata = Blocks.RAIL.getBlockData().set(BlockMinecartTrack.SHAPE, random.nextBoolean() ? BlockMinecartTrackAbstract.EnumTrackPosition.NORTH_SOUTH : BlockMinecartTrackAbstract.EnumTrackPosition.EAST_WEST);
 
-                world.setTypeAndData(blockposition, Blocks.RAIL.fromLegacyData(this.a(Blocks.RAIL, i1)), 2);
+                this.a(world, iblockdata, i, j, k, structureboundingbox);
                 EntityMinecartChest entityminecartchest = new EntityMinecartChest(world, (double) ((float) blockposition.getX() + 0.5F), (double) ((float) blockposition.getY() + 0.5F), (double) ((float) blockposition.getZ() + 0.5F));
 
-                StructurePieceTreasure.a(random, list, (IInventory) entityminecartchest, l);
+                entityminecartchest.a(minecraftkey, random.nextLong());
                 world.addEntity(entityminecartchest);
                 return true;
             } else {
@@ -513,14 +513,14 @@
                     this.a(world, structureboundingbox, random, 0.05F, 2, 2, k - 2, Blocks.WEB.getBlockData());
                     this.a(world, structureboundingbox, random, 0.05F, 0, 2, k + 2, Blocks.WEB.getBlockData());
                     this.a(world, structureboundingbox, random, 0.05F, 2, 2, k + 2, Blocks.WEB.getBlockData());
-                    this.a(world, structureboundingbox, random, 0.05F, 1, 2, k - 1, Blocks.TORCH.fromLegacyData(EnumDirection.UP.a()));
-                    this.a(world, structureboundingbox, random, 0.05F, 1, 2, k + 1, Blocks.TORCH.fromLegacyData(EnumDirection.UP.a()));
+                    this.a(world, structureboundingbox, random, 0.05F, 1, 2, k - 1, Blocks.TORCH.getBlockData().set(BlockTorch.FACING, EnumDirection.SOUTH));
+                    this.a(world, structureboundingbox, random, 0.05F, 1, 2, k + 1, Blocks.TORCH.getBlockData().set(BlockTorch.FACING, EnumDirection.NORTH));
                     if (random.nextInt(100) == 0) {
-                        this.a(world, structureboundingbox, random, 2, 0, k - 1, StructurePieceTreasure.a(WorldGenMineshaftPieces.a, new StructurePieceTreasure[] { Items.ENCHANTED_BOOK.b(random)}), 3 + random.nextInt(4));
+                        this.a(world, structureboundingbox, random, 2, 0, k - 1, LootTables.f);
                     }
 
                     if (random.nextInt(100) == 0) {
-                        this.a(world, structureboundingbox, random, 0, 0, k + 1, StructurePieceTreasure.a(WorldGenMineshaftPieces.a, new StructurePieceTreasure[] { Items.ENCHANTED_BOOK.b(random)}), 3 + random.nextInt(4));
+                        this.a(world, structureboundingbox, random, 0, 0, k + 1, LootTables.f);
                     }
 
                     if (this.b && !this.c) {
@@ -548,7 +548,7 @@
                         byte b0 = -1;
                         IBlockData iblockdata = this.a(world, j, b0, k, structureboundingbox);
 
-                        if (iblockdata.getBlock().getMaterial() == Material.AIR) {
+                        if (iblockdata.getMaterial() == Material.AIR) {
                             byte b1 = -1;
 
                             this.a(world, Blocks.PLANKS.getBlockData(), j, b1, k, structureboundingbox);
@@ -557,11 +557,13 @@
                 }
 
                 if (this.a) {
-                    for (j = 0; j <= i; ++j) {
-                        IBlockData iblockdata1 = this.a(world, 1, -1, j, structureboundingbox);
+                    IBlockData iblockdata1 = Blocks.RAIL.getBlockData().set(BlockMinecartTrack.SHAPE, BlockMinecartTrackAbstract.EnumTrackPosition.NORTH_SOUTH);
+
+                    for (k = 0; k <= i; ++k) {
+                        IBlockData iblockdata2 = this.a(world, 1, -1, k, structureboundingbox);
 
-                        if (iblockdata1.getBlock().getMaterial() != Material.AIR && iblockdata1.getBlock().o()) {
-                            this.a(world, structureboundingbox, random, 0.7F, 1, 0, j, Blocks.RAIL.fromLegacyData(this.a(Blocks.RAIL, 0)));
+                        if (iblockdata2.getMaterial() != Material.AIR && iblockdata2.b()) {
+                            this.a(world, structureboundingbox, random, 0.7F, 1, 0, k, iblockdata1);
                         }
                     }
                 }
@@ -696,7 +698,7 @@
             NBTTagList nbttaglist = nbttagcompound.getList("Entrances", 11);
 
             for (int i = 0; i < nbttaglist.size(); ++i) {
-                this.a.add(new StructureBoundingBox(nbttaglist.c(i)));
+                this.a.add(new StructureBoundingBox(nbttaglist.d(i)));
             }
 
         }
