--- a/net/minecraft/server/WorldGenTaigaStructure.java
+++ b/net/minecraft/server/WorldGenTaigaStructure.java
@@ -47,7 +47,7 @@
                 while (iterator.hasNext()) {
                     BlockPosition blockposition1 = (BlockPosition) iterator.next();
 
-                    if (blockposition1.i(blockposition) <= (double) (f * f)) {
+                    if (blockposition1.k(blockposition) <= (double) (f * f)) {
                         world.setTypeAndData(blockposition1, this.a.getBlockData(), 4);
                     }
                 }
