--- a/net/minecraft/server/AttributeInstance.java
+++ b/net/minecraft/server/AttributeInstance.java
@@ -2,6 +2,7 @@
 
 import java.util.Collection;
 import java.util.UUID;
+import javax.annotation.Nullable;
 
 public interface AttributeInstance {
 
@@ -17,6 +18,7 @@
 
     boolean a(AttributeModifier attributemodifier);
 
+    @Nullable
     AttributeModifier a(UUID uuid);
 
     void b(AttributeModifier attributemodifier);
