--- a/net/minecraft/server/BlockThin.java
+++ b/net/minecraft/server/BlockThin.java
@@ -2,6 +2,7 @@
 
 import java.util.List;
 import java.util.Random;
+import javax.annotation.Nullable;
 
 public class BlockThin extends Block {
 
@@ -19,7 +20,7 @@
         this.a(CreativeModeTab.c);
     }
 
-    public void a(IBlockData iblockdata, World world, BlockPosition blockposition, AxisAlignedBB axisalignedbb, List<AxisAlignedBB> list, Entity entity) {
+    public void a(IBlockData iblockdata, World world, BlockPosition blockposition, AxisAlignedBB axisalignedbb, List<AxisAlignedBB> list, @Nullable Entity entity) {
         iblockdata = this.updateState(iblockdata, world, blockposition);
         a(blockposition, axisalignedbb, list, BlockThin.f[0]);
         if (((Boolean) iblockdata.get(BlockThin.NORTH)).booleanValue()) {
@@ -75,6 +76,7 @@
         return iblockdata.set(BlockThin.NORTH, Boolean.valueOf(this.c(iblockaccess.getType(blockposition.north()).getBlock()))).set(BlockThin.SOUTH, Boolean.valueOf(this.c(iblockaccess.getType(blockposition.south()).getBlock()))).set(BlockThin.WEST, Boolean.valueOf(this.c(iblockaccess.getType(blockposition.west()).getBlock()))).set(BlockThin.EAST, Boolean.valueOf(this.c(iblockaccess.getType(blockposition.east()).getBlock())));
     }
 
+    @Nullable
     public Item getDropType(IBlockData iblockdata, Random random, int i) {
         return !this.a ? null : super.getDropType(iblockdata, random, i);
     }
