--- a/net/minecraft/server/BlockEndGateway.java
+++ b/net/minecraft/server/BlockEndGateway.java
@@ -1,6 +1,7 @@
 package net.minecraft.server;
 
 import java.util.Random;
+import javax.annotation.Nullable;
 
 public class BlockEndGateway extends BlockTileEntity {
 
@@ -13,6 +14,7 @@
         return new TileEntityEndGateway();
     }
 
+    @Nullable
     public AxisAlignedBB a(IBlockData iblockdata, World world, BlockPosition blockposition) {
         return BlockEndGateway.k;
     }
@@ -29,6 +31,7 @@
         return 0;
     }
 
+    @Nullable
     public ItemStack a(World world, BlockPosition blockposition, IBlockData iblockdata) {
         return null;
     }
