--- a/net/minecraft/server/BlockSlowSand.java
+++ b/net/minecraft/server/BlockSlowSand.java
@@ -1,5 +1,7 @@
 package net.minecraft.server;
 
+import javax.annotation.Nullable;
+
 public class BlockSlowSand extends Block {
 
     protected static final AxisAlignedBB a = new AxisAlignedBB(0.0D, 0.0D, 0.0D, 1.0D, 0.875D, 1.0D);
@@ -9,6 +11,7 @@
         this.a(CreativeModeTab.b);
     }
 
+    @Nullable
     public AxisAlignedBB a(IBlockData iblockdata, World world, BlockPosition blockposition) {
         return BlockSlowSand.a;
     }
