--- a/net/minecraft/server/BlockBrewingStand.java
+++ b/net/minecraft/server/BlockBrewingStand.java
@@ -2,6 +2,7 @@
 
 import java.util.List;
 import java.util.Random;
+import javax.annotation.Nullable;
 
 public class BlockBrewingStand extends BlockTileEntity {
 
@@ -34,7 +35,7 @@
         return false;
     }
 
-    public void a(IBlockData iblockdata, World world, BlockPosition blockposition, AxisAlignedBB axisalignedbb, List<AxisAlignedBB> list, Entity entity) {
+    public void a(IBlockData iblockdata, World world, BlockPosition blockposition, AxisAlignedBB axisalignedbb, List<AxisAlignedBB> list, @Nullable Entity entity) {
         a(blockposition, axisalignedbb, list, BlockBrewingStand.c);
         a(blockposition, axisalignedbb, list, BlockBrewingStand.b);
     }
@@ -43,7 +44,7 @@
         return BlockBrewingStand.b;
     }
 
-    public boolean interact(World world, BlockPosition blockposition, IBlockData iblockdata, EntityHuman entityhuman, EnumHand enumhand, ItemStack itemstack, EnumDirection enumdirection, float f, float f1, float f2) {
+    public boolean interact(World world, BlockPosition blockposition, IBlockData iblockdata, EntityHuman entityhuman, EnumHand enumhand, @Nullable ItemStack itemstack, EnumDirection enumdirection, float f, float f1, float f2) {
         if (world.isClientSide) {
             return true;
         } else {
@@ -79,6 +80,7 @@
         super.remove(world, blockposition, iblockdata);
     }
 
+    @Nullable
     public Item getDropType(IBlockData iblockdata, Random random, int i) {
         return Items.BREWING_STAND;
     }
