--- a/net/minecraft/server/ChunkProviderGenerate.java
+++ b/net/minecraft/server/ChunkProviderGenerate.java
@@ -2,6 +2,7 @@
 
 import java.util.List;
 import java.util.Random;
+import javax.annotation.Nullable;
 
 public class ChunkProviderGenerate implements ChunkGenerator {
 
@@ -418,6 +419,7 @@
         return biomebase.getMobs(enumcreaturetype);
     }
 
+    @Nullable
     public BlockPosition findNearestMapFeature(World world, String s, BlockPosition blockposition) {
         return "Stronghold".equals(s) && this.w != null ? this.w.getNearestGeneratedFeature(world, blockposition) : null;
     }
