--- a/net/minecraft/server/ChunkProviderTheEnd.java
+++ b/net/minecraft/server/ChunkProviderTheEnd.java
@@ -2,6 +2,7 @@
 
 import java.util.List;
 import java.util.Random;
+import javax.annotation.Nullable;
 
 public class ChunkProviderTheEnd implements ChunkGenerator {
 
@@ -316,6 +317,7 @@
         return this.l.getBiome(blockposition).getMobs(enumcreaturetype);
     }
 
+    @Nullable
     public BlockPosition findNearestMapFeature(World world, String s, BlockPosition blockposition) {
         return null;
     }
