--- a/net/minecraft/server/ChunkProviderHell.java
+++ b/net/minecraft/server/ChunkProviderHell.java
@@ -2,6 +2,7 @@
 
 import java.util.List;
 import java.util.Random;
+import javax.annotation.Nullable;
 
 public class ChunkProviderHell implements ChunkGenerator {
 
@@ -359,6 +360,7 @@
         return biomebase.getMobs(enumcreaturetype);
     }
 
+    @Nullable
     public BlockPosition findNearestMapFeature(World world, String s, BlockPosition blockposition) {
         return null;
     }
