--- a/net/minecraft/server/CombatMath.java
+++ b/net/minecraft/server/CombatMath.java
@@ -5,9 +5,8 @@
     public static float a(float f, float f1, float f2) {
         float f3 = 2.0F + f2 / 4.0F;
         float f4 = MathHelper.a(f1 - f / f3, f1 * 0.2F, 20.0F);
-        float f5 = f * (1.0F - f4 / 25.0F);
 
-        return f5;
+        return f * (1.0F - f4 / 25.0F);
     }
 
     public static float a(float f, float f1) {
