--- a/net/minecraft/server/CommandDispatcher.java
+++ b/net/minecraft/server/CommandDispatcher.java
@@ -51,6 +51,7 @@
         this.a((ICommand) (new CommandWorldBorder()));
         this.a((ICommand) (new CommandTitle()));
         this.a((ICommand) (new CommandEntityData()));
+        this.a((ICommand) (new CommandStopSound()));
         if (minecraftserver.aa()) {
             this.a((ICommand) (new CommandOp()));
             this.a((ICommand) (new CommandDeop()));
