--- a/net/minecraft/server/CommandTell.java
+++ b/net/minecraft/server/CommandTell.java
@@ -2,6 +2,7 @@
 
 import java.util.Arrays;
 import java.util.List;
+import javax.annotation.Nullable;
 
 public class CommandTell extends CommandAbstract {
 
@@ -44,7 +45,7 @@
         }
     }
 
-    public List<String> tabComplete(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring, BlockPosition blockposition) {
+    public List<String> tabComplete(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring, @Nullable BlockPosition blockposition) {
         return a(astring, minecraftserver.getPlayers());
     }
 
