--- a/net/minecraft/server/CommandXp.java
+++ b/net/minecraft/server/CommandXp.java
@@ -2,6 +2,7 @@
 
 import java.util.Collections;
 import java.util.List;
+import javax.annotation.Nullable;
 
 public class CommandXp extends CommandAbstract {
 
@@ -61,7 +62,7 @@
         }
     }
 
-    public List<String> tabComplete(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring, BlockPosition blockposition) {
+    public List<String> tabComplete(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring, @Nullable BlockPosition blockposition) {
         return astring.length == 2 ? a(astring, minecraftserver.getPlayers()) : Collections.emptyList();
     }
 
