--- a/net/minecraft/server/CommandBan.java
+++ b/net/minecraft/server/CommandBan.java
@@ -4,6 +4,7 @@
 import java.util.Collections;
 import java.util.Date;
 import java.util.List;
+import javax.annotation.Nullable;
 
 public class CommandBan extends CommandAbstract {
 
@@ -54,7 +55,7 @@
         }
     }
 
-    public List<String> tabComplete(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring, BlockPosition blockposition) {
+    public List<String> tabComplete(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring, @Nullable BlockPosition blockposition) {
         return astring.length >= 1 ? a(astring, minecraftserver.getPlayers()) : Collections.emptyList();
     }
 }
