--- a/net/minecraft/server/CommandMe.java
+++ b/net/minecraft/server/CommandMe.java
@@ -1,6 +1,7 @@
 package net.minecraft.server;
 
 import java.util.List;
+import javax.annotation.Nullable;
 
 public class CommandMe extends CommandAbstract {
 
@@ -28,7 +29,7 @@
         }
     }
 
-    public List<String> tabComplete(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring, BlockPosition blockposition) {
+    public List<String> tabComplete(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring, @Nullable BlockPosition blockposition) {
         return a(astring, minecraftserver.getPlayers());
     }
 }
