--- a/net/minecraft/server/CommandBanList.java
+++ b/net/minecraft/server/CommandBanList.java
@@ -2,6 +2,7 @@
 
 import java.util.Collections;
 import java.util.List;
+import javax.annotation.Nullable;
 
 public class CommandBanList extends CommandAbstract {
 
@@ -34,7 +35,7 @@
 
     }
 
-    public List<String> tabComplete(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring, BlockPosition blockposition) {
+    public List<String> tabComplete(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring, @Nullable BlockPosition blockposition) {
         return astring.length == 1 ? a(astring, new String[] { "players", "ips"}) : Collections.emptyList();
     }
 }
