--- a/net/minecraft/server/CommandTellRaw.java
+++ b/net/minecraft/server/CommandTellRaw.java
@@ -3,6 +3,7 @@
 import com.google.gson.JsonParseException;
 import java.util.Collections;
 import java.util.List;
+import javax.annotation.Nullable;
 
 public class CommandTellRaw extends CommandAbstract {
 
@@ -37,7 +38,7 @@
         }
     }
 
-    public List<String> tabComplete(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring, BlockPosition blockposition) {
+    public List<String> tabComplete(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring, @Nullable BlockPosition blockposition) {
         return astring.length == 1 ? a(astring, minecraftserver.getPlayers()) : Collections.emptyList();
     }
 
