--- a/net/minecraft/server/ContainerUtil.java
+++ b/net/minecraft/server/ContainerUtil.java
@@ -1,7 +1,10 @@
 package net.minecraft.server;
 
+import javax.annotation.Nullable;
+
 public class ContainerUtil {
 
+    @Nullable
     public static ItemStack a(ItemStack[] aitemstack, int i, int j) {
         if (i >= 0 && i < aitemstack.length && aitemstack[i] != null && j > 0) {
             ItemStack itemstack = aitemstack[i].cloneAndSubtract(j);
@@ -16,6 +19,7 @@
         }
     }
 
+    @Nullable
     public static ItemStack a(ItemStack[] aitemstack, int i) {
         if (i >= 0 && i < aitemstack.length) {
             ItemStack itemstack = aitemstack[i];
