--- a/net/minecraft/server/DataConverterRegistry.java
+++ b/net/minecraft/server/DataConverterRegistry.java
@@ -21,7 +21,7 @@
     }
 
     public static DataConverterManager a() {
-        DataConverterManager dataconvertermanager = new DataConverterManager(176);
+        DataConverterManager dataconvertermanager = new DataConverterManager(184);
 
         dataconvertermanager.a(DataConverterTypes.LEVEL, new DataInspector() {
             public NBTTagCompound a(DataConverter dataconverter, NBTTagCompound nbttagcompound, int i) {
