--- a/net/minecraft/server/DemoPlayerInteractManager.java
+++ b/net/minecraft/server/DemoPlayerInteractManager.java
@@ -1,5 +1,7 @@
 package net.minecraft.server;
 
+import javax.annotation.Nullable;
+
 public class DemoPlayerInteractManager extends PlayerInteractManager {
 
     private boolean c;
@@ -80,7 +82,7 @@
         }
     }
 
-    public EnumInteractionResult a(EntityHuman entityhuman, World world, ItemStack itemstack, EnumHand enumhand, BlockPosition blockposition, EnumDirection enumdirection, float f, float f1, float f2) {
+    public EnumInteractionResult a(EntityHuman entityhuman, World world, @Nullable ItemStack itemstack, EnumHand enumhand, BlockPosition blockposition, EnumDirection enumdirection, float f, float f1, float f2) {
         if (this.d) {
             this.f();
             return EnumInteractionResult.PASS;
