--- a/net/minecraft/server/DragonControllerPhase.java
+++ b/net/minecraft/server/DragonControllerPhase.java
@@ -49,7 +49,7 @@
         return this.o + " (#" + this.n + ")";
     }
 
-    public static DragonControllerPhase<?> a(int i) {
+    public static DragonControllerPhase<?> getById(int i) {
         return i >= 0 && i < DragonControllerPhase.l.length ? DragonControllerPhase.l[i] : DragonControllerPhase.a;
     }
 
