--- a/net/minecraft/server/EntityPlayer.java
+++ b/net/minecraft/server/EntityPlayer.java
@@ -8,12 +8,13 @@
 import java.util.HashSet;
 import java.util.Iterator;
 import java.util.List;
+import javax.annotation.Nullable;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class EntityPlayer extends EntityHuman implements ICrafting {
 
-    private static final Logger bQ = LogManager.getLogger();
+    private static final Logger bR = LogManager.getLogger();
     private String locale = "en_US";
     public PlayerConnection playerConnection;
     public final MinecraftServer server;
@@ -21,23 +22,23 @@
     public double d;
     public double e;
     public final List<Integer> removeQueue = Lists.newLinkedList();
-    private final ServerStatisticManager bT;
-    private float bU = Float.MIN_VALUE;
-    private int bV = Integer.MIN_VALUE;
+    private final ServerStatisticManager bU;
+    private float bV = Float.MIN_VALUE;
     private int bW = Integer.MIN_VALUE;
     private int bX = Integer.MIN_VALUE;
     private int bY = Integer.MIN_VALUE;
     private int bZ = Integer.MIN_VALUE;
+    private int ca = Integer.MIN_VALUE;
     private float lastHealthSent = -1.0E8F;
-    private int cb = -99999999;
-    private boolean cc = true;
+    private int cc = -99999999;
+    private boolean cd = true;
     public int lastSentExp = -99999999;
     public int invulnerableTicks = 60;
-    private EntityHuman.EnumChatVisibility cf;
-    private boolean cg = true;
-    private long ch = System.currentTimeMillis();
-    private Entity ci = null;
-    private boolean cj;
+    private EntityHuman.EnumChatVisibility cg;
+    private boolean ch = true;
+    private long ci = System.currentTimeMillis();
+    private Entity cj = null;
+    private boolean ck;
     private int containerCounter;
     public boolean f;
     public int ping;
@@ -65,7 +66,7 @@
         }
 
         this.server = minecraftserver;
-        this.bT = minecraftserver.getPlayerList().a((EntityHuman) this);
+        this.bU = minecraftserver.getPlayerList().a((EntityHuman) this);
         this.P = 0.0F;
         this.setPositionRotation(blockposition, 0.0F, 0.0F);
 
@@ -92,12 +93,12 @@
         nbttagcompound.setInt("playerGameType", this.playerInteractManager.getGameMode().getId());
         Entity entity = this.getVehicle();
 
-        if (this.by() != null && entity != this & entity.b(EntityPlayer.class).size() == 1) {
+        if (this.bz() != null && entity != this & entity.b(EntityPlayer.class).size() == 1) {
             NBTTagCompound nbttagcompound1 = new NBTTagCompound();
             NBTTagCompound nbttagcompound2 = new NBTTagCompound();
 
             entity.d(nbttagcompound2);
-            nbttagcompound1.a("Attach", this.by().getUniqueID());
+            nbttagcompound1.a("Attach", this.bz().getUniqueID());
             nbttagcompound1.set("Entity", nbttagcompound2);
             nbttagcompound.set("RootVehicle", nbttagcompound1);
         }
@@ -191,41 +192,41 @@
                 }
             }
 
-            if (this.getHealth() != this.lastHealthSent || this.cb != this.foodData.getFoodLevel() || this.foodData.getSaturationLevel() == 0.0F != this.cc) {
+            if (this.getHealth() != this.lastHealthSent || this.cc != this.foodData.getFoodLevel() || this.foodData.getSaturationLevel() == 0.0F != this.cd) {
                 this.playerConnection.sendPacket(new PacketPlayOutUpdateHealth(this.getHealth(), this.foodData.getFoodLevel(), this.foodData.getSaturationLevel()));
                 this.lastHealthSent = this.getHealth();
-                this.cb = this.foodData.getFoodLevel();
-                this.cc = this.foodData.getSaturationLevel() == 0.0F;
+                this.cc = this.foodData.getFoodLevel();
+                this.cd = this.foodData.getSaturationLevel() == 0.0F;
             }
 
-            if (this.getHealth() + this.getAbsorptionHearts() != this.bU) {
-                this.bU = this.getHealth() + this.getAbsorptionHearts();
-                this.a(IScoreboardCriteria.g, MathHelper.f(this.bU));
+            if (this.getHealth() + this.getAbsorptionHearts() != this.bV) {
+                this.bV = this.getHealth() + this.getAbsorptionHearts();
+                this.a(IScoreboardCriteria.g, MathHelper.f(this.bV));
             }
 
-            if (this.foodData.getFoodLevel() != this.bV) {
-                this.bV = this.foodData.getFoodLevel();
-                this.a(IScoreboardCriteria.h, MathHelper.f((float) this.bV));
+            if (this.foodData.getFoodLevel() != this.bW) {
+                this.bW = this.foodData.getFoodLevel();
+                this.a(IScoreboardCriteria.h, MathHelper.f((float) this.bW));
             }
 
-            if (this.getAirTicks() != this.bW) {
-                this.bW = this.getAirTicks();
-                this.a(IScoreboardCriteria.i, MathHelper.f((float) this.bW));
+            if (this.getAirTicks() != this.bX) {
+                this.bX = this.getAirTicks();
+                this.a(IScoreboardCriteria.i, MathHelper.f((float) this.bX));
             }
 
-            if (this.getArmorStrength() != this.bX) {
-                this.bX = this.getArmorStrength();
-                this.a(IScoreboardCriteria.j, MathHelper.f((float) this.bX));
+            if (this.getArmorStrength() != this.bY) {
+                this.bY = this.getArmorStrength();
+                this.a(IScoreboardCriteria.j, MathHelper.f((float) this.bY));
             }
 
-            if (this.expTotal != this.bZ) {
-                this.bZ = this.expTotal;
-                this.a(IScoreboardCriteria.k, MathHelper.f((float) this.bZ));
+            if (this.expTotal != this.ca) {
+                this.ca = this.expTotal;
+                this.a(IScoreboardCriteria.k, MathHelper.f((float) this.ca));
             }
 
-            if (this.expLevel != this.bY) {
-                this.bY = this.expLevel;
-                this.a(IScoreboardCriteria.l, MathHelper.f((float) this.bY));
+            if (this.expLevel != this.bZ) {
+                this.bZ = this.expLevel;
+                this.a(IScoreboardCriteria.l, MathHelper.f((float) this.bZ));
             }
 
             if (this.expTotal != this.lastSentExp) {
@@ -304,10 +305,10 @@
         if (flag) {
             ScoreboardTeamBase scoreboardteambase = this.aO();
 
-            if (scoreboardteambase != null && scoreboardteambase.j() != ScoreboardTeamBase.EnumNameTagVisibility.ALWAYS) {
-                if (scoreboardteambase.j() == ScoreboardTeamBase.EnumNameTagVisibility.HIDE_FOR_OTHER_TEAMS) {
+            if (scoreboardteambase != null && scoreboardteambase.getDeathMessageVisibility() != ScoreboardTeamBase.EnumNameTagVisibility.ALWAYS) {
+                if (scoreboardteambase.getDeathMessageVisibility() == ScoreboardTeamBase.EnumNameTagVisibility.HIDE_FOR_OTHER_TEAMS) {
                     this.server.getPlayerList().a((EntityHuman) this, this.getCombatTracker().getDeathMessage());
-                } else if (scoreboardteambase.j() == ScoreboardTeamBase.EnumNameTagVisibility.HIDE_FOR_OWN_TEAM) {
+                } else if (scoreboardteambase.getDeathMessageVisibility() == ScoreboardTeamBase.EnumNameTagVisibility.HIDE_FOR_OWN_TEAM) {
                     this.server.getPlayerList().b((EntityHuman) this, this.getCombatTracker().getDeathMessage());
                 }
             } else {
@@ -329,16 +330,16 @@
             scoreboardscore.incrementScore();
         }
 
-        EntityLiving entityliving = this.bV();
+        EntityLiving entityliving = this.bW();
 
         if (entityliving != null) {
             EntityTypes.MonsterEggInfo entitytypes_monsteregginfo = (EntityTypes.MonsterEggInfo) EntityTypes.eggInfo.get(EntityTypes.b((Entity) entityliving));
 
             if (entitytypes_monsteregginfo != null) {
-                this.b(entitytypes_monsteregginfo.e);
+                this.b(entitytypes_monsteregginfo.killedByEntityStatistic);
             }
 
-            entityliving.b(this, this.ba);
+            entityliving.b(this, this.bb);
         }
 
         this.b(StatisticList.A);
@@ -384,8 +385,9 @@
         return this.server.getPVP();
     }
 
+    @Nullable
     public Entity c(int i) {
-        this.cj = true;
+        this.ck = true;
         if (this.dimension == 1 && i == 1) {
             this.world.kill(this);
             if (!this.viewingCredits) {
@@ -411,7 +413,7 @@
             this.playerConnection.sendPacket(new PacketPlayOutWorldEvent(1032, BlockPosition.ZERO, 0, false));
             this.lastSentExp = -1;
             this.lastHealthSent = -1.0F;
-            this.cb = -1;
+            this.cc = -1;
             return this;
         }
     }
@@ -422,10 +424,10 @@
 
     private void a(TileEntity tileentity) {
         if (tileentity != null) {
-            Packet packet = tileentity.getUpdatePacket();
+            PacketPlayOutTileEntityData packetplayouttileentitydata = tileentity.getUpdatePacket();
 
-            if (packet != null) {
-                this.playerConnection.sendPacket(packet);
+            if (packetplayouttileentitydata != null) {
+                this.playerConnection.sendPacket(packetplayouttileentitydata);
             }
         }
 
@@ -464,12 +466,12 @@
     }
 
     public boolean a(Entity entity, boolean flag) {
-        Entity entity1 = this.by();
+        Entity entity1 = this.bz();
 
         if (!super.a(entity, flag)) {
             return false;
         } else {
-            Entity entity2 = this.by();
+            Entity entity2 = this.bz();
 
             if (entity2 != entity1 && this.playerConnection != null) {
                 this.playerConnection.a(this.locX, this.locY, this.locZ, this.yaw, this.pitch);
@@ -480,10 +482,10 @@
     }
 
     public void stopRiding() {
-        Entity entity = this.by();
+        Entity entity = this.bz();
 
         super.stopRiding();
-        Entity entity1 = this.by();
+        Entity entity1 = this.bz();
 
         if (entity1 != entity && this.playerConnection != null) {
             this.playerConnection.a(this.locX, this.locY, this.locZ, this.yaw, this.pitch);
@@ -678,26 +680,26 @@
     public void a(float f, float f1, boolean flag, boolean flag1) {
         if (this.isPassenger()) {
             if (f >= -1.0F && f <= 1.0F) {
-                this.bd = f;
+                this.be = f;
             }
 
             if (f1 >= -1.0F && f1 <= 1.0F) {
-                this.be = f1;
+                this.bf = f1;
             }
 
-            this.bc = flag;
+            this.bd = flag;
             this.setSneaking(flag1);
         }
 
     }
 
     public boolean a(Achievement achievement) {
-        return this.bT.hasAchievement(achievement);
+        return this.bU.hasAchievement(achievement);
     }
 
     public void a(Statistic statistic, int i) {
         if (statistic != null) {
-            this.bT.b(this, statistic, i);
+            this.bU.b(this, statistic, i);
             Iterator iterator = this.getScoreboard().getObjectivesForCriteria(statistic.k()).iterator();
 
             while (iterator.hasNext()) {
@@ -706,8 +708,8 @@
                 this.getScoreboard().getPlayerScoreForObjective(this.getName(), scoreboardobjective).addScore(i);
             }
 
-            if (this.bT.e()) {
-                this.bT.a(this);
+            if (this.bU.e()) {
+                this.bU.a(this);
             }
 
         }
@@ -715,7 +717,7 @@
 
     public void a(Statistic statistic) {
         if (statistic != null) {
-            this.bT.setStatistic(this, statistic, 0);
+            this.bU.setStatistic(this, statistic, 0);
             Iterator iterator = this.getScoreboard().getObjectivesForCriteria(statistic.k()).iterator();
 
             while (iterator.hasNext()) {
@@ -724,8 +726,8 @@
                 this.getScoreboard().getPlayerScoreForObjective(this.getName(), scoreboardobjective).setScore(0);
             }
 
-            if (this.bT.e()) {
-                this.bT.a(this);
+            if (this.bU.e()) {
+                this.bU.a(this);
             }
 
         }
@@ -748,7 +750,7 @@
     }
 
     protected void v() {
-        if (this.bm != null && this.cs()) {
+        if (this.bn != null && this.ct()) {
             this.playerConnection.sendPacket(new PacketPlayOutEntityStatus(this, (byte) 9));
             super.v();
         }
@@ -759,7 +761,7 @@
         super.copyTo(entityhuman, flag);
         this.lastSentExp = -1;
         this.lastHealthSent = -1.0F;
-        this.cb = -1;
+        this.cc = -1;
         this.removeQueue.addAll(((EntityPlayer) entityhuman).removeQueue);
     }
 
@@ -811,7 +813,7 @@
         }
 
         this.updateAbilities();
-        this.cq();
+        this.cr();
     }
 
     public boolean isSpectator() {
@@ -852,14 +854,14 @@
 
     public void a(PacketPlayInSettings packetplayinsettings) {
         this.locale = packetplayinsettings.a();
-        this.cf = packetplayinsettings.c();
-        this.cg = packetplayinsettings.d();
-        this.getDataWatcher().set(EntityPlayer.bp, Byte.valueOf((byte) packetplayinsettings.e()));
-        this.getDataWatcher().set(EntityPlayer.bq, Byte.valueOf((byte) (packetplayinsettings.f() == EnumMainHand.LEFT ? 0 : 1)));
+        this.cg = packetplayinsettings.c();
+        this.ch = packetplayinsettings.d();
+        this.getDataWatcher().set(EntityPlayer.bq, Byte.valueOf((byte) packetplayinsettings.e()));
+        this.getDataWatcher().set(EntityPlayer.br, Byte.valueOf((byte) (packetplayinsettings.f() == EnumMainHand.LEFT ? 0 : 1)));
     }
 
     public EntityHuman.EnumChatVisibility getChatFlags() {
-        return this.cf;
+        return this.cg;
     }
 
     public void setResourcePack(String s, String s1) {
@@ -871,11 +873,11 @@
     }
 
     public void resetIdleTimer() {
-        this.ch = MinecraftServer.av();
+        this.ci = MinecraftServer.av();
     }
 
     public ServerStatisticManager getStatisticManager() {
-        return this.bT;
+        return this.bU;
     }
 
     public void c(Entity entity) {
@@ -893,7 +895,7 @@
 
     protected void F() {
         if (this.isSpectator()) {
-            this.bM();
+            this.bN();
             this.setInvisible(true);
         } else {
             super.F();
@@ -903,22 +905,22 @@
     }
 
     public Entity getSpecatorTarget() {
-        return (Entity) (this.ci == null ? this : this.ci);
+        return (Entity) (this.cj == null ? this : this.cj);
     }
 
     public void setSpectatorTarget(Entity entity) {
         Entity entity1 = this.getSpecatorTarget();
 
-        this.ci = (Entity) (entity == null ? this : entity);
-        if (entity1 != this.ci) {
-            this.playerConnection.sendPacket(new PacketPlayOutCamera(this.ci));
-            this.enderTeleportTo(this.ci.locX, this.ci.locY, this.ci.locZ);
+        this.cj = (Entity) (entity == null ? this : entity);
+        if (entity1 != this.cj) {
+            this.playerConnection.sendPacket(new PacketPlayOutCamera(this.cj));
+            this.enderTeleportTo(this.cj.locX, this.cj.locY, this.cj.locZ);
         }
 
     }
 
     protected void H() {
-        if (this.portalCooldown > 0 && !this.cj) {
+        if (this.portalCooldown > 0 && !this.ck) {
             --this.portalCooldown;
         }
 
@@ -934,24 +936,25 @@
     }
 
     public long I() {
-        return this.ch;
+        return this.ci;
     }
 
+    @Nullable
     public IChatBaseComponent getPlayerListName() {
         return null;
     }
 
     public void a(EnumHand enumhand) {
         super.a(enumhand);
-        this.cZ();
+        this.da();
     }
 
     public boolean K() {
-        return this.cj;
+        return this.ck;
     }
 
     public void L() {
-        this.cj = false;
+        this.ck = false;
     }
 
     public void M() {
