--- a/net/minecraft/server/EntityFallingBlock.java
+++ b/net/minecraft/server/EntityFallingBlock.java
@@ -3,6 +3,7 @@
 import com.google.common.collect.Lists;
 import java.util.ArrayList;
 import java.util.Iterator;
+import javax.annotation.Nullable;
 
 public class EntityFallingBlock extends Entity {
 
@@ -102,9 +103,7 @@
                                     TileEntity tileentity = this.world.getTileEntity(blockposition);
 
                                     if (tileentity != null) {
-                                        NBTTagCompound nbttagcompound = new NBTTagCompound();
-
-                                        tileentity.save(nbttagcompound);
+                                        NBTTagCompound nbttagcompound = tileentity.save(new NBTTagCompound());
                                         Iterator iterator = this.tileEntityData.c().iterator();
 
                                         while (iterator.hasNext()) {
@@ -238,11 +237,12 @@
 
     }
 
+    @Nullable
     public IBlockData getBlock() {
         return this.block;
     }
 
-    public boolean br() {
+    public boolean bs() {
         return true;
     }
 }
