--- a/net/minecraft/server/EntityItem.java
+++ b/net/minecraft/server/EntityItem.java
@@ -2,6 +2,7 @@
 
 import com.google.common.base.Optional;
 import java.util.Iterator;
+import javax.annotation.Nullable;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
@@ -294,6 +295,7 @@
         return false;
     }
 
+    @Nullable
     public Entity c(int i) {
         Entity entity = super.c(i);
 
@@ -318,7 +320,7 @@
         }
     }
 
-    public void setItemStack(ItemStack itemstack) {
+    public void setItemStack(@Nullable ItemStack itemstack) {
         this.getDataWatcher().set(EntityItem.c, Optional.fromNullable(itemstack));
         this.getDataWatcher().markDirty(EntityItem.c);
     }
