--- a/net/minecraft/server/EntityEnderCrystal.java
+++ b/net/minecraft/server/EntityEnderCrystal.java
@@ -1,6 +1,7 @@
 package net.minecraft.server;
 
 import com.google.common.base.Optional;
+import javax.annotation.Nullable;
 
 public class EntityEnderCrystal extends Entity {
 
@@ -45,20 +46,20 @@
     }
 
     protected void b(NBTTagCompound nbttagcompound) {
-        if (this.j() != null) {
-            nbttagcompound.set("BeamTarget", GameProfileSerializer.a(this.j()));
+        if (this.getBeamTarget() != null) {
+            nbttagcompound.set("BeamTarget", GameProfileSerializer.a(this.getBeamTarget()));
         }
 
-        nbttagcompound.setBoolean("ShowBottom", this.k());
+        nbttagcompound.setBoolean("ShowBottom", this.isShowingBottom());
     }
 
     protected void a(NBTTagCompound nbttagcompound) {
         if (nbttagcompound.hasKeyOfType("BeamTarget", 10)) {
-            this.a(GameProfileSerializer.c(nbttagcompound.getCompound("BeamTarget")));
+            this.setBeamTarget(GameProfileSerializer.c(nbttagcompound.getCompound("BeamTarget")));
         }
 
         if (nbttagcompound.hasKeyOfType("ShowBottom", 1)) {
-            this.a(nbttagcompound.getBoolean("ShowBottom"));
+            this.setShowingBottom(nbttagcompound.getBoolean("ShowBottom"));
         }
 
     }
@@ -102,19 +103,20 @@
 
     }
 
-    public void a(BlockPosition blockposition) {
+    public void setBeamTarget(@Nullable BlockPosition blockposition) {
         this.getDataWatcher().set(EntityEnderCrystal.b, Optional.fromNullable(blockposition));
     }
 
-    public BlockPosition j() {
+    @Nullable
+    public BlockPosition getBeamTarget() {
         return (BlockPosition) ((Optional) this.getDataWatcher().get(EntityEnderCrystal.b)).orNull();
     }
 
-    public void a(boolean flag) {
+    public void setShowingBottom(boolean flag) {
         this.getDataWatcher().set(EntityEnderCrystal.c, Boolean.valueOf(flag));
     }
 
-    public boolean k() {
+    public boolean isShowingBottom() {
         return ((Boolean) this.getDataWatcher().get(EntityEnderCrystal.c)).booleanValue();
     }
 }
