--- a/net/minecraft/server/ItemMonsterEgg.java
+++ b/net/minecraft/server/ItemMonsterEgg.java
@@ -1,6 +1,7 @@
 package net.minecraft.server;
 
 import java.util.UUID;
+import javax.annotation.Nullable;
 
 public class ItemMonsterEgg extends Item {
 
@@ -68,20 +69,18 @@
         }
     }
 
-    public static void a(World world, EntityHuman entityhuman, ItemStack itemstack, Entity entity) {
+    public static void a(World world, @Nullable EntityHuman entityhuman, ItemStack itemstack, @Nullable Entity entity) {
         MinecraftServer minecraftserver = world.getMinecraftServer();
 
         if (minecraftserver != null && entity != null) {
             NBTTagCompound nbttagcompound = itemstack.getTag();
 
             if (nbttagcompound != null && nbttagcompound.hasKeyOfType("EntityTag", 10)) {
-                if (!world.isClientSide && entity.br() && (entityhuman == null || !minecraftserver.getPlayerList().isOp(entityhuman.getProfile()))) {
+                if (!world.isClientSide && entity.bs() && (entityhuman == null || !minecraftserver.getPlayerList().isOp(entityhuman.getProfile()))) {
                     return;
                 }
 
-                NBTTagCompound nbttagcompound1 = new NBTTagCompound();
-
-                entity.e(nbttagcompound1);
+                NBTTagCompound nbttagcompound1 = entity.e(new NBTTagCompound());
                 UUID uuid = entity.getUniqueID();
 
                 nbttagcompound1.a(nbttagcompound.getCompound("EntityTag"));
@@ -130,7 +129,8 @@
         }
     }
 
-    public static Entity a(World world, String s, double d0, double d1, double d2) {
+    @Nullable
+    public static Entity a(World world, @Nullable String s, double d0, double d1, double d2) {
         if (s != null && EntityTypes.eggInfo.containsKey(s)) {
             Entity entity = null;
 
@@ -140,8 +140,8 @@
                     EntityInsentient entityinsentient = (EntityInsentient) entity;
 
                     entity.setPositionRotation(d0, d1, d2, MathHelper.g(world.random.nextFloat() * 360.0F), 0.0F);
-                    entityinsentient.aO = entityinsentient.yaw;
-                    entityinsentient.aM = entityinsentient.yaw;
+                    entityinsentient.aP = entityinsentient.yaw;
+                    entityinsentient.aN = entityinsentient.yaw;
                     entityinsentient.prepare(world.D(new BlockPosition(entityinsentient)), (GroupDataEntity) null);
                     world.addEntity(entity);
                     entityinsentient.D();
@@ -154,6 +154,7 @@
         }
     }
 
+    @Nullable
     public static String h(ItemStack itemstack) {
         NBTTagCompound nbttagcompound = itemstack.getTag();
 
