--- a/net/minecraft/server/ItemBow.java
+++ b/net/minecraft/server/ItemBow.java
@@ -1,5 +1,7 @@
 package net.minecraft.server;
 
+import javax.annotation.Nullable;
+
 public class ItemBow extends Item {
 
     public ItemBow() {
@@ -30,7 +32,7 @@
         }
     }
 
-    protected boolean h_(ItemStack itemstack) {
+    protected boolean h_(@Nullable ItemStack itemstack) {
         return itemstack != null && itemstack.getItem() instanceof ItemArrow;
     }
 
