--- a/net/minecraft/server/ItemWorldMapBase.java
+++ b/net/minecraft/server/ItemWorldMapBase.java
@@ -1,5 +1,7 @@
 package net.minecraft.server;
 
+import javax.annotation.Nullable;
+
 public class ItemWorldMapBase extends Item {
 
     protected ItemWorldMapBase() {}
@@ -8,6 +10,7 @@
         return true;
     }
 
+    @Nullable
     public Packet<?> a(ItemStack itemstack, World world, EntityHuman entityhuman) {
         return null;
     }
