--- a/net/minecraft/server/ItemPotion.java
+++ b/net/minecraft/server/ItemPotion.java
@@ -2,6 +2,7 @@
 
 import java.util.Iterator;
 import java.util.List;
+import javax.annotation.Nullable;
 
 public class ItemPotion extends Item {
 
@@ -10,6 +11,7 @@
         this.a(CreativeModeTab.k);
     }
 
+    @Nullable
     public ItemStack a(ItemStack itemstack, World world, EntityLiving entityliving) {
         EntityHuman entityhuman = entityliving instanceof EntityHuman ? (EntityHuman) entityliving : null;
 
