--- a/net/minecraft/server/ItemNameTag.java
+++ b/net/minecraft/server/ItemNameTag.java
@@ -13,7 +13,7 @@
             EntityInsentient entityinsentient = (EntityInsentient) entityliving;
 
             entityinsentient.setCustomName(itemstack.getName());
-            entityinsentient.cL();
+            entityinsentient.cM();
             --itemstack.count;
             return true;
         } else {
