--- a/net/minecraft/server/IteratorUtils.java
+++ b/net/minecraft/server/IteratorUtils.java
@@ -11,6 +11,7 @@
 import java.util.Iterator;
 import java.util.List;
 import java.util.NoSuchElementException;
+import javax.annotation.Nullable;
 
 public class IteratorUtils {
 
@@ -158,7 +159,7 @@
 
         private ArrayToList() {}
 
-        public List<T> a(Object[] aobject) {
+        public List<T> a(@Nullable Object[] aobject) {
             return Arrays.asList((Object[]) aobject);
         }
 
