--- a/net/minecraft/server/IWorldAccess.java
+++ b/net/minecraft/server/IWorldAccess.java
@@ -1,5 +1,7 @@
 package net.minecraft.server;
 
+import javax.annotation.Nullable;
+
 public interface IWorldAccess {
 
     void a(World world, BlockPosition blockposition, IBlockData iblockdata, IBlockData iblockdata1, int i);
@@ -8,7 +10,7 @@
 
     void a(int i, int j, int k, int l, int i1, int j1);
 
-    void a(EntityHuman entityhuman, SoundEffect soundeffect, SoundCategory soundcategory, double d0, double d1, double d2, float f, float f1);
+    void a(@Nullable EntityHuman entityhuman, SoundEffect soundeffect, SoundCategory soundcategory, double d0, double d1, double d2, float f, float f1);
 
     void a(SoundEffect soundeffect, BlockPosition blockposition);
 
