--- a/net/minecraft/server/PacketPrepender.java
+++ b/net/minecraft/server/PacketPrepender.java
@@ -2,8 +2,10 @@
 
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelHandler.Sharable;
 import io.netty.handler.codec.MessageToByteEncoder;
 
+@Sharable
 public class PacketPrepender extends MessageToByteEncoder<ByteBuf> {
 
     public PacketPrepender() {}
