--- a/net/minecraft/server/PacketListenerPlayOut.java
+++ b/net/minecraft/server/PacketListenerPlayOut.java
@@ -116,8 +116,6 @@
 
     void a(PacketPlayOutUpdateTime packetplayoutupdatetime);
 
-    void a(PacketPlayOutUpdateSign packetplayoutupdatesign);
-
     void a(PacketPlayOutNamedSoundEffect packetplayoutnamedsoundeffect);
 
     void a(PacketPlayOutCustomSoundEffect packetplayoutcustomsoundeffect);
