--- a/net/minecraft/server/PathfinderGoalTakeFlower.java
+++ b/net/minecraft/server/PathfinderGoalTakeFlower.java
@@ -31,7 +31,7 @@
                 while (iterator.hasNext()) {
                     EntityIronGolem entityirongolem = (EntityIronGolem) iterator.next();
 
-                    if (entityirongolem.da() > 0) {
+                    if (entityirongolem.db() > 0) {
                         this.b = entityirongolem;
                         break;
                     }
@@ -43,7 +43,7 @@
     }
 
     public boolean b() {
-        return this.b.da() > 0;
+        return this.b.db() > 0;
     }
 
     public void c() {
@@ -59,7 +59,7 @@
 
     public void e() {
         this.a.getControllerLook().a(this.b, 30.0F, 30.0F);
-        if (this.b.da() == this.c) {
+        if (this.b.db() == this.c) {
             this.a.getNavigation().a((Entity) this.b, 0.5D);
             this.d = true;
         }
