--- a/net/minecraft/server/RecipeRepair.java
+++ b/net/minecraft/server/RecipeRepair.java
@@ -2,6 +2,7 @@
 
 import com.google.common.collect.Lists;
 import java.util.ArrayList;
+import javax.annotation.Nullable;
 
 public class RecipeRepair implements IRecipe {
 
@@ -28,6 +29,7 @@
         return arraylist.size() == 2;
     }
 
+    @Nullable
     public ItemStack craftItem(InventoryCrafting inventorycrafting) {
         ArrayList arraylist = Lists.newArrayList();
 
@@ -73,6 +75,7 @@
         return 4;
     }
 
+    @Nullable
     public ItemStack b() {
         return null;
     }
