--- a/net/minecraft/server/RecipeArmorDye.java
+++ b/net/minecraft/server/RecipeArmorDye.java
@@ -2,6 +2,7 @@
 
 import com.google.common.collect.Lists;
 import java.util.ArrayList;
+import javax.annotation.Nullable;
 
 public class RecipeArmorDye implements IRecipe {
 
@@ -36,6 +37,7 @@
         return itemstack != null && !arraylist.isEmpty();
     }
 
+    @Nullable
     public ItemStack craftItem(InventoryCrafting inventorycrafting) {
         ItemStack itemstack = null;
         int[] aint = new int[3];
@@ -115,6 +117,7 @@
         return 10;
     }
 
+    @Nullable
     public ItemStack b() {
         return null;
     }
