--- a/net/minecraft/server/RegistryDefault.java
+++ b/net/minecraft/server/RegistryDefault.java
@@ -1,5 +1,8 @@
 package net.minecraft.server;
 
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
 public class RegistryDefault<K, V> extends RegistrySimple<K, V> {
 
     private final V a;
@@ -8,7 +11,8 @@
         this.a = v0;
     }
 
-    public V get(K k0) {
+    @Nonnull
+    public V get(@Nullable K k0) {
         Object object = super.get(k0);
 
         return object == null ? this.a : object;
