--- a/net/minecraft/server/SlotFurnaceFuel.java
+++ b/net/minecraft/server/SlotFurnaceFuel.java
@@ -1,12 +1,14 @@
 package net.minecraft.server;
 
+import javax.annotation.Nullable;
+
 public class SlotFurnaceFuel extends Slot {
 
     public SlotFurnaceFuel(IInventory iinventory, int i, int j, int k) {
         super(iinventory, i, j, k);
     }
 
-    public boolean isAllowed(ItemStack itemstack) {
+    public boolean isAllowed(@Nullable ItemStack itemstack) {
         return TileEntityFurnace.isFuel(itemstack) || d_(itemstack);
     }
 
