--- a/net/minecraft/server/SystemUtils.java
+++ b/net/minecraft/server/SystemUtils.java
@@ -3,10 +3,12 @@
 import java.util.List;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.FutureTask;
+import javax.annotation.Nullable;
 import org.apache.logging.log4j.Logger;
 
 public class SystemUtils {
 
+    @Nullable
     public static <V> V a(FutureTask<V> futuretask, Logger logger) {
         try {
             futuretask.run();
