--- a/net/minecraft/server/TileEntity.java
+++ b/net/minecraft/server/TileEntity.java
@@ -2,7 +2,7 @@
 
 import com.google.common.collect.Maps;
 import java.util.Map;
-import java.util.concurrent.Callable;
+import javax.annotation.Nullable;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
@@ -47,7 +47,11 @@
         this.position = new BlockPosition(nbttagcompound.getInt("x"), nbttagcompound.getInt("y"), nbttagcompound.getInt("z"));
     }
 
-    public void save(NBTTagCompound nbttagcompound) {
+    public NBTTagCompound save(NBTTagCompound nbttagcompound) {
+        return this.d(nbttagcompound);
+    }
+
+    private NBTTagCompound d(NBTTagCompound nbttagcompound) {
         String s = (String) TileEntity.g.get(this.getClass());
 
         if (s == null) {
@@ -57,10 +61,11 @@
             nbttagcompound.setInt("x", this.position.getX());
             nbttagcompound.setInt("y", this.position.getY());
             nbttagcompound.setInt("z", this.position.getZ());
+            return nbttagcompound;
         }
     }
 
-    public static TileEntity a(MinecraftServer minecraftserver, NBTTagCompound nbttagcompound) {
+    public static TileEntity c(NBTTagCompound nbttagcompound) {
         TileEntity tileentity = null;
         String s = nbttagcompound.getString("id");
 
@@ -123,10 +128,15 @@
         return this.e;
     }
 
-    public Packet<?> getUpdatePacket() {
+    @Nullable
+    public PacketPlayOutTileEntityData getUpdatePacket() {
         return null;
     }
 
+    public NBTTagCompound E_() {
+        return this.d(new NBTTagCompound());
+    }
+
     public boolean x() {
         return this.d;
     }
@@ -149,7 +159,7 @@
     }
 
     public void a(CrashReportSystemDetails crashreportsystemdetails) {
-        crashreportsystemdetails.a("Name", new Callable() {
+        crashreportsystemdetails.a("Name", new CrashReportCallable() {
             public String a() throws Exception {
                 return (String) TileEntity.g.get(TileEntity.this.getClass()) + " // " + TileEntity.this.getClass().getCanonicalName();
             }
@@ -160,7 +170,7 @@
         });
         if (this.world != null) {
             CrashReportSystemDetails.a(crashreportsystemdetails, this.position, this.getBlock(), this.u());
-            crashreportsystemdetails.a("Actual block type", new Callable() {
+            crashreportsystemdetails.a("Actual block type", new CrashReportCallable() {
                 public String a() throws Exception {
                     int i = Block.getId(TileEntity.this.world.getType(TileEntity.this.position).getBlock());
 
@@ -175,7 +185,7 @@
                     return this.a();
                 }
             });
-            crashreportsystemdetails.a("Actual block data value", new Callable() {
+            crashreportsystemdetails.a("Actual block data value", new CrashReportCallable() {
                 public String a() throws Exception {
                     IBlockData iblockdata = TileEntity.this.world.getType(TileEntity.this.position);
                     int i = iblockdata.getBlock().toLegacyData(iblockdata);
