--- a/net/minecraft/server/WorldNBTStorage.java
+++ b/net/minecraft/server/WorldNBTStorage.java
@@ -8,6 +8,7 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
+import javax.annotation.Nullable;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
@@ -98,7 +99,7 @@
         return file.exists() ? WorldLoader.a(file, this.a) : null;
     }
 
-    public void saveWorldData(WorldData worlddata, NBTTagCompound nbttagcompound) {
+    public void saveWorldData(WorldData worlddata, @Nullable NBTTagCompound nbttagcompound) {
         NBTTagCompound nbttagcompound1 = worlddata.a(nbttagcompound);
         NBTTagCompound nbttagcompound2 = new NBTTagCompound();
 
@@ -135,11 +136,9 @@
 
     public void save(EntityHuman entityhuman) {
         try {
-            NBTTagCompound nbttagcompound = new NBTTagCompound();
-
-            entityhuman.e(nbttagcompound);
-            File file = new File(this.playerDir, entityhuman.getUniqueID().toString() + ".dat.tmp");
-            File file1 = new File(this.playerDir, entityhuman.getUniqueID().toString() + ".dat");
+            NBTTagCompound nbttagcompound = entityhuman.e(new NBTTagCompound());
+            File file = new File(this.playerDir, entityhuman.bd() + ".dat.tmp");
+            File file1 = new File(this.playerDir, entityhuman.bd() + ".dat");
 
             NBTCompressedStreamTools.a(nbttagcompound, (OutputStream) (new FileOutputStream(file)));
             if (file1.exists()) {
@@ -157,7 +156,7 @@
         NBTTagCompound nbttagcompound = null;
 
         try {
-            File file = new File(this.playerDir, entityhuman.getUniqueID().toString() + ".dat");
+            File file = new File(this.playerDir, entityhuman.bd() + ".dat");
 
             if (file.exists() && file.isFile()) {
                 nbttagcompound = NBTCompressedStreamTools.a((InputStream) (new FileInputStream(file)));
