--- a/net/minecraft/server/WorldGenLiquids.java
+++ b/net/minecraft/server/WorldGenLiquids.java
@@ -15,51 +15,57 @@
             return false;
         } else if (world.getType(blockposition.down()).getBlock() != Blocks.STONE) {
             return false;
-        } else if (world.getType(blockposition).getMaterial() != Material.AIR && world.getType(blockposition).getBlock() != Blocks.STONE) {
-            return false;
         } else {
-            int i = 0;
+            IBlockData iblockdata = world.getType(blockposition);
 
-            if (world.getType(blockposition.west()).getBlock() == Blocks.STONE) {
-                ++i;
-            }
+            if (iblockdata.getMaterial() != Material.AIR && iblockdata.getBlock() != Blocks.STONE) {
+                return false;
+            } else {
+                int i = 0;
 
-            if (world.getType(blockposition.east()).getBlock() == Blocks.STONE) {
-                ++i;
-            }
+                if (world.getType(blockposition.west()).getBlock() == Blocks.STONE) {
+                    ++i;
+                }
 
-            if (world.getType(blockposition.north()).getBlock() == Blocks.STONE) {
-                ++i;
-            }
+                if (world.getType(blockposition.east()).getBlock() == Blocks.STONE) {
+                    ++i;
+                }
 
-            if (world.getType(blockposition.south()).getBlock() == Blocks.STONE) {
-                ++i;
-            }
+                if (world.getType(blockposition.north()).getBlock() == Blocks.STONE) {
+                    ++i;
+                }
 
-            int j = 0;
+                if (world.getType(blockposition.south()).getBlock() == Blocks.STONE) {
+                    ++i;
+                }
 
-            if (world.isEmpty(blockposition.west())) {
-                ++j;
-            }
+                int j = 0;
 
-            if (world.isEmpty(blockposition.east())) {
-                ++j;
-            }
+                if (world.isEmpty(blockposition.west())) {
+                    ++j;
+                }
 
-            if (world.isEmpty(blockposition.north())) {
-                ++j;
-            }
+                if (world.isEmpty(blockposition.east())) {
+                    ++j;
+                }
 
-            if (world.isEmpty(blockposition.south())) {
-                ++j;
-            }
+                if (world.isEmpty(blockposition.north())) {
+                    ++j;
+                }
 
-            if (i == 3 && j == 1) {
-                world.setTypeAndData(blockposition, this.a.getBlockData(), 2);
-                world.a(this.a, blockposition, random);
-            }
+                if (world.isEmpty(blockposition.south())) {
+                    ++j;
+                }
 
-            return true;
+                if (i == 3 && j == 1) {
+                    IBlockData iblockdata1 = this.a.getBlockData();
+
+                    world.setTypeAndData(blockposition, iblockdata1, 2);
+                    world.a(blockposition, iblockdata1, random);
+                }
+
+                return true;
+            }
         }
     }
 }
