--- a/net/minecraft/server/CommandList.java
+++ b/net/minecraft/server/CommandList.java
@@ -17,9 +17,9 @@
     }
 
     public void execute(ICommandListener icommandlistener, String[] astring) throws CommandException {
-        int i = MinecraftServer.getServer().H();
+        int i = MinecraftServer.getServer().I();
 
-        icommandlistener.sendMessage(new ChatMessage("commands.players.list", new Object[] { Integer.valueOf(i), Integer.valueOf(MinecraftServer.getServer().I())}));
+        icommandlistener.sendMessage(new ChatMessage("commands.players.list", new Object[] { Integer.valueOf(i), Integer.valueOf(MinecraftServer.getServer().J())}));
         icommandlistener.sendMessage(new ChatComponentText(MinecraftServer.getServer().getPlayerList().b(astring.length > 0 && "uuids".equalsIgnoreCase(astring[0]))));
         icommandlistener.a(CommandObjectiveExecutor.EnumCommandResult.QUERY_RESULT, i);
     }
