--- a/net/minecraft/server/CommandSeed.java
+++ b/net/minecraft/server/CommandSeed.java
@@ -5,7 +5,7 @@
     public CommandSeed() {}
 
     public boolean canUse(ICommandListener icommandlistener) {
-        return MinecraftServer.getServer().S() || super.canUse(icommandlistener);
+        return MinecraftServer.getServer().T() || super.canUse(icommandlistener);
     }
 
     public String getCommand() {
