--- a/net/minecraft/server/ContainerEnchantTable.java
+++ b/net/minecraft/server/ContainerEnchantTable.java
@@ -173,7 +173,7 @@
                 boolean flag = itemstack.getItem() == Items.BOOK;
 
                 if (list != null) {
-                    entityhuman.b(j);
+                    entityhuman.enchantDone(j);
                     if (flag) {
                         itemstack.setItem(Items.ENCHANTED_BOOK);
                     }
