--- a/net/minecraft/server/ControllerMove.java
+++ b/net/minecraft/server/ControllerMove.java
@@ -46,7 +46,7 @@
                 float f = (float) (MathHelper.b(d1, d0) * 180.0D / 3.1415927410125732D) - 90.0F;
 
                 this.a.yaw = this.a(this.a.yaw, f, 30.0F);
-                this.a.k((float) (this.e * this.a.getAttributeInstance(GenericAttributes.d).getValue()));
+                this.a.k((float) (this.e * this.a.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).getValue()));
                 if (d2 > 0.0D && d0 * d0 + d1 * d1 < 1.0D) {
                     this.a.getControllerJump().a();
                 }
