--- a/net/minecraft/server/CraftingManager.java
+++ b/net/minecraft/server/CraftingManager.java
@@ -12,13 +12,13 @@
 public class CraftingManager {
 
     private static final CraftingManager a = new CraftingManager();
-    private final List<IRecipe> recipes = Lists.newArrayList();
+    public List<IRecipe> recipes = Lists.newArrayList();
 
     public static CraftingManager getInstance() {
         return CraftingManager.a;
     }
 
-    private CraftingManager() {
+    public CraftingManager() {
         (new RecipesTools()).a(this);
         (new RecipesWeapons()).a(this);
         (new RecipeIngots()).a(this);
@@ -280,7 +280,7 @@
             irecipe = (IRecipe) iterator.next();
         } while (!irecipe.a(inventorycrafting, world));
 
-        return irecipe.a(inventorycrafting);
+        return irecipe.craftItem(inventorycrafting);
     }
 
     public ItemStack[] b(InventoryCrafting inventorycrafting, World world) {
