--- a/net/minecraft/server/DedicatedPlayerList.java
+++ b/net/minecraft/server/DedicatedPlayerList.java
@@ -14,7 +14,7 @@
         this.a(dedicatedserver.a("view-distance", 10));
         this.maxPlayers = dedicatedserver.a("max-players", 20);
         this.setHasWhitelist(dedicatedserver.a("white-list", false));
-        if (!dedicatedserver.S()) {
+        if (!dedicatedserver.T()) {
             this.getProfileBans().a(true);
             this.getIPBans().a(true);
         }
@@ -142,6 +142,10 @@
         return (DedicatedServer) super.getServer();
     }
 
+    public boolean f(GameProfile gameprofile) {
+        return this.getOPs().b(gameprofile);
+    }
+
     public MinecraftServer getServer() {
         return this.getServer();
     }
