--- a/net/minecraft/server/DispenseBehaviorProjectile.java
+++ b/net/minecraft/server/DispenseBehaviorProjectile.java
@@ -5,19 +5,19 @@
     public DispenseBehaviorProjectile() {}
 
     public ItemStack b(ISourceBlock isourceblock, ItemStack itemstack) {
-        World world = isourceblock.i();
+        World world = isourceblock.getWorld();
         IPosition iposition = BlockDispenser.a(isourceblock);
         EnumDirection enumdirection = BlockDispenser.b(isourceblock.f());
         IProjectile iprojectile = this.a(world, iposition);
 
-        iprojectile.shoot((double) enumdirection.getAdjacentX(), (double) ((float) enumdirection.getAdjacentY() + 0.1F), (double) enumdirection.getAdjacentZ(), this.b(), this.a());
+        iprojectile.shoot((double) enumdirection.getAdjacentX(), (double) ((float) enumdirection.getAdjacentY() + 0.1F), (double) enumdirection.getAdjacentZ(), this.getPower(), this.a());
         world.addEntity((Entity) iprojectile);
-        itemstack.a(1);
+        itemstack.cloneAndSubtract(1);
         return itemstack;
     }
 
     protected void a(ISourceBlock isourceblock) {
-        isourceblock.i().triggerEffect(1002, isourceblock.getBlockPosition(), 0);
+        isourceblock.getWorld().triggerEffect(1002, isourceblock.getBlockPosition(), 0);
     }
 
     protected abstract IProjectile a(World world, IPosition iposition);
@@ -26,7 +26,7 @@
         return 6.0F;
     }
 
-    protected float b() {
+    protected float getPower() {
         return 1.1F;
     }
 }
