--- a/net/minecraft/server/EntityPigZombie.java
+++ b/net/minecraft/server/EntityPigZombie.java
@@ -6,7 +6,7 @@
 
     private static final UUID b = UUID.fromString("49455A49-7EC5-45BA-B886-3B90B23A1718");
     private static final AttributeModifier c = (new AttributeModifier(EntityPigZombie.b, "Attacking speed boost", 0.05D, 0)).a(false);
-    private int angerLevel;
+    public int angerLevel;
     private int soundDelay;
     private UUID hurtBy;
 
@@ -31,8 +31,8 @@
     protected void initAttributes() {
         super.initAttributes();
         this.getAttributeInstance(EntityPigZombie.a).setValue(0.0D);
-        this.getAttributeInstance(GenericAttributes.d).setValue(0.23000000417232513D);
-        this.getAttributeInstance(GenericAttributes.e).setValue(5.0D);
+        this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).setValue(0.23000000417232513D);
+        this.getAttributeInstance(GenericAttributes.ATTACK_DAMAGE).setValue(5.0D);
     }
 
     public void t_() {
@@ -40,7 +40,7 @@
     }
 
     protected void E() {
-        AttributeInstance attributeinstance = this.getAttributeInstance(GenericAttributes.d);
+        AttributeInstance attributeinstance = this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED);
 
         if (this.cm()) {
             if (!this.isBaby() && !attributeinstance.a(EntityPigZombie.c)) {
