--- a/net/minecraft/server/EntityEnderman.java
+++ b/net/minecraft/server/EntityEnderman.java
@@ -42,9 +42,9 @@
     protected void initAttributes() {
         super.initAttributes();
         this.getAttributeInstance(GenericAttributes.maxHealth).setValue(40.0D);
-        this.getAttributeInstance(GenericAttributes.d).setValue(0.30000001192092896D);
-        this.getAttributeInstance(GenericAttributes.e).setValue(7.0D);
-        this.getAttributeInstance(GenericAttributes.b).setValue(64.0D);
+        this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).setValue(0.30000001192092896D);
+        this.getAttributeInstance(GenericAttributes.ATTACK_DAMAGE).setValue(7.0D);
+        this.getAttributeInstance(GenericAttributes.FOLLOW_RANGE).setValue(64.0D);
     }
 
     protected void h() {
@@ -407,7 +407,7 @@
         public void d() {
             this.g = null;
             this.j.a(false);
-            AttributeInstance attributeinstance = this.j.getAttributeInstance(GenericAttributes.d);
+            AttributeInstance attributeinstance = this.j.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED);
 
             attributeinstance.c(EntityEnderman.b);
             super.d();
@@ -435,7 +435,7 @@
                     super.c();
                     this.j.makeSound("mob.endermen.stare", 1.0F, 1.0F);
                     this.j.a(true);
-                    AttributeInstance attributeinstance = this.j.getAttributeInstance(GenericAttributes.d);
+                    AttributeInstance attributeinstance = this.j.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED);
 
                     attributeinstance.b(EntityEnderman.b);
                 }
