--- a/net/minecraft/server/EntityMinecartContainer.java
+++ b/net/minecraft/server/EntityMinecartContainer.java
@@ -34,7 +34,7 @@
                 this.items[i] = null;
                 return itemstack;
             } else {
-                itemstack = this.items[i].a(j);
+                itemstack = this.items[i].cloneAndSubtract(j);
                 if (this.items[i].count == 0) {
                     this.items[i] = null;
                 }
