--- a/net/minecraft/server/EntityWither.java
+++ b/net/minecraft/server/EntityWither.java
@@ -432,8 +432,8 @@
     protected void initAttributes() {
         super.initAttributes();
         this.getAttributeInstance(GenericAttributes.maxHealth).setValue(300.0D);
-        this.getAttributeInstance(GenericAttributes.d).setValue(0.6000000238418579D);
-        this.getAttributeInstance(GenericAttributes.b).setValue(40.0D);
+        this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).setValue(0.6000000238418579D);
+        this.getAttributeInstance(GenericAttributes.FOLLOW_RANGE).setValue(40.0D);
     }
 
     public int cl() {
