--- a/net/minecraft/server/EntityMonster.java
+++ b/net/minecraft/server/EntityMonster.java
@@ -59,7 +59,7 @@
     }
 
     public boolean r(Entity entity) {
-        float f = (float) this.getAttributeInstance(GenericAttributes.e).getValue();
+        float f = (float) this.getAttributeInstance(GenericAttributes.ATTACK_DAMAGE).getValue();
         int i = 0;
 
         if (entity instanceof EntityLiving) {
@@ -118,7 +118,7 @@
 
     protected void initAttributes() {
         super.initAttributes();
-        this.getAttributeMap().b(GenericAttributes.e);
+        this.getAttributeMap().b(GenericAttributes.ATTACK_DAMAGE);
     }
 
     protected boolean ba() {
