--- a/net/minecraft/server/EntityMagmaCube.java
+++ b/net/minecraft/server/EntityMagmaCube.java
@@ -9,7 +9,7 @@
 
     protected void initAttributes() {
         super.initAttributes();
-        this.getAttributeInstance(GenericAttributes.d).setValue(0.20000000298023224D);
+        this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).setValue(0.20000000298023224D);
     }
 
     public boolean bR() {
