--- a/net/minecraft/server/EntityZombie.java
+++ b/net/minecraft/server/EntityZombie.java
@@ -40,9 +40,9 @@
 
     protected void initAttributes() {
         super.initAttributes();
-        this.getAttributeInstance(GenericAttributes.b).setValue(35.0D);
-        this.getAttributeInstance(GenericAttributes.d).setValue(0.23000000417232513D);
-        this.getAttributeInstance(GenericAttributes.e).setValue(3.0D);
+        this.getAttributeInstance(GenericAttributes.FOLLOW_RANGE).setValue(35.0D);
+        this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).setValue(0.23000000417232513D);
+        this.getAttributeInstance(GenericAttributes.ATTACK_DAMAGE).setValue(3.0D);
         this.getAttributeMap().b(EntityZombie.a).setValue(this.random.nextDouble() * 0.10000000149011612D);
     }
 
@@ -94,7 +94,7 @@
     public void setBaby(boolean flag) {
         this.getDataWatcher().watch(12, Byte.valueOf((byte) (flag ? 1 : 0)));
         if (this.world != null && !this.world.isClientSide) {
-            AttributeInstance attributeinstance = this.getAttributeInstance(GenericAttributes.d);
+            AttributeInstance attributeinstance = this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED);
 
             attributeinstance.c(EntityZombie.c);
             if (flag) {
@@ -397,7 +397,7 @@
         double d0 = this.random.nextDouble() * 1.5D * (double) f;
 
         if (d0 > 1.0D) {
-            this.getAttributeInstance(GenericAttributes.b).b(new AttributeModifier("Random zombie-spawn bonus", d0, 2));
+            this.getAttributeInstance(GenericAttributes.FOLLOW_RANGE).b(new AttributeModifier("Random zombie-spawn bonus", d0, 2));
         }
 
         if (this.random.nextFloat() < f * 0.05F) {
