--- a/net/minecraft/server/EntitySkeleton.java
+++ b/net/minecraft/server/EntitySkeleton.java
@@ -27,7 +27,7 @@
 
     protected void initAttributes() {
         super.initAttributes();
-        this.getAttributeInstance(GenericAttributes.d).setValue(0.25D);
+        this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).setValue(0.25D);
     }
 
     protected void h() {
@@ -176,7 +176,7 @@
             this.goalSelector.a(4, this.b);
             this.setSkeletonType(1);
             this.setEquipment(0, new ItemStack(Items.STONE_SWORD));
-            this.getAttributeInstance(GenericAttributes.e).setValue(4.0D);
+            this.getAttributeInstance(GenericAttributes.ATTACK_DAMAGE).setValue(4.0D);
         } else {
             this.goalSelector.a(4, this.a);
             this.a(difficultydamagescaler);
