--- a/net/minecraft/server/InventorySubcontainer.java
+++ b/net/minecraft/server/InventorySubcontainer.java
@@ -7,7 +7,7 @@
 
     private String a;
     private int b;
-    private ItemStack[] items;
+    public ItemStack[] items;
     private List<IInventoryListener> d;
     private boolean e;
 
@@ -44,7 +44,7 @@
                 this.update();
                 return itemstack;
             } else {
-                itemstack = this.items[i].a(j);
+                itemstack = this.items[i].cloneAndSubtract(j);
                 if (this.items[i].count == 0) {
                     this.items[i] = null;
                 }
