--- a/net/minecraft/server/ItemBlock.java
+++ b/net/minecraft/server/ItemBlock.java
@@ -32,7 +32,7 @@
             if (world.setTypeAndData(blockposition, iblockdata1, 3)) {
                 iblockdata1 = world.getType(blockposition);
                 if (iblockdata1.getBlock() == this.a) {
-                    a(world, blockposition, itemstack);
+                    a(world, entityhuman, blockposition, itemstack);
                     this.a.postPlace(world, blockposition, iblockdata1, entityhuman, itemstack);
                 }
 
@@ -46,30 +46,40 @@
         }
     }
 
-    public static boolean a(World world, BlockPosition blockposition, ItemStack itemstack) {
-        if (itemstack.hasTag() && itemstack.getTag().hasKeyOfType("BlockEntityTag", 10)) {
-            TileEntity tileentity = world.getTileEntity(blockposition);
-
-            if (tileentity != null) {
-                NBTTagCompound nbttagcompound = new NBTTagCompound();
-                NBTTagCompound nbttagcompound1 = (NBTTagCompound) nbttagcompound.clone();
-
-                tileentity.b(nbttagcompound);
-                NBTTagCompound nbttagcompound2 = (NBTTagCompound) itemstack.getTag().get("BlockEntityTag");
-
-                nbttagcompound.a(nbttagcompound2);
-                nbttagcompound.setInt("x", blockposition.getX());
-                nbttagcompound.setInt("y", blockposition.getY());
-                nbttagcompound.setInt("z", blockposition.getZ());
-                if (!nbttagcompound.equals(nbttagcompound1)) {
-                    tileentity.a(nbttagcompound);
-                    tileentity.update();
-                    return true;
+    public static boolean a(World world, EntityHuman entityhuman, BlockPosition blockposition, ItemStack itemstack) {
+        MinecraftServer minecraftserver = MinecraftServer.getServer();
+
+        if (minecraftserver == null) {
+            return false;
+        } else {
+            if (itemstack.hasTag() && itemstack.getTag().hasKeyOfType("BlockEntityTag", 10)) {
+                TileEntity tileentity = world.getTileEntity(blockposition);
+
+                if (tileentity != null) {
+                    if (!world.isClientSide && tileentity.F() && !minecraftserver.getPlayerList().isOp(entityhuman.getProfile())) {
+                        return false;
+                    }
+
+                    NBTTagCompound nbttagcompound = new NBTTagCompound();
+                    NBTTagCompound nbttagcompound1 = (NBTTagCompound) nbttagcompound.clone();
+
+                    tileentity.b(nbttagcompound);
+                    NBTTagCompound nbttagcompound2 = (NBTTagCompound) itemstack.getTag().get("BlockEntityTag");
+
+                    nbttagcompound.a(nbttagcompound2);
+                    nbttagcompound.setInt("x", blockposition.getX());
+                    nbttagcompound.setInt("y", blockposition.getY());
+                    nbttagcompound.setInt("z", blockposition.getZ());
+                    if (!nbttagcompound.equals(nbttagcompound1)) {
+                        tileentity.a(nbttagcompound);
+                        tileentity.update();
+                        return true;
+                    }
                 }
             }
-        }
 
-        return false;
+            return false;
+        }
     }
 
     public String e_(ItemStack itemstack) {
