--- a/net/minecraft/server/MojangStatisticsGenerator.java
+++ b/net/minecraft/server/MojangStatisticsGenerator.java
@@ -71,7 +71,7 @@
         this.b("os_version", System.getProperty("os.version"));
         this.b("os_architecture", System.getProperty("os.arch"));
         this.b("java_version", System.getProperty("java.version"));
-        this.a("version", "1.8.3");
+        this.a("version", "1.8.8");
         this.e.b(this);
     }
 
