--- a/net/minecraft/server/PathfinderGoalNearestAttackableTargetInsentient.java
+++ b/net/minecraft/server/PathfinderGoalNearestAttackableTargetInsentient.java
@@ -78,7 +78,7 @@
     }
 
     protected double f() {
-        AttributeInstance attributeinstance = this.b.getAttributeInstance(GenericAttributes.b);
+        AttributeInstance attributeinstance = this.b.getAttributeInstance(GenericAttributes.FOLLOW_RANGE);
 
         return attributeinstance == null ? 16.0D : attributeinstance.getValue();
     }
