--- a/net/minecraft/server/PathfinderGoalFollowOwner.java
+++ b/net/minecraft/server/PathfinderGoalFollowOwner.java
@@ -30,6 +30,8 @@
 
         if (entityliving == null) {
             return false;
+        } else if (entityliving instanceof EntityHuman && ((EntityHuman) entityliving).isSpectator()) {
+            return false;
         } else if (this.d.isSitting()) {
             return false;
         } else if (this.d.h(entityliving) < (double) (this.c * this.c)) {
