--- a/net/minecraft/server/PlayerInventory.java
+++ b/net/minecraft/server/PlayerInventory.java
@@ -264,7 +264,7 @@
                 aitemstack[i] = null;
                 return itemstack;
             } else {
-                itemstack = aitemstack[i].a(j);
+                itemstack = aitemstack[i].cloneAndSubtract(j);
                 if (aitemstack[i].count == 0) {
                     aitemstack[i] = null;
                 }
