--- a/net/minecraft/server/PortalTravelAgent.java
+++ b/net/minecraft/server/PortalTravelAgent.java
@@ -109,11 +109,9 @@
             }
 
             if (shapedetector_shapedetectorcollection.b().k() == EnumDirection.EnumAxis.X) {
-                d2 += (double) shapedetector_shapedetectorcollection.b().getAdjacentX();
                 d4 = d5 + (1.0D - entity.aG().a) * (double) shapedetector_shapedetectorcollection.d() * (double) shapedetector_shapedetectorcollection.b().e().c().a();
             } else {
                 d2 = d5 + (1.0D - entity.aG().a) * (double) shapedetector_shapedetectorcollection.d() * (double) shapedetector_shapedetectorcollection.b().e().c().a();
-                d4 += (double) shapedetector_shapedetectorcollection.b().getAdjacentZ();
             }
 
             float f1 = 0.0F;
