--- a/net/minecraft/server/TileEntityDispenser.java
+++ b/net/minecraft/server/TileEntityDispenser.java
@@ -28,7 +28,7 @@
                 this.update();
                 return itemstack;
             } else {
-                itemstack = this.items[i].a(j);
+                itemstack = this.items[i].cloneAndSubtract(j);
                 if (this.items[i].count == 0) {
                     this.items[i] = null;
                 }
