--- a/net/minecraft/server/TileEntityChest.java
+++ b/net/minecraft/server/TileEntityChest.java
@@ -38,7 +38,7 @@
                 this.update();
                 return itemstack;
             } else {
-                itemstack = this.items[i].a(j);
+                itemstack = this.items[i].cloneAndSubtract(j);
                 if (this.items[i].count == 0) {
                     this.items[i] = null;
                 }
@@ -300,7 +300,7 @@
     }
 
     public void startOpen(EntityHuman entityhuman) {
-        if (!entityhuman.v()) {
+        if (!entityhuman.isSpectator()) {
             if (this.l < 0) {
                 this.l = 0;
             }
@@ -314,7 +314,7 @@
     }
 
     public void closeContainer(EntityHuman entityhuman) {
-        if (!entityhuman.v() && this.w() instanceof BlockChest) {
+        if (!entityhuman.isSpectator() && this.w() instanceof BlockChest) {
             --this.l;
             this.world.playBlockAction(this.position, this.w(), 1, this.l);
             this.world.applyPhysics(this.position, this.w());
