--- a/net/minecraft/server/TileEntityBrewingStand.java
+++ b/net/minecraft/server/TileEntityBrewingStand.java
@@ -8,7 +8,7 @@
     private static final int[] a = new int[] { 3};
     private static final int[] f = new int[] { 0, 1, 2};
     private ItemStack[] items = new ItemStack[4];
-    private int brewTime;
+    public int brewTime;
     private boolean[] i;
     private Item j;
     private String k;
