--- a/net/minecraft/server/WorldData.java
+++ b/net/minecraft/server/WorldData.java
@@ -292,7 +292,7 @@
         nbttagcompound.setLong("Time", this.h);
         nbttagcompound.setLong("DayTime", this.i);
         nbttagcompound.setLong("SizeOnDisk", this.k);
-        nbttagcompound.setLong("LastPlayed", MinecraftServer.ay());
+        nbttagcompound.setLong("LastPlayed", MinecraftServer.az());
         nbttagcompound.setString("LevelName", this.n);
         nbttagcompound.setInt("version", this.o);
         nbttagcompound.setInt("clearWeatherTime", this.p);
@@ -550,7 +550,7 @@
         this.J = i;
     }
 
-    public EnumDifficulty y() {
+    public EnumDifficulty getDifficulty() {
         return this.z;
     }
 
@@ -558,7 +558,7 @@
         this.z = enumdifficulty;
     }
 
-    public boolean z() {
+    public boolean isDifficultyLocked() {
         return this.A;
     }
 
