--- a/net/minecraft/server/BiomeMushrooms.java
+++ b/net/minecraft/server/BiomeMushrooms.java
@@ -5,10 +5,10 @@
     public BiomeMushrooms(BiomeBase.a biomebase_a) {
         super(biomebase_a);
         this.t.z = -100;
-        this.t.A = -100;
         this.t.B = -100;
-        this.t.D = 1;
-        this.t.J = 1;
+        this.t.C = -100;
+        this.t.E = 1;
+        this.t.K = 1;
         this.r = Blocks.MYCELIUM.getBlockData();
         this.u.clear();
         this.v.clear();
