--- a/net/minecraft/server/BiomeDesert.java
+++ b/net/minecraft/server/BiomeDesert.java
@@ -10,9 +10,9 @@
         this.r = Blocks.SAND.getBlockData();
         this.s = Blocks.SAND.getBlockData();
         this.t.z = -999;
-        this.t.C = 2;
-        this.t.E = 50;
-        this.t.F = 10;
+        this.t.D = 2;
+        this.t.F = 50;
+        this.t.G = 10;
         this.v.clear();
         this.v.add(new BiomeBase.BiomeMeta(EntityRabbit.class, 4, 2, 3));
     }
@@ -27,5 +27,9 @@
             (new WorldGenDesertWell()).generate(world, random, blockposition1);
         }
 
+        if (random.nextInt(64) == 0) {
+            (new WorldGenFossils()).generate(world, random, blockposition);
+        }
+
     }
 }
