--- a/net/minecraft/server/BlockCobbleWall.java
+++ b/net/minecraft/server/BlockCobbleWall.java
@@ -111,7 +111,7 @@
         private static final BlockCobbleWall.EnumCobbleVariant[] c = new BlockCobbleWall.EnumCobbleVariant[values().length];
         private final int d;
         private final String e;
-        private String f;
+        private final String f;
 
         private EnumCobbleVariant(int i, String s, String s1) {
             this.d = i;
