--- a/net/minecraft/server/BlockSand.java
+++ b/net/minecraft/server/BlockSand.java
@@ -9,11 +9,11 @@
     }
 
     public int getDropData(IBlockData iblockdata) {
-        return ((BlockSand.EnumSandVariant) iblockdata.get(BlockSand.VARIANT)).a();
+        return ((BlockSand.EnumSandVariant) iblockdata.get(BlockSand.VARIANT)).b();
     }
 
     public MaterialMapColor r(IBlockData iblockdata) {
-        return ((BlockSand.EnumSandVariant) iblockdata.get(BlockSand.VARIANT)).c();
+        return ((BlockSand.EnumSandVariant) iblockdata.get(BlockSand.VARIANT)).d();
     }
 
     public IBlockData fromLegacyData(int i) {
@@ -21,7 +21,7 @@
     }
 
     public int toLegacyData(IBlockData iblockdata) {
-        return ((BlockSand.EnumSandVariant) iblockdata.get(BlockSand.VARIANT)).a();
+        return ((BlockSand.EnumSandVariant) iblockdata.get(BlockSand.VARIANT)).b();
     }
 
     protected BlockStateList getStateList() {
@@ -30,22 +30,24 @@
 
     public static enum EnumSandVariant implements INamable {
 
-        SAND(0, "sand", "default", MaterialMapColor.d), RED_SAND(1, "red_sand", "red", MaterialMapColor.q);
+        SAND(0, "sand", "default", MaterialMapColor.d, -2370656), RED_SAND(1, "red_sand", "red", MaterialMapColor.q, -5679071);
 
         private static final BlockSand.EnumSandVariant[] c = new BlockSand.EnumSandVariant[values().length];
         private final int d;
         private final String e;
         private final MaterialMapColor f;
         private final String g;
+        private final int h;
 
-        private EnumSandVariant(int i, String s, String s1, MaterialMapColor materialmapcolor) {
+        private EnumSandVariant(int i, String s, String s1, MaterialMapColor materialmapcolor, int j) {
             this.d = i;
             this.e = s;
             this.f = materialmapcolor;
             this.g = s1;
+            this.h = j;
         }
 
-        public int a() {
+        public int b() {
             return this.d;
         }
 
@@ -53,7 +55,7 @@
             return this.e;
         }
 
-        public MaterialMapColor c() {
+        public MaterialMapColor d() {
             return this.f;
         }
 
@@ -69,7 +71,7 @@
             return this.e;
         }
 
-        public String d() {
+        public String e() {
             return this.g;
         }
 
@@ -80,7 +82,7 @@
             for (int j = 0; j < i; ++j) {
                 BlockSand.EnumSandVariant blocksand_enumsandvariant = ablocksand_enumsandvariant[j];
 
-                BlockSand.EnumSandVariant.c[blocksand_enumsandvariant.a()] = blocksand_enumsandvariant;
+                BlockSand.EnumSandVariant.c[blocksand_enumsandvariant.b()] = blocksand_enumsandvariant;
             }
 
         }
