--- a/net/minecraft/server/CommandWhitelist.java
+++ b/net/minecraft/server/CommandWhitelist.java
@@ -25,13 +25,13 @@
         if (astring.length < 1) {
             throw new ExceptionUsage("commands.whitelist.usage", new Object[0]);
         } else {
-            if (astring[0].equals("on")) {
+            if ("on".equals(astring[0])) {
                 minecraftserver.getPlayerList().setHasWhitelist(true);
                 a(icommandlistener, (ICommand) this, "commands.whitelist.enabled", new Object[0]);
-            } else if (astring[0].equals("off")) {
+            } else if ("off".equals(astring[0])) {
                 minecraftserver.getPlayerList().setHasWhitelist(false);
                 a(icommandlistener, (ICommand) this, "commands.whitelist.disabled", new Object[0]);
-            } else if (astring[0].equals("list")) {
+            } else if ("list".equals(astring[0])) {
                 icommandlistener.sendMessage(new ChatMessage("commands.whitelist.list", new Object[] { Integer.valueOf(minecraftserver.getPlayerList().getWhitelisted().length), Integer.valueOf(minecraftserver.getPlayerList().getSeenPlayers().length)}));
                 String[] astring1 = minecraftserver.getPlayerList().getWhitelisted();
 
@@ -39,7 +39,7 @@
             } else {
                 GameProfile gameprofile;
 
-                if (astring[0].equals("add")) {
+                if ("add".equals(astring[0])) {
                     if (astring.length < 2) {
                         throw new ExceptionUsage("commands.whitelist.add.usage", new Object[0]);
                     }
@@ -51,7 +51,7 @@
 
                     minecraftserver.getPlayerList().addWhitelist(gameprofile);
                     a(icommandlistener, (ICommand) this, "commands.whitelist.add.success", new Object[] { astring[1]});
-                } else if (astring[0].equals("remove")) {
+                } else if ("remove".equals(astring[0])) {
                     if (astring.length < 2) {
                         throw new ExceptionUsage("commands.whitelist.remove.usage", new Object[0]);
                     }
@@ -63,7 +63,7 @@
 
                     minecraftserver.getPlayerList().removeWhitelist(gameprofile);
                     a(icommandlistener, (ICommand) this, "commands.whitelist.remove.success", new Object[] { astring[1]});
-                } else if (astring[0].equals("reload")) {
+                } else if ("reload".equals(astring[0])) {
                     minecraftserver.getPlayerList().reloadWhitelist();
                     a(icommandlistener, (ICommand) this, "commands.whitelist.reloaded", new Object[0]);
                 }
@@ -77,11 +77,11 @@
             return a(astring, new String[] { "on", "off", "list", "add", "remove", "reload"});
         } else {
             if (astring.length == 2) {
-                if (astring[0].equals("remove")) {
+                if ("remove".equals(astring[0])) {
                     return a(astring, minecraftserver.getPlayerList().getWhitelisted());
                 }
 
-                if (astring[0].equals("add")) {
+                if ("add".equals(astring[0])) {
                     return a(astring, minecraftserver.getUserCache().a());
                 }
             }
