--- a/net/minecraft/server/CommandGamemode.java
+++ b/net/minecraft/server/CommandGamemode.java
@@ -24,29 +24,29 @@
         if (astring.length <= 0) {
             throw new ExceptionUsage("commands.gamemode.usage", new Object[0]);
         } else {
-            WorldSettings.EnumGamemode worldsettings_enumgamemode = this.c(icommandlistener, astring[0]);
+            EnumGamemode enumgamemode = this.c(icommandlistener, astring[0]);
             EntityPlayer entityplayer = astring.length >= 2 ? a(minecraftserver, icommandlistener, astring[1]) : a(icommandlistener);
 
-            entityplayer.a(worldsettings_enumgamemode);
-            ChatMessage chatmessage = new ChatMessage("gameMode." + worldsettings_enumgamemode.b(), new Object[0]);
+            entityplayer.a(enumgamemode);
+            ChatMessage chatmessage = new ChatMessage("gameMode." + enumgamemode.b(), new Object[0]);
 
             if (icommandlistener.getWorld().getGameRules().getBoolean("sendCommandFeedback")) {
                 entityplayer.sendMessage(new ChatMessage("gameMode.changed", new Object[] { chatmessage}));
             }
 
-            if (entityplayer != icommandlistener) {
-                a(icommandlistener, this, 1, "commands.gamemode.success.other", new Object[] { entityplayer.getName(), chatmessage});
-            } else {
+            if (entityplayer == icommandlistener) {
                 a(icommandlistener, this, 1, "commands.gamemode.success.self", new Object[] { chatmessage});
+            } else {
+                a(icommandlistener, this, 1, "commands.gamemode.success.other", new Object[] { entityplayer.getName(), chatmessage});
             }
 
         }
     }
 
-    protected WorldSettings.EnumGamemode c(ICommandListener icommandlistener, String s) throws ExceptionInvalidNumber {
-        WorldSettings.EnumGamemode worldsettings_enumgamemode = WorldSettings.EnumGamemode.a(s, WorldSettings.EnumGamemode.NOT_SET);
+    protected EnumGamemode c(ICommandListener icommandlistener, String s) throws ExceptionInvalidNumber {
+        EnumGamemode enumgamemode = EnumGamemode.a(s, EnumGamemode.NOT_SET);
 
-        return worldsettings_enumgamemode == WorldSettings.EnumGamemode.NOT_SET ? WorldSettings.a(a(s, 0, WorldSettings.EnumGamemode.values().length - 2)) : worldsettings_enumgamemode;
+        return enumgamemode == EnumGamemode.NOT_SET ? WorldSettings.a(a(s, 0, EnumGamemode.values().length - 2)) : enumgamemode;
     }
 
     public List<String> tabComplete(MinecraftServer minecraftserver, ICommandListener icommandlistener, String[] astring, @Nullable BlockPosition blockposition) {
