--- a/net/minecraft/server/ContainerMerchant.java
+++ b/net/minecraft/server/ContainerMerchant.java
@@ -4,8 +4,8 @@
 
 public class ContainerMerchant extends Container {
 
-    private IMerchant merchant;
-    private InventoryMerchant f;
+    private final IMerchant merchant;
+    private final InventoryMerchant f;
     private final World g;
 
     public ContainerMerchant(PlayerInventory playerinventory, IMerchant imerchant, World world) {
