--- a/net/minecraft/server/CrashReportSystemDetails.java
+++ b/net/minecraft/server/CrashReportSystemDetails.java
@@ -158,7 +158,7 @@
                 StackTraceElement stacktraceelement = astacktraceelement[j];
 
                 stringbuilder.append("\n\tat ");
-                stringbuilder.append(stacktraceelement.toString());
+                stringbuilder.append(stacktraceelement);
             }
         }
 
