--- a/net/minecraft/server/DemoWorldServer.java
+++ b/net/minecraft/server/DemoWorldServer.java
@@ -3,7 +3,7 @@
 public class DemoWorldServer extends WorldServer {
 
     private static final long I = (long) "North Carolina".hashCode();
-    public static final WorldSettings a = (new WorldSettings(DemoWorldServer.I, WorldSettings.EnumGamemode.SURVIVAL, true, false, WorldType.NORMAL)).a();
+    public static final WorldSettings a = (new WorldSettings(DemoWorldServer.I, EnumGamemode.SURVIVAL, true, false, WorldType.NORMAL)).a();
 
     public DemoWorldServer(MinecraftServer minecraftserver, IDataManager idatamanager, WorldData worlddata, int i, MethodProfiler methodprofiler) {
         super(minecraftserver, idatamanager, worlddata, i, methodprofiler);
