--- a/net/minecraft/server/DifficultyDamageScaler.java
+++ b/net/minecraft/server/DifficultyDamageScaler.java
@@ -17,7 +17,11 @@
         return this.b;
     }
 
-    public float c() {
+    public boolean c() {
+        return this.b >= (float) EnumDifficulty.HARD.ordinal();
+    }
+
+    public float d() {
         return this.b < 2.0F ? 0.0F : (this.b > 4.0F ? 1.0F : (this.b - 2.0F) / 2.0F);
     }
 
