--- a/net/minecraft/server/DragonControllerLandedSearch.java
+++ b/net/minecraft/server/DragonControllerLandedSearch.java
@@ -22,11 +22,11 @@
                 float f1 = (float) (Math.acos((double) f) * 57.2957763671875D) + 0.5F;
 
                 if (f1 < 0.0F || f1 > 10.0F) {
-                    double d0 = entityhuman.locX - this.a.bv.locX;
-                    double d1 = entityhuman.locZ - this.a.bv.locZ;
+                    double d0 = entityhuman.locX - this.a.bw.locX;
+                    double d1 = entityhuman.locZ - this.a.bw.locZ;
                     double d2 = MathHelper.a(MathHelper.g(180.0D - MathHelper.b(d0, d1) * 57.2957763671875D - (double) this.a.yaw), -100.0D, 100.0D);
 
-                    this.a.bg *= 0.8F;
+                    this.a.bh *= 0.8F;
                     float f2 = MathHelper.sqrt(d0 * d0 + d1 * d1) + 1.0F;
                     float f3 = f2;
 
@@ -34,8 +34,8 @@
                         f2 = 40.0F;
                     }
 
-                    this.a.bg = (float) ((double) this.a.bg + d2 * (double) (0.7F / f2 / f3));
-                    this.a.yaw += this.a.bg;
+                    this.a.bh = (float) ((double) this.a.bh + d2 * (double) (0.7F / f2 / f3));
+                    this.a.yaw += this.a.bh;
                 }
             }
         } else if (this.b >= 100) {
