--- a/net/minecraft/server/EntityAgeable.java
+++ b/net/minecraft/server/EntityAgeable.java
@@ -4,12 +4,12 @@
 
 public abstract class EntityAgeable extends EntityCreature {
 
-    private static final DataWatcherObject<Boolean> bw = DataWatcher.a(EntityAgeable.class, DataWatcherRegistry.h);
+    private static final DataWatcherObject<Boolean> bx = DataWatcher.a(EntityAgeable.class, DataWatcherRegistry.h);
     protected int a;
     protected int b;
     protected int c;
-    private float bx = -1.0F;
-    private float by;
+    private float by = -1.0F;
+    private float bz;
 
     public EntityAgeable(World world) {
         super(world);
@@ -48,11 +48,11 @@
 
     protected void i() {
         super.i();
-        this.datawatcher.register(EntityAgeable.bw, Boolean.valueOf(false));
+        this.datawatcher.register(EntityAgeable.bx, Boolean.valueOf(false));
     }
 
     public int getAge() {
-        return this.world.isClientSide ? (((Boolean) this.datawatcher.get(EntityAgeable.bw)).booleanValue() ? -1 : 1) : this.a;
+        return this.world.isClientSide ? (((Boolean) this.datawatcher.get(EntityAgeable.bx)).booleanValue() ? -1 : 1) : this.a;
     }
 
     public void setAge(int i, boolean flag) {
@@ -88,7 +88,7 @@
     }
 
     public void setAgeRaw(int i) {
-        this.datawatcher.set(EntityAgeable.bw, Boolean.valueOf(i < 0));
+        this.datawatcher.set(EntityAgeable.bx, Boolean.valueOf(i < 0));
         this.a = i;
         this.a(this.isBaby());
     }
@@ -106,7 +106,7 @@
     }
 
     public void a(DataWatcherObject<?> datawatcherobject) {
-        if (EntityAgeable.bw.equals(datawatcherobject)) {
+        if (EntityAgeable.bx.equals(datawatcherobject)) {
             this.a(this.isBaby());
         }
 
@@ -151,10 +151,10 @@
     }
 
     public final void setSize(float f, float f1) {
-        boolean flag = this.bx > 0.0F;
+        boolean flag = this.by > 0.0F;
 
-        this.bx = f;
-        this.by = f1;
+        this.by = f;
+        this.bz = f1;
         if (!flag) {
             this.a(1.0F);
         }
@@ -162,6 +162,6 @@
     }
 
     protected final void a(float f) {
-        super.setSize(this.bx * f, this.by * f);
+        super.setSize(this.by * f, this.bz * f);
     }
 }
