--- a/net/minecraft/server/EntityMinecartHopper.java
+++ b/net/minecraft/server/EntityMinecartHopper.java
@@ -7,7 +7,7 @@
 
     private boolean a = true;
     private int b = -1;
-    private BlockPosition c;
+    private final BlockPosition c;
 
     public EntityMinecartHopper(World world) {
         super(world);
@@ -27,7 +27,7 @@
         return Blocks.HOPPER.getBlockData();
     }
 
-    public int A() {
+    public int z() {
         return 1;
     }
 
@@ -120,6 +120,10 @@
 
     }
 
+    public static void a(DataConverterManager dataconvertermanager) {
+        EntityMinecartContainer.b(dataconvertermanager, "MinecartHopper");
+    }
+
     protected void b(NBTTagCompound nbttagcompound) {
         super.b(nbttagcompound);
         nbttagcompound.setInt("TransferCooldown", this.b);
