--- a/net/minecraft/server/EntityHorse.java
+++ b/net/minecraft/server/EntityHorse.java
@@ -9,9 +9,9 @@
 
 public class EntityHorse extends EntityAnimal implements IInventoryListener, IJumpable {
 
-    private static final Predicate<Entity> bC = new Predicate() {
+    private static final Predicate<Entity> bD = new Predicate() {
         public boolean a(@Nullable Entity entity) {
-            return entity instanceof EntityHorse && ((EntityHorse) entity).dp();
+            return entity instanceof EntityHorse && ((EntityHorse) entity).dt();
         }
 
         public boolean apply(Object object) {
@@ -19,40 +19,39 @@
         }
     };
     public static final IAttribute attributeJumpStrength = (new AttributeRanged((IAttribute) null, "horse.jumpStrength", 0.7D, 0.0D, 2.0D)).a("Jump Strength").a(true);
-    private static final UUID bE = UUID.fromString("556E1665-8B10-40C8-8F9D-CF9B1667F295");
-    private static final DataWatcherObject<Byte> bF = DataWatcher.a(EntityHorse.class, DataWatcherRegistry.a);
-    private static final DataWatcherObject<Integer> bG = DataWatcher.a(EntityHorse.class, DataWatcherRegistry.b);
+    private static final UUID bF = UUID.fromString("556E1665-8B10-40C8-8F9D-CF9B1667F295");
+    private static final DataWatcherObject<Byte> bG = DataWatcher.a(EntityHorse.class, DataWatcherRegistry.a);
     private static final DataWatcherObject<Integer> bH = DataWatcher.a(EntityHorse.class, DataWatcherRegistry.b);
-    private static final DataWatcherObject<Optional<UUID>> bI = DataWatcher.a(EntityHorse.class, DataWatcherRegistry.m);
-    private static final DataWatcherObject<Integer> bJ = DataWatcher.a(EntityHorse.class, DataWatcherRegistry.b);
-    private static final String[] bK = new String[] { "textures/entity/horse/horse_white.png", "textures/entity/horse/horse_creamy.png", "textures/entity/horse/horse_chestnut.png", "textures/entity/horse/horse_brown.png", "textures/entity/horse/horse_black.png", "textures/entity/horse/horse_gray.png", "textures/entity/horse/horse_darkbrown.png"};
-    private static final String[] bL = new String[] { "hwh", "hcr", "hch", "hbr", "hbl", "hgr", "hdb"};
-    private static final String[] bM = new String[] { null, "textures/entity/horse/horse_markings_white.png", "textures/entity/horse/horse_markings_whitefield.png", "textures/entity/horse/horse_markings_whitedots.png", "textures/entity/horse/horse_markings_blackdots.png"};
-    private static final String[] bN = new String[] { "", "wo_", "wmo", "wdo", "bdo"};
-    private final PathfinderGoalHorseTrap bO = new PathfinderGoalHorseTrap(this);
-    private int bP;
+    private static final DataWatcherObject<Integer> bI = DataWatcher.a(EntityHorse.class, DataWatcherRegistry.b);
+    private static final DataWatcherObject<Optional<UUID>> bJ = DataWatcher.a(EntityHorse.class, DataWatcherRegistry.m);
+    private static final DataWatcherObject<Integer> bK = DataWatcher.a(EntityHorse.class, DataWatcherRegistry.b);
+    private static final String[] bL = new String[] { "textures/entity/horse/horse_white.png", "textures/entity/horse/horse_creamy.png", "textures/entity/horse/horse_chestnut.png", "textures/entity/horse/horse_brown.png", "textures/entity/horse/horse_black.png", "textures/entity/horse/horse_gray.png", "textures/entity/horse/horse_darkbrown.png"};
+    private static final String[] bM = new String[] { "hwh", "hcr", "hch", "hbr", "hbl", "hgr", "hdb"};
+    private static final String[] bN = new String[] { null, "textures/entity/horse/horse_markings_white.png", "textures/entity/horse/horse_markings_whitefield.png", "textures/entity/horse/horse_markings_whitedots.png", "textures/entity/horse/horse_markings_blackdots.png"};
+    private static final String[] bO = new String[] { "", "wo_", "wmo", "wdo", "bdo"};
+    private final PathfinderGoalHorseTrap bP = new PathfinderGoalHorseTrap(this);
     private int bQ;
     private int bR;
-    public int bw;
+    private int bS;
     public int bx;
-    protected boolean by;
+    public int by;
+    protected boolean bz;
     public InventoryHorseChest inventoryChest;
-    private boolean bT;
-    protected int bA;
+    private boolean bU;
+    protected int bB;
     protected float jumpPower;
     private boolean canSlide;
-    private boolean bV;
-    private int bW = 0;
-    private float bX;
+    private boolean bW;
+    private int bX;
     private float bY;
     private float bZ;
     private float ca;
     private float cb;
     private float cc;
-    private int cd;
-    private String ce;
-    private String[] cf = new String[3];
-    private boolean cg = false;
+    private float cd;
+    private int ce;
+    private String cf;
+    private final String[] cg = new String[3];
 
     public EntityHorse(World world) {
         super(world);
@@ -76,29 +75,29 @@
 
     protected void i() {
         super.i();
-        this.datawatcher.register(EntityHorse.bF, Byte.valueOf((byte) 0));
-        this.datawatcher.register(EntityHorse.bG, Integer.valueOf(EnumHorseType.HORSE.k()));
-        this.datawatcher.register(EntityHorse.bH, Integer.valueOf(0));
-        this.datawatcher.register(EntityHorse.bI, Optional.absent());
-        this.datawatcher.register(EntityHorse.bJ, Integer.valueOf(EnumHorseArmor.NONE.a()));
+        this.datawatcher.register(EntityHorse.bG, Byte.valueOf((byte) 0));
+        this.datawatcher.register(EntityHorse.bH, Integer.valueOf(EnumHorseType.HORSE.k()));
+        this.datawatcher.register(EntityHorse.bI, Integer.valueOf(0));
+        this.datawatcher.register(EntityHorse.bJ, Optional.absent());
+        this.datawatcher.register(EntityHorse.bK, Integer.valueOf(EnumHorseArmor.NONE.a()));
     }
 
     public void setType(EnumHorseType enumhorsetype) {
-        this.datawatcher.set(EntityHorse.bG, Integer.valueOf(enumhorsetype.k()));
-        this.dM();
+        this.datawatcher.set(EntityHorse.bH, Integer.valueOf(enumhorsetype.k()));
+        this.dQ();
     }
 
     public EnumHorseType getType() {
-        return EnumHorseType.a(((Integer) this.datawatcher.get(EntityHorse.bG)).intValue());
+        return EnumHorseType.a(((Integer) this.datawatcher.get(EntityHorse.bH)).intValue());
     }
 
     public void setVariant(int i) {
-        this.datawatcher.set(EntityHorse.bH, Integer.valueOf(i));
-        this.dM();
+        this.datawatcher.set(EntityHorse.bI, Integer.valueOf(i));
+        this.dQ();
     }
 
     public int getVariant() {
-        return ((Integer) this.datawatcher.get(EntityHorse.bH)).intValue();
+        return ((Integer) this.datawatcher.get(EntityHorse.bI)).intValue();
     }
 
     public String getName() {
@@ -106,21 +105,21 @@
     }
 
     private boolean o(int i) {
-        return (((Byte) this.datawatcher.get(EntityHorse.bF)).byteValue() & i) != 0;
+        return (((Byte) this.datawatcher.get(EntityHorse.bG)).byteValue() & i) != 0;
     }
 
     private void c(int i, boolean flag) {
-        byte b0 = ((Byte) this.datawatcher.get(EntityHorse.bF)).byteValue();
+        byte b0 = ((Byte) this.datawatcher.get(EntityHorse.bG)).byteValue();
 
         if (flag) {
-            this.datawatcher.set(EntityHorse.bF, Byte.valueOf((byte) (b0 | i)));
+            this.datawatcher.set(EntityHorse.bG, Byte.valueOf((byte) (b0 | i)));
         } else {
-            this.datawatcher.set(EntityHorse.bF, Byte.valueOf((byte) (b0 & ~i)));
+            this.datawatcher.set(EntityHorse.bG, Byte.valueOf((byte) (b0 & ~i)));
         }
 
     }
 
-    public boolean dc() {
+    public boolean dg() {
         return !this.isBaby();
     }
 
@@ -128,42 +127,42 @@
         return this.o(2);
     }
 
-    public boolean de() {
-        return this.dc();
+    public boolean di() {
+        return this.dg();
     }
 
     @Nullable
     public UUID getOwnerUUID() {
-        return (UUID) ((Optional) this.datawatcher.get(EntityHorse.bI)).orNull();
+        return (UUID) ((Optional) this.datawatcher.get(EntityHorse.bJ)).orNull();
     }
 
     public void setOwnerUUID(@Nullable UUID uuid) {
-        this.datawatcher.set(EntityHorse.bI, Optional.fromNullable(uuid));
+        this.datawatcher.set(EntityHorse.bJ, Optional.fromNullable(uuid));
     }
 
-    public float dj() {
+    public float dn() {
         return 0.5F;
     }
 
     public void a(boolean flag) {
         if (flag) {
-            this.a(this.dj());
+            this.a(this.dn());
         } else {
             this.a(1.0F);
         }
 
     }
 
-    public boolean dk() {
-        return this.by;
+    public boolean do_() {
+        return this.bz;
     }
 
     public void setTame(boolean flag) {
         this.c(2, flag);
     }
 
-    public void p(boolean flag) {
-        this.by = flag;
+    public void q(boolean flag) {
+        this.bz = flag;
     }
 
     public boolean a(EntityHuman entityhuman) {
@@ -171,8 +170,8 @@
     }
 
     protected void q(float f) {
-        if (f > 6.0F && this.dn()) {
-            this.u(false);
+        if (f > 6.0F && this.dr()) {
+            this.v(false);
         }
 
     }
@@ -181,43 +180,43 @@
         return this.getType().f() && this.o(8);
     }
 
-    public EnumHorseArmor dm() {
-        return EnumHorseArmor.a(((Integer) this.datawatcher.get(EntityHorse.bJ)).intValue());
+    public EnumHorseArmor dq() {
+        return EnumHorseArmor.a(((Integer) this.datawatcher.get(EntityHorse.bK)).intValue());
     }
 
-    public boolean dn() {
+    public boolean dr() {
         return this.o(32);
     }
 
-    public boolean do_() {
+    public boolean ds() {
         return this.o(64);
     }
 
-    public boolean dp() {
+    public boolean dt() {
         return this.o(16);
     }
 
     public boolean hasReproduced() {
-        return this.bT;
+        return this.bU;
     }
 
     public void f(ItemStack itemstack) {
         EnumHorseArmor enumhorsearmor = EnumHorseArmor.a(itemstack);
 
-        this.datawatcher.set(EntityHorse.bJ, Integer.valueOf(enumhorsearmor.a()));
-        this.dM();
+        this.datawatcher.set(EntityHorse.bK, Integer.valueOf(enumhorsearmor.a()));
+        this.dQ();
         if (!this.world.isClientSide) {
-            this.getAttributeInstance(GenericAttributes.g).b(EntityHorse.bE);
+            this.getAttributeInstance(GenericAttributes.g).b(EntityHorse.bF);
             int i = enumhorsearmor.c();
 
             if (i != 0) {
-                this.getAttributeInstance(GenericAttributes.g).b((new AttributeModifier(EntityHorse.bE, "Horse armor bonus", (double) i, 0)).a(false));
+                this.getAttributeInstance(GenericAttributes.g).b((new AttributeModifier(EntityHorse.bF, "Horse armor bonus", (double) i, 0)).a(false));
             }
         }
 
     }
 
-    public void q(boolean flag) {
+    public void r(boolean flag) {
         this.c(16, flag);
     }
 
@@ -225,20 +224,20 @@
         this.c(8, flag);
     }
 
-    public void s(boolean flag) {
-        this.bT = flag;
+    public void t(boolean flag) {
+        this.bU = flag;
     }
 
-    public void t(boolean flag) {
+    public void u(boolean flag) {
         this.c(4, flag);
     }
 
     public int getTemper() {
-        return this.bA;
+        return this.bB;
     }
 
     public void setTemper(int i) {
-        this.bA = i;
+        this.bB = i;
     }
 
     public int n(int i) {
@@ -258,7 +257,7 @@
         return !this.isVehicle();
     }
 
-    public boolean ds() {
+    public boolean dw() {
         int i = MathHelper.floor(this.locX);
         int j = MathHelper.floor(this.locZ);
 
@@ -266,24 +265,24 @@
         return true;
     }
 
-    public void dt() {
+    public void dx() {
         if (!this.world.isClientSide && this.hasChest()) {
             this.a(Item.getItemOf(Blocks.CHEST), 1);
             this.setHasChest(false);
         }
     }
 
-    private void dI() {
-        this.dP();
-        if (!this.ad()) {
-            this.world.a((EntityHuman) null, this.locX, this.locY, this.locZ, SoundEffects.ct, this.bA(), 1.0F, 1.0F + (this.random.nextFloat() - this.random.nextFloat()) * 0.2F);
+    private void dM() {
+        this.dT();
+        if (!this.isSilent()) {
+            this.world.a((EntityHuman) null, this.locX, this.locY, this.locZ, SoundEffects.cu, this.bC(), 1.0F, 1.0F + (this.random.nextFloat() - this.random.nextFloat()) * 0.2F);
         }
 
     }
 
     public void e(float f, float f1) {
         if (f > 1.0F) {
-            this.a(SoundEffects.cx, 0.4F, 1.0F);
+            this.a(SoundEffects.cy, 0.4F, 1.0F);
         }
 
         int i = MathHelper.f((f * 0.5F - 3.0F) * f1);
@@ -291,7 +290,7 @@
         if (i > 0) {
             this.damageEntity(DamageSource.FALL, (float) i);
             if (this.isVehicle()) {
-                Iterator iterator = this.bw().iterator();
+                Iterator iterator = this.by().iterator();
 
                 while (iterator.hasNext()) {
                     Entity entity = (Entity) iterator.next();
@@ -303,16 +302,16 @@
             IBlockData iblockdata = this.world.getType(new BlockPosition(this.locX, this.locY - 0.2D - (double) this.lastYaw, this.locZ));
             Block block = iblockdata.getBlock();
 
-            if (iblockdata.getMaterial() != Material.AIR && !this.ad()) {
+            if (iblockdata.getMaterial() != Material.AIR && !this.isSilent()) {
                 SoundEffectType soundeffecttype = block.w();
 
-                this.world.a((EntityHuman) null, this.locX, this.locY, this.locZ, soundeffecttype.d(), this.bA(), soundeffecttype.a() * 0.5F, soundeffecttype.b() * 0.75F);
+                this.world.a((EntityHuman) null, this.locX, this.locY, this.locZ, soundeffecttype.d(), this.bC(), soundeffecttype.a() * 0.5F, soundeffecttype.b() * 0.75F);
             }
 
         }
     }
 
-    private int dJ() {
+    private int dN() {
         EnumHorseType enumhorsetype = this.getType();
 
         return this.hasChest() && enumhorsetype.f() ? 17 : 2;
@@ -321,7 +320,7 @@
     public void loadChest() {
         InventoryHorseChest inventoryhorsechest = this.inventoryChest;
 
-        this.inventoryChest = new InventoryHorseChest("HorseChest", this.dJ());
+        this.inventoryChest = new InventoryHorseChest("HorseChest", this.dN());
         this.inventoryChest.a(this.getName());
         if (inventoryhorsechest != null) {
             inventoryhorsechest.b(this);
@@ -337,12 +336,12 @@
         }
 
         this.inventoryChest.a((IInventoryListener) this);
-        this.dL();
+        this.dP();
     }
 
-    private void dL() {
+    private void dP() {
         if (!this.world.isClientSide) {
-            this.t(this.inventoryChest.getItem(0) != null);
+            this.u(this.inventoryChest.getItem(0) != null);
             if (this.getType().j()) {
                 this.f(this.inventoryChest.getItem(1));
             }
@@ -351,33 +350,33 @@
     }
 
     public void a(InventorySubcontainer inventorysubcontainer) {
-        EnumHorseArmor enumhorsearmor = this.dm();
-        boolean flag = this.dv();
+        EnumHorseArmor enumhorsearmor = this.dq();
+        boolean flag = this.dz();
 
-        this.dL();
+        this.dP();
         if (this.ticksLived > 20) {
-            if (enumhorsearmor == EnumHorseArmor.NONE && enumhorsearmor != this.dm()) {
-                this.a(SoundEffects.cq, 0.5F, 1.0F);
-            } else if (enumhorsearmor != this.dm()) {
-                this.a(SoundEffects.cq, 0.5F, 1.0F);
+            if (enumhorsearmor == EnumHorseArmor.NONE && enumhorsearmor != this.dq()) {
+                this.a(SoundEffects.cr, 0.5F, 1.0F);
+            } else if (enumhorsearmor != this.dq()) {
+                this.a(SoundEffects.cr, 0.5F, 1.0F);
             }
 
-            if (!flag && this.dv()) {
-                this.a(SoundEffects.cy, 0.5F, 1.0F);
+            if (!flag && this.dz()) {
+                this.a(SoundEffects.cz, 0.5F, 1.0F);
             }
         }
 
     }
 
-    public boolean cG() {
-        this.ds();
-        return super.cG();
+    public boolean cK() {
+        this.dw();
+        return super.cK();
     }
 
     protected EntityHorse a(Entity entity, double d0) {
         double d1 = Double.MAX_VALUE;
         Entity entity1 = null;
-        List list = this.world.getEntities(entity, entity.getBoundingBox().a(d0, d0, d0), EntityHorse.bC);
+        List list = this.world.getEntities(entity, entity.getBoundingBox().a(d0, d0, d0), EntityHorse.bD);
         Iterator iterator = list.iterator();
 
         while (iterator.hasNext()) {
@@ -397,13 +396,13 @@
         return this.getAttributeInstance(EntityHorse.attributeJumpStrength).getValue();
     }
 
-    protected SoundEffect bT() {
-        this.dP();
+    protected SoundEffect bW() {
+        this.dT();
         return this.getType().c();
     }
 
-    protected SoundEffect bS() {
-        this.dP();
+    protected SoundEffect bV() {
+        this.dT();
         if (this.random.nextInt(3) == 0) {
             this.setStanding();
         }
@@ -411,13 +410,13 @@
         return this.getType().b();
     }
 
-    public boolean dv() {
+    public boolean dz() {
         return this.o(4);
     }
 
     protected SoundEffect G() {
-        this.dP();
-        if (this.random.nextInt(10) == 0 && !this.cg()) {
+        this.dT();
+        if (this.random.nextInt(10) == 0 && !this.cj()) {
             this.setStanding();
         }
 
@@ -425,12 +424,12 @@
     }
 
     @Nullable
-    protected SoundEffect dw() {
-        this.dP();
+    protected SoundEffect dA() {
+        this.dT();
         this.setStanding();
         EnumHorseType enumhorsetype = this.getType();
 
-        return enumhorsetype.h() ? null : (enumhorsetype.g() ? SoundEffects.ay : SoundEffects.cp);
+        return enumhorsetype.h() ? null : (enumhorsetype.g() ? SoundEffects.ay : SoundEffects.cq);
     }
 
     protected void a(BlockPosition blockposition, Block block) {
@@ -444,19 +443,19 @@
             EnumHorseType enumhorsetype = this.getType();
 
             if (this.isVehicle() && !enumhorsetype.g()) {
-                ++this.cd;
-                if (this.cd > 5 && this.cd % 3 == 0) {
-                    this.a(SoundEffects.cu, soundeffecttype.a() * 0.15F, soundeffecttype.b());
+                ++this.ce;
+                if (this.ce > 5 && this.ce % 3 == 0) {
+                    this.a(SoundEffects.cv, soundeffecttype.a() * 0.15F, soundeffecttype.b());
                     if (enumhorsetype == EnumHorseType.HORSE && this.random.nextInt(10) == 0) {
-                        this.a(SoundEffects.cr, soundeffecttype.a() * 0.6F, soundeffecttype.b());
+                        this.a(SoundEffects.cs, soundeffecttype.a() * 0.6F, soundeffecttype.b());
                     }
-                } else if (this.cd <= 5) {
-                    this.a(SoundEffects.cA, soundeffecttype.a() * 0.15F, soundeffecttype.b());
+                } else if (this.ce <= 5) {
+                    this.a(SoundEffects.cB, soundeffecttype.a() * 0.15F, soundeffecttype.b());
                 }
             } else if (soundeffecttype == SoundEffectType.a) {
-                this.a(SoundEffects.cA, soundeffecttype.a() * 0.15F, soundeffecttype.b());
+                this.a(SoundEffects.cB, soundeffecttype.a() * 0.15F, soundeffecttype.b());
             } else {
-                this.a(SoundEffects.cz, soundeffecttype.a() * 0.15F, soundeffecttype.b());
+                this.a(SoundEffects.cA, soundeffecttype.a() * 0.15F, soundeffecttype.b());
             }
         }
 
@@ -469,7 +468,7 @@
         this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).setValue(0.22499999403953552D);
     }
 
-    public int cK() {
+    public int cO() {
         return 6;
     }
 
@@ -477,7 +476,7 @@
         return 100;
     }
 
-    protected float ce() {
+    protected float ch() {
         return 0.8F;
     }
 
@@ -485,8 +484,8 @@
         return 400;
     }
 
-    private void dM() {
-        this.ce = null;
+    private void dQ() {
+        this.cf = null;
     }
 
     public void f(EntityHuman entityhuman) {
@@ -502,10 +501,10 @@
             return super.a(entityhuman, enumhand, itemstack);
         } else if (!this.isTamed() && this.getType().h()) {
             return false;
-        } else if (this.isTamed() && this.dc() && entityhuman.isSneaking()) {
+        } else if (this.isTamed() && this.dg() && entityhuman.isSneaking()) {
             this.f(entityhuman);
             return true;
-        } else if (this.de() && this.isVehicle()) {
+        } else if (this.di() && this.isVehicle()) {
             return super.a(entityhuman, enumhand, itemstack);
         } else {
             if (itemstack != null) {
@@ -514,7 +513,7 @@
 
                     if (enumhorsearmor != EnumHorseArmor.NONE) {
                         if (!this.isTamed()) {
-                            this.dF();
+                            this.dJ();
                             return true;
                         }
 
@@ -568,7 +567,7 @@
                         flag = true;
                     }
 
-                    if (!this.dc() && short0 > 0) {
+                    if (!this.dg() && short0 > 0) {
                         if (!this.world.isClientSide) {
                             this.setAge(short0);
                         }
@@ -584,7 +583,7 @@
                     }
 
                     if (flag) {
-                        this.dI();
+                        this.dM();
                     }
                 }
 
@@ -593,7 +592,7 @@
                         return true;
                     }
 
-                    this.dF();
+                    this.dJ();
                     return true;
                 }
 
@@ -604,7 +603,7 @@
                     this.loadChest();
                 }
 
-                if (!flag && this.de() && !this.dv() && itemstack.getItem() == Items.SADDLE) {
+                if (!flag && this.di() && !this.dz() && itemstack.getItem() == Items.SADDLE) {
                     this.f(entityhuman);
                     return true;
                 }
@@ -618,7 +617,7 @@
                 }
             }
 
-            if (this.de() && !this.isVehicle()) {
+            if (this.di() && !this.isVehicle()) {
                 if (itemstack != null && itemstack.a(entityhuman, (EntityLiving) this, enumhand)) {
                     return true;
                 } else {
@@ -634,24 +633,24 @@
     private void h(EntityHuman entityhuman) {
         entityhuman.yaw = this.yaw;
         entityhuman.pitch = this.pitch;
-        this.u(false);
         this.v(false);
+        this.w(false);
         if (!this.world.isClientSide) {
             entityhuman.startRiding(this);
         }
 
     }
 
-    protected boolean cg() {
-        return this.isVehicle() && this.dv() ? true : this.dn() || this.do_();
+    protected boolean cj() {
+        return this.isVehicle() && this.dz() ? true : this.dr() || this.ds();
     }
 
     public boolean e(@Nullable ItemStack itemstack) {
         return false;
     }
 
-    private void dO() {
-        this.bw = 1;
+    private void dS() {
+        this.bx = 1;
     }
 
     public void die(DamageSource damagesource) {
@@ -664,7 +663,7 @@
 
     public void n() {
         if (this.random.nextInt(200) == 0) {
-            this.dO();
+            this.dS();
         }
 
         super.n();
@@ -673,16 +672,16 @@
                 this.heal(1.0F);
             }
 
-            if (!this.dn() && !this.isVehicle() && this.random.nextInt(300) == 0 && this.world.getType(new BlockPosition(MathHelper.floor(this.locX), MathHelper.floor(this.locY) - 1, MathHelper.floor(this.locZ))).getBlock() == Blocks.GRASS) {
-                this.u(true);
+            if (!this.dr() && !this.isVehicle() && this.random.nextInt(300) == 0 && this.world.getType(new BlockPosition(MathHelper.floor(this.locX), MathHelper.floor(this.locY) - 1, MathHelper.floor(this.locZ))).getBlock() == Blocks.GRASS) {
+                this.v(true);
             }
 
-            if (this.dn() && ++this.bP > 50) {
-                this.bP = 0;
-                this.u(false);
+            if (this.dr() && ++this.bQ > 50) {
+                this.bQ = 0;
+                this.v(false);
             }
 
-            if (this.dp() && !this.dc() && !this.dn()) {
+            if (this.dt() && !this.dg() && !this.dr()) {
                 EntityHorse entityhorse = this.a(this, 16.0D);
 
                 if (entityhorse != null && this.h((Entity) entityhorse) > 4.0D) {
@@ -690,7 +689,7 @@
                 }
             }
 
-            if (this.dH() && this.bW++ >= 18000) {
+            if (this.dL() && this.bX++ >= 18000) {
                 this.die();
             }
         }
@@ -701,118 +700,119 @@
         super.m();
         if (this.world.isClientSide && this.datawatcher.a()) {
             this.datawatcher.e();
-            this.dM();
+            this.dQ();
         }
 
-        if (this.bQ > 0 && ++this.bQ > 30) {
-            this.bQ = 0;
+        if (this.bR > 0 && ++this.bR > 30) {
+            this.bR = 0;
             this.c(128, false);
         }
 
-        if (this.by() && this.bR > 0 && ++this.bR > 20) {
-            this.bR = 0;
-            this.v(false);
+        if (this.bA() && this.bS > 0 && ++this.bS > 20) {
+            this.bS = 0;
+            this.w(false);
         }
 
-        if (this.bw > 0 && ++this.bw > 8) {
-            this.bw = 0;
+        if (this.bx > 0 && ++this.bx > 8) {
+            this.bx = 0;
         }
 
-        if (this.bx > 0) {
-            ++this.bx;
-            if (this.bx > 300) {
-                this.bx = 0;
+        if (this.by > 0) {
+            ++this.by;
+            if (this.by > 300) {
+                this.by = 0;
             }
         }
 
-        this.bY = this.bX;
-        if (this.dn()) {
-            this.bX += (1.0F - this.bX) * 0.4F + 0.05F;
-            if (this.bX > 1.0F) {
-                this.bX = 1.0F;
+        this.bZ = this.bY;
+        if (this.dr()) {
+            this.bY += (1.0F - this.bY) * 0.4F + 0.05F;
+            if (this.bY > 1.0F) {
+                this.bY = 1.0F;
             }
         } else {
-            this.bX += (0.0F - this.bX) * 0.4F - 0.05F;
-            if (this.bX < 0.0F) {
-                this.bX = 0.0F;
+            this.bY += (0.0F - this.bY) * 0.4F - 0.05F;
+            if (this.bY < 0.0F) {
+                this.bY = 0.0F;
             }
         }
 
-        this.ca = this.bZ;
-        if (this.do_()) {
-            this.bY = this.bX = 0.0F;
-            this.bZ += (1.0F - this.bZ) * 0.4F + 0.05F;
-            if (this.bZ > 1.0F) {
-                this.bZ = 1.0F;
+        this.cb = this.ca;
+        if (this.ds()) {
+            this.bY = 0.0F;
+            this.bZ = this.bY;
+            this.ca += (1.0F - this.ca) * 0.4F + 0.05F;
+            if (this.ca > 1.0F) {
+                this.ca = 1.0F;
             }
         } else {
             this.canSlide = false;
-            this.bZ += (0.8F * this.bZ * this.bZ * this.bZ - this.bZ) * 0.6F - 0.05F;
-            if (this.bZ < 0.0F) {
-                this.bZ = 0.0F;
+            this.ca += (0.8F * this.ca * this.ca * this.ca - this.ca) * 0.6F - 0.05F;
+            if (this.ca < 0.0F) {
+                this.ca = 0.0F;
             }
         }
 
-        this.cc = this.cb;
+        this.cd = this.cc;
         if (this.o(128)) {
-            this.cb += (1.0F - this.cb) * 0.7F + 0.05F;
-            if (this.cb > 1.0F) {
-                this.cb = 1.0F;
+            this.cc += (1.0F - this.cc) * 0.7F + 0.05F;
+            if (this.cc > 1.0F) {
+                this.cc = 1.0F;
             }
         } else {
-            this.cb += (0.0F - this.cb) * 0.7F - 0.05F;
-            if (this.cb < 0.0F) {
-                this.cb = 0.0F;
+            this.cc += (0.0F - this.cc) * 0.7F - 0.05F;
+            if (this.cc < 0.0F) {
+                this.cc = 0.0F;
             }
         }
 
     }
 
-    private void dP() {
+    private void dT() {
         if (!this.world.isClientSide) {
-            this.bQ = 1;
+            this.bR = 1;
             this.c(128, true);
         }
 
     }
 
-    private boolean dQ() {
-        return !this.isVehicle() && !this.isPassenger() && this.isTamed() && this.dc() && this.getType().i() && this.getHealth() >= this.getMaxHealth() && this.isInLove();
+    private boolean dU() {
+        return !this.isVehicle() && !this.isPassenger() && this.isTamed() && this.dg() && this.getType().i() && this.getHealth() >= this.getMaxHealth() && this.isInLove();
     }
 
-    public void u(boolean flag) {
+    public void v(boolean flag) {
         this.c(32, flag);
     }
 
-    public void v(boolean flag) {
+    public void w(boolean flag) {
         if (flag) {
-            this.u(false);
+            this.v(false);
         }
 
         this.c(64, flag);
     }
 
     private void setStanding() {
-        if (this.by()) {
-            this.bR = 1;
-            this.v(true);
+        if (this.bA()) {
+            this.bS = 1;
+            this.w(true);
         }
 
     }
 
-    public void dF() {
+    public void dJ() {
         this.setStanding();
-        SoundEffect soundeffect = this.dw();
+        SoundEffect soundeffect = this.dA();
 
         if (soundeffect != null) {
-            this.a(soundeffect, this.ce(), this.cf());
+            this.a(soundeffect, this.ch(), this.ci());
         }
 
     }
 
     public void dropChest() {
         this.a((Entity) this, this.inventoryChest);
-        this.dt();
+        this.dx();
     }
 
     private void a(Entity entity, InventoryHorseChest inventoryhorsechest) {
@@ -835,32 +835,34 @@
     }
 
     public void g(float f, float f1) {
-        if (this.isVehicle() && this.cL() && this.dv()) {
-            EntityLiving entityliving = (EntityLiving) this.bu();
+        if (this.isVehicle() && this.cP() && this.dz()) {
+            EntityLiving entityliving = (EntityLiving) this.bw();
 
-            this.lastYaw = this.yaw = entityliving.yaw;
+            this.yaw = entityliving.yaw;
+            this.lastYaw = this.yaw;
             this.pitch = entityliving.pitch * 0.5F;
             this.setYawPitch(this.yaw, this.pitch);
-            this.aP = this.aN = this.yaw;
-            f = entityliving.be * 0.5F;
-            f1 = entityliving.bf;
+            this.aO = this.yaw;
+            this.aQ = this.aO;
+            f = entityliving.bf * 0.5F;
+            f1 = entityliving.bg;
             if (f1 <= 0.0F) {
                 f1 *= 0.25F;
-                this.cd = 0;
+                this.ce = 0;
             }
 
-            if (this.onGround && this.jumpPower == 0.0F && this.do_() && !this.canSlide) {
+            if (this.onGround && this.jumpPower == 0.0F && this.ds() && !this.canSlide) {
                 f = 0.0F;
                 f1 = 0.0F;
             }
 
-            if (this.jumpPower > 0.0F && !this.dk() && this.onGround) {
+            if (this.jumpPower > 0.0F && !this.do_() && this.onGround) {
                 this.motY = this.getJumpStrength() * (double) this.jumpPower;
                 if (this.hasEffect(MobEffects.JUMP)) {
                     this.motY += (double) ((float) (this.getEffect(MobEffects.JUMP).getAmplifier() + 1) * 0.1F);
                 }
 
-                this.p(true);
+                this.q(true);
                 this.impulse = true;
                 if (f1 > 0.0F) {
                     float f2 = MathHelper.sin(this.yaw * 0.017453292F);
@@ -868,14 +870,14 @@
 
                     this.motX += (double) (-0.4F * f2 * this.jumpPower);
                     this.motZ += (double) (0.4F * f3 * this.jumpPower);
-                    this.a(SoundEffects.cw, 0.4F, 1.0F);
+                    this.a(SoundEffects.cx, 0.4F, 1.0F);
                 }
 
                 this.jumpPower = 0.0F;
             }
 
-            this.aR = this.cl() * 0.1F;
-            if (this.by()) {
+            this.aS = this.cp() * 0.1F;
+            if (this.bA()) {
                 this.l((float) this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).getValue());
                 super.g(f, f1);
             } else if (entityliving instanceof EntityHuman) {
@@ -886,10 +888,10 @@
 
             if (this.onGround) {
                 this.jumpPower = 0.0F;
-                this.p(false);
+                this.q(false);
             }
 
-            this.aF = this.aG;
+            this.aG = this.aH;
             double d0 = this.locX - this.lastX;
             double d1 = this.locZ - this.lastZ;
             float f4 = MathHelper.sqrt(d0 * d0 + d1 * d1) * 4.0F;
@@ -898,26 +900,32 @@
                 f4 = 1.0F;
             }
 
-            this.aG += (f4 - this.aG) * 0.4F;
-            this.aH += this.aG;
+            this.aH += (f4 - this.aH) * 0.4F;
+            this.aI += this.aH;
         } else {
-            this.aR = 0.02F;
+            this.aS = 0.02F;
             super.g(f, f1);
         }
     }
 
+    public static void b(DataConverterManager dataconvertermanager) {
+        EntityInsentient.a(dataconvertermanager, "EntityHorse");
+        dataconvertermanager.a(DataConverterTypes.ENTITY, (DataInspector) (new DataInspectorItemList("EntityHorse", new String[] { "Items"})));
+        dataconvertermanager.a(DataConverterTypes.ENTITY, (DataInspector) (new DataInspectorItem("EntityHorse", new String[] { "ArmorItem", "SaddleItem"})));
+    }
+
     public void b(NBTTagCompound nbttagcompound) {
         super.b(nbttagcompound);
-        nbttagcompound.setBoolean("EatingHaystack", this.dn());
+        nbttagcompound.setBoolean("EatingHaystack", this.dr());
         nbttagcompound.setBoolean("ChestedHorse", this.hasChest());
         nbttagcompound.setBoolean("HasReproduced", this.hasReproduced());
-        nbttagcompound.setBoolean("Bred", this.dp());
+        nbttagcompound.setBoolean("Bred", this.dt());
         nbttagcompound.setInt("Type", this.getType().k());
         nbttagcompound.setInt("Variant", this.getVariant());
         nbttagcompound.setInt("Temper", this.getTemper());
         nbttagcompound.setBoolean("Tame", this.isTamed());
-        nbttagcompound.setBoolean("SkeletonTrap", this.dH());
-        nbttagcompound.setInt("SkeletonTrapTime", this.bW);
+        nbttagcompound.setBoolean("SkeletonTrap", this.dL());
+        nbttagcompound.setInt("SkeletonTrapTime", this.bX);
         if (this.getOwnerUUID() != null) {
             nbttagcompound.setString("OwnerUUID", this.getOwnerUUID().toString());
         }
@@ -952,17 +960,17 @@
 
     public void a(NBTTagCompound nbttagcompound) {
         super.a(nbttagcompound);
-        this.u(nbttagcompound.getBoolean("EatingHaystack"));
-        this.q(nbttagcompound.getBoolean("Bred"));
+        this.v(nbttagcompound.getBoolean("EatingHaystack"));
+        this.r(nbttagcompound.getBoolean("Bred"));
         this.setHasChest(nbttagcompound.getBoolean("ChestedHorse"));
-        this.s(nbttagcompound.getBoolean("HasReproduced"));
+        this.t(nbttagcompound.getBoolean("HasReproduced"));
         this.setType(EnumHorseType.a(nbttagcompound.getInt("Type")));
         this.setVariant(nbttagcompound.getInt("Variant"));
         this.setTemper(nbttagcompound.getInt("Temper"));
         this.setTame(nbttagcompound.getBoolean("Tame"));
-        this.x(nbttagcompound.getBoolean("SkeletonTrap"));
-        this.bW = nbttagcompound.getInt("SkeletonTrapTime");
-        String s = "";
+        this.y(nbttagcompound.getBoolean("SkeletonTrap"));
+        this.bX = nbttagcompound.getInt("SkeletonTrapTime");
+        String s;
 
         if (nbttagcompound.hasKeyOfType("OwnerUUID", 8)) {
             s = nbttagcompound.getString("OwnerUUID");
@@ -1013,7 +1021,7 @@
             }
         }
 
-        this.dL();
+        this.dP();
     }
 
     public boolean mate(EntityAnimal entityanimal) {
@@ -1024,7 +1032,7 @@
         } else {
             EntityHorse entityhorse = (EntityHorse) entityanimal;
 
-            if (this.dQ() && entityhorse.dQ()) {
+            if (this.dU() && entityhorse.dU()) {
                 EnumHorseType enumhorsetype = this.getType();
                 EnumHorseType enumhorsetype1 = entityhorse.getType();
 
@@ -1074,13 +1082,13 @@
         }
 
         entityhorse1.setType(enumhorsetype2);
-        double d0 = this.getAttributeInstance(GenericAttributes.maxHealth).b() + entityageable.getAttributeInstance(GenericAttributes.maxHealth).b() + (double) this.dS();
+        double d0 = this.getAttributeInstance(GenericAttributes.maxHealth).b() + entityageable.getAttributeInstance(GenericAttributes.maxHealth).b() + (double) this.dW();
 
         entityhorse1.getAttributeInstance(GenericAttributes.maxHealth).setValue(d0 / 3.0D);
-        double d1 = this.getAttributeInstance(EntityHorse.attributeJumpStrength).b() + entityageable.getAttributeInstance(EntityHorse.attributeJumpStrength).b() + this.dT();
+        double d1 = this.getAttributeInstance(EntityHorse.attributeJumpStrength).b() + entityageable.getAttributeInstance(EntityHorse.attributeJumpStrength).b() + this.dX();
 
         entityhorse1.getAttributeInstance(EntityHorse.attributeJumpStrength).setValue(d1 / 3.0D);
-        double d2 = this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).b() + entityageable.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).b() + this.dU();
+        double d2 = this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).b() + entityageable.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).b() + this.dY();
 
         entityhorse1.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).setValue(d2 / 3.0D);
         return entityhorse1;
@@ -1089,8 +1097,8 @@
     @Nullable
     public GroupDataEntity prepare(DifficultyDamageScaler difficultydamagescaler, @Nullable GroupDataEntity groupdataentity) {
         Object object = super.prepare(difficultydamagescaler, groupdataentity);
-        EnumHorseType enumhorsetype = EnumHorseType.HORSE;
         int i = 0;
+        EnumHorseType enumhorsetype;
 
         if (object instanceof EntityHorse.a) {
             enumhorsetype = ((EntityHorse.a) object).a;
@@ -1119,9 +1127,9 @@
             this.getAttributeInstance(GenericAttributes.maxHealth).setValue(15.0D);
             this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).setValue(0.20000000298023224D);
         } else {
-            this.getAttributeInstance(GenericAttributes.maxHealth).setValue((double) this.dS());
+            this.getAttributeInstance(GenericAttributes.maxHealth).setValue((double) this.dW());
             if (enumhorsetype == EnumHorseType.HORSE) {
-                this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).setValue(this.dU());
+                this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).setValue(this.dY());
             } else {
                 this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).setValue(0.17499999701976776D);
             }
@@ -1130,21 +1138,21 @@
         if (enumhorsetype.g()) {
             this.getAttributeInstance(EntityHorse.attributeJumpStrength).setValue(0.5D);
         } else {
-            this.getAttributeInstance(EntityHorse.attributeJumpStrength).setValue(this.dT());
+            this.getAttributeInstance(EntityHorse.attributeJumpStrength).setValue(this.dX());
         }
 
         this.setHealth(this.getMaxHealth());
         return (GroupDataEntity) object;
     }
 
-    public boolean cL() {
-        Entity entity = this.bu();
+    public boolean cP() {
+        Entity entity = this.bw();
 
         return entity instanceof EntityLiving;
     }
 
     public boolean b() {
-        return this.dv();
+        return this.dz();
     }
 
     public void b(int i) {
@@ -1159,18 +1167,18 @@
         if (entity instanceof EntityInsentient) {
             EntityInsentient entityinsentient = (EntityInsentient) entity;
 
-            this.aN = entityinsentient.aN;
+            this.aO = entityinsentient.aO;
         }
 
-        if (this.ca > 0.0F) {
-            float f = MathHelper.sin(this.aN * 0.017453292F);
-            float f1 = MathHelper.cos(this.aN * 0.017453292F);
-            float f2 = 0.7F * this.ca;
-            float f3 = 0.15F * this.ca;
+        if (this.cb > 0.0F) {
+            float f = MathHelper.sin(this.aO * 0.017453292F);
+            float f1 = MathHelper.cos(this.aO * 0.017453292F);
+            float f2 = 0.7F * this.cb;
+            float f3 = 0.15F * this.cb;
 
             entity.setPosition(this.locX + (double) (f2 * f), this.locY + this.ay() + entity.ax() + (double) f3, this.locZ - (double) (f2 * f1));
             if (entity instanceof EntityLiving) {
-                ((EntityLiving) entity).aN = this.aN;
+                ((EntityLiving) entity).aO = this.aO;
             }
         }
 
@@ -1188,35 +1196,35 @@
         return d0;
     }
 
-    private float dS() {
+    private float dW() {
         return 15.0F + (float) this.random.nextInt(8) + (float) this.random.nextInt(9);
     }
 
-    private double dT() {
+    private double dX() {
         return 0.4000000059604645D + this.random.nextDouble() * 0.2D + this.random.nextDouble() * 0.2D + this.random.nextDouble() * 0.2D;
     }
 
-    private double dU() {
+    private double dY() {
         return (0.44999998807907104D + this.random.nextDouble() * 0.3D + this.random.nextDouble() * 0.3D + this.random.nextDouble() * 0.3D) * 0.25D;
     }
 
-    public boolean dH() {
-        return this.bV;
+    public boolean dL() {
+        return this.bW;
     }
 
-    public void x(boolean flag) {
-        if (flag != this.bV) {
-            this.bV = flag;
+    public void y(boolean flag) {
+        if (flag != this.bW) {
+            this.bW = flag;
             if (flag) {
-                this.goalSelector.a(1, this.bO);
+                this.goalSelector.a(1, this.bP);
             } else {
-                this.goalSelector.a((PathfinderGoal) this.bO);
+                this.goalSelector.a((PathfinderGoal) this.bP);
             }
         }
 
     }
 
-    public boolean n_() {
+    public boolean m_() {
         return false;
     }
 
@@ -1248,7 +1256,7 @@
                 return false;
             } else {
                 this.inventoryChest.setItem(j, itemstack);
-                this.dL();
+                this.dP();
                 return true;
             }
         } else {
@@ -1264,8 +1272,8 @@
     }
 
     @Nullable
-    public Entity bu() {
-        return this.bv().isEmpty() ? null : (Entity) this.bv().get(0);
+    public Entity bw() {
+        return this.bx().isEmpty() ? null : (Entity) this.bx().get(0);
     }
 
     public EnumMonsterType getMonsterType() {
