--- a/net/minecraft/server/EntityFireball.java
+++ b/net/minecraft/server/EntityFireball.java
@@ -6,10 +6,10 @@
     private int f = -1;
     private int g = -1;
     private Block h;
-    private boolean at;
+    private boolean au;
     public EntityLiving shooter;
-    private int au;
     private int av;
+    private int aw;
     public double dirX;
     public double dirY;
     public double dirZ;
@@ -39,7 +39,9 @@
         this.setSize(1.0F, 1.0F);
         this.setPositionRotation(entityliving.locX, entityliving.locY, entityliving.locZ, entityliving.yaw, entityliving.pitch);
         this.setPosition(this.locX, this.locY, this.locZ);
-        this.motX = this.motY = this.motZ = 0.0D;
+        this.motX = 0.0D;
+        this.motY = 0.0D;
+        this.motZ = 0.0D;
         d0 += this.random.nextGaussian() * 0.4D;
         d1 += this.random.nextGaussian() * 0.4D;
         d2 += this.random.nextGaussian() * 0.4D;
@@ -59,27 +61,27 @@
                 this.setOnFire(1);
             }
 
-            if (this.at) {
+            if (this.au) {
                 if (this.world.getType(new BlockPosition(this.e, this.f, this.g)).getBlock() == this.h) {
-                    ++this.au;
-                    if (this.au == 600) {
+                    ++this.av;
+                    if (this.av == 600) {
                         this.die();
                     }
 
                     return;
                 }
 
-                this.at = false;
+                this.au = false;
                 this.motX *= (double) (this.random.nextFloat() * 0.2F);
                 this.motY *= (double) (this.random.nextFloat() * 0.2F);
                 this.motZ *= (double) (this.random.nextFloat() * 0.2F);
-                this.au = 0;
                 this.av = 0;
+                this.aw = 0;
             } else {
-                ++this.av;
+                ++this.aw;
             }
 
-            MovingObjectPosition movingobjectposition = ProjectileHelper.a(this, true, this.av >= 25, this.shooter);
+            MovingObjectPosition movingobjectposition = ProjectileHelper.a(this, true, this.aw >= 25, this.shooter);
 
             if (movingobjectposition != null) {
                 this.a(movingobjectposition);
@@ -95,7 +97,7 @@
                 for (int i = 0; i < 4; ++i) {
                     float f1 = 0.25F;
 
-                    this.world.addParticle(EnumParticle.WATER_BUBBLE, this.locX - this.motX * (double) f1, this.locY - this.motY * (double) f1, this.locZ - this.motZ * (double) f1, this.motX, this.motY, this.motZ, new int[0]);
+                    this.world.addParticle(EnumParticle.WATER_BUBBLE, this.locX - this.motX * 0.25D, this.locY - this.motY * 0.25D, this.locZ - this.motZ * 0.25D, this.motX, this.motY, this.motZ, new int[0]);
                 }
 
                 f = 0.8F;
@@ -126,6 +128,8 @@
 
     protected abstract void a(MovingObjectPosition movingobjectposition);
 
+    public static void a(DataConverterManager dataconvertermanager, String s) {}
+
     public void b(NBTTagCompound nbttagcompound) {
         nbttagcompound.setInt("xTile", this.e);
         nbttagcompound.setInt("yTile", this.f);
@@ -133,10 +137,10 @@
         MinecraftKey minecraftkey = (MinecraftKey) Block.REGISTRY.b(this.h);
 
         nbttagcompound.setString("inTile", minecraftkey == null ? "" : minecraftkey.toString());
-        nbttagcompound.setByte("inGround", (byte) (this.at ? 1 : 0));
+        nbttagcompound.setByte("inGround", (byte) (this.au ? 1 : 0));
         nbttagcompound.set("direction", this.a(new double[] { this.motX, this.motY, this.motZ}));
         nbttagcompound.set("power", this.a(new double[] { this.dirX, this.dirY, this.dirZ}));
-        nbttagcompound.setInt("life", this.au);
+        nbttagcompound.setInt("life", this.av);
     }
 
     public void a(NBTTagCompound nbttagcompound) {
@@ -149,7 +153,7 @@
             this.h = Block.getById(nbttagcompound.getByte("inTile") & 255);
         }
 
-        this.at = nbttagcompound.getByte("inGround") == 1;
+        this.au = nbttagcompound.getByte("inGround") == 1;
         NBTTagList nbttaglist;
 
         if (nbttagcompound.hasKeyOfType("power", 9)) {
@@ -161,7 +165,7 @@
             }
         }
 
-        this.au = nbttagcompound.getInt("life");
+        this.av = nbttagcompound.getInt("life");
         if (nbttagcompound.hasKeyOfType("direction", 9) && nbttagcompound.getList("direction", 6).size() == 3) {
             nbttaglist = nbttagcompound.getList("direction", 6);
             this.motX = nbttaglist.e(0);
@@ -185,7 +189,7 @@
         if (this.isInvulnerable(damagesource)) {
             return false;
         } else {
-            this.ao();
+            this.ap();
             if (damagesource.getEntity() != null) {
                 Vec3D vec3d = damagesource.getEntity().aB();
 
