--- a/net/minecraft/server/EntitySmallFireball.java
+++ b/net/minecraft/server/EntitySmallFireball.java
@@ -17,6 +17,10 @@
         this.setSize(0.3125F, 0.3125F);
     }
 
+    public static void a(DataConverterManager dataconvertermanager) {
+        EntityFireball.a(dataconvertermanager, "SmallFireball");
+    }
+
     protected void a(MovingObjectPosition movingobjectposition) {
         if (!this.world.isClientSide) {
             boolean flag;
