--- a/net/minecraft/server/EntityWaterAnimal.java
+++ b/net/minecraft/server/EntityWaterAnimal.java
@@ -6,11 +6,11 @@
         super(world);
     }
 
-    public boolean bC() {
+    public boolean bE() {
         return true;
     }
 
-    public boolean cG() {
+    public boolean cK() {
         return true;
     }
 
@@ -47,7 +47,7 @@
 
     }
 
-    public boolean be() {
+    public boolean bg() {
         return false;
     }
 }
