--- a/net/minecraft/server/HttpUtilities.java
+++ b/net/minecraft/server/HttpUtilities.java
@@ -95,7 +95,7 @@
             return stringbuffer.toString();
         } catch (Exception exception) {
             if (!flag) {
-                HttpUtilities.c.error("Could not post to " + url, exception);
+                HttpUtilities.c.error("Could not post to {}", new Object[] { url, exception});
             }
 
             return "";
