--- a/net/minecraft/server/InventoryMerchant.java
+++ b/net/minecraft/server/InventoryMerchant.java
@@ -5,7 +5,7 @@
 public class InventoryMerchant implements IInventory {
 
     private final IMerchant merchant;
-    private ItemStack[] itemsInSlots = new ItemStack[3];
+    private final ItemStack[] itemsInSlots = new ItemStack[3];
     private final EntityHuman player;
     private MerchantRecipe recipe;
     public int selectedIndex;
