--- a/net/minecraft/server/ItemFlintAndSteel.java
+++ b/net/minecraft/server/ItemFlintAndSteel.java
@@ -14,7 +14,7 @@
             return EnumInteractionResult.FAIL;
         } else {
             if (world.getType(blockposition).getMaterial() == Material.AIR) {
-                world.a(entityhuman, blockposition, SoundEffects.bx, SoundCategory.BLOCKS, 1.0F, ItemFlintAndSteel.j.nextFloat() * 0.4F + 0.8F);
+                world.a(entityhuman, blockposition, SoundEffects.by, SoundCategory.BLOCKS, 1.0F, ItemFlintAndSteel.j.nextFloat() * 0.4F + 0.8F);
                 world.setTypeAndData(blockposition, Blocks.FIRE.getBlockData(), 11);
             }
 
