--- a/net/minecraft/server/ItemLeash.java
+++ b/net/minecraft/server/ItemLeash.java
@@ -30,7 +30,7 @@
         int i = blockposition.getX();
         int j = blockposition.getY();
         int k = blockposition.getZ();
-        List list = world.a(EntityInsentient.class, new AxisAlignedBB((double) i - d0, (double) j - d0, (double) k - d0, (double) i + d0, (double) j + d0, (double) k + d0));
+        List list = world.a(EntityInsentient.class, new AxisAlignedBB((double) i - 7.0D, (double) j - 7.0D, (double) k - 7.0D, (double) i + 7.0D, (double) j + 7.0D, (double) k + 7.0D));
         Iterator iterator = list.iterator();
 
         while (iterator.hasNext()) {
