--- a/net/minecraft/server/LegacyPingHandler.java
+++ b/net/minecraft/server/LegacyPingHandler.java
@@ -13,7 +13,7 @@
 public class LegacyPingHandler extends ChannelInboundHandlerAdapter {
 
     private static final Logger a = LogManager.getLogger();
-    private ServerConnection b;
+    private final ServerConnection b;
 
     public LegacyPingHandler(ServerConnection serverconnection) {
         this.b = serverconnection;
