--- a/net/minecraft/server/MethodProfiler.java
+++ b/net/minecraft/server/MethodProfiler.java
@@ -54,10 +54,10 @@
             }
 
             if (k > 100000000L) {
-                MethodProfiler.b.warn("Something\'s taking too long! \'" + this.e + "\' took aprox " + (double) k / 1000000.0D + " ms");
+                MethodProfiler.b.warn("Something\'s taking too long! \'{}\' took aprox {} ms", new Object[] { this.e, Double.valueOf((double) k / 1000000.0D)});
             }
 
-            this.e = !this.c.isEmpty() ? (String) this.c.get(this.c.size() - 1) : "";
+            this.e = this.c.isEmpty() ? "" : (String) this.c.get(this.c.size() - 1);
         }
     }
 
