--- a/net/minecraft/server/MojangStatisticsGenerator.java
+++ b/net/minecraft/server/MojangStatisticsGenerator.java
@@ -73,7 +73,7 @@
         this.b("os_version", System.getProperty("os.version"));
         this.b("os_architecture", System.getProperty("os.arch"));
         this.b("java_version", System.getProperty("java.version"));
-        this.a("version", "1.9.4");
+        this.a("version", "1.10");
         this.e.b(this);
     }
 
