--- a/net/minecraft/server/NBTTagByte.java
+++ b/net/minecraft/server/NBTTagByte.java
@@ -4,7 +4,7 @@
 import java.io.DataOutput;
 import java.io.IOException;
 
-public class NBTTagByte extends NBTBase.NBTNumber {
+public class NBTTagByte extends NBTNumber {
 
     private byte data;
 
@@ -31,7 +31,7 @@
         return "" + this.data + "b";
     }
 
-    public NBTBase clone() {
+    public NBTTagByte c() {
         return new NBTTagByte(this.data);
     }
 
@@ -49,27 +49,31 @@
         return super.hashCode() ^ this.data;
     }
 
-    public long c() {
+    public long d() {
         return (long) this.data;
     }
 
-    public int d() {
+    public int e() {
         return this.data;
     }
 
-    public short e() {
+    public short f() {
         return (short) this.data;
     }
 
-    public byte f() {
+    public byte g() {
         return this.data;
     }
 
-    public double g() {
+    public double h() {
         return (double) this.data;
     }
 
-    public float h() {
+    public float i() {
         return (float) this.data;
     }
+
+    public NBTBase clone() {
+        return this.c();
+    }
 }
