--- a/net/minecraft/server/NBTTagIntArray.java
+++ b/net/minecraft/server/NBTTagIntArray.java
@@ -17,9 +17,13 @@
 
     void write(DataOutput dataoutput) throws IOException {
         dataoutput.writeInt(this.data.length);
+        int[] aint = this.data;
+        int i = aint.length;
+
+        for (int j = 0; j < i; ++j) {
+            int k = aint[j];
 
-        for (int i = 0; i < this.data.length; ++i) {
-            dataoutput.writeInt(this.data[i]);
+            dataoutput.writeInt(k);
         }
 
     }
@@ -55,7 +59,7 @@
         return s + "]";
     }
 
-    public NBTBase clone() {
+    public NBTTagIntArray c() {
         int[] aint = new int[this.data.length];
 
         System.arraycopy(this.data, 0, aint, 0, this.data.length);
@@ -70,7 +74,11 @@
         return super.hashCode() ^ Arrays.hashCode(this.data);
     }
 
-    public int[] c() {
+    public int[] d() {
         return this.data;
     }
+
+    public NBTBase clone() {
+        return this.c();
+    }
 }
