--- a/net/minecraft/server/NoiseGenerator3Handler.java
+++ b/net/minecraft/server/NoiseGenerator3Handler.java
@@ -4,9 +4,9 @@
 
 public class NoiseGenerator3Handler {
 
-    private static int[][] e = new int[][] { { 1, 1, 0}, { -1, 1, 0}, { 1, -1, 0}, { -1, -1, 0}, { 1, 0, 1}, { -1, 0, 1}, { 1, 0, -1}, { -1, 0, -1}, { 0, 1, 1}, { 0, -1, 1}, { 0, 1, -1}, { 0, -1, -1}};
+    private static final int[][] e = new int[][] { { 1, 1, 0}, { -1, 1, 0}, { 1, -1, 0}, { -1, -1, 0}, { 1, 0, 1}, { -1, 0, 1}, { 1, 0, -1}, { -1, 0, -1}, { 0, 1, 1}, { 0, -1, 1}, { 0, 1, -1}, { 0, -1, -1}};
     public static final double a = Math.sqrt(3.0D);
-    private int[] f;
+    private final int[] f;
     public double b;
     public double c;
     public double d;
