--- a/net/minecraft/server/PacketPlayOutCustomSoundEffect.java
+++ b/net/minecraft/server/PacketPlayOutCustomSoundEffect.java
@@ -11,7 +11,7 @@
     private int d = Integer.MAX_VALUE;
     private int e;
     private float f;
-    private int g;
+    private float g;
 
     public PacketPlayOutCustomSoundEffect() {}
 
@@ -23,8 +23,7 @@
         this.d = (int) (d1 * 8.0D);
         this.e = (int) (d2 * 8.0D);
         this.f = f;
-        this.g = (int) (f1 * 63.0F);
-        f1 = MathHelper.a(f1, 0.0F, 255.0F);
+        this.g = f1;
     }
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
@@ -34,7 +33,7 @@
         this.d = packetdataserializer.readInt();
         this.e = packetdataserializer.readInt();
         this.f = packetdataserializer.readFloat();
-        this.g = packetdataserializer.readUnsignedByte();
+        this.g = packetdataserializer.readFloat();
     }
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
@@ -44,7 +43,7 @@
         packetdataserializer.writeInt(this.d);
         packetdataserializer.writeInt(this.e);
         packetdataserializer.writeFloat(this.f);
-        packetdataserializer.writeByte(this.g);
+        packetdataserializer.writeFloat(this.g);
     }
 
     public void a(PacketListenerPlayOut packetlistenerplayout) {
