--- a/net/minecraft/server/PathfinderGoalEatTile.java
+++ b/net/minecraft/server/PathfinderGoalEatTile.java
@@ -6,8 +6,8 @@
 public class PathfinderGoalEatTile extends PathfinderGoal {
 
     private static final Predicate<IBlockData> b = BlockStatePredicate.a((Block) Blocks.TALLGRASS).a(BlockLongGrass.TYPE, Predicates.equalTo(BlockLongGrass.EnumTallGrassType.GRASS));
-    private EntityInsentient c;
-    private World d;
+    private final EntityInsentient c;
+    private final World d;
     int a;
 
     public PathfinderGoalEatTile(EntityInsentient entityinsentient) {
@@ -54,7 +54,7 @@
                     this.d.setAir(blockposition, false);
                 }
 
-                this.c.B();
+                this.c.A();
             } else {
                 BlockPosition blockposition1 = blockposition.down();
 
@@ -64,7 +64,7 @@
                         this.d.setTypeAndData(blockposition1, Blocks.DIRT.getBlockData(), 2);
                     }
 
-                    this.c.B();
+                    this.c.A();
                 }
             }
 
