--- a/net/minecraft/server/PathfinderGoalRandomLookaround.java
+++ b/net/minecraft/server/PathfinderGoalRandomLookaround.java
@@ -2,7 +2,7 @@
 
 public class PathfinderGoalRandomLookaround extends PathfinderGoal {
 
-    private EntityInsentient a;
+    private final EntityInsentient a;
     private double b;
     private double c;
     private int d;
@@ -30,6 +30,6 @@
 
     public void e() {
         --this.d;
-        this.a.getControllerLook().a(this.a.locX + this.b, this.a.locY + (double) this.a.getHeadHeight(), this.a.locZ + this.c, (float) this.a.cF(), (float) this.a.N());
+        this.a.getControllerLook().a(this.a.locX + this.b, this.a.locY + (double) this.a.getHeadHeight(), this.a.locZ + this.c, (float) this.a.cJ(), (float) this.a.N());
     }
 }
