--- a/net/minecraft/server/PathfinderGoalDefendVillage.java
+++ b/net/minecraft/server/PathfinderGoalDefendVillage.java
@@ -20,15 +20,13 @@
             this.b = village.b((EntityLiving) this.a);
             if (this.b instanceof EntityCreeper) {
                 return false;
-            } else if (!this.a(this.b, false)) {
-                if (this.e.getRandom().nextInt(20) == 0) {
-                    this.b = village.c((EntityLiving) this.a);
-                    return this.a(this.b, false);
-                } else {
-                    return false;
-                }
-            } else {
+            } else if (this.a(this.b, false)) {
                 return true;
+            } else if (this.e.getRandom().nextInt(20) == 0) {
+                this.b = village.c((EntityLiving) this.a);
+                return this.a(this.b, false);
+            } else {
+                return false;
             }
         }
     }
