--- a/net/minecraft/server/PathfinderGoalRandomTargetNonTamed.java
+++ b/net/minecraft/server/PathfinderGoalRandomTargetNonTamed.java
@@ -4,7 +4,7 @@
 
 public class PathfinderGoalRandomTargetNonTamed<T extends EntityLiving> extends PathfinderGoalNearestAttackableTarget<T> {
 
-    private EntityTameableAnimal i;
+    private final EntityTameableAnimal i;
 
     public PathfinderGoalRandomTargetNonTamed(EntityTameableAnimal entitytameableanimal, Class<T> oclass, boolean flag, Predicate<? super T> predicate) {
         super(entitytameableanimal, oclass, 10, flag, false, predicate);
