--- a/net/minecraft/server/PathfinderGoalHurtByTarget.java
+++ b/net/minecraft/server/PathfinderGoalHurtByTarget.java
@@ -17,7 +17,7 @@
     }
 
     public boolean a() {
-        int i = this.e.bI();
+        int i = this.e.bK();
         EntityLiving entityliving = this.e.getLastDamager();
 
         return i != this.b && entityliving != null && this.a(entityliving, false);
@@ -26,38 +26,43 @@
     public void c() {
         this.e.setGoalTarget(this.e.getLastDamager());
         this.g = this.e.getGoalTarget();
-        this.b = this.e.bI();
+        this.b = this.e.bK();
         this.h = 300;
         if (this.a) {
-            double d0 = this.f();
-            List list = this.e.world.a(this.e.getClass(), (new AxisAlignedBB(this.e.locX, this.e.locY, this.e.locZ, this.e.locX + 1.0D, this.e.locY + 1.0D, this.e.locZ + 1.0D)).grow(d0, 10.0D, d0));
-            Iterator iterator = list.iterator();
-
-            while (iterator.hasNext()) {
-                EntityCreature entitycreature = (EntityCreature) iterator.next();
-
-                if (this.e != entitycreature && entitycreature.getGoalTarget() == null && (!(this.e instanceof EntityTameableAnimal) || ((EntityTameableAnimal) this.e).getOwner() == ((EntityTameableAnimal) entitycreature).getOwner()) && !entitycreature.r(this.e.getLastDamager())) {
-                    boolean flag = false;
-                    Class[] aclass = this.c;
-                    int i = aclass.length;
-
-                    for (int j = 0; j < i; ++j) {
-                        Class oclass = aclass[j];
-
-                        if (entitycreature.getClass() == oclass) {
-                            flag = true;
-                            break;
-                        }
-                    }
+            this.f();
+        }
+
+        super.c();
+    }
 
-                    if (!flag) {
-                        this.a(entitycreature, this.e.getLastDamager());
+    protected void f() {
+        double d0 = this.i();
+        List list = this.e.world.a(this.e.getClass(), (new AxisAlignedBB(this.e.locX, this.e.locY, this.e.locZ, this.e.locX + 1.0D, this.e.locY + 1.0D, this.e.locZ + 1.0D)).grow(d0, 10.0D, d0));
+        Iterator iterator = list.iterator();
+
+        while (iterator.hasNext()) {
+            EntityCreature entitycreature = (EntityCreature) iterator.next();
+
+            if (this.e != entitycreature && entitycreature.getGoalTarget() == null && (!(this.e instanceof EntityTameableAnimal) || ((EntityTameableAnimal) this.e).getOwner() == ((EntityTameableAnimal) entitycreature).getOwner()) && !entitycreature.r(this.e.getLastDamager())) {
+                boolean flag = false;
+                Class[] aclass = this.c;
+                int i = aclass.length;
+
+                for (int j = 0; j < i; ++j) {
+                    Class oclass = aclass[j];
+
+                    if (entitycreature.getClass() == oclass) {
+                        flag = true;
+                        break;
                     }
                 }
+
+                if (!flag) {
+                    this.a(entitycreature, this.e.getLastDamager());
+                }
             }
         }
 
-        super.c();
     }
 
     protected void a(EntityCreature entitycreature, EntityLiving entityliving) {
