--- a/net/minecraft/server/PathfinderGoalBowShoot.java
+++ b/net/minecraft/server/PathfinderGoalBowShoot.java
@@ -42,11 +42,11 @@
     }
 
     public void d() {
-        super.c();
+        super.d();
         this.a.a(false);
         this.f = 0;
         this.e = -1;
-        this.a.cA();
+        this.a.cE();
     }
 
     public void e() {
@@ -100,14 +100,14 @@
                 this.a.getControllerLook().a(entityliving, 30.0F, 30.0F);
             }
 
-            if (this.a.ct()) {
+            if (this.a.cx()) {
                 if (!flag && this.f < -60) {
-                    this.a.cA();
+                    this.a.cE();
                 } else if (flag) {
-                    int i = this.a.cy();
+                    int i = this.a.cC();
 
                     if (i >= 20) {
-                        this.a.cA();
+                        this.a.cE();
                         this.a.a(entityliving, ItemBow.b(i));
                         this.e = this.c;
                     }
