--- a/net/minecraft/server/RecipesWeapons.java
+++ b/net/minecraft/server/RecipesWeapons.java
@@ -2,8 +2,8 @@
 
 public class RecipesWeapons {
 
-    private String[][] a = new String[][] { { "X", "X", "#"}};
-    private Object[][] b;
+    private final String[][] a = new String[][] { { "X", "X", "#"}};
+    private final Object[][] b;
 
     public RecipesWeapons() {
         this.b = new Object[][] { { Blocks.PLANKS, Blocks.COBBLESTONE, Items.IRON_INGOT, Items.DIAMOND, Items.GOLD_INGOT}, { Items.WOODEN_SWORD, Items.STONE_SWORD, Items.IRON_SWORD, Items.DIAMOND_SWORD, Items.GOLDEN_SWORD}};
