--- a/net/minecraft/server/RecipeBookClone.java
+++ b/net/minecraft/server/RecipeBookClone.java
@@ -61,7 +61,7 @@
         if (itemstack != null && i >= 1 && ItemWrittenBook.h(itemstack) < 2) {
             ItemStack itemstack2 = new ItemStack(Items.WRITTEN_BOOK, i);
 
-            itemstack2.setTag((NBTTagCompound) itemstack.getTag().clone());
+            itemstack2.setTag(itemstack.getTag().g());
             itemstack2.getTag().setInt("generation", ItemWrittenBook.h(itemstack) + 1);
             if (itemstack.hasName()) {
                 itemstack2.c(itemstack.getName());
