--- a/net/minecraft/server/SharedConstants.java
+++ b/net/minecraft/server/SharedConstants.java
@@ -6,6 +6,7 @@
 public class SharedConstants {
 
     public static final Level a = Level.DISABLED;
+    public static final char[] b = new char[] { '.', '\n', '\r', '\t', '\u0000', '\f', '`', '?', '*', '\\', '<', '>', '|', '\"'};
     public static final char[] allowedCharacters = new char[] { '/', '\n', '\r', '\t', '\u0000', '\f', '`', '?', '*', '\\', '<', '>', '|', '\"', ':'};
 
     public static boolean isAllowedChatCharacter(char c0) {
