--- a/net/minecraft/server/WorldGenDungeons.java
+++ b/net/minecraft/server/WorldGenDungeons.java
@@ -125,7 +125,7 @@
             if (tileentity1 instanceof TileEntityMobSpawner) {
                 ((TileEntityMobSpawner) tileentity1).getSpawner().setMobName(this.a(random));
             } else {
-                WorldGenDungeons.a.error("Failed to fetch mob spawner entity at (" + blockposition.getX() + ", " + blockposition.getY() + ", " + blockposition.getZ() + ")");
+                WorldGenDungeons.a.error("Failed to fetch mob spawner entity at ({}, {}, {})", new Object[] { Integer.valueOf(blockposition.getX()), Integer.valueOf(blockposition.getY()), Integer.valueOf(blockposition.getZ())});
             }
 
             return true;
