--- a/net/minecraft/server/WorldNBTStorage.java
+++ b/net/minecraft/server/WorldNBTStorage.java
@@ -137,8 +137,8 @@
     public void save(EntityHuman entityhuman) {
         try {
             NBTTagCompound nbttagcompound = entityhuman.e(new NBTTagCompound());
-            File file = new File(this.playerDir, entityhuman.bd() + ".dat.tmp");
-            File file1 = new File(this.playerDir, entityhuman.bd() + ".dat");
+            File file = new File(this.playerDir, entityhuman.bf() + ".dat.tmp");
+            File file1 = new File(this.playerDir, entityhuman.bf() + ".dat");
 
             NBTCompressedStreamTools.a(nbttagcompound, (OutputStream) (new FileOutputStream(file)));
             if (file1.exists()) {
@@ -147,7 +147,7 @@
 
             file.renameTo(file1);
         } catch (Exception exception) {
-            WorldNBTStorage.b.warn("Failed to save player data for " + entityhuman.getName());
+            WorldNBTStorage.b.warn("Failed to save player data for {}", new Object[] { entityhuman.getName()});
         }
 
     }
@@ -156,13 +156,13 @@
         NBTTagCompound nbttagcompound = null;
 
         try {
-            File file = new File(this.playerDir, entityhuman.bd() + ".dat");
+            File file = new File(this.playerDir, entityhuman.bf() + ".dat");
 
             if (file.exists() && file.isFile()) {
                 nbttagcompound = NBTCompressedStreamTools.a((InputStream) (new FileInputStream(file)));
             }
         } catch (Exception exception) {
-            WorldNBTStorage.b.warn("Failed to load player data for " + entityhuman.getName());
+            WorldNBTStorage.b.warn("Failed to load player data for {}", new Object[] { entityhuman.getName()});
         }
 
         if (nbttagcompound != null) {
