--- a/net/minecraft/server/WorldGenEndCityPieces.java
+++ b/net/minecraft/server/WorldGenEndCityPieces.java
@@ -8,7 +8,7 @@
 
 public class WorldGenEndCityPieces {
 
-    public static final DefinedStructureManager a = new DefinedStructureManager();
+    public static final DefinedStructureManager a = new DefinedStructureManager("structures");
     private static final DefinedStructureInfo b = (new DefinedStructureInfo()).a(true);
     private static final DefinedStructureInfo c = (new DefinedStructureInfo()).a(true).a(Blocks.AIR);
     private static final WorldGenEndCityPieces.PieceGenerator d = new WorldGenEndCityPieces.PieceGenerator() {
@@ -19,21 +19,19 @@
                 return false;
             } else {
                 EnumBlockRotation enumblockrotation = worldgenendcitypieces_piece.b.c();
-                WorldGenEndCityPieces.Piece worldgenendcitypieces_piece1;
-
-                list.add(worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece, blockposition, "base_floor", enumblockrotation, true));
+                WorldGenEndCityPieces.Piece worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece, blockposition, "base_floor", enumblockrotation, true));
                 int j = random.nextInt(3);
 
                 if (j == 0) {
-                    list.add(WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-1, 4, -1), "base_roof", enumblockrotation, true));
+                    WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-1, 4, -1), "base_roof", enumblockrotation, true));
                 } else if (j == 1) {
-                    list.add(worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-1, 0, -1), "second_floor_2", enumblockrotation, false));
-                    list.add(worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-1, 8, -1), "second_roof", enumblockrotation, false));
+                    worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-1, 0, -1), "second_floor_2", enumblockrotation, false));
+                    worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-1, 8, -1), "second_roof", enumblockrotation, false));
                     WorldGenEndCityPieces.b(WorldGenEndCityPieces.f, i + 1, worldgenendcitypieces_piece1, (BlockPosition) null, list, random);
                 } else if (j == 2) {
-                    list.add(worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-1, 0, -1), "second_floor_2", enumblockrotation, false));
-                    list.add(worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-1, 4, -1), "third_floor_c", enumblockrotation, false));
-                    list.add(worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-1, 8, -1), "third_roof", enumblockrotation, true));
+                    worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-1, 0, -1), "second_floor_2", enumblockrotation, false));
+                    worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-1, 4, -1), "third_floor_c", enumblockrotation, false));
+                    worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-1, 8, -1), "third_roof", enumblockrotation, true));
                     WorldGenEndCityPieces.b(WorldGenEndCityPieces.f, i + 1, worldgenendcitypieces_piece1, (BlockPosition) null, list, random);
                 }
 
@@ -47,15 +45,14 @@
 
         public boolean a(int i, WorldGenEndCityPieces.Piece worldgenendcitypieces_piece, BlockPosition blockposition, List<StructurePiece> list, Random random) {
             EnumBlockRotation enumblockrotation = worldgenendcitypieces_piece.b.c();
-            WorldGenEndCityPieces.Piece worldgenendcitypieces_piece1;
+            WorldGenEndCityPieces.Piece worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece, new BlockPosition(3 + random.nextInt(2), -3, 3 + random.nextInt(2)), "tower_base", enumblockrotation, true));
 
-            list.add(worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece, new BlockPosition(3 + random.nextInt(2), -3, 3 + random.nextInt(2)), "tower_base", enumblockrotation, true));
-            list.add(worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(0, 7, 0), "tower_piece", enumblockrotation, true));
+            worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(0, 7, 0), "tower_piece", enumblockrotation, true));
             WorldGenEndCityPieces.Piece worldgenendcitypieces_piece2 = random.nextInt(3) == 0 ? worldgenendcitypieces_piece1 : null;
             int j = 1 + random.nextInt(3);
 
             for (int k = 0; k < j; ++k) {
-                list.add(worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(0, 4, 0), "tower_piece", enumblockrotation, true));
+                worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(0, 4, 0), "tower_piece", enumblockrotation, true));
                 if (k < j - 1 && random.nextBoolean()) {
                     worldgenendcitypieces_piece2 = worldgenendcitypieces_piece1;
                 }
@@ -68,20 +65,19 @@
                     Tuple tuple = (Tuple) iterator.next();
 
                     if (random.nextBoolean()) {
-                        WorldGenEndCityPieces.Piece worldgenendcitypieces_piece3;
+                        WorldGenEndCityPieces.Piece worldgenendcitypieces_piece3 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece2, (BlockPosition) tuple.b(), "bridge_end", enumblockrotation.a((EnumBlockRotation) tuple.a()), true));
 
-                        list.add(worldgenendcitypieces_piece3 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece2, (BlockPosition) tuple.b(), "bridge_end", enumblockrotation.a((EnumBlockRotation) tuple.a()), true));
                         WorldGenEndCityPieces.b(WorldGenEndCityPieces.g, i + 1, worldgenendcitypieces_piece3, (BlockPosition) null, list, random);
                     }
                 }
 
-                list.add(WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-1, 4, -1), "tower_top", enumblockrotation, true));
+                WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-1, 4, -1), "tower_top", enumblockrotation, true));
             } else {
                 if (i != 7) {
                     return WorldGenEndCityPieces.b(WorldGenEndCityPieces.i, i + 1, worldgenendcitypieces_piece1, (BlockPosition) null, list, random);
                 }
 
-                list.add(WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-1, 4, -1), "tower_top", enumblockrotation, true));
+                WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-1, 4, -1), "tower_top", enumblockrotation, true));
             }
 
             return true;
@@ -97,21 +93,20 @@
         public boolean a(int i, WorldGenEndCityPieces.Piece worldgenendcitypieces_piece, BlockPosition blockposition, List<StructurePiece> list, Random random) {
             EnumBlockRotation enumblockrotation = worldgenendcitypieces_piece.b.c();
             int j = random.nextInt(4) + 1;
-            WorldGenEndCityPieces.Piece worldgenendcitypieces_piece1;
+            WorldGenEndCityPieces.Piece worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece, new BlockPosition(0, 0, -4), "bridge_piece", enumblockrotation, true));
 
-            list.add(worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece, new BlockPosition(0, 0, -4), "bridge_piece", enumblockrotation, true));
             worldgenendcitypieces_piece1.m = -1;
             byte b0 = 0;
 
             for (int k = 0; k < j; ++k) {
                 if (random.nextBoolean()) {
-                    list.add(worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(0, b0, -4), "bridge_piece", enumblockrotation, true));
+                    worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(0, b0, -4), "bridge_piece", enumblockrotation, true));
                     b0 = 0;
                 } else {
                     if (random.nextBoolean()) {
-                        list.add(worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(0, b0, -4), "bridge_steep_stairs", enumblockrotation, true));
+                        worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(0, b0, -4), "bridge_steep_stairs", enumblockrotation, true));
                     } else {
-                        list.add(worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(0, b0, -8), "bridge_gentle_stairs", enumblockrotation, true));
+                        worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(0, b0, -8), "bridge_gentle_stairs", enumblockrotation, true));
                     }
 
                     b0 = 4;
@@ -119,13 +114,13 @@
             }
 
             if (!this.a && random.nextInt(10 - i) == 0) {
-                list.add(WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-8 + random.nextInt(8), b0, -70 + random.nextInt(10)), "ship", enumblockrotation, true));
+                WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-8 + random.nextInt(8), b0, -70 + random.nextInt(10)), "ship", enumblockrotation, true));
                 this.a = true;
             } else if (!WorldGenEndCityPieces.b(WorldGenEndCityPieces.d, i + 1, worldgenendcitypieces_piece1, new BlockPosition(-3, b0 + 1, -11), list, random)) {
                 return false;
             }
 
-            list.add(worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(4, b0, 0), "bridge_end", enumblockrotation.a(EnumBlockRotation.CLOCKWISE_180), true));
+            worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(4, b0, 0), "bridge_end", enumblockrotation.a(EnumBlockRotation.CLOCKWISE_180), true));
             worldgenendcitypieces_piece1.m = -1;
             return true;
         }
@@ -136,28 +131,26 @@
 
         public boolean a(int i, WorldGenEndCityPieces.Piece worldgenendcitypieces_piece, BlockPosition blockposition, List<StructurePiece> list, Random random) {
             EnumBlockRotation enumblockrotation = worldgenendcitypieces_piece.b.c();
-            WorldGenEndCityPieces.Piece worldgenendcitypieces_piece1;
+            WorldGenEndCityPieces.Piece worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece, new BlockPosition(-3, 4, -3), "fat_tower_base", enumblockrotation, true));
 
-            list.add(worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece, new BlockPosition(-3, 4, -3), "fat_tower_base", enumblockrotation, true));
-            list.add(worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(0, 4, 0), "fat_tower_middle", enumblockrotation, true));
+            worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(0, 4, 0), "fat_tower_middle", enumblockrotation, true));
 
             for (int j = 0; j < 2 && random.nextInt(3) != 0; ++j) {
-                list.add(worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(0, 8, 0), "fat_tower_middle", enumblockrotation, true));
+                worldgenendcitypieces_piece1 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(0, 8, 0), "fat_tower_middle", enumblockrotation, true));
                 Iterator iterator = WorldGenEndCityPieces.h.iterator();
 
                 while (iterator.hasNext()) {
                     Tuple tuple = (Tuple) iterator.next();
 
                     if (random.nextBoolean()) {
-                        WorldGenEndCityPieces.Piece worldgenendcitypieces_piece2;
+                        WorldGenEndCityPieces.Piece worldgenendcitypieces_piece2 = WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, (BlockPosition) tuple.b(), "bridge_end", enumblockrotation.a((EnumBlockRotation) tuple.a()), true));
 
-                        list.add(worldgenendcitypieces_piece2 = WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, (BlockPosition) tuple.b(), "bridge_end", enumblockrotation.a((EnumBlockRotation) tuple.a()), true));
                         WorldGenEndCityPieces.b(WorldGenEndCityPieces.g, i + 1, worldgenendcitypieces_piece2, (BlockPosition) null, list, random);
                     }
                 }
             }
 
-            list.add(WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-2, 8, -2), "fat_tower_top", enumblockrotation, true));
+            WorldGenEndCityPieces.b(list, WorldGenEndCityPieces.b(worldgenendcitypieces_piece1, new BlockPosition(-2, 8, -2), "fat_tower_top", enumblockrotation, true));
             return true;
         }
     };
@@ -179,15 +172,17 @@
         WorldGenEndCityPieces.d.a();
         WorldGenEndCityPieces.g.a();
         WorldGenEndCityPieces.f.a();
-        WorldGenEndCityPieces.Piece worldgenendcitypieces_piece = new WorldGenEndCityPieces.Piece("base_floor", blockposition, enumblockrotation, true);
+        WorldGenEndCityPieces.Piece worldgenendcitypieces_piece = b(list, new WorldGenEndCityPieces.Piece("base_floor", blockposition, enumblockrotation, true));
 
-        list.add(worldgenendcitypieces_piece);
-        WorldGenEndCityPieces.Piece worldgenendcitypieces_piece1;
+        worldgenendcitypieces_piece = b(list, b(worldgenendcitypieces_piece, new BlockPosition(-1, 0, -1), "second_floor", enumblockrotation, false));
+        worldgenendcitypieces_piece = b(list, b(worldgenendcitypieces_piece, new BlockPosition(-1, 4, -1), "third_floor", enumblockrotation, false));
+        worldgenendcitypieces_piece = b(list, b(worldgenendcitypieces_piece, new BlockPosition(-1, 8, -1), "third_roof", enumblockrotation, true));
+        b(WorldGenEndCityPieces.f, 1, worldgenendcitypieces_piece, (BlockPosition) null, list, random);
+    }
 
-        list.add(worldgenendcitypieces_piece1 = b(worldgenendcitypieces_piece, new BlockPosition(-1, 0, -1), "second_floor", enumblockrotation, false));
-        list.add(worldgenendcitypieces_piece1 = b(worldgenendcitypieces_piece1, new BlockPosition(-1, 4, -1), "third_floor", enumblockrotation, false));
-        list.add(worldgenendcitypieces_piece1 = b(worldgenendcitypieces_piece1, new BlockPosition(-1, 8, -1), "third_roof", enumblockrotation, true));
-        b(WorldGenEndCityPieces.f, 1, worldgenendcitypieces_piece1, (BlockPosition) null, list, random);
+    private static WorldGenEndCityPieces.Piece b(List<StructurePiece> list, WorldGenEndCityPieces.Piece worldgenendcitypieces_piece) {
+        list.add(worldgenendcitypieces_piece);
+        return worldgenendcitypieces_piece;
     }
 
     private static boolean b(WorldGenEndCityPieces.PieceGenerator worldgenendcitypieces_piecegenerator, int i, WorldGenEndCityPieces.Piece worldgenendcitypieces_piece, BlockPosition blockposition, List<StructurePiece> list, Random random) {
@@ -205,7 +200,7 @@
                     StructurePiece structurepiece = (StructurePiece) iterator.next();
 
                     structurepiece.m = j;
-                    StructurePiece structurepiece1 = StructurePiece.a(list, structurepiece.c());
+                    StructurePiece structurepiece1 = StructurePiece.a(list, structurepiece.d());
 
                     if (structurepiece1 != null && structurepiece1.m != worldgenendcitypieces_piece.m) {
                         flag = true;
