--- a/net/minecraft/server/WorldGenClay.java
+++ b/net/minecraft/server/WorldGenClay.java
@@ -4,8 +4,8 @@
 
 public class WorldGenClay extends WorldGenerator {
 
-    private Block a;
-    private int b;
+    private final Block a;
+    private final int b;
 
     public WorldGenClay(int i) {
         this.a = Blocks.CLAY;
@@ -17,7 +17,7 @@
             return false;
         } else {
             int i = random.nextInt(this.b - 2) + 2;
-            byte b0 = 1;
+            boolean flag = true;
 
             for (int j = blockposition.getX() - i; j <= blockposition.getX() + i; ++j) {
                 for (int k = blockposition.getZ() - i; k <= blockposition.getZ() + i; ++k) {
@@ -25,7 +25,7 @@
                     int i1 = k - blockposition.getZ();
 
                     if (l * l + i1 * i1 <= i * i) {
-                        for (int j1 = blockposition.getY() - b0; j1 <= blockposition.getY() + b0; ++j1) {
+                        for (int j1 = blockposition.getY() - 1; j1 <= blockposition.getY() + 1; ++j1) {
                             BlockPosition blockposition1 = new BlockPosition(j, j1, k);
                             Block block = world.getType(blockposition1).getBlock();
 
