--- a/net/minecraft/server/AttributeMapBase.java
+++ b/net/minecraft/server/AttributeMapBase.java
@@ -7,6 +7,7 @@
 import java.util.Iterator;
 import java.util.Map;
 import java.util.Map.Entry;
+import javax.annotation.Nullable;
 
 public abstract class AttributeMapBase {
 
@@ -20,6 +21,7 @@
         return (AttributeInstance) this.a.get(iattribute);
     }
 
+    @Nullable
     public AttributeInstance a(String s) {
         return (AttributeInstance) this.b.get(s);
     }
