--- a/net/minecraft/server/BiomeMesa.java
+++ b/net/minecraft/server/BiomeMesa.java
@@ -39,17 +39,13 @@
     }
 
     protected BiomeDecorator a() {
-        return new BiomeMesa.a((BiomeMesa.SyntheticClass_1) null);
+        return new BiomeMesa.a(null);
     }
 
     public WorldGenTreeAbstract a(Random random) {
         return BiomeMesa.n;
     }
 
-    public void a(World world, Random random, BlockPosition blockposition) {
-        super.a(world, random, blockposition);
-    }
-
     public void a(World world, Random random, ChunkSnapshot chunksnapshot, int i, int j, double d0) {
         if (this.E == null || this.F != world.getSeed()) {
             this.a(world.getSeed());
@@ -105,7 +101,7 @@
 
             if (i2 <= random.nextInt(5)) {
                 chunksnapshot.a(l, i2, k, BiomeMesa.c);
-            } else if (l1 < 15) {
+            } else if (l1 < 15 || this.J) {
                 IBlockData iblockdata2 = chunksnapshot.a(l, i2, k);
 
                 if (iblockdata2.getMaterial() == Material.AIR) {
@@ -257,8 +253,6 @@
         return this.E[(j + l + 64) % 64];
     }
 
-    static class SyntheticClass_1 {    }
-
     class a extends BiomeDecorator {
 
         private a() {}
@@ -268,7 +262,7 @@
             this.a(world, random, 20, this.n, 32, 80);
         }
 
-        a(BiomeMesa.SyntheticClass_1 biomemesa_syntheticclass_1) {
+        a(Object object) {
             this();
         }
     }
