--- a/net/minecraft/server/BiomeHell.java
+++ b/net/minecraft/server/BiomeHell.java
@@ -1,7 +1,5 @@
 package net.minecraft.server;
 
-import java.util.Random;
-
 public class BiomeHell extends BiomeBase {
 
     public BiomeHell(BiomeBase.a biomebase_a) {
@@ -16,8 +14,4 @@
         this.u.add(new BiomeBase.BiomeMeta(EntityEnderman.class, 1, 4, 4));
         this.t = new BiomeDecoratorHell();
     }
-
-    public void a(World world, Random random, BlockPosition blockposition) {
-        super.a(world, random, blockposition);
-    }
 }
