--- a/net/minecraft/server/BlockDoubleStoneStepAbstract.java
+++ b/net/minecraft/server/BlockDoubleStoneStepAbstract.java
@@ -1,7 +1,6 @@
 package net.minecraft.server;
 
 import java.util.Random;
-import javax.annotation.Nullable;
 
 public abstract class BlockDoubleStoneStepAbstract extends BlockStepAbstract {
 
@@ -18,7 +17,7 @@
             iblockdata = iblockdata.set(BlockDoubleStoneStepAbstract.HALF, BlockStepAbstract.EnumSlabHalf.BOTTOM);
         }
 
-        this.w(iblockdata.set(BlockDoubleStoneStepAbstract.VARIANT, BlockDoubleStoneStepAbstract.EnumStoneSlab2Variant.RED_SANDSTONE));
+        this.y(iblockdata.set(BlockDoubleStoneStepAbstract.VARIANT, BlockDoubleStoneStepAbstract.EnumStoneSlab2Variant.RED_SANDSTONE));
         this.a(CreativeModeTab.b);
     }
 
@@ -26,7 +25,6 @@
         return LocaleI18n.get(this.a() + ".red_sandstone.name");
     }
 
-    @Nullable
     public Item getDropType(IBlockData iblockdata, Random random, int i) {
         return Item.getItemOf(Blocks.STONE_SLAB2);
     }
