--- a/net/minecraft/server/BlockWallSign.java
+++ b/net/minecraft/server/BlockWallSign.java
@@ -9,27 +9,27 @@
     protected static final AxisAlignedBB f = new AxisAlignedBB(0.0D, 0.28125D, 0.875D, 1.0D, 0.78125D, 1.0D);
 
     public BlockWallSign() {
-        this.w(this.blockStateList.getBlockData().set(BlockWallSign.FACING, EnumDirection.NORTH));
+        this.y(this.blockStateList.getBlockData().set(BlockWallSign.FACING, EnumDirection.NORTH));
     }
 
-    public AxisAlignedBB a(IBlockData iblockdata, IBlockAccess iblockaccess, BlockPosition blockposition) {
-        switch (BlockWallSign.SyntheticClass_1.a[((EnumDirection) iblockdata.get(BlockWallSign.FACING)).ordinal()]) {
-        case 1:
+    public AxisAlignedBB b(IBlockData iblockdata, IBlockAccess iblockaccess, BlockPosition blockposition) {
+        switch ((EnumDirection) iblockdata.get(BlockWallSign.FACING)) {
+        case NORTH:
         default:
             return BlockWallSign.f;
 
-        case 2:
+        case SOUTH:
             return BlockWallSign.e;
 
-        case 3:
+        case WEST:
             return BlockWallSign.d;
 
-        case 4:
+        case EAST:
             return BlockWallSign.c;
         }
     }
 
-    public void a(IBlockData iblockdata, World world, BlockPosition blockposition, Block block) {
+    public void a(IBlockData iblockdata, World world, BlockPosition blockposition, Block block, BlockPosition blockposition1) {
         EnumDirection enumdirection = (EnumDirection) iblockdata.get(BlockWallSign.FACING);
 
         if (!world.getType(blockposition.shift(enumdirection.opposite())).getMaterial().isBuildable()) {
@@ -37,7 +37,7 @@
             world.setAir(blockposition);
         }
 
-        super.a(iblockdata, world, blockposition, block);
+        super.a(iblockdata, world, blockposition, block, blockposition1);
     }
 
     public IBlockData fromLegacyData(int i) {
@@ -65,36 +65,4 @@
     protected BlockStateList getStateList() {
         return new BlockStateList(this, new IBlockState[] { BlockWallSign.FACING});
     }
-
-    static class SyntheticClass_1 {
-
-        static final int[] a = new int[EnumDirection.values().length];
-
-        static {
-            try {
-                BlockWallSign.SyntheticClass_1.a[EnumDirection.NORTH.ordinal()] = 1;
-            } catch (NoSuchFieldError nosuchfielderror) {
-                ;
-            }
-
-            try {
-                BlockWallSign.SyntheticClass_1.a[EnumDirection.SOUTH.ordinal()] = 2;
-            } catch (NoSuchFieldError nosuchfielderror1) {
-                ;
-            }
-
-            try {
-                BlockWallSign.SyntheticClass_1.a[EnumDirection.WEST.ordinal()] = 3;
-            } catch (NoSuchFieldError nosuchfielderror2) {
-                ;
-            }
-
-            try {
-                BlockWallSign.SyntheticClass_1.a[EnumDirection.EAST.ordinal()] = 4;
-            } catch (NoSuchFieldError nosuchfielderror3) {
-                ;
-            }
-
-        }
-    }
 }
