--- a/net/minecraft/server/BlockStainedGlass.java
+++ b/net/minecraft/server/BlockStainedGlass.java
@@ -8,7 +8,7 @@
 
     public BlockStainedGlass(Material material) {
         super(material, false);
-        this.w(this.blockStateList.getBlockData().set(BlockStainedGlass.COLOR, EnumColor.WHITE));
+        this.y(this.blockStateList.getBlockData().set(BlockStainedGlass.COLOR, EnumColor.WHITE));
         this.a(CreativeModeTab.b);
     }
 
@@ -24,7 +24,7 @@
         return 0;
     }
 
-    protected boolean o() {
+    protected boolean n() {
         return true;
     }
 
