--- a/net/minecraft/server/BlockEndGateway.java
+++ b/net/minecraft/server/BlockEndGateway.java
@@ -15,7 +15,7 @@
     }
 
     @Nullable
-    public AxisAlignedBB a(IBlockData iblockdata, World world, BlockPosition blockposition) {
+    public AxisAlignedBB a(IBlockData iblockdata, IBlockAccess iblockaccess, BlockPosition blockposition) {
         return BlockEndGateway.k;
     }
 
@@ -31,9 +31,8 @@
         return 0;
     }
 
-    @Nullable
     public ItemStack a(World world, BlockPosition blockposition, IBlockData iblockdata) {
-        return null;
+        return ItemStack.a;
     }
 
     public MaterialMapColor r(IBlockData iblockdata) {
