--- a/net/minecraft/server/BlockWorkbench.java
+++ b/net/minecraft/server/BlockWorkbench.java
@@ -1,7 +1,5 @@
 package net.minecraft.server;
 
-import javax.annotation.Nullable;
-
 public class BlockWorkbench extends Block {
 
     protected BlockWorkbench() {
@@ -9,7 +7,7 @@
         this.a(CreativeModeTab.c);
     }
 
-    public boolean interact(World world, BlockPosition blockposition, IBlockData iblockdata, EntityHuman entityhuman, EnumHand enumhand, @Nullable ItemStack itemstack, EnumDirection enumdirection, float f, float f1, float f2) {
+    public boolean interact(World world, BlockPosition blockposition, IBlockData iblockdata, EntityHuman entityhuman, EnumHand enumhand, EnumDirection enumdirection, float f, float f1, float f2) {
         if (world.isClientSide) {
             return true;
         } else {
@@ -30,7 +28,7 @@
         }
 
         public String getName() {
-            return null;
+            return "crafting_table";
         }
 
         public boolean hasCustomName() {
