--- a/net/minecraft/server/BlockFalling.java
+++ b/net/minecraft/server/BlockFalling.java
@@ -19,18 +19,18 @@
         world.a(blockposition, (Block) this, this.a(world));
     }
 
-    public void a(IBlockData iblockdata, World world, BlockPosition blockposition, Block block) {
+    public void a(IBlockData iblockdata, World world, BlockPosition blockposition, Block block, BlockPosition blockposition1) {
         world.a(blockposition, (Block) this, this.a(world));
     }
 
     public void b(World world, BlockPosition blockposition, IBlockData iblockdata, Random random) {
         if (!world.isClientSide) {
-            this.b(world, blockposition);
+            this.c(world, blockposition);
         }
 
     }
 
-    private void b(World world, BlockPosition blockposition) {
+    private void c(World world, BlockPosition blockposition) {
         if (i(world.getType(blockposition.down())) && blockposition.getY() >= 0) {
             boolean flag = true;
 
@@ -72,4 +72,6 @@
     }
 
     public void a_(World world, BlockPosition blockposition) {}
+
+    public void b(World world, BlockPosition blockposition) {}
 }
