--- a/net/minecraft/server/BlockSandStone.java
+++ b/net/minecraft/server/BlockSandStone.java
@@ -6,7 +6,7 @@
 
     public BlockSandStone() {
         super(Material.STONE);
-        this.w(this.blockStateList.getBlockData().set(BlockSandStone.TYPE, BlockSandStone.EnumSandstoneVariant.DEFAULT));
+        this.y(this.blockStateList.getBlockData().set(BlockSandStone.TYPE, BlockSandStone.EnumSandstoneVariant.DEFAULT));
         this.a(CreativeModeTab.b);
     }
 
