--- a/net/minecraft/server/BlockTileEntity.java
+++ b/net/minecraft/server/BlockTileEntity.java
@@ -1,5 +1,7 @@
 package net.minecraft.server;
 
+import javax.annotation.Nullable;
+
 public abstract class BlockTileEntity extends Block implements ITileEntity {
 
     protected BlockTileEntity(Material material) {
@@ -28,6 +30,31 @@
         world.s(blockposition);
     }
 
+    public void a(World world, EntityHuman entityhuman, BlockPosition blockposition, IBlockData iblockdata, @Nullable TileEntity tileentity, ItemStack itemstack) {
+        if (tileentity instanceof INamableTileEntity && ((INamableTileEntity) tileentity).hasCustomName()) {
+            entityhuman.b(StatisticList.a((Block) this));
+            entityhuman.applyExhaustion(0.005F);
+            if (world.isClientSide) {
+                return;
+            }
+
+            int i = EnchantmentManager.getEnchantmentLevel(Enchantments.LOOT_BONUS_BLOCKS, itemstack);
+            Item item = this.getDropType(iblockdata, world.random, i);
+
+            if (item == Items.a) {
+                return;
+            }
+
+            ItemStack itemstack1 = new ItemStack(item, this.a(world.random));
+
+            itemstack1.g(((INamableTileEntity) tileentity).getName());
+            a(world, blockposition, itemstack1);
+        } else {
+            super.a(world, entityhuman, blockposition, iblockdata, (TileEntity) null, itemstack);
+        }
+
+    }
+
     public boolean a(IBlockData iblockdata, World world, BlockPosition blockposition, int i, int j) {
         super.a(iblockdata, world, blockposition, i, j);
         TileEntity tileentity = world.getTileEntity(blockposition);
