--- a/net/minecraft/server/BlockIce.java
+++ b/net/minecraft/server/BlockIce.java
@@ -12,15 +12,11 @@
         this.a(CreativeModeTab.b);
     }
 
-    public void a(World world, EntityHuman entityhuman, BlockPosition blockposition, IBlockData iblockdata, @Nullable TileEntity tileentity, @Nullable ItemStack itemstack) {
+    public void a(World world, EntityHuman entityhuman, BlockPosition blockposition, IBlockData iblockdata, @Nullable TileEntity tileentity, ItemStack itemstack) {
         entityhuman.b(StatisticList.a((Block) this));
-        entityhuman.applyExhaustion(0.025F);
-        if (this.o() && EnchantmentManager.getEnchantmentLevel(Enchantments.SILK_TOUCH, itemstack) > 0) {
-            ItemStack itemstack1 = this.u(iblockdata);
-
-            if (itemstack1 != null) {
-                a(world, blockposition, itemstack1);
-            }
+        entityhuman.applyExhaustion(0.005F);
+        if (this.n() && EnchantmentManager.getEnchantmentLevel(Enchantments.SILK_TOUCH, itemstack) > 0) {
+            a(world, blockposition, this.w(iblockdata));
         } else {
             if (world.worldProvider.l()) {
                 world.setAir(blockposition);
@@ -44,7 +40,7 @@
     }
 
     public void b(World world, BlockPosition blockposition, IBlockData iblockdata, Random random) {
-        if (world.b(EnumSkyBlock.BLOCK, blockposition) > 11 - this.getBlockData().c()) {
+        if (world.getBrightness(EnumSkyBlock.BLOCK, blockposition) > 11 - this.getBlockData().c()) {
             this.b(world, blockposition);
         }
 
@@ -56,7 +52,7 @@
         } else {
             this.b(world, blockposition, world.getType(blockposition), 0);
             world.setTypeUpdate(blockposition, Blocks.WATER.getBlockData());
-            world.e(blockposition, Blocks.WATER);
+            world.a(blockposition, (Block) Blocks.WATER, blockposition);
         }
     }
 
