--- a/net/minecraft/server/BlockStainedGlassPane.java
+++ b/net/minecraft/server/BlockStainedGlassPane.java
@@ -6,7 +6,7 @@
 
     public BlockStainedGlassPane() {
         super(Material.SHATTERABLE, false);
-        this.w(this.blockStateList.getBlockData().set(BlockStainedGlassPane.NORTH, Boolean.valueOf(false)).set(BlockStainedGlassPane.EAST, Boolean.valueOf(false)).set(BlockStainedGlassPane.SOUTH, Boolean.valueOf(false)).set(BlockStainedGlassPane.WEST, Boolean.valueOf(false)).set(BlockStainedGlassPane.COLOR, EnumColor.WHITE));
+        this.y(this.blockStateList.getBlockData().set(BlockStainedGlassPane.NORTH, Boolean.valueOf(false)).set(BlockStainedGlassPane.EAST, Boolean.valueOf(false)).set(BlockStainedGlassPane.SOUTH, Boolean.valueOf(false)).set(BlockStainedGlassPane.WEST, Boolean.valueOf(false)).set(BlockStainedGlassPane.COLOR, EnumColor.WHITE));
         this.a(CreativeModeTab.c);
     }
 
@@ -27,14 +27,14 @@
     }
 
     public IBlockData a(IBlockData iblockdata, EnumBlockRotation enumblockrotation) {
-        switch (BlockStainedGlassPane.SyntheticClass_1.a[enumblockrotation.ordinal()]) {
-        case 1:
+        switch (enumblockrotation) {
+        case CLOCKWISE_180:
             return iblockdata.set(BlockStainedGlassPane.NORTH, iblockdata.get(BlockStainedGlassPane.SOUTH)).set(BlockStainedGlassPane.EAST, iblockdata.get(BlockStainedGlassPane.WEST)).set(BlockStainedGlassPane.SOUTH, iblockdata.get(BlockStainedGlassPane.NORTH)).set(BlockStainedGlassPane.WEST, iblockdata.get(BlockStainedGlassPane.EAST));
 
-        case 2:
+        case COUNTERCLOCKWISE_90:
             return iblockdata.set(BlockStainedGlassPane.NORTH, iblockdata.get(BlockStainedGlassPane.EAST)).set(BlockStainedGlassPane.EAST, iblockdata.get(BlockStainedGlassPane.SOUTH)).set(BlockStainedGlassPane.SOUTH, iblockdata.get(BlockStainedGlassPane.WEST)).set(BlockStainedGlassPane.WEST, iblockdata.get(BlockStainedGlassPane.NORTH));
 
-        case 3:
+        case CLOCKWISE_90:
             return iblockdata.set(BlockStainedGlassPane.NORTH, iblockdata.get(BlockStainedGlassPane.WEST)).set(BlockStainedGlassPane.EAST, iblockdata.get(BlockStainedGlassPane.NORTH)).set(BlockStainedGlassPane.SOUTH, iblockdata.get(BlockStainedGlassPane.EAST)).set(BlockStainedGlassPane.WEST, iblockdata.get(BlockStainedGlassPane.SOUTH));
 
         default:
@@ -43,11 +43,11 @@
     }
 
     public IBlockData a(IBlockData iblockdata, EnumBlockMirror enumblockmirror) {
-        switch (BlockStainedGlassPane.SyntheticClass_1.b[enumblockmirror.ordinal()]) {
-        case 1:
+        switch (enumblockmirror) {
+        case LEFT_RIGHT:
             return iblockdata.set(BlockStainedGlassPane.NORTH, iblockdata.get(BlockStainedGlassPane.SOUTH)).set(BlockStainedGlassPane.SOUTH, iblockdata.get(BlockStainedGlassPane.NORTH));
 
-        case 2:
+        case FRONT_BACK:
             return iblockdata.set(BlockStainedGlassPane.EAST, iblockdata.get(BlockStainedGlassPane.WEST)).set(BlockStainedGlassPane.WEST, iblockdata.get(BlockStainedGlassPane.EAST));
 
         default:
@@ -72,45 +72,4 @@
         }
 
     }
-
-    static class SyntheticClass_1 {
-
-        static final int[] a;
-        static final int[] b = new int[EnumBlockMirror.values().length];
-
-        static {
-            try {
-                BlockStainedGlassPane.SyntheticClass_1.b[EnumBlockMirror.LEFT_RIGHT.ordinal()] = 1;
-            } catch (NoSuchFieldError nosuchfielderror) {
-                ;
-            }
-
-            try {
-                BlockStainedGlassPane.SyntheticClass_1.b[EnumBlockMirror.FRONT_BACK.ordinal()] = 2;
-            } catch (NoSuchFieldError nosuchfielderror1) {
-                ;
-            }
-
-            a = new int[EnumBlockRotation.values().length];
-
-            try {
-                BlockStainedGlassPane.SyntheticClass_1.a[EnumBlockRotation.CLOCKWISE_180.ordinal()] = 1;
-            } catch (NoSuchFieldError nosuchfielderror2) {
-                ;
-            }
-
-            try {
-                BlockStainedGlassPane.SyntheticClass_1.a[EnumBlockRotation.COUNTERCLOCKWISE_90.ordinal()] = 2;
-            } catch (NoSuchFieldError nosuchfielderror3) {
-                ;
-            }
-
-            try {
-                BlockStainedGlassPane.SyntheticClass_1.a[EnumBlockRotation.CLOCKWISE_90.ordinal()] = 3;
-            } catch (NoSuchFieldError nosuchfielderror4) {
-                ;
-            }
-
-        }
-    }
 }
