--- a/net/minecraft/server/BlockBeacon.java
+++ b/net/minecraft/server/BlockBeacon.java
@@ -1,7 +1,5 @@
 package net.minecraft.server;
 
-import javax.annotation.Nullable;
-
 public class BlockBeacon extends BlockTileEntity {
 
     public BlockBeacon() {
@@ -14,7 +12,7 @@
         return new TileEntityBeacon();
     }
 
-    public boolean interact(World world, BlockPosition blockposition, IBlockData iblockdata, EntityHuman entityhuman, EnumHand enumhand, @Nullable ItemStack itemstack, EnumDirection enumdirection, float f, float f1, float f2) {
+    public boolean interact(World world, BlockPosition blockposition, IBlockData iblockdata, EntityHuman entityhuman, EnumHand enumhand, EnumDirection enumdirection, float f, float f1, float f2) {
         if (world.isClientSide) {
             return true;
         } else {
@@ -53,11 +51,11 @@
 
     }
 
-    public void a(IBlockData iblockdata, World world, BlockPosition blockposition, Block block) {
+    public void a(IBlockData iblockdata, World world, BlockPosition blockposition, Block block, BlockPosition blockposition1) {
         TileEntity tileentity = world.getTileEntity(blockposition);
 
         if (tileentity instanceof TileEntityBeacon) {
-            ((TileEntityBeacon) tileentity).m();
+            ((TileEntityBeacon) tileentity).n();
             world.playBlockAction(blockposition, this, 1, 0);
         }
 
@@ -83,7 +81,7 @@
                                 TileEntity tileentity = world.getTileEntity(blockposition);
 
                                 if (tileentity instanceof TileEntityBeacon) {
-                                    ((TileEntityBeacon) tileentity).m();
+                                    ((TileEntityBeacon) tileentity).n();
                                     world.playBlockAction(blockposition, Blocks.BEACON, 1, 0);
                                 }
 
