--- a/net/minecraft/server/CommandSetBlock.java
+++ b/net/minecraft/server/CommandSetBlock.java
@@ -28,10 +28,12 @@
             icommandlistener.a(CommandObjectiveExecutor.EnumCommandResult.AFFECTED_BLOCKS, 0);
             BlockPosition blockposition = a(icommandlistener, astring, 0, false);
             Block block = CommandAbstract.b(icommandlistener, astring[3]);
-            int i = 0;
+            IBlockData iblockdata;
 
             if (astring.length >= 5) {
-                i = a(astring[4], 0, 15);
+                iblockdata = a(block, astring[4]);
+            } else {
+                iblockdata = block.getBlockData();
             }
 
             World world = icommandlistener.getWorld();
@@ -69,14 +71,12 @@
 
                 if (tileentity != null) {
                     if (tileentity instanceof IInventory) {
-                        ((IInventory) tileentity).l();
+                        ((IInventory) tileentity).clear();
                     }
 
                     world.setTypeAndData(blockposition, Blocks.AIR.getBlockData(), block == Blocks.AIR ? 2 : 4);
                 }
 
-                IBlockData iblockdata = block.fromLegacyData(i);
-
                 if (!world.setTypeAndData(blockposition, iblockdata, 2)) {
                     throw new CommandException("commands.setblock.noChange", new Object[0]);
                 } else {
@@ -91,7 +91,7 @@
                         }
                     }
 
-                    world.update(blockposition, iblockdata.getBlock());
+                    world.update(blockposition, iblockdata.getBlock(), false);
                     icommandlistener.a(CommandObjectiveExecutor.EnumCommandResult.AFFECTED_BLOCKS, 1);
                     a(icommandlistener, (ICommand) this, "commands.setblock.success", new Object[0]);
                 }
