--- a/net/minecraft/server/ControllerLook.java
+++ b/net/minecraft/server/ControllerLook.java
@@ -45,24 +45,24 @@
             double d1 = this.f - (this.a.locY + (double) this.a.getHeadHeight());
             double d2 = this.g - this.a.locZ;
             double d3 = (double) MathHelper.sqrt(d0 * d0 + d2 * d2);
-            float f = (float) (MathHelper.b(d2, d0) * 57.2957763671875D) - 90.0F;
-            float f1 = (float) (-(MathHelper.b(d1, d3) * 57.2957763671875D));
+            float f = (float) (MathHelper.c(d2, d0) * 57.2957763671875D) - 90.0F;
+            float f1 = (float) (-(MathHelper.c(d1, d3) * 57.2957763671875D));
 
             this.a.pitch = this.a(this.a.pitch, f1, this.c);
-            this.a.aQ = this.a(this.a.aQ, f, this.b);
+            this.a.aP = this.a(this.a.aP, f, this.b);
         } else {
-            this.a.aQ = this.a(this.a.aQ, this.a.aO, 10.0F);
+            this.a.aP = this.a(this.a.aP, this.a.aN, 10.0F);
         }
 
-        float f2 = MathHelper.g(this.a.aQ - this.a.aO);
+        float f2 = MathHelper.g(this.a.aP - this.a.aN);
 
         if (!this.a.getNavigation().n()) {
             if (f2 < -75.0F) {
-                this.a.aQ = this.a.aO - 75.0F;
+                this.a.aP = this.a.aN - 75.0F;
             }
 
             if (f2 > 75.0F) {
-                this.a.aQ = this.a.aO + 75.0F;
+                this.a.aP = this.a.aN + 75.0F;
             }
         }
 
