--- a/net/minecraft/server/ContainerFurnace.java
+++ b/net/minecraft/server/ContainerFurnace.java
@@ -1,7 +1,5 @@
 package net.minecraft.server;
 
-import javax.annotation.Nullable;
-
 public class ContainerFurnace extends Container {
 
     private final IInventory furnace;
@@ -68,9 +66,8 @@
         return this.furnace.a(entityhuman);
     }
 
-    @Nullable
     public ItemStack b(EntityHuman entityhuman, int i) {
-        ItemStack itemstack = null;
+        ItemStack itemstack = ItemStack.a;
         Slot slot = (Slot) this.c.get(i);
 
         if (slot != null && slot.hasItem()) {
@@ -79,38 +76,38 @@
             itemstack = itemstack1.cloneItemStack();
             if (i == 2) {
                 if (!this.a(itemstack1, 3, 39, true)) {
-                    return null;
+                    return ItemStack.a;
                 }
 
                 slot.a(itemstack1, itemstack);
             } else if (i != 1 && i != 0) {
-                if (RecipesFurnace.getInstance().getResult(itemstack1) != null) {
+                if (!RecipesFurnace.getInstance().getResult(itemstack1).isEmpty()) {
                     if (!this.a(itemstack1, 0, 1, false)) {
-                        return null;
+                        return ItemStack.a;
                     }
                 } else if (TileEntityFurnace.isFuel(itemstack1)) {
                     if (!this.a(itemstack1, 1, 2, false)) {
-                        return null;
+                        return ItemStack.a;
                     }
                 } else if (i >= 3 && i < 30) {
                     if (!this.a(itemstack1, 30, 39, false)) {
-                        return null;
+                        return ItemStack.a;
                     }
                 } else if (i >= 30 && i < 39 && !this.a(itemstack1, 3, 30, false)) {
-                    return null;
+                    return ItemStack.a;
                 }
             } else if (!this.a(itemstack1, 3, 39, false)) {
-                return null;
+                return ItemStack.a;
             }
 
-            if (itemstack1.count == 0) {
-                slot.set((ItemStack) null);
+            if (itemstack1.isEmpty()) {
+                slot.set(ItemStack.a);
             } else {
                 slot.f();
             }
 
-            if (itemstack1.count == itemstack.count) {
-                return null;
+            if (itemstack1.getCount() == itemstack.getCount()) {
+                return ItemStack.a;
             }
 
             slot.a(entityhuman, itemstack1);
