--- a/net/minecraft/server/CreativeModeTab.java
+++ b/net/minecraft/server/CreativeModeTab.java
@@ -11,32 +11,34 @@
     };
     public static final CreativeModeTab e = new CreativeModeTab(3, "transportation") {
     };
-    public static final CreativeModeTab f = (new CreativeModeTab(4, "misc") {
-    }).a(new EnchantmentSlotType[] { EnchantmentSlotType.ALL});
+    public static final CreativeModeTab f = new CreativeModeTab(4, "misc") {
+    };
     public static final CreativeModeTab g = (new CreativeModeTab(5, "search") {
     }).a("item_search.png");
     public static final CreativeModeTab h = new CreativeModeTab(6, "food") {
     };
     public static final CreativeModeTab i = (new CreativeModeTab(7, "tools") {
-    }).a(new EnchantmentSlotType[] { EnchantmentSlotType.DIGGER, EnchantmentSlotType.FISHING_ROD, EnchantmentSlotType.BREAKABLE});
+    }).a(new EnchantmentSlotType[] { EnchantmentSlotType.ALL, EnchantmentSlotType.DIGGER, EnchantmentSlotType.FISHING_ROD, EnchantmentSlotType.BREAKABLE});
     public static final CreativeModeTab j = (new CreativeModeTab(8, "combat") {
-    }).a(new EnchantmentSlotType[] { EnchantmentSlotType.ARMOR, EnchantmentSlotType.ARMOR_FEET, EnchantmentSlotType.ARMOR_HEAD, EnchantmentSlotType.ARMOR_LEGS, EnchantmentSlotType.ARMOR_CHEST, EnchantmentSlotType.BOW, EnchantmentSlotType.WEAPON});
+    }).a(new EnchantmentSlotType[] { EnchantmentSlotType.ALL, EnchantmentSlotType.ARMOR, EnchantmentSlotType.ARMOR_FEET, EnchantmentSlotType.ARMOR_HEAD, EnchantmentSlotType.ARMOR_LEGS, EnchantmentSlotType.ARMOR_CHEST, EnchantmentSlotType.BOW, EnchantmentSlotType.WEAPON, EnchantmentSlotType.WEARABLE, EnchantmentSlotType.BREAKABLE});
     public static final CreativeModeTab k = new CreativeModeTab(9, "brewing") {
     };
     public static final CreativeModeTab l = new CreativeModeTab(10, "materials") {
     };
     public static final CreativeModeTab m = (new CreativeModeTab(11, "inventory") {
-    }).a("inventory.png").k().i();
+    }).a("inventory.png").j().h();
     private final int n;
     private final String o;
     private String p = "items.png";
     private boolean q = true;
     private boolean r = true;
     private EnchantmentSlotType[] s;
+    private ItemStack t;
 
     public CreativeModeTab(int i, String s) {
         this.n = i;
         this.o = s;
+        this.t = ItemStack.a;
         CreativeModeTab.a[i] = this;
     }
 
@@ -45,12 +47,12 @@
         return this;
     }
 
-    public CreativeModeTab i() {
+    public CreativeModeTab h() {
         this.r = false;
         return this;
     }
 
-    public CreativeModeTab k() {
+    public CreativeModeTab j() {
         this.q = false;
         return this;
     }
