--- a/net/minecraft/server/DataInspectorTagged.java
+++ b/net/minecraft/server/DataInspectorTagged.java
@@ -2,16 +2,21 @@
 
 public abstract class DataInspectorTagged implements DataInspector {
 
-    private final String a;
-    private final String b;
+    private final MinecraftKey a;
+
+    public DataInspectorTagged(Class<?> oclass) {
+        if (Entity.class.isAssignableFrom(oclass)) {
+            this.a = EntityTypes.getName(oclass);
+        } else if (TileEntity.class.isAssignableFrom(oclass)) {
+            this.a = TileEntity.a(oclass);
+        } else {
+            this.a = null;
+        }
 
-    public DataInspectorTagged(String s, String s1) {
-        this.a = s;
-        this.b = s1;
     }
 
     public NBTTagCompound a(DataConverter dataconverter, NBTTagCompound nbttagcompound, int i) {
-        if (nbttagcompound.getString(this.a).equals(this.b)) {
+        if ((new MinecraftKey(nbttagcompound.getString("id"))).equals(this.a)) {
             nbttagcompound = this.b(dataconverter, nbttagcompound, i);
         }
 
