--- a/net/minecraft/server/DispenseBehaviorProjectile.java
+++ b/net/minecraft/server/DispenseBehaviorProjectile.java
@@ -12,7 +12,7 @@
 
         iprojectile.shoot((double) enumdirection.getAdjacentX(), (double) ((float) enumdirection.getAdjacentY() + 0.1F), (double) enumdirection.getAdjacentZ(), this.getPower(), this.a());
         world.addEntity((Entity) iprojectile);
-        itemstack.cloneAndSubtract(1);
+        itemstack.subtract(1);
         return itemstack;
     }
 
