--- a/net/minecraft/server/EnchantmentThorns.java
+++ b/net/minecraft/server/EnchantmentThorns.java
@@ -34,10 +34,10 @@
                 entity.damageEntity(DamageSource.a(entityliving), (float) b(i, random));
             }
 
-            if (itemstack != null) {
+            if (!itemstack.isEmpty()) {
                 itemstack.damage(3, entityliving);
             }
-        } else if (itemstack != null) {
+        } else if (!itemstack.isEmpty()) {
             itemstack.damage(1, entityliving);
         }
 
