--- a/net/minecraft/server/EntityShulkerBullet.java
+++ b/net/minecraft/server/EntityShulkerBullet.java
@@ -19,10 +19,10 @@
     private double g;
     @Nullable
     private UUID h;
-    private BlockPosition au;
+    private BlockPosition at;
     @Nullable
-    private UUID av;
-    private BlockPosition aw;
+    private UUID au;
+    private BlockPosition av;
 
     public EntityShulkerBullet(World world) {
         super(world);
@@ -94,13 +94,13 @@
         if (nbttagcompound.hasKeyOfType("Owner", 10)) {
             nbttagcompound1 = nbttagcompound.getCompound("Owner");
             this.h = GameProfileSerializer.b(nbttagcompound1);
-            this.au = new BlockPosition(nbttagcompound1.getInt("X"), nbttagcompound1.getInt("Y"), nbttagcompound1.getInt("Z"));
+            this.at = new BlockPosition(nbttagcompound1.getInt("X"), nbttagcompound1.getInt("Y"), nbttagcompound1.getInt("Z"));
         }
 
         if (nbttagcompound.hasKeyOfType("Target", 10)) {
             nbttagcompound1 = nbttagcompound.getCompound("Target");
-            this.av = GameProfileSerializer.b(nbttagcompound1);
-            this.aw = new BlockPosition(nbttagcompound1.getInt("X"), nbttagcompound1.getInt("Y"), nbttagcompound1.getInt("Z"));
+            this.au = GameProfileSerializer.b(nbttagcompound1);
+            this.av = new BlockPosition(nbttagcompound1.getInt("X"), nbttagcompound1.getInt("Y"), nbttagcompound1.getInt("Z"));
         }
 
     }
@@ -189,33 +189,33 @@
         this.d = 10 + this.random.nextInt(5) * 10;
     }
 
-    public void m() {
+    public void A_() {
         if (!this.world.isClientSide && this.world.getDifficulty() == EnumDifficulty.PEACEFUL) {
             this.die();
         } else {
-            super.m();
+            super.A_();
             if (!this.world.isClientSide) {
                 List list;
                 Iterator iterator;
                 EntityLiving entityliving;
 
-                if (this.target == null && this.av != null) {
-                    list = this.world.a(EntityLiving.class, new AxisAlignedBB(this.aw.a(-2, -2, -2), this.aw.a(2, 2, 2)));
+                if (this.target == null && this.au != null) {
+                    list = this.world.a(EntityLiving.class, new AxisAlignedBB(this.av.a(-2, -2, -2), this.av.a(2, 2, 2)));
                     iterator = list.iterator();
 
                     while (iterator.hasNext()) {
                         entityliving = (EntityLiving) iterator.next();
-                        if (entityliving.getUniqueID().equals(this.av)) {
+                        if (entityliving.getUniqueID().equals(this.au)) {
                             this.target = entityliving;
                             break;
                         }
                     }
 
-                    this.av = null;
+                    this.au = null;
                 }
 
                 if (this.shooter == null && this.h != null) {
-                    list = this.world.a(EntityLiving.class, new AxisAlignedBB(this.au.a(-2, -2, -2), this.au.a(2, 2, 2)));
+                    list = this.world.a(EntityLiving.class, new AxisAlignedBB(this.at.a(-2, -2, -2), this.at.a(2, 2, 2)));
                     iterator = list.iterator();
 
                     while (iterator.hasNext()) {
@@ -289,7 +289,7 @@
     protected void a(MovingObjectPosition movingobjectposition) {
         if (movingobjectposition.entity == null) {
             ((WorldServer) this.world).a(EnumParticle.EXPLOSION_LARGE, this.locX, this.locY, this.locZ, 2, 0.2D, 0.2D, 0.2D, 0.0D, new int[0]);
-            this.a(SoundEffects.fg, 1.0F, 1.0F);
+            this.a(SoundEffects.fC, 1.0F, 1.0F);
         } else {
             boolean flag = movingobjectposition.entity.damageEntity(DamageSource.a(this, this.shooter).b(), 4.0F);
 
@@ -310,7 +310,7 @@
 
     public boolean damageEntity(DamageSource damagesource, float f) {
         if (!this.world.isClientSide) {
-            this.a(SoundEffects.fh, 1.0F, 1.0F);
+            this.a(SoundEffects.fD, 1.0F, 1.0F);
             ((WorldServer) this.world).a(EnumParticle.CRIT, this.locX, this.locY, this.locZ, 15, 0.2D, 0.2D, 0.2D, 0.0D, new int[0]);
             this.die();
         }
