--- a/net/minecraft/server/EntityCaveSpider.java
+++ b/net/minecraft/server/EntityCaveSpider.java
@@ -10,7 +10,7 @@
     }
 
     public static void b(DataConverterManager dataconvertermanager) {
-        EntityInsentient.a(dataconvertermanager, "CaveSpider");
+        EntityInsentient.a(dataconvertermanager, EntityCaveSpider.class);
     }
 
     protected void initAttributes() {
@@ -51,6 +51,6 @@
 
     @Nullable
     protected MinecraftKey J() {
-        return LootTables.s;
+        return LootTables.t;
     }
 }
