--- a/net/minecraft/server/EntityAnimal.java
+++ b/net/minecraft/server/EntityAnimal.java
@@ -1,21 +1,19 @@
 package net.minecraft.server;
 
-import javax.annotation.Nullable;
-
 public abstract class EntityAnimal extends EntityAgeable implements IAnimal {
 
-    protected Block bA;
-    private int bx;
-    private EntityHuman by;
+    protected Block bz;
+    private int bw;
+    private EntityHuman bx;
 
     public EntityAnimal(World world) {
         super(world);
-        this.bA = Blocks.GRASS;
+        this.bz = Blocks.GRASS;
     }
 
     protected void M() {
         if (this.getAge() != 0) {
-            this.bx = 0;
+            this.bw = 0;
         }
 
         super.M();
@@ -24,12 +22,12 @@
     public void n() {
         super.n();
         if (this.getAge() != 0) {
-            this.bx = 0;
+            this.bw = 0;
         }
 
-        if (this.bx > 0) {
-            --this.bx;
-            if (this.bx % 10 == 0) {
+        if (this.bw > 0) {
+            --this.bw;
+            if (this.bw % 10 == 0) {
                 double d0 = this.random.nextGaussian() * 0.02D;
                 double d1 = this.random.nextGaussian() * 0.02D;
                 double d2 = this.random.nextGaussian() * 0.02D;
@@ -44,36 +42,36 @@
         if (this.isInvulnerable(damagesource)) {
             return false;
         } else {
-            this.bx = 0;
+            this.bw = 0;
             return super.damageEntity(damagesource, f);
         }
     }
 
     public float a(BlockPosition blockposition) {
-        return this.world.getType(blockposition.down()).getBlock() == Blocks.GRASS ? 10.0F : this.world.n(blockposition) - 0.5F;
+        return this.world.getType(blockposition.down()).getBlock() == this.bz ? 10.0F : this.world.n(blockposition) - 0.5F;
     }
 
     public void b(NBTTagCompound nbttagcompound) {
         super.b(nbttagcompound);
-        nbttagcompound.setInt("InLove", this.bx);
+        nbttagcompound.setInt("InLove", this.bw);
     }
 
     public double ax() {
-        return 0.29D;
+        return 0.14D;
     }
 
     public void a(NBTTagCompound nbttagcompound) {
         super.a(nbttagcompound);
-        this.bx = nbttagcompound.getInt("InLove");
+        this.bw = nbttagcompound.getInt("InLove");
     }
 
-    public boolean cK() {
+    public boolean cM() {
         int i = MathHelper.floor(this.locX);
         int j = MathHelper.floor(this.getBoundingBox().b);
         int k = MathHelper.floor(this.locZ);
         BlockPosition blockposition = new BlockPosition(i, j, k);
 
-        return this.world.getType(blockposition.down()).getBlock() == this.bA && this.world.j(blockposition) > 8 && super.cK();
+        return this.world.getType(blockposition.down()).getBlock() == this.bz && this.world.j(blockposition) > 8 && super.cM();
     }
 
     public int C() {
@@ -88,13 +86,15 @@
         return 1 + this.world.random.nextInt(3);
     }
 
-    public boolean e(@Nullable ItemStack itemstack) {
-        return itemstack == null ? false : itemstack.getItem() == Items.WHEAT;
+    public boolean e(ItemStack itemstack) {
+        return itemstack.getItem() == Items.WHEAT;
     }
 
-    public boolean a(EntityHuman entityhuman, EnumHand enumhand, @Nullable ItemStack itemstack) {
-        if (itemstack != null) {
-            if (this.e(itemstack) && this.getAge() == 0 && this.bx <= 0) {
+    public boolean a(EntityHuman entityhuman, EnumHand enumhand) {
+        ItemStack itemstack = entityhuman.b(enumhand);
+
+        if (!itemstack.isEmpty()) {
+            if (this.e(itemstack) && this.getAge() == 0 && this.bw <= 0) {
                 this.a(entityhuman, itemstack);
                 this.c(entityhuman);
                 return true;
@@ -107,32 +107,32 @@
             }
         }
 
-        return super.a(entityhuman, enumhand, itemstack);
+        return super.a(entityhuman, enumhand);
     }
 
     protected void a(EntityHuman entityhuman, ItemStack itemstack) {
         if (!entityhuman.abilities.canInstantlyBuild) {
-            --itemstack.count;
+            itemstack.subtract(1);
         }
 
     }
 
     public void c(EntityHuman entityhuman) {
-        this.bx = 600;
-        this.by = entityhuman;
+        this.bw = 600;
+        this.bx = entityhuman;
         this.world.broadcastEntityEffect(this, (byte) 18);
     }
 
     public EntityHuman getBreedCause() {
-        return this.by;
+        return this.bx;
     }
 
     public boolean isInLove() {
-        return this.bx > 0;
+        return this.bw > 0;
     }
 
     public void resetLove() {
-        this.bx = 0;
+        this.bw = 0;
     }
 
     public boolean mate(EntityAnimal entityanimal) {
