--- a/net/minecraft/server/EntityGiantZombie.java
+++ b/net/minecraft/server/EntityGiantZombie.java
@@ -10,7 +10,7 @@
     }
 
     public static void b(DataConverterManager dataconvertermanager) {
-        EntityInsentient.a(dataconvertermanager, "Giant");
+        EntityInsentient.a(dataconvertermanager, EntityGiantZombie.class);
     }
 
     public float getHeadHeight() {
@@ -30,6 +30,6 @@
 
     @Nullable
     protected MinecraftKey J() {
-        return LootTables.t;
+        return LootTables.u;
     }
 }
