--- a/net/minecraft/server/EntityDamageSource.java
+++ b/net/minecraft/server/EntityDamageSource.java
@@ -4,42 +4,43 @@
 
 public class EntityDamageSource extends DamageSource {
 
-    protected Entity t;
-    private boolean u;
+    @Nullable
+    protected Entity u;
+    private boolean v;
 
-    public EntityDamageSource(String s, Entity entity) {
+    public EntityDamageSource(String s, @Nullable Entity entity) {
         super(s);
-        this.t = entity;
+        this.u = entity;
     }
 
     public EntityDamageSource w() {
-        this.u = true;
+        this.v = true;
         return this;
     }
 
     public boolean x() {
-        return this.u;
+        return this.v;
     }
 
     @Nullable
     public Entity getEntity() {
-        return this.t;
+        return this.u;
     }
 
     public IChatBaseComponent getLocalizedDeathMessage(EntityLiving entityliving) {
-        ItemStack itemstack = this.t instanceof EntityLiving ? ((EntityLiving) this.t).getItemInMainHand() : null;
+        ItemStack itemstack = this.u instanceof EntityLiving ? ((EntityLiving) this.u).getItemInMainHand() : ItemStack.a;
         String s = "death.attack." + this.translationIndex;
         String s1 = s + ".item";
 
-        return itemstack != null && itemstack.hasName() && LocaleI18n.c(s1) ? new ChatMessage(s1, new Object[] { entityliving.getScoreboardDisplayName(), this.t.getScoreboardDisplayName(), itemstack.B()}) : new ChatMessage(s, new Object[] { entityliving.getScoreboardDisplayName(), this.t.getScoreboardDisplayName()});
+        return !itemstack.isEmpty() && itemstack.hasName() && LocaleI18n.c(s1) ? new ChatMessage(s1, new Object[] { entityliving.getScoreboardDisplayName(), this.u.getScoreboardDisplayName(), itemstack.C()}) : new ChatMessage(s, new Object[] { entityliving.getScoreboardDisplayName(), this.u.getScoreboardDisplayName()});
     }
 
     public boolean r() {
-        return this.t != null && this.t instanceof EntityLiving && !(this.t instanceof EntityHuman);
+        return this.u != null && this.u instanceof EntityLiving && !(this.u instanceof EntityHuman);
     }
 
     @Nullable
     public Vec3D v() {
-        return new Vec3D(this.t.locX, this.t.locY, this.t.locZ);
+        return new Vec3D(this.u.locX, this.u.locY, this.u.locZ);
     }
 }
