--- a/net/minecraft/server/EnumBlockMirror.java
+++ b/net/minecraft/server/EnumBlockMirror.java
@@ -15,11 +15,11 @@
         int k = j / 2;
         int l = i > k ? i - j : i;
 
-        switch (EnumBlockMirror.SyntheticClass_1.a[this.ordinal()]) {
-        case 1:
+        switch (this) {
+        case FRONT_BACK:
             return (j - l) % j;
 
-        case 2:
+        case LEFT_RIGHT:
             return (k - l + j) % j;
 
         default:
@@ -34,8 +34,8 @@
     }
 
     public EnumDirection b(EnumDirection enumdirection) {
-        switch (EnumBlockMirror.SyntheticClass_1.a[this.ordinal()]) {
-        case 1:
+        switch (this) {
+        case FRONT_BACK:
             if (enumdirection == EnumDirection.WEST) {
                 return EnumDirection.EAST;
             } else {
@@ -46,7 +46,7 @@
                 return enumdirection;
             }
 
-        case 2:
+        case LEFT_RIGHT:
             if (enumdirection == EnumDirection.NORTH) {
                 return EnumDirection.SOUTH;
             } else {
@@ -74,24 +74,4 @@
         }
 
     }
-
-    static class SyntheticClass_1 {
-
-        static final int[] a = new int[EnumBlockMirror.values().length];
-
-        static {
-            try {
-                EnumBlockMirror.SyntheticClass_1.a[EnumBlockMirror.FRONT_BACK.ordinal()] = 1;
-            } catch (NoSuchFieldError nosuchfielderror) {
-                ;
-            }
-
-            try {
-                EnumBlockMirror.SyntheticClass_1.a[EnumBlockMirror.LEFT_RIGHT.ordinal()] = 2;
-            } catch (NoSuchFieldError nosuchfielderror1) {
-                ;
-            }
-
-        }
-    }
 }
