--- a/net/minecraft/server/EnumChatFormat.java
+++ b/net/minecraft/server/EnumChatFormat.java
@@ -4,6 +4,7 @@
 import com.google.common.collect.Maps;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Locale;
 import java.util.Map;
 import java.util.regex.Pattern;
 import javax.annotation.Nullable;
@@ -13,7 +14,7 @@
     BLACK("BLACK", '0', 0), DARK_BLUE("DARK_BLUE", '1', 1), DARK_GREEN("DARK_GREEN", '2', 2), DARK_AQUA("DARK_AQUA", '3', 3), DARK_RED("DARK_RED", '4', 4), DARK_PURPLE("DARK_PURPLE", '5', 5), GOLD("GOLD", '6', 6), GRAY("GRAY", '7', 7), DARK_GRAY("DARK_GRAY", '8', 8), BLUE("BLUE", '9', 9), GREEN("GREEN", 'a', 10), AQUA("AQUA", 'b', 11), RED("RED", 'c', 12), LIGHT_PURPLE("LIGHT_PURPLE", 'd', 13), YELLOW("YELLOW", 'e', 14), WHITE("WHITE", 'f', 15), OBFUSCATED("OBFUSCATED", 'k', true), BOLD("BOLD", 'l', true), STRIKETHROUGH("STRIKETHROUGH", 'm', true), UNDERLINE("UNDERLINE", 'n', true), ITALIC("ITALIC", 'o', true), RESET("RESET", 'r', -1);
 
     private static final Map<String, EnumChatFormat> w = Maps.newHashMap();
-    private static final Pattern x = Pattern.compile("(?i)" + String.valueOf('\u00a7') + "[0-9A-FK-OR]");
+    private static final Pattern x = Pattern.compile("(?i)\u00a7[0-9A-FK-OR]");
     private final String y;
     private final char z;
     private final boolean A;
@@ -21,7 +22,7 @@
     private final int C;
 
     private static String c(String s) {
-        return s.toLowerCase().replaceAll("[^a-z]", "");
+        return s.toLowerCase(Locale.ROOT).replaceAll("[^a-z]", "");
     }
 
     private EnumChatFormat(String s, char c0, int i) {
@@ -53,7 +54,7 @@
     }
 
     public String e() {
-        return this.name().toLowerCase();
+        return this.name().toLowerCase(Locale.ROOT);
     }
 
     public String toString() {
