--- a/net/minecraft/server/ExceptionInvalidBlockState.java
+++ b/net/minecraft/server/ExceptionInvalidBlockState.java
@@ -0,0 +1,12 @@
+package net.minecraft.server;
+
+public class ExceptionInvalidBlockState extends CommandException {
+
+    public ExceptionInvalidBlockState() {
+        this("commands.generic.blockstate.invalid", new Object[0]);
+    }
+
+    public ExceptionInvalidBlockState(String s, Object... aobject) {
+        super(s, aobject);
+    }
+}
