--- a/net/minecraft/server/InsensitiveStringMap.java
+++ b/net/minecraft/server/InsensitiveStringMap.java
@@ -3,6 +3,7 @@
 import com.google.common.collect.Maps;
 import java.util.Collection;
 import java.util.Iterator;
+import java.util.Locale;
 import java.util.Map;
 import java.util.Set;
 import java.util.Map.Entry;
@@ -22,7 +23,7 @@
     }
 
     public boolean containsKey(Object object) {
-        return this.a.containsKey(object.toString().toLowerCase());
+        return this.a.containsKey(object.toString().toLowerCase(Locale.ROOT));
     }
 
     public boolean containsValue(Object object) {
@@ -30,15 +31,15 @@
     }
 
     public V get(Object object) {
-        return this.a.get(object.toString().toLowerCase());
+        return this.a.get(object.toString().toLowerCase(Locale.ROOT));
     }
 
     public V a(String s, V v0) {
-        return this.a.put(s.toLowerCase(), v0);
+        return this.a.put(s.toLowerCase(Locale.ROOT), v0);
     }
 
     public V remove(Object object) {
-        return this.a.remove(object.toString().toLowerCase());
+        return this.a.remove(object.toString().toLowerCase(Locale.ROOT));
     }
 
     public void putAll(Map<? extends String, ? extends V> map) {
