--- a/net/minecraft/server/ItemLingeringPotion.java
+++ b/net/minecraft/server/ItemLingeringPotion.java
@@ -4,18 +4,17 @@
 
     public ItemLingeringPotion() {}
 
-    public String a(ItemStack itemstack) {
-        return LocaleI18n.get(PotionUtil.c(itemstack).b("lingering_potion.effect."));
+    public String b(ItemStack itemstack) {
+        return LocaleI18n.get(PotionUtil.d(itemstack).b("lingering_potion.effect."));
     }
 
-    public InteractionResultWrapper<ItemStack> a(ItemStack itemstack, World world, EntityHuman entityhuman, EnumHand enumhand) {
-        if (!entityhuman.abilities.canInstantlyBuild) {
-            --itemstack.count;
-        }
+    public InteractionResultWrapper<ItemStack> a(World world, EntityHuman entityhuman, EnumHand enumhand) {
+        ItemStack itemstack = entityhuman.b(enumhand);
+        ItemStack itemstack1 = entityhuman.abilities.canInstantlyBuild ? itemstack.cloneItemStack() : itemstack.cloneAndSubtract(1);
 
-        world.a((EntityHuman) null, entityhuman.locX, entityhuman.locY, entityhuman.locZ, SoundEffects.do, SoundCategory.NEUTRAL, 0.5F, 0.4F / (ItemLingeringPotion.j.nextFloat() * 0.4F + 0.8F));
+        world.a((EntityHuman) null, entityhuman.locX, entityhuman.locY, entityhuman.locZ, SoundEffects.dy, SoundCategory.NEUTRAL, 0.5F, 0.4F / (ItemLingeringPotion.j.nextFloat() * 0.4F + 0.8F));
         if (!world.isClientSide) {
-            EntityPotion entitypotion = new EntityPotion(world, entityhuman, itemstack);
+            EntityPotion entitypotion = new EntityPotion(world, entityhuman, itemstack1);
 
             entitypotion.a(entityhuman, entityhuman.pitch, entityhuman.yaw, -20.0F, 0.5F, 1.0F);
             world.addEntity(entitypotion);
