--- a/net/minecraft/server/ItemSeeds.java
+++ b/net/minecraft/server/ItemSeeds.java
@@ -11,10 +11,12 @@
         this.a(CreativeModeTab.l);
     }
 
-    public EnumInteractionResult a(ItemStack itemstack, EntityHuman entityhuman, World world, BlockPosition blockposition, EnumHand enumhand, EnumDirection enumdirection, float f, float f1, float f2) {
+    public EnumInteractionResult a(EntityHuman entityhuman, World world, BlockPosition blockposition, EnumHand enumhand, EnumDirection enumdirection, float f, float f1, float f2) {
+        ItemStack itemstack = entityhuman.b(enumhand);
+
         if (enumdirection == EnumDirection.UP && entityhuman.a(blockposition.shift(enumdirection), enumdirection, itemstack) && world.getType(blockposition).getBlock() == this.b && world.isEmpty(blockposition.up())) {
             world.setTypeUpdate(blockposition.up(), this.a.getBlockData());
-            --itemstack.count;
+            itemstack.subtract(1);
             return EnumInteractionResult.SUCCESS;
         } else {
             return EnumInteractionResult.FAIL;
