--- a/net/minecraft/server/ItemLeash.java
+++ b/net/minecraft/server/ItemLeash.java
@@ -9,7 +9,7 @@
         this.a(CreativeModeTab.i);
     }
 
-    public EnumInteractionResult a(ItemStack itemstack, EntityHuman entityhuman, World world, BlockPosition blockposition, EnumHand enumhand, EnumDirection enumdirection, float f, float f1, float f2) {
+    public EnumInteractionResult a(EntityHuman entityhuman, World world, BlockPosition blockposition, EnumHand enumhand, EnumDirection enumdirection, float f, float f1, float f2) {
         Block block = world.getType(blockposition).getBlock();
 
         if (!(block instanceof BlockFence)) {
