--- a/net/minecraft/server/ItemSnow.java
+++ b/net/minecraft/server/ItemSnow.java
@@ -5,11 +5,12 @@
     public ItemSnow(Block block) {
         super(block);
         this.setMaxDurability(0);
-        this.a(true);
     }
 
-    public EnumInteractionResult a(ItemStack itemstack, EntityHuman entityhuman, World world, BlockPosition blockposition, EnumHand enumhand, EnumDirection enumdirection, float f, float f1, float f2) {
-        if (itemstack.count != 0 && entityhuman.a(blockposition, enumdirection, itemstack)) {
+    public EnumInteractionResult a(EntityHuman entityhuman, World world, BlockPosition blockposition, EnumHand enumhand, EnumDirection enumdirection, float f, float f1, float f2) {
+        ItemStack itemstack = entityhuman.b(enumhand);
+
+        if (!itemstack.isEmpty() && entityhuman.a(blockposition, enumdirection, itemstack)) {
             IBlockData iblockdata = world.getType(blockposition);
             Block block = iblockdata.getBlock();
             BlockPosition blockposition1 = blockposition;
@@ -23,21 +24,21 @@
             if (block == this.a) {
                 int i = ((Integer) iblockdata.get(BlockSnow.LAYERS)).intValue();
 
-                if (i <= 7) {
+                if (i < 8) {
                     IBlockData iblockdata1 = iblockdata.set(BlockSnow.LAYERS, Integer.valueOf(i + 1));
-                    AxisAlignedBB axisalignedbb = iblockdata1.d(world, blockposition1);
+                    AxisAlignedBB axisalignedbb = iblockdata1.c(world, blockposition1);
 
                     if (axisalignedbb != Block.k && world.c(axisalignedbb.a(blockposition1)) && world.setTypeAndData(blockposition1, iblockdata1, 10)) {
-                        SoundEffectType soundeffecttype = this.a.w();
+                        SoundEffectType soundeffecttype = this.a.getStepSound();
 
                         world.a(entityhuman, blockposition1, soundeffecttype.e(), SoundCategory.BLOCKS, (soundeffecttype.a() + 1.0F) / 2.0F, soundeffecttype.b() * 0.8F);
-                        --itemstack.count;
+                        itemstack.subtract(1);
                         return EnumInteractionResult.SUCCESS;
                     }
                 }
             }
 
-            return super.a(itemstack, entityhuman, world, blockposition1, enumhand, enumdirection, f, f1, f2);
+            return super.a(entityhuman, world, blockposition, enumhand, enumdirection, f, f1, f2);
         } else {
             return EnumInteractionResult.FAIL;
         }
