--- a/net/minecraft/server/ItemMultiTexture.java
+++ b/net/minecraft/server/ItemMultiTexture.java
@@ -1,25 +1,21 @@
 package net.minecraft.server;
 
-import com.google.common.base.Function;
-import javax.annotation.Nullable;
-
 public class ItemMultiTexture extends ItemBlock {
 
     protected final Block b;
-    protected final Function<ItemStack, String> c;
+    protected final ItemMultiTexture.a c;
 
-    public ItemMultiTexture(Block block, Block block1, Function<ItemStack, String> function) {
+    public ItemMultiTexture(Block block, Block block1, ItemMultiTexture.a itemmultitexture_a) {
         super(block);
         this.b = block1;
-        this.c = function;
+        this.c = itemmultitexture_a;
         this.setMaxDurability(0);
         this.a(true);
     }
 
     public ItemMultiTexture(Block block, Block block1, final String[] astring) {
-        this(block, block1, new Function() {
-            @Nullable
-            public String a(@Nullable ItemStack itemstack) {
+        this(block, block1, new ItemMultiTexture.a() {
+            public String a(ItemStack itemstack) {
                 int i = itemstack.getData();
 
                 if (i < 0 || i >= astring.length) {
@@ -28,10 +24,6 @@
 
                 return astring[i];
             }
-
-            public Object apply(Object object) {
-                return this.a((ItemStack) object);
-            }
         });
     }
 
@@ -39,7 +31,12 @@
         return i;
     }
 
-    public String f_(ItemStack itemstack) {
-        return super.getName() + "." + (String) this.c.apply(itemstack);
+    public String a(ItemStack itemstack) {
+        return super.getName() + "." + this.c.a(itemstack);
+    }
+
+    public interface a {
+
+        String a(ItemStack itemstack);
     }
 }
