--- a/net/minecraft/server/IWorldAccess.java
+++ b/net/minecraft/server/IWorldAccess.java
@@ -16,6 +16,8 @@
 
     void a(int i, boolean flag, double d0, double d1, double d2, double d3, double d4, double d5, int... aint);
 
+    void a(int i, boolean flag, boolean flag1, double d0, double d1, double d2, double d3, double d4, double d5, int... aint);
+
     void a(Entity entity);
 
     void b(Entity entity);
