--- a/net/minecraft/server/LootItemFunctionSmelt.java
+++ b/net/minecraft/server/LootItemFunctionSmelt.java
@@ -18,13 +18,13 @@
     public ItemStack a(ItemStack itemstack, Random random, LootTableInfo loottableinfo) {
         ItemStack itemstack1 = RecipesFurnace.getInstance().getResult(itemstack);
 
-        if (itemstack1 == null) {
+        if (itemstack1.isEmpty()) {
             LootItemFunctionSmelt.a.warn("Couldn\'t smelt {} because there is no smelting recipe", new Object[] { itemstack});
             return itemstack;
         } else {
             ItemStack itemstack2 = itemstack1.cloneItemStack();
 
-            itemstack2.count = itemstack.count;
+            itemstack2.setCount(itemstack.getCount());
             return itemstack2;
         }
     }
