--- a/net/minecraft/server/MapIcon.java
+++ b/net/minecraft/server/MapIcon.java
@@ -2,26 +2,23 @@
 
 public class MapIcon {
 
-    private byte type;
+    private final MapIcon.Type type;
     private byte x;
     private byte y;
     private byte rotation;
 
-    public MapIcon(byte b0, byte b1, byte b2, byte b3) {
-        this.type = b0;
-        this.x = b1;
-        this.y = b2;
-        this.rotation = b3;
+    public MapIcon(MapIcon.Type mapicon_type, byte b0, byte b1, byte b2) {
+        this.type = mapicon_type;
+        this.x = b0;
+        this.y = b1;
+        this.rotation = b2;
     }
 
-    public MapIcon(MapIcon mapicon) {
-        this.type = mapicon.type;
-        this.x = mapicon.x;
-        this.y = mapicon.y;
-        this.rotation = mapicon.rotation;
+    public byte getType() {
+        return this.type.a();
     }
 
-    public byte getType() {
+    public MapIcon.Type b() {
         return this.type;
     }
 
@@ -50,11 +47,46 @@
     }
 
     public int hashCode() {
-        byte b0 = this.type;
+        byte b0 = this.type.a();
         int i = 31 * b0 + this.x;
 
         i = 31 * i + this.y;
         i = 31 * i + this.rotation;
         return i;
     }
+
+    public static enum Type {
+
+        PLAYER(false), FRAME(true), RED_MARKER(false), BLUE_MARKER(false), TARGET_X(true), TARGET_POINT(true), PLAYER_OFF_MAP(false), PLAYER_OFF_LIMITS(false), MANSION(true, 5393476), MONUMENT(true, 3830373);
+
+        private final byte k;
+        private final boolean l;
+        private final int m;
+
+        private Type(boolean flag) {
+            this(flag, -1);
+        }
+
+        private Type(boolean flag, int i) {
+            this.k = (byte) this.ordinal();
+            this.l = flag;
+            this.m = i;
+        }
+
+        public byte a() {
+            return this.k;
+        }
+
+        public boolean c() {
+            return this.m >= 0;
+        }
+
+        public int d() {
+            return this.m;
+        }
+
+        public static MapIcon.Type a(byte b0) {
+            return values()[b0];
+        }
+    }
 }
