--- a/net/minecraft/server/MojangStatisticsGenerator.java
+++ b/net/minecraft/server/MojangStatisticsGenerator.java
@@ -58,7 +58,7 @@
 
                         MinecraftServer minecraftserver = MojangStatisticsGenerator.this.e instanceof MinecraftServer ? (MinecraftServer) MojangStatisticsGenerator.this.e : null;
 
-                        HttpUtilities.a(MojangStatisticsGenerator.this.d, (Map) hashmap, true, minecraftserver == null ? null : minecraftserver.au());
+                        HttpUtilities.a(MojangStatisticsGenerator.this.d, (Map) hashmap, true, minecraftserver == null ? null : minecraftserver.av());
                     }
                 }
             }, 0L, 900000L);
@@ -73,7 +73,7 @@
         this.b("os_version", System.getProperty("os.version"));
         this.b("os_architecture", System.getProperty("os.arch"));
         this.b("java_version", System.getProperty("java.version"));
-        this.a("version", "1.10.2");
+        this.a("version", "1.11");
         this.e.b(this);
     }
 
