--- a/net/minecraft/server/PacketPlayInSetCreativeSlot.java
+++ b/net/minecraft/server/PacketPlayInSetCreativeSlot.java
@@ -7,7 +7,9 @@
     private int slot;
     private ItemStack b;
 
-    public PacketPlayInSetCreativeSlot() {}
+    public PacketPlayInSetCreativeSlot() {
+        this.b = ItemStack.a;
+    }
 
     public void a(PacketListenerPlayIn packetlistenerplayin) {
         packetlistenerplayin.a(this);
