--- a/net/minecraft/server/PathfinderGoalBeg.java
+++ b/net/minecraft/server/PathfinderGoalBeg.java
@@ -47,14 +47,12 @@
             EnumHand enumhand = aenumhand[j];
             ItemStack itemstack = entityhuman.b(enumhand);
 
-            if (itemstack != null) {
-                if (this.a.isTamed() && itemstack.getItem() == Items.BONE) {
-                    return true;
-                }
+            if (this.a.isTamed() && itemstack.getItem() == Items.BONE) {
+                return true;
+            }
 
-                if (this.a.e(itemstack)) {
-                    return true;
-                }
+            if (this.a.e(itemstack)) {
+                return true;
             }
         }
 
