--- a/net/minecraft/server/PathfinderGoalTame.java
+++ b/net/minecraft/server/PathfinderGoalTame.java
@@ -2,14 +2,14 @@
 
 public class PathfinderGoalTame extends PathfinderGoal {
 
-    private final EntityHorse entity;
+    private final EntityHorseAbstract entity;
     private final double b;
     private double c;
     private double d;
     private double e;
 
-    public PathfinderGoalTame(EntityHorse entityhorse, double d0) {
-        this.entity = entityhorse;
+    public PathfinderGoalTame(EntityHorseAbstract entityhorseabstract, double d0) {
+        this.entity = entityhorseabstract;
         this.b = d0;
         this.a(1);
     }
@@ -52,8 +52,7 @@
                 int j = this.entity.getMaxDomestication();
 
                 if (j > 0 && this.entity.getRandom().nextInt(j) < i) {
-                    this.entity.g((EntityHuman) entity);
-                    this.entity.world.broadcastEntityEffect(this.entity, (byte) 7);
+                    this.entity.h((EntityHuman) entity);
                     return;
                 }
 
@@ -61,7 +60,7 @@
             }
 
             this.entity.az();
-            this.entity.dJ();
+            this.entity.dF();
             this.entity.world.broadcastEntityEffect(this.entity, (byte) 6);
         }
 
