--- a/net/minecraft/server/PathfinderGoalPlay.java
+++ b/net/minecraft/server/PathfinderGoalPlay.java
@@ -29,7 +29,7 @@
             while (iterator.hasNext()) {
                 EntityVillager entityvillager = (EntityVillager) iterator.next();
 
-                if (entityvillager != this.a && !entityvillager.dg() && entityvillager.getAge() < 0) {
+                if (entityvillager != this.a && !entityvillager.dj() && entityvillager.getAge() < 0) {
                     double d1 = entityvillager.h(this.a);
 
                     if (d1 <= d0) {
