--- a/net/minecraft/server/PistonExtendsChecker.java
+++ b/net/minecraft/server/PistonExtendsChecker.java
@@ -32,7 +32,7 @@
         IBlockData iblockdata = this.a.getType(this.c);
 
         if (!BlockPiston.a(iblockdata, this.a, this.c, this.d, false)) {
-            if (iblockdata.o() == EnumPistonReaction.DESTROY) {
+            if (iblockdata.p() == EnumPistonReaction.DESTROY) {
                 this.f.add(this.c);
                 return true;
             } else {
@@ -124,7 +124,7 @@
                         return false;
                     }
 
-                    if (iblockdata.o() == EnumPistonReaction.DESTROY) {
+                    if (iblockdata.p() == EnumPistonReaction.DESTROY) {
                         this.f.add(blockposition2);
                         return true;
                     }
