--- a/net/minecraft/server/RemoteControlSession.java
+++ b/net/minecraft/server/RemoteControlSession.java
@@ -35,20 +35,20 @@
         while (true) {
             try {
                 if (!this.a) {
-                    break;
+                    return;
                 }
 
                 BufferedInputStream bufferedinputstream = new BufferedInputStream(this.j.getInputStream());
                 int i = bufferedinputstream.read(this.k, 0, 1460);
 
-                if (10 <= i) {
-                    byte b0 = 0;
-                    int j = StatusChallengeUtils.b(this.k, 0, i);
+                if (10 > i) {
+                    return;
+                }
 
-                    if (j != i - 4) {
-                        return;
-                    }
+                byte b0 = 0;
+                int j = StatusChallengeUtils.b(this.k, 0, i);
 
+                if (j == i - 4) {
                     int k = b0 + 4;
                     int l = StatusChallengeUtils.b(this.k, k, i);
 
@@ -92,19 +92,18 @@
                     }
                 }
             } catch (SocketTimeoutException sockettimeoutexception) {
-                break;
+                return;
             } catch (IOException ioexception) {
-                break;
+                return;
             } catch (Exception exception1) {
                 RemoteControlSession.h.error("Exception whilst parsing RCON input", exception1);
-                break;
+                return;
             } finally {
                 this.g();
             }
 
             return;
         }
-
     }
 
     private void a(int i, int j, String s) throws IOException {
