--- a/net/minecraft/server/SlotFurnaceFuel.java
+++ b/net/minecraft/server/SlotFurnaceFuel.java
@@ -1,14 +1,12 @@
 package net.minecraft.server;
 
-import javax.annotation.Nullable;
-
 public class SlotFurnaceFuel extends Slot {
 
     public SlotFurnaceFuel(IInventory iinventory, int i, int j, int k) {
         super(iinventory, i, j, k);
     }
 
-    public boolean isAllowed(@Nullable ItemStack itemstack) {
+    public boolean isAllowed(ItemStack itemstack) {
         return TileEntityFurnace.isFuel(itemstack) || d_(itemstack);
     }
 
@@ -17,6 +15,6 @@
     }
 
     public static boolean d_(ItemStack itemstack) {
-        return itemstack != null && itemstack.getItem() != null && itemstack.getItem() == Items.BUCKET;
+        return itemstack.getItem() == Items.BUCKET;
     }
 }
