--- a/net/minecraft/server/SoundEffect.java
+++ b/net/minecraft/server/SoundEffect.java
@@ -96,6 +96,8 @@
         a("block.sand.hit");
         a("block.sand.place");
         a("block.sand.step");
+        a("block.shulker_box.close");
+        a("block.shulker_box.open");
         a("block.slime.break");
         a("block.slime.fall");
         a("block.slime.hit");
@@ -185,6 +187,7 @@
         a("entity.elder_guardian.death");
         a("entity.elder_guardian.death_land");
         a("entity.elder_guardian.hurt");
+        a("entity.elder_guardian.flop");
         a("entity.elder_guardian.hurt_land");
         a("entity.enderdragon.ambient");
         a("entity.enderdragon.death");
@@ -205,9 +208,16 @@
         a("entity.endermite.hurt");
         a("entity.endermite.step");
         a("entity.enderpearl.throw");
+        a("entity.evocation_fangs.attack");
+        a("entity.evocation_illager.ambient");
+        a("entity.evocation_illager.cast_spell");
+        a("entity.evocation_illager.death");
+        a("entity.evocation_illager.hurt");
+        a("entity.evocation_illager.prepare_attack");
+        a("entity.evocation_illager.prepare_summon");
+        a("entity.evocation_illager.prepare_wololo");
         a("entity.experience_bottle.throw");
         a("entity.experience_orb.pickup");
-        a("entity.experience_orb.touch");
         a("entity.firework.blast");
         a("entity.firework.blast_far");
         a("entity.firework.large_blast");
@@ -280,6 +290,15 @@
         a("entity.lightning.impact");
         a("entity.lightning.thunder");
         a("entity.lingeringpotion.throw");
+        a("entity.llama.ambient");
+        a("entity.llama.angry");
+        a("entity.llama.chest");
+        a("entity.llama.death");
+        a("entity.llama.eat");
+        a("entity.llama.hurt");
+        a("entity.llama.spit");
+        a("entity.llama.step");
+        a("entity.llama.swag");
         a("entity.magmacube.death");
         a("entity.magmacube.hurt");
         a("entity.magmacube.jump");
@@ -288,6 +307,7 @@
         a("entity.minecart.riding");
         a("entity.mooshroom.shear");
         a("entity.mule.ambient");
+        a("entity.mule.chest");
         a("entity.mule.death");
         a("entity.mule.hurt");
         a("entity.painting.break");
@@ -381,12 +401,19 @@
         a("entity.stray.hurt");
         a("entity.stray.step");
         a("entity.tnt.primed");
+        a("entity.vex.ambient");
+        a("entity.vex.charge");
+        a("entity.vex.death");
+        a("entity.vex.hurt");
         a("entity.villager.ambient");
         a("entity.villager.death");
         a("entity.villager.hurt");
         a("entity.villager.no");
         a("entity.villager.trading");
         a("entity.villager.yes");
+        a("entity.vindication_illager.ambient");
+        a("entity.vindication_illager.death");
+        a("entity.vindication_illager.hurt");
         a("entity.witch.ambient");
         a("entity.witch.death");
         a("entity.witch.drink");
@@ -434,10 +461,12 @@
         a("entity.zombie_villager.step");
         a("item.armor.equip_chain");
         a("item.armor.equip_diamond");
+        a("item.armor.equip_elytra");
         a("item.armor.equip_generic");
         a("item.armor.equip_gold");
         a("item.armor.equip_iron");
         a("item.armor.equip_leather");
+        a("item.bottle.empty");
         a("item.bottle.fill");
         a("item.bottle.fill_dragonbreath");
         a("item.bucket.empty");
@@ -452,6 +481,7 @@
         a("item.shield.block");
         a("item.shield.break");
         a("item.shovel.flatten");
+        a("item.totem.use");
         a("music.creative");
         a("music.credits");
         a("music.dragon");
