--- a/net/minecraft/server/StructurePiece.java
+++ b/net/minecraft/server/StructurePiece.java
@@ -44,10 +44,10 @@
 
         this.a(i == -1 ? null : EnumDirection.fromType2(i));
         this.m = nbttagcompound.getInt("GD");
-        this.b(nbttagcompound);
+        this.a(nbttagcompound, world.getDataManager().h());
     }
 
-    protected abstract void b(NBTTagCompound nbttagcompound);
+    protected abstract void a(NBTTagCompound nbttagcompound, DefinedStructureManager definedstructuremanager);
 
     public void a(StructurePiece structurepiece, List<StructurePiece> list, Random random) {}
 
@@ -77,10 +77,6 @@
         return structurepiece;
     }
 
-    public BlockPosition a() {
-        return new BlockPosition(this.l.f());
-    }
-
     protected boolean a(World world, StructureBoundingBox structureboundingbox) {
         int i = Math.max(this.l.a - 1, structureboundingbox.a);
         int j = Math.max(this.l.b - 1, structureboundingbox.b);
@@ -138,15 +134,15 @@
         if (enumdirection == null) {
             return i;
         } else {
-            switch (StructurePiece.SyntheticClass_1.a[enumdirection.ordinal()]) {
-            case 1:
-            case 2:
+            switch (enumdirection) {
+            case NORTH:
+            case SOUTH:
                 return this.l.a + i;
 
-            case 3:
+            case WEST:
                 return this.l.d - j;
 
-            case 4:
+            case EAST:
                 return this.l.a + j;
 
             default:
@@ -165,15 +161,15 @@
         if (enumdirection == null) {
             return j;
         } else {
-            switch (StructurePiece.SyntheticClass_1.a[enumdirection.ordinal()]) {
-            case 1:
+            switch (enumdirection) {
+            case NORTH:
                 return this.l.f - j;
 
-            case 2:
+            case SOUTH:
                 return this.l.c + j;
 
-            case 3:
-            case 4:
+            case WEST:
+            case EAST:
                 return this.l.c + i;
 
             default:
@@ -213,7 +209,7 @@
         int j1 = this.b(i, k);
         BlockPosition blockposition = new BlockPosition(l, i1, j1);
 
-        return !structureboundingbox.b((BaseBlockPosition) blockposition) ? EnumSkyBlock.SKY.c : world.b(EnumSkyBlock.SKY, blockposition);
+        return !structureboundingbox.b((BaseBlockPosition) blockposition) ? EnumSkyBlock.SKY.c : world.getBrightness(EnumSkyBlock.SKY, blockposition);
     }
 
     protected void a(World world, StructureBoundingBox structureboundingbox, int i, int j, int k, int l, int i1, int j1) {
@@ -340,10 +336,16 @@
     protected boolean a(World world, StructureBoundingBox structureboundingbox, Random random, int i, int j, int k, MinecraftKey minecraftkey) {
         BlockPosition blockposition = new BlockPosition(this.a(i, k), this.d(j), this.b(i, k));
 
+        return this.a(world, structureboundingbox, random, blockposition, minecraftkey, (IBlockData) null);
+    }
+
+    protected boolean a(World world, StructureBoundingBox structureboundingbox, Random random, BlockPosition blockposition, MinecraftKey minecraftkey, @Nullable IBlockData iblockdata) {
         if (structureboundingbox.b((BaseBlockPosition) blockposition) && world.getType(blockposition).getBlock() != Blocks.CHEST) {
-            IBlockData iblockdata = Blocks.CHEST.getBlockData();
+            if (iblockdata == null) {
+                iblockdata = Blocks.CHEST.f(world, blockposition, Blocks.CHEST.getBlockData());
+            }
 
-            world.setTypeAndData(blockposition, Blocks.CHEST.f(world, blockposition, iblockdata), 2);
+            world.setTypeAndData(blockposition, iblockdata, 2);
             TileEntity tileentity = world.getTileEntity(blockposition);
 
             if (tileentity instanceof TileEntityChest) {
@@ -393,18 +395,18 @@
             this.c = EnumBlockRotation.NONE;
             this.b = EnumBlockMirror.NONE;
         } else {
-            switch (StructurePiece.SyntheticClass_1.a[enumdirection.ordinal()]) {
-            case 2:
+            switch (enumdirection) {
+            case SOUTH:
                 this.b = EnumBlockMirror.LEFT_RIGHT;
                 this.c = EnumBlockRotation.NONE;
                 break;
 
-            case 3:
+            case WEST:
                 this.b = EnumBlockMirror.LEFT_RIGHT;
                 this.c = EnumBlockRotation.CLOCKWISE_90;
                 break;
 
-            case 4:
+            case EAST:
                 this.b = EnumBlockMirror.NONE;
                 this.c = EnumBlockRotation.CLOCKWISE_90;
                 break;
@@ -417,38 +419,6 @@
 
     }
 
-    static class SyntheticClass_1 {
-
-        static final int[] a = new int[EnumDirection.values().length];
-
-        static {
-            try {
-                StructurePiece.SyntheticClass_1.a[EnumDirection.NORTH.ordinal()] = 1;
-            } catch (NoSuchFieldError nosuchfielderror) {
-                ;
-            }
-
-            try {
-                StructurePiece.SyntheticClass_1.a[EnumDirection.SOUTH.ordinal()] = 2;
-            } catch (NoSuchFieldError nosuchfielderror1) {
-                ;
-            }
-
-            try {
-                StructurePiece.SyntheticClass_1.a[EnumDirection.WEST.ordinal()] = 3;
-            } catch (NoSuchFieldError nosuchfielderror2) {
-                ;
-            }
-
-            try {
-                StructurePiece.SyntheticClass_1.a[EnumDirection.EAST.ordinal()] = 4;
-            } catch (NoSuchFieldError nosuchfielderror3) {
-                ;
-            }
-
-        }
-    }
-
     public abstract static class StructurePieceBlockSelector {
 
         protected IBlockData a;
