--- a/net/minecraft/server/TileEntityLightDetector.java
+++ b/net/minecraft/server/TileEntityLightDetector.java
@@ -4,7 +4,7 @@
 
     public TileEntityLightDetector() {}
 
-    public void E_() {
+    public void F_() {
         if (this.world != null && !this.world.isClientSide && this.world.getTime() % 20L == 0L) {
             this.e = this.getBlock();
             if (this.e instanceof BlockDaylightDetector) {
