--- a/net/minecraft/server/TileEntityEnchantTable.java
+++ b/net/minecraft/server/TileEntityEnchantTable.java
@@ -36,7 +36,7 @@
 
     }
 
-    public void E_() {
+    public void F_() {
         this.k = this.j;
         this.m = this.l;
         EntityHuman entityhuman = this.world.a((double) ((float) this.position.getX() + 0.5F), (double) ((float) this.position.getY() + 0.5F), (double) ((float) this.position.getZ() + 0.5F), 3.0D, false);
@@ -45,7 +45,7 @@
             double d0 = entityhuman.locX - (double) ((float) this.position.getX() + 0.5F);
             double d1 = entityhuman.locZ - (double) ((float) this.position.getZ() + 0.5F);
 
-            this.n = (float) MathHelper.b(d1, d0);
+            this.n = (float) MathHelper.c(d1, d0);
             this.j += 0.1F;
             if (this.j < 0.5F || TileEntityEnchantTable.o.nextInt(40) == 0) {
                 float f = this.h;
