--- a/net/minecraft/server/TileEntityEnderChest.java
+++ b/net/minecraft/server/TileEntityEnderChest.java
@@ -9,7 +9,7 @@
 
     public TileEntityEnderChest() {}
 
-    public void E_() {
+    public void F_() {
         if (++this.h % 20 * 4 == 0) {
             this.world.playBlockAction(this.position, Blocks.ENDER_CHEST, 1, this.g);
         }
@@ -25,7 +25,7 @@
             double d1 = (double) i + 0.5D;
 
             d0 = (double) k + 0.5D;
-            this.world.a((EntityHuman) null, d1, (double) j + 0.5D, d0, SoundEffects.aN, SoundCategory.BLOCKS, 0.5F, this.world.random.nextFloat() * 0.1F + 0.9F);
+            this.world.a((EntityHuman) null, d1, (double) j + 0.5D, d0, SoundEffects.aQ, SoundCategory.BLOCKS, 0.5F, this.world.random.nextFloat() * 0.1F + 0.9F);
         }
 
         if (this.g == 0 && this.a > 0.0F || this.g > 0 && this.a < 1.0F) {
@@ -47,7 +47,7 @@
                 d0 = (double) i + 0.5D;
                 double d2 = (double) k + 0.5D;
 
-                this.world.a((EntityHuman) null, d0, (double) j + 0.5D, d2, SoundEffects.aM, SoundCategory.BLOCKS, 0.5F, this.world.random.nextFloat() * 0.1F + 0.9F);
+                this.world.a((EntityHuman) null, d0, (double) j + 0.5D, d2, SoundEffects.aP, SoundCategory.BLOCKS, 0.5F, this.world.random.nextFloat() * 0.1F + 0.9F);
             }
 
             if (this.a < 0.0F) {
@@ -66,12 +66,12 @@
         }
     }
 
-    public void y() {
+    public void z() {
         this.invalidateBlockCache();
-        super.y();
+        super.z();
     }
 
-    public void d() {
+    public void a() {
         ++this.g;
         this.world.playBlockAction(this.position, Blocks.ENDER_CHEST, 1, this.g);
     }
@@ -82,6 +82,6 @@
     }
 
     public boolean a(EntityHuman entityhuman) {
-        return this.world.getTileEntity(this.position) != this ? false : entityhuman.e((double) this.position.getX() + 0.5D, (double) this.position.getY() + 0.5D, (double) this.position.getZ() + 0.5D) <= 64.0D;
+        return this.world.getTileEntity(this.position) != this ? false : entityhuman.d((double) this.position.getX() + 0.5D, (double) this.position.getY() + 0.5D, (double) this.position.getZ() + 0.5D) <= 64.0D;
     }
 }
