--- a/net/minecraft/server/TileEntityDropper.java
+++ b/net/minecraft/server/TileEntityDropper.java
@@ -5,11 +5,11 @@
     public TileEntityDropper() {}
 
     public static void b(DataConverterManager dataconvertermanager) {
-        dataconvertermanager.a(DataConverterTypes.BLOCK_ENTITY, (DataInspector) (new DataInspectorItemList("Dropper", new String[] { "Items"})));
+        dataconvertermanager.a(DataConverterTypes.BLOCK_ENTITY, (DataInspector) (new DataInspectorItemList(TileEntityDropper.class, new String[] { "Items"})));
     }
 
     public String getName() {
-        return this.hasCustomName() ? this.a : "container.dropper";
+        return this.hasCustomName() ? this.o : "container.dropper";
     }
 
     public String getContainerName() {
