--- a/net/minecraft/server/WorldManager.java
+++ b/net/minecraft/server/WorldManager.java
@@ -15,6 +15,8 @@
 
     public void a(int i, boolean flag, double d0, double d1, double d2, double d3, double d4, double d5, int... aint) {}
 
+    public void a(int i, boolean flag, boolean flag1, double d0, double d1, double d2, double d3, double d4, double d5, int... aint) {}
+
     public void a(Entity entity) {
         this.world.getTracker().track(entity);
         if (entity instanceof EntityPlayer) {
