--- a/net/minecraft/server/BlockStoneButton.java
+++ b/net/minecraft/server/BlockStoneButton.java
@@ -7,10 +7,10 @@
     }
 
     protected void a(EntityHuman entityhuman, World world, BlockPosition blockposition) {
-        world.a(entityhuman, blockposition, SoundEffects.gb, SoundCategory.BLOCKS, 0.3F, 0.6F);
+        world.a(entityhuman, blockposition, SoundEffects.gc, SoundCategory.BLOCKS, 0.3F, 0.6F);
     }
 
     protected void b(World world, BlockPosition blockposition) {
-        world.a((EntityHuman) null, blockposition, SoundEffects.ga, SoundCategory.BLOCKS, 0.3F, 0.5F);
+        world.a((EntityHuman) null, blockposition, SoundEffects.gb, SoundCategory.BLOCKS, 0.3F, 0.5F);
     }
 }
