--- a/net/minecraft/server/DataConverterManager.java
+++ b/net/minecraft/server/DataConverterManager.java
@@ -21,7 +21,7 @@
     public NBTTagCompound a(DataConverterType dataconvertertype, NBTTagCompound nbttagcompound) {
         int i = nbttagcompound.hasKeyOfType("DataVersion", 99) ? nbttagcompound.getInt("DataVersion") : -1;
 
-        return i >= 169 ? nbttagcompound : this.a(dataconvertertype, nbttagcompound, i);
+        return i >= 176 ? nbttagcompound : this.a(dataconvertertype, nbttagcompound, i);
     }
 
     public NBTTagCompound a(DataConverterType dataconvertertype, NBTTagCompound nbttagcompound, int i) {
