--- a/net/minecraft/server/DragonControllerLandedAttack.java
+++ b/net/minecraft/server/DragonControllerLandedAttack.java
@@ -9,7 +9,7 @@
     }
 
     public void b() {
-        this.a.world.a(this.a.locX, this.a.locY, this.a.locZ, SoundEffects.aQ, this.a.bz(), 2.5F, 0.8F + this.a.getRandom().nextFloat() * 0.3F, false);
+        this.a.world.a(this.a.locX, this.a.locY, this.a.locZ, SoundEffects.aR, this.a.bz(), 2.5F, 0.8F + this.a.getRandom().nextFloat() * 0.3F, false);
     }
 
     public void c() {
