--- a/net/minecraft/server/EntityPig.java
+++ b/net/minecraft/server/EntityPig.java
@@ -73,19 +73,19 @@
     }
 
     protected SoundEffect G() {
-        return SoundEffects.dP;
+        return SoundEffects.dQ;
     }
 
     protected SoundEffect bR() {
-        return SoundEffects.dR;
+        return SoundEffects.dS;
     }
 
     protected SoundEffect bS() {
-        return SoundEffects.dQ;
+        return SoundEffects.dR;
     }
 
     protected void a(BlockPosition blockposition, Block block) {
-        this.a(SoundEffects.dT, 0.15F, 1.0F);
+        this.a(SoundEffects.dU, 0.15F, 1.0F);
     }
 
     public boolean a(EntityHuman entityhuman, EnumHand enumhand, ItemStack itemstack) {
@@ -110,7 +110,7 @@
     }
 
     protected MinecraftKey J() {
-        return LootTables.C;
+        return LootTables.D;
     }
 
     public boolean hasSaddle() {
