--- a/net/minecraft/server/EntityPainting.java
+++ b/net/minecraft/server/EntityPainting.java
@@ -69,7 +69,7 @@
 
     public void a(Entity entity) {
         if (this.world.getGameRules().getBoolean("doEntityDrops")) {
-            this.a(SoundEffects.dN, 1.0F, 1.0F);
+            this.a(SoundEffects.dO, 1.0F, 1.0F);
             if (entity instanceof EntityHuman) {
                 EntityHuman entityhuman = (EntityHuman) entity;
 
@@ -83,7 +83,7 @@
     }
 
     public void o() {
-        this.a(SoundEffects.dO, 1.0F, 1.0F);
+        this.a(SoundEffects.dP, 1.0F, 1.0F);
     }
 
     public void setPositionRotation(double d0, double d1, double d2, float f, float f1) {
