--- a/net/minecraft/server/ItemArmor.java
+++ b/net/minecraft/server/ItemArmor.java
@@ -7,8 +7,8 @@
 
 public class ItemArmor extends Item {
 
-    private static final int[] m = new int[] { 13, 15, 16, 11};
-    private static final UUID[] n = new UUID[] { UUID.fromString("845DB27C-C624-495F-8C9F-6020A9A58B6B"), UUID.fromString("D8499B04-0E66-4726-AB29-64469D734E0D"), UUID.fromString("9F3D476D-C118-4544-8365-64846904B48E"), UUID.fromString("2AD3F246-FEE1-4E67-B886-69FD380BB150")};
+    private static final int[] n = new int[] { 13, 15, 16, 11};
+    private static final UUID[] o = new UUID[] { UUID.fromString("845DB27C-C624-495F-8C9F-6020A9A58B6B"), UUID.fromString("D8499B04-0E66-4726-AB29-64469D734E0D"), UUID.fromString("9F3D476D-C118-4544-8365-64846904B48E"), UUID.fromString("2AD3F246-FEE1-4E67-B886-69FD380BB150")};
     public static final String[] a = new String[] { "minecraft:items/empty_armor_slot_boots", "minecraft:items/empty_armor_slot_leggings", "minecraft:items/empty_armor_slot_chestplate", "minecraft:items/empty_armor_slot_helmet"};
     public static final IDispenseBehavior b = new DispenseBehaviorItem() {
         protected ItemStack b(ISourceBlock isourceblock, ItemStack itemstack) {
@@ -19,8 +19,9 @@
     };
     public final EnumItemSlot c;
     public final int d;
-    public final int e;
-    private final ItemArmor.EnumArmorMaterial o;
+    public final float e;
+    public final int f;
+    private final ItemArmor.EnumArmorMaterial p;
 
     public static ItemStack a(ISourceBlock isourceblock, ItemStack itemstack) {
         BlockPosition blockposition = isourceblock.getBlockPosition().shift(BlockDispenser.e(isourceblock.f()));
@@ -49,26 +50,27 @@
     }
 
     public ItemArmor(ItemArmor.EnumArmorMaterial itemarmor_enumarmormaterial, int i, EnumItemSlot enumitemslot) {
-        this.o = itemarmor_enumarmormaterial;
+        this.p = itemarmor_enumarmormaterial;
         this.c = enumitemslot;
-        this.e = i;
+        this.f = i;
         this.d = itemarmor_enumarmormaterial.b(enumitemslot);
         this.setMaxDurability(itemarmor_enumarmormaterial.a(enumitemslot));
+        this.e = itemarmor_enumarmormaterial.e();
         this.maxStackSize = 1;
         this.a(CreativeModeTab.j);
         BlockDispenser.REGISTRY.a(this, ItemArmor.b);
     }
 
     public int c() {
-        return this.o.a();
+        return this.p.a();
     }
 
     public ItemArmor.EnumArmorMaterial d() {
-        return this.o;
+        return this.p;
     }
 
     public boolean e_(ItemStack itemstack) {
-        if (this.o != ItemArmor.EnumArmorMaterial.LEATHER) {
+        if (this.p != ItemArmor.EnumArmorMaterial.LEATHER) {
             return false;
         } else {
             NBTTagCompound nbttagcompound = itemstack.getTag();
@@ -78,7 +80,7 @@
     }
 
     public int b(ItemStack itemstack) {
-        if (this.o != ItemArmor.EnumArmorMaterial.LEATHER) {
+        if (this.p != ItemArmor.EnumArmorMaterial.LEATHER) {
             return 16777215;
         } else {
             NBTTagCompound nbttagcompound = itemstack.getTag();
@@ -96,7 +98,7 @@
     }
 
     public void c(ItemStack itemstack) {
-        if (this.o == ItemArmor.EnumArmorMaterial.LEATHER) {
+        if (this.p == ItemArmor.EnumArmorMaterial.LEATHER) {
             NBTTagCompound nbttagcompound = itemstack.getTag();
 
             if (nbttagcompound != null) {
@@ -111,7 +113,7 @@
     }
 
     public void a(ItemStack itemstack, int i) {
-        if (this.o != ItemArmor.EnumArmorMaterial.LEATHER) {
+        if (this.p != ItemArmor.EnumArmorMaterial.LEATHER) {
             throw new UnsupportedOperationException("Can\'t dye non-leather!");
         } else {
             NBTTagCompound nbttagcompound = itemstack.getTag();
@@ -132,7 +134,7 @@
     }
 
     public boolean a(ItemStack itemstack, ItemStack itemstack1) {
-        return this.o.c() == itemstack1.getItem() ? true : super.a(itemstack, itemstack1);
+        return this.p.c() == itemstack1.getItem() ? true : super.a(itemstack, itemstack1);
     }
 
     public InteractionResultWrapper<ItemStack> a(ItemStack itemstack, World world, EntityHuman entityhuman, EnumHand enumhand) {
@@ -152,7 +154,8 @@
         Multimap multimap = super.a(enumitemslot);
 
         if (enumitemslot == this.c) {
-            multimap.put(GenericAttributes.g.getName(), new AttributeModifier(ItemArmor.n[enumitemslot.b()], "Armor modifier", (double) this.d, 0));
+            multimap.put(GenericAttributes.g.getName(), new AttributeModifier(ItemArmor.o[enumitemslot.b()], "Armor modifier", (double) this.d, 0));
+            multimap.put(GenericAttributes.h.getName(), new AttributeModifier(ItemArmor.o[enumitemslot.b()], "Armor toughness", (double) this.e, 0));
         }
 
         return multimap;
@@ -160,24 +163,26 @@
 
     public static enum EnumArmorMaterial {
 
-        LEATHER("leather", 5, new int[] { 1, 2, 3, 1}, 15, SoundEffects.s), CHAIN("chainmail", 15, new int[] { 1, 4, 5, 2}, 12, SoundEffects.n), IRON("iron", 15, new int[] { 2, 5, 6, 2}, 9, SoundEffects.r), GOLD("gold", 7, new int[] { 1, 3, 5, 2}, 25, SoundEffects.q), DIAMOND("diamond", 33, new int[] { 3, 6, 8, 3}, 10, SoundEffects.o);
+        LEATHER("leather", 5, new int[] { 1, 2, 3, 1}, 15, SoundEffects.s, 0.0F), CHAIN("chainmail", 15, new int[] { 1, 4, 5, 2}, 12, SoundEffects.n, 0.0F), IRON("iron", 15, new int[] { 2, 5, 6, 2}, 9, SoundEffects.r, 0.0F), GOLD("gold", 7, new int[] { 1, 3, 5, 2}, 25, SoundEffects.q, 0.0F), DIAMOND("diamond", 33, new int[] { 3, 6, 8, 3}, 10, SoundEffects.o, 2.0F);
 
         private final String f;
         private final int g;
         private final int[] h;
         private final int i;
         private final SoundEffect j;
+        private final float k;
 
-        private EnumArmorMaterial(String s, int i, int[] aint, int j, SoundEffect soundeffect) {
+        private EnumArmorMaterial(String s, int i, int[] aint, int j, SoundEffect soundeffect, float f) {
             this.f = s;
             this.g = i;
             this.h = aint;
             this.i = j;
             this.j = soundeffect;
+            this.k = f;
         }
 
         public int a(EnumItemSlot enumitemslot) {
-            return ItemArmor.m[enumitemslot.b()] * this.g;
+            return ItemArmor.n[enumitemslot.b()] * this.g;
         }
 
         public int b(EnumItemSlot enumitemslot) {
@@ -195,5 +200,9 @@
         public Item c() {
             return this == ItemArmor.EnumArmorMaterial.LEATHER ? Items.LEATHER : (this == ItemArmor.EnumArmorMaterial.CHAIN ? Items.IRON_INGOT : (this == ItemArmor.EnumArmorMaterial.GOLD ? Items.GOLD_INGOT : (this == ItemArmor.EnumArmorMaterial.IRON ? Items.IRON_INGOT : (this == ItemArmor.EnumArmorMaterial.DIAMOND ? Items.DIAMOND : null))));
         }
+
+        public float e() {
+            return this.k;
+        }
     }
 }
