--- a/net/minecraft/server/PacketStatusOutServerInfo.java
+++ b/net/minecraft/server/PacketStatusOutServerInfo.java
@@ -16,7 +16,7 @@
     }
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
-        this.b = (ServerPing) ChatDeserializer.a(PacketStatusOutServerInfo.a, packetdataserializer.c(32767), ServerPing.class);
+        this.b = (ServerPing) ChatDeserializer.a(PacketStatusOutServerInfo.a, packetdataserializer.e(32767), ServerPing.class);
     }
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
