--- a/net/minecraft/server/PacketPlayOutRespawn.java
+++ b/net/minecraft/server/PacketPlayOutRespawn.java
@@ -26,7 +26,7 @@
         this.a = packetdataserializer.readInt();
         this.b = EnumDifficulty.getById(packetdataserializer.readUnsignedByte());
         this.c = WorldSettings.EnumGamemode.getById(packetdataserializer.readUnsignedByte());
-        this.d = WorldType.getType(packetdataserializer.c(16));
+        this.d = WorldType.getType(packetdataserializer.e(16));
         if (this.d == null) {
             this.d = WorldType.NORMAL;
         }
