--- a/net/minecraft/server/PacketPlayOutEntityVelocity.java
+++ b/net/minecraft/server/PacketPlayOutEntityVelocity.java
@@ -56,7 +56,7 @@
     }
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
-        packetdataserializer.b(this.a);
+        packetdataserializer.d(this.a);
         packetdataserializer.writeShort(this.b);
         packetdataserializer.writeShort(this.c);
         packetdataserializer.writeShort(this.d);
