--- a/net/minecraft/server/PacketPlayOutCombatEvent.java
+++ b/net/minecraft/server/PacketPlayOutCombatEvent.java
@@ -54,10 +54,10 @@
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
         packetdataserializer.a((Enum) this.a);
         if (this.a == PacketPlayOutCombatEvent.EnumCombatEventType.END_COMBAT) {
-            packetdataserializer.b(this.d);
+            packetdataserializer.d(this.d);
             packetdataserializer.writeInt(this.c);
         } else if (this.a == PacketPlayOutCombatEvent.EnumCombatEventType.ENTITY_DIED) {
-            packetdataserializer.b(this.b);
+            packetdataserializer.d(this.b);
             packetdataserializer.writeInt(this.c);
             packetdataserializer.a(this.e);
         }
