--- a/net/minecraft/server/PacketPlayInTabComplete.java
+++ b/net/minecraft/server/PacketPlayInTabComplete.java
@@ -12,7 +12,7 @@
     public PacketPlayInTabComplete() {}
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
-        this.a = packetdataserializer.c(32767);
+        this.a = packetdataserializer.e(32767);
         this.b = packetdataserializer.readBoolean();
         boolean flag = packetdataserializer.readBoolean();
 
