--- a/net/minecraft/server/PacketPlayInSettings.java
+++ b/net/minecraft/server/PacketPlayInSettings.java
@@ -14,7 +14,7 @@
     public PacketPlayInSettings() {}
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
-        this.a = packetdataserializer.c(7);
+        this.a = packetdataserializer.e(7);
         this.b = packetdataserializer.readByte();
         this.c = (EntityHuman.EnumChatVisibility) packetdataserializer.a(EntityHuman.EnumChatVisibility.class);
         this.d = packetdataserializer.readBoolean();
