--- a/net/minecraft/server/PacketPlayInCustomPayload.java
+++ b/net/minecraft/server/PacketPlayInCustomPayload.java
@@ -11,7 +11,7 @@
     public PacketPlayInCustomPayload() {}
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
-        this.a = packetdataserializer.c(20);
+        this.a = packetdataserializer.e(20);
         int i = packetdataserializer.readableBytes();
 
         if (i >= 0 && i <= 32767) {
