--- a/net/minecraft/server/PacketPlayOutBlockAction.java
+++ b/net/minecraft/server/PacketPlayOutBlockAction.java
@@ -29,7 +29,7 @@
         packetdataserializer.a(this.a);
         packetdataserializer.writeByte(this.b);
         packetdataserializer.writeByte(this.c);
-        packetdataserializer.b(Block.getId(this.d) & 4095);
+        packetdataserializer.d(Block.getId(this.d) & 4095);
     }
 
     public void a(PacketListenerPlayOut packetlistenerplayout) {
