--- a/net/minecraft/server/PacketLoginOutDisconnect.java
+++ b/net/minecraft/server/PacketLoginOutDisconnect.java
@@ -13,7 +13,7 @@
     }
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
-        this.a = IChatBaseComponent.ChatSerializer.b(packetdataserializer.c(32767));
+        this.a = IChatBaseComponent.ChatSerializer.b(packetdataserializer.e(32767));
     }
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
