--- a/net/minecraft/server/PacketPlayOutSpawnEntityPainting.java
+++ b/net/minecraft/server/PacketPlayOutSpawnEntityPainting.java
@@ -24,13 +24,13 @@
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
         this.a = packetdataserializer.g();
         this.b = packetdataserializer.i();
-        this.e = packetdataserializer.c(EntityPainting.EnumArt.A);
+        this.e = packetdataserializer.e(EntityPainting.EnumArt.A);
         this.c = packetdataserializer.e();
         this.d = EnumDirection.fromType2(packetdataserializer.readUnsignedByte());
     }
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
-        packetdataserializer.b(this.a);
+        packetdataserializer.d(this.a);
         packetdataserializer.a(this.b);
         packetdataserializer.a(this.e);
         packetdataserializer.a(this.c);
