--- a/net/minecraft/server/PacketPlayInUseEntity.java
+++ b/net/minecraft/server/PacketPlayInUseEntity.java
@@ -30,7 +30,7 @@
     }
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
-        packetdataserializer.b(this.a);
+        packetdataserializer.d(this.a);
         packetdataserializer.a((Enum) this.action);
         if (this.action == PacketPlayInUseEntity.EnumEntityUseAction.INTERACT_AT) {
             packetdataserializer.writeFloat((float) this.c.x);
