--- a/net/minecraft/server/PacketPlayOutStatistic.java
+++ b/net/minecraft/server/PacketPlayOutStatistic.java
@@ -26,7 +26,7 @@
         this.a = Maps.newHashMap();
 
         for (int j = 0; j < i; ++j) {
-            Statistic statistic = StatisticList.getStatistic(packetdataserializer.c(32767));
+            Statistic statistic = StatisticList.getStatistic(packetdataserializer.e(32767));
             int k = packetdataserializer.g();
 
             if (statistic != null) {
@@ -37,14 +37,14 @@
     }
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
-        packetdataserializer.b(this.a.size());
+        packetdataserializer.d(this.a.size());
         Iterator iterator = this.a.entrySet().iterator();
 
         while (iterator.hasNext()) {
             Entry entry = (Entry) iterator.next();
 
             packetdataserializer.a(((Statistic) entry.getKey()).name);
-            packetdataserializer.b(((Integer) entry.getValue()).intValue());
+            packetdataserializer.d(((Integer) entry.getValue()).intValue());
         }
 
     }
