--- a/net/minecraft/server/PacketPlayOutMap.java
+++ b/net/minecraft/server/PacketPlayOutMap.java
@@ -59,10 +59,10 @@
     }
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
-        packetdataserializer.b(this.a);
+        packetdataserializer.d(this.a);
         packetdataserializer.writeByte(this.b);
         packetdataserializer.writeBoolean(this.c);
-        packetdataserializer.b(this.d.length);
+        packetdataserializer.d(this.d.length);
         MapIcon[] amapicon = this.d;
         int i = amapicon.length;
 
